{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "'''attempting to find similar number of clinvar variants Mar 23, filter out crap at first'''\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/pbs.487569.hn-10-03/ipykernel_54108/3835090091.py:1: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv('./variant_summary.txt', sep='\\t')\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv('./variant_summary.txt', sep='\\t')"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2829012, 34)\n"
     ]
    }
   ],
   "source": [
    "df = df[(df['Type'] == 'single nucleotide variant')]\n",
    "df_meta = df.copy()\n",
    "print(df_meta.shape)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "outputs": [
    {
     "data": {
      "text/plain": "         #AlleleID                       Type  \\\n4            15043  single nucleotide variant   \n5            15043  single nucleotide variant   \n6            15044  single nucleotide variant   \n7            15044  single nucleotide variant   \n8            15045  single nucleotide variant   \n...            ...                        ...   \n3251302    1863476  single nucleotide variant   \n3251305    1863478  single nucleotide variant   \n3251306    1863478  single nucleotide variant   \n3251307    1863479  single nucleotide variant   \n3251308    1863479  single nucleotide variant   \n\n                                                 Name  GeneID GeneSymbol  \\\n4        NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n5        NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n6         NM_017547.4(FOXRED1):c.694C>T (p.Gln232Ter)   55572    FOXRED1   \n7         NM_017547.4(FOXRED1):c.694C>T (p.Gln232Ter)   55572    FOXRED1   \n8        NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)   55572    FOXRED1   \n...                                               ...     ...        ...   \n3251302     NM_024616.3(C3orf52):c.34G>T (p.Glu12Ter)   79669    C3orf52   \n3251305      NM_004456.5(EZH2):c.395C>T (p.Pro132Leu)    2146       EZH2   \n3251306      NM_004456.5(EZH2):c.395C>T (p.Pro132Leu)    2146       EZH2   \n3251307     NM_001972.4(ELANE):c.640G>C (p.Gly214Arg)    1991      ELANE   \n3251308     NM_001972.4(ELANE):c.640G>C (p.Gly214Arg)    1991      ELANE   \n\n            HGNC_ID    ClinicalSignificance  ClinSigSimple LastEvaluated  \\\n4        HGNC:28986  Uncertain significance              0  Jun 29, 2015   \n5        HGNC:28986  Uncertain significance              0  Jun 29, 2015   \n6        HGNC:26927              Pathogenic              1  Dec 30, 2019   \n7        HGNC:26927              Pathogenic              1  Dec 30, 2019   \n8        HGNC:26927              Pathogenic              1  Oct 01, 2010   \n...             ...                     ...            ...           ...   \n3251302  HGNC:26255              Pathogenic              1  Dec 21, 2022   \n3251305   HGNC:3527            not provided              0             -   \n3251306   HGNC:3527            not provided              0             -   \n3251307   HGNC:3309            not provided              0             -   \n3251308   HGNC:3309            not provided              0             -   \n\n         RS# (dbSNP)  ...                                       ReviewStatus  \\\n4          150829393  ...                     no assertion criteria provided   \n5          150829393  ...                     no assertion criteria provided   \n6          267606829  ...  criteria provided, multiple submitters, no con...   \n7          267606829  ...  criteria provided, multiple submitters, no con...   \n8          267606830  ...                     no assertion criteria provided   \n...              ...  ...                                                ...   \n3251302           -1  ...                     no assertion criteria provided   \n3251305           -1  ...                              no assertion provided   \n3251306           -1  ...                              no assertion provided   \n3251307           -1  ...                              no assertion provided   \n3251308           -1  ...                              no assertion provided   \n\n        NumberSubmitters Guidelines TestedInGTR  \\\n4                      1          -           N   \n5                      1          -           N   \n6                      3          -           N   \n7                      3          -           N   \n8                      1          -           N   \n...                  ...        ...         ...   \n3251302                1          -           N   \n3251305                1          -           N   \n3251306                1          -           N   \n3251307                1          -           N   \n3251308                1          -           N   \n\n                                                  OtherIDs  \\\n4        OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...   \n5        OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...   \n6                        ClinGen:CA113792,OMIM:613622.0001   \n7                        ClinGen:CA113792,OMIM:613622.0001   \n8        ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...   \n...                                                    ...   \n3251302                                   OMIM:611956.0002   \n3251305                                                  -   \n3251306                                                  -   \n3251307                                                  -   \n3251308                                                  -   \n\n        SubmitterCategories VariationID PositionVCF ReferenceAlleleVCF  \\\n4                         1           4    85342440                  G   \n5                         1           4    84799209                  G   \n6                         3           5   126145284                  C   \n7                         3           5   126275389                  C   \n8                         1           6   126147412                  A   \n...                     ...         ...         ...                ...   \n3251302                   1     1806469   112086441                  G   \n3251305                   1     1806471   148526909                  G   \n3251306                   1     1806471   148829817                  G   \n3251307                   1     1806472      856000                  G   \n3251308                   1     1806472      856000                  G   \n\n         AlternateAlleleVCF  \n4                         A  \n5                         A  \n6                         T  \n7                         T  \n8                         G  \n...                     ...  \n3251302                   T  \n3251305                   A  \n3251306                   A  \n3251307                   C  \n3251308                   C  \n\n[2829012 rows x 34 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>#AlleleID</th>\n      <th>Type</th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>HGNC_ID</th>\n      <th>ClinicalSignificance</th>\n      <th>ClinSigSimple</th>\n      <th>LastEvaluated</th>\n      <th>RS# (dbSNP)</th>\n      <th>...</th>\n      <th>ReviewStatus</th>\n      <th>NumberSubmitters</th>\n      <th>Guidelines</th>\n      <th>TestedInGTR</th>\n      <th>OtherIDs</th>\n      <th>SubmitterCategories</th>\n      <th>VariationID</th>\n      <th>PositionVCF</th>\n      <th>ReferenceAlleleVCF</th>\n      <th>AlternateAlleleVCF</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>4</th>\n      <td>15043</td>\n      <td>single nucleotide variant</td>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>9640</td>\n      <td>ZNF592</td>\n      <td>HGNC:28986</td>\n      <td>Uncertain significance</td>\n      <td>0</td>\n      <td>Jun 29, 2015</td>\n      <td>150829393</td>\n      <td>...</td>\n      <td>no assertion criteria provided</td>\n      <td>1</td>\n      <td>-</td>\n      <td>N</td>\n      <td>OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...</td>\n      <td>1</td>\n      <td>4</td>\n      <td>85342440</td>\n      <td>G</td>\n      <td>A</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>15043</td>\n      <td>single nucleotide variant</td>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>9640</td>\n      <td>ZNF592</td>\n      <td>HGNC:28986</td>\n      <td>Uncertain significance</td>\n      <td>0</td>\n      <td>Jun 29, 2015</td>\n      <td>150829393</td>\n      <td>...</td>\n      <td>no assertion criteria provided</td>\n      <td>1</td>\n      <td>-</td>\n      <td>N</td>\n      <td>OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...</td>\n      <td>1</td>\n      <td>4</td>\n      <td>84799209</td>\n      <td>G</td>\n      <td>A</td>\n    </tr>\n    <tr>\n      <th>6</th>\n      <td>15044</td>\n      <td>single nucleotide variant</td>\n      <td>NM_017547.4(FOXRED1):c.694C&gt;T (p.Gln232Ter)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>HGNC:26927</td>\n      <td>Pathogenic</td>\n      <td>1</td>\n      <td>Dec 30, 2019</td>\n      <td>267606829</td>\n      <td>...</td>\n      <td>criteria provided, multiple submitters, no con...</td>\n      <td>3</td>\n      <td>-</td>\n      <td>N</td>\n      <td>ClinGen:CA113792,OMIM:613622.0001</td>\n      <td>3</td>\n      <td>5</td>\n      <td>126145284</td>\n      <td>C</td>\n      <td>T</td>\n    </tr>\n    <tr>\n      <th>7</th>\n      <td>15044</td>\n      <td>single nucleotide variant</td>\n      <td>NM_017547.4(FOXRED1):c.694C&gt;T (p.Gln232Ter)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>HGNC:26927</td>\n      <td>Pathogenic</td>\n      <td>1</td>\n      <td>Dec 30, 2019</td>\n      <td>267606829</td>\n      <td>...</td>\n      <td>criteria provided, multiple submitters, no con...</td>\n      <td>3</td>\n      <td>-</td>\n      <td>N</td>\n      <td>ClinGen:CA113792,OMIM:613622.0001</td>\n      <td>3</td>\n      <td>5</td>\n      <td>126275389</td>\n      <td>C</td>\n      <td>T</td>\n    </tr>\n    <tr>\n      <th>8</th>\n      <td>15045</td>\n      <td>single nucleotide variant</td>\n      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>HGNC:26927</td>\n      <td>Pathogenic</td>\n      <td>1</td>\n      <td>Oct 01, 2010</td>\n      <td>267606830</td>\n      <td>...</td>\n      <td>no assertion criteria provided</td>\n      <td>1</td>\n      <td>-</td>\n      <td>N</td>\n      <td>ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...</td>\n      <td>1</td>\n      <td>6</td>\n      <td>126147412</td>\n      <td>A</td>\n      <td>G</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>3251302</th>\n      <td>1863476</td>\n      <td>single nucleotide variant</td>\n      <td>NM_024616.3(C3orf52):c.34G&gt;T (p.Glu12Ter)</td>\n      <td>79669</td>\n      <td>C3orf52</td>\n      <td>HGNC:26255</td>\n      <td>Pathogenic</td>\n      <td>1</td>\n      <td>Dec 21, 2022</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>no assertion criteria provided</td>\n      <td>1</td>\n      <td>-</td>\n      <td>N</td>\n      <td>OMIM:611956.0002</td>\n      <td>1</td>\n      <td>1806469</td>\n      <td>112086441</td>\n      <td>G</td>\n      <td>T</td>\n    </tr>\n    <tr>\n      <th>3251305</th>\n      <td>1863478</td>\n      <td>single nucleotide variant</td>\n      <td>NM_004456.5(EZH2):c.395C&gt;T (p.Pro132Leu)</td>\n      <td>2146</td>\n      <td>EZH2</td>\n      <td>HGNC:3527</td>\n      <td>not provided</td>\n      <td>0</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>no assertion provided</td>\n      <td>1</td>\n      <td>-</td>\n      <td>N</td>\n      <td>-</td>\n      <td>1</td>\n      <td>1806471</td>\n      <td>148526909</td>\n      <td>G</td>\n      <td>A</td>\n    </tr>\n    <tr>\n      <th>3251306</th>\n      <td>1863478</td>\n      <td>single nucleotide variant</td>\n      <td>NM_004456.5(EZH2):c.395C&gt;T (p.Pro132Leu)</td>\n      <td>2146</td>\n      <td>EZH2</td>\n      <td>HGNC:3527</td>\n      <td>not provided</td>\n      <td>0</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>no assertion provided</td>\n      <td>1</td>\n      <td>-</td>\n      <td>N</td>\n      <td>-</td>\n      <td>1</td>\n      <td>1806471</td>\n      <td>148829817</td>\n      <td>G</td>\n      <td>A</td>\n    </tr>\n    <tr>\n      <th>3251307</th>\n      <td>1863479</td>\n      <td>single nucleotide variant</td>\n      <td>NM_001972.4(ELANE):c.640G&gt;C (p.Gly214Arg)</td>\n      <td>1991</td>\n      <td>ELANE</td>\n      <td>HGNC:3309</td>\n      <td>not provided</td>\n      <td>0</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>no assertion provided</td>\n      <td>1</td>\n      <td>-</td>\n      <td>N</td>\n      <td>-</td>\n      <td>1</td>\n      <td>1806472</td>\n      <td>856000</td>\n      <td>G</td>\n      <td>C</td>\n    </tr>\n    <tr>\n      <th>3251308</th>\n      <td>1863479</td>\n      <td>single nucleotide variant</td>\n      <td>NM_001972.4(ELANE):c.640G&gt;C (p.Gly214Arg)</td>\n      <td>1991</td>\n      <td>ELANE</td>\n      <td>HGNC:3309</td>\n      <td>not provided</td>\n      <td>0</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>no assertion provided</td>\n      <td>1</td>\n      <td>-</td>\n      <td>N</td>\n      <td>-</td>\n      <td>1</td>\n      <td>1806472</td>\n      <td>856000</td>\n      <td>G</td>\n      <td>C</td>\n    </tr>\n  </tbody>\n</table>\n<p>2829012 rows × 34 columns</p>\n</div>"
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "outputs": [
    {
     "data": {
      "text/plain": "   #AlleleID                       Type  \\\n7      15044  single nucleotide variant   \n\n                                          Name  GeneID GeneSymbol     HGNC_ID  \\\n7  NM_017547.4(FOXRED1):c.694C>T (p.Gln232Ter)   55572    FOXRED1  HGNC:26927   \n\n  ClinicalSignificance  ClinSigSimple LastEvaluated  RS# (dbSNP)  ...  \\\n7           Pathogenic              1  Dec 30, 2019    267606829  ...   \n\n                                        ReviewStatus NumberSubmitters  \\\n7  criteria provided, multiple submitters, no con...                3   \n\n  Guidelines TestedInGTR                           OtherIDs  \\\n7          -           N  ClinGen:CA113792,OMIM:613622.0001   \n\n  SubmitterCategories VariationID PositionVCF ReferenceAlleleVCF  \\\n7                   3           5   126275389                  C   \n\n   AlternateAlleleVCF  \n7                   T  \n\n[1 rows x 34 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>#AlleleID</th>\n      <th>Type</th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>HGNC_ID</th>\n      <th>ClinicalSignificance</th>\n      <th>ClinSigSimple</th>\n      <th>LastEvaluated</th>\n      <th>RS# (dbSNP)</th>\n      <th>...</th>\n      <th>ReviewStatus</th>\n      <th>NumberSubmitters</th>\n      <th>Guidelines</th>\n      <th>TestedInGTR</th>\n      <th>OtherIDs</th>\n      <th>SubmitterCategories</th>\n      <th>VariationID</th>\n      <th>PositionVCF</th>\n      <th>ReferenceAlleleVCF</th>\n      <th>AlternateAlleleVCF</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>7</th>\n      <td>15044</td>\n      <td>single nucleotide variant</td>\n      <td>NM_017547.4(FOXRED1):c.694C&gt;T (p.Gln232Ter)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>HGNC:26927</td>\n      <td>Pathogenic</td>\n      <td>1</td>\n      <td>Dec 30, 2019</td>\n      <td>267606829</td>\n      <td>...</td>\n      <td>criteria provided, multiple submitters, no con...</td>\n      <td>3</td>\n      <td>-</td>\n      <td>N</td>\n      <td>ClinGen:CA113792,OMIM:613622.0001</td>\n      <td>3</td>\n      <td>5</td>\n      <td>126275389</td>\n      <td>C</td>\n      <td>T</td>\n    </tr>\n  </tbody>\n</table>\n<p>1 rows × 34 columns</p>\n</div>"
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta.loc[[7]]"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "outputs": [
    {
     "data": {
      "text/plain": "         #AlleleID                       Type  \\\n4            15043  single nucleotide variant   \n5            15043  single nucleotide variant   \n3251244    1863445  single nucleotide variant   \n3251246    1863446  single nucleotide variant   \n3211451    1843263  single nucleotide variant   \n3211449    1843262  single nucleotide variant   \n3211581    1843328  single nucleotide variant   \n825282      532971  single nucleotide variant   \n328697      257098  single nucleotide variant   \n15383        23448  single nucleotide variant   \n16017        23801  single nucleotide variant   \n12772        22034  single nucleotide variant   \n46304        47999  single nucleotide variant   \n47999        49126  single nucleotide variant   \n552269      378229  single nucleotide variant   \n\n                                                   Name  GeneID GeneSymbol  \\\n4          NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n5          NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n3251244                    NM_004006.3(DMD):c.9224+2T>A    1756        DMD   \n3251246                    NM_004006.3(DMD):c.9361+1G>T    1756        DMD   \n3211451  NM_000179.3(MSH6):c.1956G>C (p.Gly652_Val653=)    2956       MSH6   \n3211449  NM_198253.3(TERT):c.1956G>A (p.Glu652_Arg653=)    7015       TERT   \n3211581     NM_001035.3(RYR2):c.10711A>T (p.Arg3571Ter)    6262       RYR2   \n825282      NM_000540.3(RYR1):c.15115T>A (p.Ter5039Arg)    6261       RYR1   \n328697        NM_000540.3(RYR1):c.15099T>C (p.Tyr5033=)    6261       RYR1   \n15383                                          CYP2C9*2    1559     CYP2C9   \n16017                     NM_199168.4(CXCL12):c.*531G>A    6387     CXCL12   \n12772                         NM_002543.4(OLR1):c.*188=    4973       OLR1   \n46304       NM_000106.5(CYP2D6):c.*264G>A (p.Gly169Arg)    1565     CYP2D6   \n47999          NM_005633.4(SOS1):c.571G>A (p.Glu191Lys)    6654       SOS1   \n552269           NM_003098.3(SNTA1):c.117G>A (p.Val39=)    6640      SNTA1   \n\n            HGNC_ID                 ClinicalSignificance  ClinSigSimple  \\\n4        HGNC:28986               Uncertain significance              0   \n5        HGNC:28986               Uncertain significance              0   \n3251244   HGNC:2928                    Likely pathogenic              1   \n3251246   HGNC:2928                    Likely pathogenic              1   \n3211451   HGNC:7329                        Likely benign              0   \n3211449  HGNC:11730                        Likely benign              0   \n3211581  HGNC:10484               Uncertain significance              0   \n825282   HGNC:10483               Uncertain significance              0   \n328697   HGNC:10483                        Likely benign              0   \n15383     HGNC:2623  Likely benign; drug response; other              0   \n16017    HGNC:10672                           protective              0   \n12772     HGNC:8133                          risk factor              0   \n46304     HGNC:2625                 Likely benign; other              0   \n47999    HGNC:11187               Uncertain significance              0   \n552269   HGNC:11167                        Likely benign              0   \n\n        LastEvaluated  RS# (dbSNP)  ...  \\\n4        Jun 29, 2015    150829393  ...   \n5        Jun 29, 2015    150829393  ...   \n3251244  Dec 16, 2022           -1  ...   \n3251246  Dec 16, 2022           -1  ...   \n3211451  Apr 09, 2022           -1  ...   \n3211449  Feb 13, 2022           -1  ...   \n3211581  Jan 20, 2022           -1  ...   \n825282   Sep 01, 2021   1555806206  ...   \n328697   Nov 05, 2015    751549832  ...   \n15383    Feb 11, 2019      1799853  ...   \n16017    Jan 16, 1998    387906400  ...   \n12772    Dec 01, 2003     12316150  ...   \n46304    Aug 06, 2018      5030865  ...   \n47999    May 04, 2022    886041241  ...   \n552269   Jan 28, 2016   1057520474  ...   \n\n                                              ReviewStatus NumberSubmitters  \\\n4                           no assertion criteria provided                1   \n5                           no assertion criteria provided                1   \n3251244                criteria provided, single submitter                1   \n3251246                criteria provided, single submitter                1   \n3211451                criteria provided, single submitter                1   \n3211449                criteria provided, single submitter                1   \n3211581                criteria provided, single submitter                1   \n825282                 criteria provided, single submitter                1   \n328697                 criteria provided, single submitter                1   \n15383    criteria provided, multiple submitters, no con...                5   \n16017                       no assertion criteria provided                1   \n12772                       no assertion criteria provided                1   \n46304    criteria provided, multiple submitters, no con...                2   \n47999    criteria provided, multiple submitters, no con...                8   \n552269                 criteria provided, single submitter                1   \n\n                                  Guidelines TestedInGTR  \\\n4                                          -           N   \n5                                          -           N   \n3251244                                    -           N   \n3251246                                    -           N   \n3211451  ACMG2013,ACMG2016,ACMG2021,ACMG2022           N   \n3211449                                    -           N   \n3211581  ACMG2013,ACMG2016,ACMG2021,ACMG2022           N   \n825282   ACMG2013,ACMG2016,ACMG2021,ACMG2022           N   \n328697   ACMG2013,ACMG2016,ACMG2021,ACMG2022           N   \n15383                                      -           Y   \n16017                                      -           N   \n12772                                      -           N   \n46304                                      -           N   \n47999                                      -           N   \n552269                                     -           N   \n\n                                                  OtherIDs  \\\n4        OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...   \n5        OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...   \n3251244                                                  -   \n3251246                                                  -   \n3211451                                                  -   \n3211449                                                  -   \n3211581                                                  -   \n825282                                 ClinGen:CA405694802   \n328697                                  ClinGen:CA10587338   \n15383    PharmGKB Clinical Annotation:637879781,ClinGen...   \n16017                    ClinGen:CA119898,OMIM:600835.0001   \n12772                    ClinGen:CA118584,OMIM:602601.0003   \n46304                                                    -   \n47999                                   ClinGen:CA10602366   \n552269                                  ClinGen:CA16608435   \n\n        SubmitterCategories VariationID PositionVCF ReferenceAlleleVCF  \\\n4                         1           4    85342440                  G   \n5                         1           4    84799209                  G   \n3251244                   2     1806436    31323596                  A   \n3251246                   2     1806437    31223046                  C   \n3211451                   2     1783318    47799939                  G   \n3211449                   2     1783317     1279465                  C   \n3211581                   2     1783383   237726294                  A   \n825282                    2      544396    38587418                  T   \n328697                    2      256456    38587402                  T   \n15383                     3        8409    94942290                  C   \n16017                     1        8762    44378102                  C   \n12772                     1        6995    10159692                  A   \n46304                     2       39399    42129033                  C   \n47999                     2       40656    39054763                  C   \n552269                    2      379036    32031310                  C   \n\n         AlternateAlleleVCF  \n4                         A  \n5                         A  \n3251244                   T  \n3251246                   A  \n3211451                   C  \n3211449                   T  \n3211581                   T  \n825282                    A  \n328697                    C  \n15383                     T  \n16017                     T  \n12772                     A  \n46304                     T  \n47999                     T  \n552269                    T  \n\n[15 rows x 34 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>#AlleleID</th>\n      <th>Type</th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>HGNC_ID</th>\n      <th>ClinicalSignificance</th>\n      <th>ClinSigSimple</th>\n      <th>LastEvaluated</th>\n      <th>RS# (dbSNP)</th>\n      <th>...</th>\n      <th>ReviewStatus</th>\n      <th>NumberSubmitters</th>\n      <th>Guidelines</th>\n      <th>TestedInGTR</th>\n      <th>OtherIDs</th>\n      <th>SubmitterCategories</th>\n      <th>VariationID</th>\n      <th>PositionVCF</th>\n      <th>ReferenceAlleleVCF</th>\n      <th>AlternateAlleleVCF</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>4</th>\n      <td>15043</td>\n      <td>single nucleotide variant</td>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>9640</td>\n      <td>ZNF592</td>\n      <td>HGNC:28986</td>\n      <td>Uncertain significance</td>\n      <td>0</td>\n      <td>Jun 29, 2015</td>\n      <td>150829393</td>\n      <td>...</td>\n      <td>no assertion criteria provided</td>\n      <td>1</td>\n      <td>-</td>\n      <td>N</td>\n      <td>OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...</td>\n      <td>1</td>\n      <td>4</td>\n      <td>85342440</td>\n      <td>G</td>\n      <td>A</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>15043</td>\n      <td>single nucleotide variant</td>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>9640</td>\n      <td>ZNF592</td>\n      <td>HGNC:28986</td>\n      <td>Uncertain significance</td>\n      <td>0</td>\n      <td>Jun 29, 2015</td>\n      <td>150829393</td>\n      <td>...</td>\n      <td>no assertion criteria provided</td>\n      <td>1</td>\n      <td>-</td>\n      <td>N</td>\n      <td>OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...</td>\n      <td>1</td>\n      <td>4</td>\n      <td>84799209</td>\n      <td>G</td>\n      <td>A</td>\n    </tr>\n    <tr>\n      <th>3251244</th>\n      <td>1863445</td>\n      <td>single nucleotide variant</td>\n      <td>NM_004006.3(DMD):c.9224+2T&gt;A</td>\n      <td>1756</td>\n      <td>DMD</td>\n      <td>HGNC:2928</td>\n      <td>Likely pathogenic</td>\n      <td>1</td>\n      <td>Dec 16, 2022</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>criteria provided, single submitter</td>\n      <td>1</td>\n      <td>-</td>\n      <td>N</td>\n      <td>-</td>\n      <td>2</td>\n      <td>1806436</td>\n      <td>31323596</td>\n      <td>A</td>\n      <td>T</td>\n    </tr>\n    <tr>\n      <th>3251246</th>\n      <td>1863446</td>\n      <td>single nucleotide variant</td>\n      <td>NM_004006.3(DMD):c.9361+1G&gt;T</td>\n      <td>1756</td>\n      <td>DMD</td>\n      <td>HGNC:2928</td>\n      <td>Likely pathogenic</td>\n      <td>1</td>\n      <td>Dec 16, 2022</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>criteria provided, single submitter</td>\n      <td>1</td>\n      <td>-</td>\n      <td>N</td>\n      <td>-</td>\n      <td>2</td>\n      <td>1806437</td>\n      <td>31223046</td>\n      <td>C</td>\n      <td>A</td>\n    </tr>\n    <tr>\n      <th>3211451</th>\n      <td>1843263</td>\n      <td>single nucleotide variant</td>\n      <td>NM_000179.3(MSH6):c.1956G&gt;C (p.Gly652_Val653=)</td>\n      <td>2956</td>\n      <td>MSH6</td>\n      <td>HGNC:7329</td>\n      <td>Likely benign</td>\n      <td>0</td>\n      <td>Apr 09, 2022</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>criteria provided, single submitter</td>\n      <td>1</td>\n      <td>ACMG2013,ACMG2016,ACMG2021,ACMG2022</td>\n      <td>N</td>\n      <td>-</td>\n      <td>2</td>\n      <td>1783318</td>\n      <td>47799939</td>\n      <td>G</td>\n      <td>C</td>\n    </tr>\n    <tr>\n      <th>3211449</th>\n      <td>1843262</td>\n      <td>single nucleotide variant</td>\n      <td>NM_198253.3(TERT):c.1956G&gt;A (p.Glu652_Arg653=)</td>\n      <td>7015</td>\n      <td>TERT</td>\n      <td>HGNC:11730</td>\n      <td>Likely benign</td>\n      <td>0</td>\n      <td>Feb 13, 2022</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>criteria provided, single submitter</td>\n      <td>1</td>\n      <td>-</td>\n      <td>N</td>\n      <td>-</td>\n      <td>2</td>\n      <td>1783317</td>\n      <td>1279465</td>\n      <td>C</td>\n      <td>T</td>\n    </tr>\n    <tr>\n      <th>3211581</th>\n      <td>1843328</td>\n      <td>single nucleotide variant</td>\n      <td>NM_001035.3(RYR2):c.10711A&gt;T (p.Arg3571Ter)</td>\n      <td>6262</td>\n      <td>RYR2</td>\n      <td>HGNC:10484</td>\n      <td>Uncertain significance</td>\n      <td>0</td>\n      <td>Jan 20, 2022</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>criteria provided, single submitter</td>\n      <td>1</td>\n      <td>ACMG2013,ACMG2016,ACMG2021,ACMG2022</td>\n      <td>N</td>\n      <td>-</td>\n      <td>2</td>\n      <td>1783383</td>\n      <td>237726294</td>\n      <td>A</td>\n      <td>T</td>\n    </tr>\n    <tr>\n      <th>825282</th>\n      <td>532971</td>\n      <td>single nucleotide variant</td>\n      <td>NM_000540.3(RYR1):c.15115T&gt;A (p.Ter5039Arg)</td>\n      <td>6261</td>\n      <td>RYR1</td>\n      <td>HGNC:10483</td>\n      <td>Uncertain significance</td>\n      <td>0</td>\n      <td>Sep 01, 2021</td>\n      <td>1555806206</td>\n      <td>...</td>\n      <td>criteria provided, single submitter</td>\n      <td>1</td>\n      <td>ACMG2013,ACMG2016,ACMG2021,ACMG2022</td>\n      <td>N</td>\n      <td>ClinGen:CA405694802</td>\n      <td>2</td>\n      <td>544396</td>\n      <td>38587418</td>\n      <td>T</td>\n      <td>A</td>\n    </tr>\n    <tr>\n      <th>328697</th>\n      <td>257098</td>\n      <td>single nucleotide variant</td>\n      <td>NM_000540.3(RYR1):c.15099T&gt;C (p.Tyr5033=)</td>\n      <td>6261</td>\n      <td>RYR1</td>\n      <td>HGNC:10483</td>\n      <td>Likely benign</td>\n      <td>0</td>\n      <td>Nov 05, 2015</td>\n      <td>751549832</td>\n      <td>...</td>\n      <td>criteria provided, single submitter</td>\n      <td>1</td>\n      <td>ACMG2013,ACMG2016,ACMG2021,ACMG2022</td>\n      <td>N</td>\n      <td>ClinGen:CA10587338</td>\n      <td>2</td>\n      <td>256456</td>\n      <td>38587402</td>\n      <td>T</td>\n      <td>C</td>\n    </tr>\n    <tr>\n      <th>15383</th>\n      <td>23448</td>\n      <td>single nucleotide variant</td>\n      <td>CYP2C9*2</td>\n      <td>1559</td>\n      <td>CYP2C9</td>\n      <td>HGNC:2623</td>\n      <td>Likely benign; drug response; other</td>\n      <td>0</td>\n      <td>Feb 11, 2019</td>\n      <td>1799853</td>\n      <td>...</td>\n      <td>criteria provided, multiple submitters, no con...</td>\n      <td>5</td>\n      <td>-</td>\n      <td>Y</td>\n      <td>PharmGKB Clinical Annotation:637879781,ClinGen...</td>\n      <td>3</td>\n      <td>8409</td>\n      <td>94942290</td>\n      <td>C</td>\n      <td>T</td>\n    </tr>\n    <tr>\n      <th>16017</th>\n      <td>23801</td>\n      <td>single nucleotide variant</td>\n      <td>NM_199168.4(CXCL12):c.*531G&gt;A</td>\n      <td>6387</td>\n      <td>CXCL12</td>\n      <td>HGNC:10672</td>\n      <td>protective</td>\n      <td>0</td>\n      <td>Jan 16, 1998</td>\n      <td>387906400</td>\n      <td>...</td>\n      <td>no assertion criteria provided</td>\n      <td>1</td>\n      <td>-</td>\n      <td>N</td>\n      <td>ClinGen:CA119898,OMIM:600835.0001</td>\n      <td>1</td>\n      <td>8762</td>\n      <td>44378102</td>\n      <td>C</td>\n      <td>T</td>\n    </tr>\n    <tr>\n      <th>12772</th>\n      <td>22034</td>\n      <td>single nucleotide variant</td>\n      <td>NM_002543.4(OLR1):c.*188=</td>\n      <td>4973</td>\n      <td>OLR1</td>\n      <td>HGNC:8133</td>\n      <td>risk factor</td>\n      <td>0</td>\n      <td>Dec 01, 2003</td>\n      <td>12316150</td>\n      <td>...</td>\n      <td>no assertion criteria provided</td>\n      <td>1</td>\n      <td>-</td>\n      <td>N</td>\n      <td>ClinGen:CA118584,OMIM:602601.0003</td>\n      <td>1</td>\n      <td>6995</td>\n      <td>10159692</td>\n      <td>A</td>\n      <td>A</td>\n    </tr>\n    <tr>\n      <th>46304</th>\n      <td>47999</td>\n      <td>single nucleotide variant</td>\n      <td>NM_000106.5(CYP2D6):c.*264G&gt;A (p.Gly169Arg)</td>\n      <td>1565</td>\n      <td>CYP2D6</td>\n      <td>HGNC:2625</td>\n      <td>Likely benign; other</td>\n      <td>0</td>\n      <td>Aug 06, 2018</td>\n      <td>5030865</td>\n      <td>...</td>\n      <td>criteria provided, multiple submitters, no con...</td>\n      <td>2</td>\n      <td>-</td>\n      <td>N</td>\n      <td>-</td>\n      <td>2</td>\n      <td>39399</td>\n      <td>42129033</td>\n      <td>C</td>\n      <td>T</td>\n    </tr>\n    <tr>\n      <th>47999</th>\n      <td>49126</td>\n      <td>single nucleotide variant</td>\n      <td>NM_005633.4(SOS1):c.571G&gt;A (p.Glu191Lys)</td>\n      <td>6654</td>\n      <td>SOS1</td>\n      <td>HGNC:11187</td>\n      <td>Uncertain significance</td>\n      <td>0</td>\n      <td>May 04, 2022</td>\n      <td>886041241</td>\n      <td>...</td>\n      <td>criteria provided, multiple submitters, no con...</td>\n      <td>8</td>\n      <td>-</td>\n      <td>N</td>\n      <td>ClinGen:CA10602366</td>\n      <td>2</td>\n      <td>40656</td>\n      <td>39054763</td>\n      <td>C</td>\n      <td>T</td>\n    </tr>\n    <tr>\n      <th>552269</th>\n      <td>378229</td>\n      <td>single nucleotide variant</td>\n      <td>NM_003098.3(SNTA1):c.117G&gt;A (p.Val39=)</td>\n      <td>6640</td>\n      <td>SNTA1</td>\n      <td>HGNC:11167</td>\n      <td>Likely benign</td>\n      <td>0</td>\n      <td>Jan 28, 2016</td>\n      <td>1057520474</td>\n      <td>...</td>\n      <td>criteria provided, single submitter</td>\n      <td>1</td>\n      <td>-</td>\n      <td>N</td>\n      <td>ClinGen:CA16608435</td>\n      <td>2</td>\n      <td>379036</td>\n      <td>32031310</td>\n      <td>C</td>\n      <td>T</td>\n    </tr>\n  </tbody>\n</table>\n<p>15 rows × 34 columns</p>\n</div>"
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "testing_df = df_meta.loc[[4, 5, 3251244,3251246]+[3211451,3211449, 3211581, 825282, 328697, 15383, 16017, 12772, 46304, 47999, 552269]]\n",
    "testing_df"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/pbs.478431.hn-10-03/ipykernel_96285/2928025377.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  testing_df['CDS_position'] = testing_df['Name'].str.extract(r'c\\.\\*?(\\d+)')\n",
      "/var/tmp/pbs.478431.hn-10-03/ipykernel_96285/2928025377.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  testing_df['AlleleChange'] = testing_df['Name'].str.extract(r'c\\.\\d+([A-Z])>') + '/' + testing_df['Name'].str.extract(r'>([A-Z])')\n",
      "/var/tmp/pbs.478431.hn-10-03/ipykernel_96285/2928025377.py:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  testing_df['Protein_Patterns'] = testing_df['Name'].str.extract(r'p\\.(\\w+\\d+[A-Z][a-z]+)', expand=False).fillna(np.nan)\n"
     ]
    },
    {
     "data": {
      "text/plain": "                                                   Name GeneSymbol Assembly  \\\n4          NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)     ZNF592   GRCh37   \n5          NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)     ZNF592   GRCh38   \n3251244                    NM_004006.3(DMD):c.9224+2T>A        DMD   GRCh38   \n3251246                    NM_004006.3(DMD):c.9361+1G>T        DMD   GRCh38   \n3211451  NM_000179.3(MSH6):c.1956G>C (p.Gly652_Val653=)       MSH6   GRCh38   \n3211449  NM_198253.3(TERT):c.1956G>A (p.Glu652_Arg653=)       TERT   GRCh38   \n3211581     NM_001035.3(RYR2):c.10711A>T (p.Arg3571Ter)       RYR2   GRCh38   \n825282      NM_000540.3(RYR1):c.15115T>A (p.Ter5039Arg)       RYR1   GRCh38   \n328697        NM_000540.3(RYR1):c.15099T>C (p.Tyr5033=)       RYR1   GRCh38   \n15383                                          CYP2C9*2     CYP2C9   GRCh38   \n16017                     NM_199168.4(CXCL12):c.*531G>A     CXCL12   GRCh38   \n12772                         NM_002543.4(OLR1):c.*188=       OLR1   GRCh38   \n46304       NM_000106.5(CYP2D6):c.*264G>A (p.Gly169Arg)     CYP2D6   GRCh38   \n47999          NM_005633.4(SOS1):c.571G>A (p.Glu191Lys)       SOS1   GRCh38   \n552269           NM_003098.3(SNTA1):c.117G>A (p.Val39=)      SNTA1   GRCh37   \n\n        ChromosomeAccession ReferenceAlleleVCF AlternateAlleleVCF  \\\n4               NC_000015.9                  G                  A   \n5              NC_000015.10                  G                  A   \n3251244        NC_000023.11                  A                  T   \n3251246        NC_000023.11                  C                  A   \n3211451        NC_000002.12                  G                  C   \n3211449        NC_000005.10                  C                  T   \n3211581        NC_000001.11                  A                  T   \n825282         NC_000019.10                  T                  A   \n328697         NC_000019.10                  T                  C   \n15383          NC_000010.11                  C                  T   \n16017          NC_000010.11                  C                  T   \n12772          NC_000012.12                  A                  A   \n46304          NC_000022.11                  C                  T   \n47999          NC_000002.12                  C                  T   \n552269         NC_000020.10                  C                  T   \n\n        CDS_position AlleleChange Protein_Patterns  \n4               3136          G/A       Gly1046Arg  \n5               3136          G/A       Gly1046Arg  \n3251244         9224          NaN              NaN  \n3251246         9361          NaN              NaN  \n3211451         1956          G/C              NaN  \n3211449         1956          G/A              NaN  \n3211581        10711          A/T       Arg3571Ter  \n825282         15115          T/A       Ter5039Arg  \n328697         15099          T/C              NaN  \n15383            NaN          NaN              NaN  \n16017            531          NaN              NaN  \n12772            188          NaN              NaN  \n46304            264          NaN        Gly169Arg  \n47999            571          G/A        Glu191Lys  \n552269           117          G/A              NaN  ",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ChromosomeAccession</th>\n      <th>ReferenceAlleleVCF</th>\n      <th>AlternateAlleleVCF</th>\n      <th>CDS_position</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>4</th>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>ZNF592</td>\n      <td>GRCh37</td>\n      <td>NC_000015.9</td>\n      <td>G</td>\n      <td>A</td>\n      <td>3136</td>\n      <td>G/A</td>\n      <td>Gly1046Arg</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>ZNF592</td>\n      <td>GRCh38</td>\n      <td>NC_000015.10</td>\n      <td>G</td>\n      <td>A</td>\n      <td>3136</td>\n      <td>G/A</td>\n      <td>Gly1046Arg</td>\n    </tr>\n    <tr>\n      <th>3251244</th>\n      <td>NM_004006.3(DMD):c.9224+2T&gt;A</td>\n      <td>DMD</td>\n      <td>GRCh38</td>\n      <td>NC_000023.11</td>\n      <td>A</td>\n      <td>T</td>\n      <td>9224</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>3251246</th>\n      <td>NM_004006.3(DMD):c.9361+1G&gt;T</td>\n      <td>DMD</td>\n      <td>GRCh38</td>\n      <td>NC_000023.11</td>\n      <td>C</td>\n      <td>A</td>\n      <td>9361</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>3211451</th>\n      <td>NM_000179.3(MSH6):c.1956G&gt;C (p.Gly652_Val653=)</td>\n      <td>MSH6</td>\n      <td>GRCh38</td>\n      <td>NC_000002.12</td>\n      <td>G</td>\n      <td>C</td>\n      <td>1956</td>\n      <td>G/C</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>3211449</th>\n      <td>NM_198253.3(TERT):c.1956G&gt;A (p.Glu652_Arg653=)</td>\n      <td>TERT</td>\n      <td>GRCh38</td>\n      <td>NC_000005.10</td>\n      <td>C</td>\n      <td>T</td>\n      <td>1956</td>\n      <td>G/A</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>3211581</th>\n      <td>NM_001035.3(RYR2):c.10711A&gt;T (p.Arg3571Ter)</td>\n      <td>RYR2</td>\n      <td>GRCh38</td>\n      <td>NC_000001.11</td>\n      <td>A</td>\n      <td>T</td>\n      <td>10711</td>\n      <td>A/T</td>\n      <td>Arg3571Ter</td>\n    </tr>\n    <tr>\n      <th>825282</th>\n      <td>NM_000540.3(RYR1):c.15115T&gt;A (p.Ter5039Arg)</td>\n      <td>RYR1</td>\n      <td>GRCh38</td>\n      <td>NC_000019.10</td>\n      <td>T</td>\n      <td>A</td>\n      <td>15115</td>\n      <td>T/A</td>\n      <td>Ter5039Arg</td>\n    </tr>\n    <tr>\n      <th>328697</th>\n      <td>NM_000540.3(RYR1):c.15099T&gt;C (p.Tyr5033=)</td>\n      <td>RYR1</td>\n      <td>GRCh38</td>\n      <td>NC_000019.10</td>\n      <td>T</td>\n      <td>C</td>\n      <td>15099</td>\n      <td>T/C</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>15383</th>\n      <td>CYP2C9*2</td>\n      <td>CYP2C9</td>\n      <td>GRCh38</td>\n      <td>NC_000010.11</td>\n      <td>C</td>\n      <td>T</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>16017</th>\n      <td>NM_199168.4(CXCL12):c.*531G&gt;A</td>\n      <td>CXCL12</td>\n      <td>GRCh38</td>\n      <td>NC_000010.11</td>\n      <td>C</td>\n      <td>T</td>\n      <td>531</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>12772</th>\n      <td>NM_002543.4(OLR1):c.*188=</td>\n      <td>OLR1</td>\n      <td>GRCh38</td>\n      <td>NC_000012.12</td>\n      <td>A</td>\n      <td>A</td>\n      <td>188</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>46304</th>\n      <td>NM_000106.5(CYP2D6):c.*264G&gt;A (p.Gly169Arg)</td>\n      <td>CYP2D6</td>\n      <td>GRCh38</td>\n      <td>NC_000022.11</td>\n      <td>C</td>\n      <td>T</td>\n      <td>264</td>\n      <td>NaN</td>\n      <td>Gly169Arg</td>\n    </tr>\n    <tr>\n      <th>47999</th>\n      <td>NM_005633.4(SOS1):c.571G&gt;A (p.Glu191Lys)</td>\n      <td>SOS1</td>\n      <td>GRCh38</td>\n      <td>NC_000002.12</td>\n      <td>C</td>\n      <td>T</td>\n      <td>571</td>\n      <td>G/A</td>\n      <td>Glu191Lys</td>\n    </tr>\n    <tr>\n      <th>552269</th>\n      <td>NM_003098.3(SNTA1):c.117G&gt;A (p.Val39=)</td>\n      <td>SNTA1</td>\n      <td>GRCh37</td>\n      <td>NC_000020.10</td>\n      <td>C</td>\n      <td>T</td>\n      <td>117</td>\n      <td>G/A</td>\n      <td>NaN</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# showing wield input regex have to extract\n",
    "testing_df = testing_df[['Name', 'GeneSymbol', 'Assembly', 'ChromosomeAccession','ReferenceAlleleVCF','AlternateAlleleVCF' ]]\n",
    "testing_df['CDS_position'] = testing_df['Name'].str.extract(r'c\\.\\*?(\\d+)')\n",
    "testing_df['AlleleChange'] = testing_df['Name'].str.extract(r'c\\.\\d+([A-Z])>') + '/' + testing_df['Name'].str.extract(r'>([A-Z])')\n",
    "# put na for those entires of intron changes\n",
    "import numpy as np\n",
    "testing_df['Protein_Patterns'] = testing_df['Name'].str.extract(r'p\\.(\\w+\\d+[A-Z][a-z]+)', expand=False).fillna(np.nan)\n",
    "testing_df"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/pbs.478431.hn-10-03/ipykernel_96285/3721374663.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  testing_df['Amino_acids_before_3letters'] = testing_df['Protein_Patterns'].apply(extract_amino_acid, before = True)\n",
      "/var/tmp/pbs.478431.hn-10-03/ipykernel_96285/3721374663.py:30: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  testing_df['Protein_positions'] = testing_df['Protein_Patterns'].apply(extract_amino_acid, pos = True)\n",
      "/var/tmp/pbs.478431.hn-10-03/ipykernel_96285/3721374663.py:31: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  testing_df['Amino_acids_after_3letters'] = testing_df['Protein_Patterns'].apply(extract_amino_acid, after = True)\n",
      "/var/tmp/pbs.478431.hn-10-03/ipykernel_96285/3721374663.py:32: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  testing_df['Protein_positions'] = testing_df['Protein_positions'].astype('Int64')\n"
     ]
    },
    {
     "data": {
      "text/plain": "                                                   Name GeneSymbol Assembly  \\\n4          NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)     ZNF592   GRCh37   \n5          NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)     ZNF592   GRCh38   \n3251244                    NM_004006.3(DMD):c.9224+2T>A        DMD   GRCh38   \n3251246                    NM_004006.3(DMD):c.9361+1G>T        DMD   GRCh38   \n3211451  NM_000179.3(MSH6):c.1956G>C (p.Gly652_Val653=)       MSH6   GRCh38   \n3211449  NM_198253.3(TERT):c.1956G>A (p.Glu652_Arg653=)       TERT   GRCh38   \n3211581     NM_001035.3(RYR2):c.10711A>T (p.Arg3571Ter)       RYR2   GRCh38   \n825282      NM_000540.3(RYR1):c.15115T>A (p.Ter5039Arg)       RYR1   GRCh38   \n328697        NM_000540.3(RYR1):c.15099T>C (p.Tyr5033=)       RYR1   GRCh38   \n15383                                          CYP2C9*2     CYP2C9   GRCh38   \n16017                     NM_199168.4(CXCL12):c.*531G>A     CXCL12   GRCh38   \n12772                         NM_002543.4(OLR1):c.*188=       OLR1   GRCh38   \n46304       NM_000106.5(CYP2D6):c.*264G>A (p.Gly169Arg)     CYP2D6   GRCh38   \n47999          NM_005633.4(SOS1):c.571G>A (p.Glu191Lys)       SOS1   GRCh38   \n552269           NM_003098.3(SNTA1):c.117G>A (p.Val39=)      SNTA1   GRCh37   \n\n        ChromosomeAccession ReferenceAlleleVCF AlternateAlleleVCF  \\\n4               NC_000015.9                  G                  A   \n5              NC_000015.10                  G                  A   \n3251244        NC_000023.11                  A                  T   \n3251246        NC_000023.11                  C                  A   \n3211451        NC_000002.12                  G                  C   \n3211449        NC_000005.10                  C                  T   \n3211581        NC_000001.11                  A                  T   \n825282         NC_000019.10                  T                  A   \n328697         NC_000019.10                  T                  C   \n15383          NC_000010.11                  C                  T   \n16017          NC_000010.11                  C                  T   \n12772          NC_000012.12                  A                  A   \n46304          NC_000022.11                  C                  T   \n47999          NC_000002.12                  C                  T   \n552269         NC_000020.10                  C                  T   \n\n        CDS_position AlleleChange Protein_Patterns  \\\n4               3136          G/A       Gly1046Arg   \n5               3136          G/A       Gly1046Arg   \n3251244         9224          NaN              NaN   \n3251246         9361          NaN              NaN   \n3211451         1956          G/C              NaN   \n3211449         1956          G/A              NaN   \n3211581        10711          A/T       Arg3571Ter   \n825282         15115          T/A       Ter5039Arg   \n328697         15099          T/C              NaN   \n15383            NaN          NaN              NaN   \n16017            531          NaN              NaN   \n12772            188          NaN              NaN   \n46304            264          NaN        Gly169Arg   \n47999            571          G/A        Glu191Lys   \n552269           117          G/A              NaN   \n\n        Amino_acids_before_3letters  Protein_positions  \\\n4                               Gly               1046   \n5                               Gly               1046   \n3251244                         NaN               <NA>   \n3251246                         NaN               <NA>   \n3211451                         NaN               <NA>   \n3211449                         NaN               <NA>   \n3211581                         Arg               3571   \n825282                          Ter               5039   \n328697                          NaN               <NA>   \n15383                           NaN               <NA>   \n16017                           NaN               <NA>   \n12772                           NaN               <NA>   \n46304                           Gly                169   \n47999                           Glu                191   \n552269                          NaN               <NA>   \n\n        Amino_acids_after_3letters  \n4                              Arg  \n5                              Arg  \n3251244                        NaN  \n3251246                        NaN  \n3211451                        NaN  \n3211449                        NaN  \n3211581                        Ter  \n825282                         Arg  \n328697                         NaN  \n15383                          NaN  \n16017                          NaN  \n12772                          NaN  \n46304                          Arg  \n47999                          Lys  \n552269                         NaN  ",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ChromosomeAccession</th>\n      <th>ReferenceAlleleVCF</th>\n      <th>AlternateAlleleVCF</th>\n      <th>CDS_position</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>4</th>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>ZNF592</td>\n      <td>GRCh37</td>\n      <td>NC_000015.9</td>\n      <td>G</td>\n      <td>A</td>\n      <td>3136</td>\n      <td>G/A</td>\n      <td>Gly1046Arg</td>\n      <td>Gly</td>\n      <td>1046</td>\n      <td>Arg</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>ZNF592</td>\n      <td>GRCh38</td>\n      <td>NC_000015.10</td>\n      <td>G</td>\n      <td>A</td>\n      <td>3136</td>\n      <td>G/A</td>\n      <td>Gly1046Arg</td>\n      <td>Gly</td>\n      <td>1046</td>\n      <td>Arg</td>\n    </tr>\n    <tr>\n      <th>3251244</th>\n      <td>NM_004006.3(DMD):c.9224+2T&gt;A</td>\n      <td>DMD</td>\n      <td>GRCh38</td>\n      <td>NC_000023.11</td>\n      <td>A</td>\n      <td>T</td>\n      <td>9224</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>&lt;NA&gt;</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>3251246</th>\n      <td>NM_004006.3(DMD):c.9361+1G&gt;T</td>\n      <td>DMD</td>\n      <td>GRCh38</td>\n      <td>NC_000023.11</td>\n      <td>C</td>\n      <td>A</td>\n      <td>9361</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>&lt;NA&gt;</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>3211451</th>\n      <td>NM_000179.3(MSH6):c.1956G&gt;C (p.Gly652_Val653=)</td>\n      <td>MSH6</td>\n      <td>GRCh38</td>\n      <td>NC_000002.12</td>\n      <td>G</td>\n      <td>C</td>\n      <td>1956</td>\n      <td>G/C</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>&lt;NA&gt;</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>3211449</th>\n      <td>NM_198253.3(TERT):c.1956G&gt;A (p.Glu652_Arg653=)</td>\n      <td>TERT</td>\n      <td>GRCh38</td>\n      <td>NC_000005.10</td>\n      <td>C</td>\n      <td>T</td>\n      <td>1956</td>\n      <td>G/A</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>&lt;NA&gt;</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>3211581</th>\n      <td>NM_001035.3(RYR2):c.10711A&gt;T (p.Arg3571Ter)</td>\n      <td>RYR2</td>\n      <td>GRCh38</td>\n      <td>NC_000001.11</td>\n      <td>A</td>\n      <td>T</td>\n      <td>10711</td>\n      <td>A/T</td>\n      <td>Arg3571Ter</td>\n      <td>Arg</td>\n      <td>3571</td>\n      <td>Ter</td>\n    </tr>\n    <tr>\n      <th>825282</th>\n      <td>NM_000540.3(RYR1):c.15115T&gt;A (p.Ter5039Arg)</td>\n      <td>RYR1</td>\n      <td>GRCh38</td>\n      <td>NC_000019.10</td>\n      <td>T</td>\n      <td>A</td>\n      <td>15115</td>\n      <td>T/A</td>\n      <td>Ter5039Arg</td>\n      <td>Ter</td>\n      <td>5039</td>\n      <td>Arg</td>\n    </tr>\n    <tr>\n      <th>328697</th>\n      <td>NM_000540.3(RYR1):c.15099T&gt;C (p.Tyr5033=)</td>\n      <td>RYR1</td>\n      <td>GRCh38</td>\n      <td>NC_000019.10</td>\n      <td>T</td>\n      <td>C</td>\n      <td>15099</td>\n      <td>T/C</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>&lt;NA&gt;</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>15383</th>\n      <td>CYP2C9*2</td>\n      <td>CYP2C9</td>\n      <td>GRCh38</td>\n      <td>NC_000010.11</td>\n      <td>C</td>\n      <td>T</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>&lt;NA&gt;</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>16017</th>\n      <td>NM_199168.4(CXCL12):c.*531G&gt;A</td>\n      <td>CXCL12</td>\n      <td>GRCh38</td>\n      <td>NC_000010.11</td>\n      <td>C</td>\n      <td>T</td>\n      <td>531</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>&lt;NA&gt;</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>12772</th>\n      <td>NM_002543.4(OLR1):c.*188=</td>\n      <td>OLR1</td>\n      <td>GRCh38</td>\n      <td>NC_000012.12</td>\n      <td>A</td>\n      <td>A</td>\n      <td>188</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>&lt;NA&gt;</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>46304</th>\n      <td>NM_000106.5(CYP2D6):c.*264G&gt;A (p.Gly169Arg)</td>\n      <td>CYP2D6</td>\n      <td>GRCh38</td>\n      <td>NC_000022.11</td>\n      <td>C</td>\n      <td>T</td>\n      <td>264</td>\n      <td>NaN</td>\n      <td>Gly169Arg</td>\n      <td>Gly</td>\n      <td>169</td>\n      <td>Arg</td>\n    </tr>\n    <tr>\n      <th>47999</th>\n      <td>NM_005633.4(SOS1):c.571G&gt;A (p.Glu191Lys)</td>\n      <td>SOS1</td>\n      <td>GRCh38</td>\n      <td>NC_000002.12</td>\n      <td>C</td>\n      <td>T</td>\n      <td>571</td>\n      <td>G/A</td>\n      <td>Glu191Lys</td>\n      <td>Glu</td>\n      <td>191</td>\n      <td>Lys</td>\n    </tr>\n    <tr>\n      <th>552269</th>\n      <td>NM_003098.3(SNTA1):c.117G&gt;A (p.Val39=)</td>\n      <td>SNTA1</td>\n      <td>GRCh37</td>\n      <td>NC_000020.10</td>\n      <td>C</td>\n      <td>T</td>\n      <td>117</td>\n      <td>G/A</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>&lt;NA&gt;</td>\n      <td>NaN</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# need to consider conditions of:\n",
    "# p.Tyr5033=  synonymous mutation\n",
    "# p.Ter5039*  premature stop codon\n",
    "\n",
    "def extract_amino_acid(row, before = False, pos = False, after = False):\n",
    "    import re\n",
    "    try:\n",
    "        if before:\n",
    "            first_aa = row[:3]\n",
    "            return first_aa\n",
    "        if pos:\n",
    "            pos = int(re.search(r'\\d+', row).group())\n",
    "            return pos\n",
    "        # the following command extract the index within the string where the position number ends\n",
    "        if after:\n",
    "            first_aa = row[:3]\n",
    "            position_end_index = re.search(r'\\d+', row).span()[-1]\n",
    "            later_part = row[position_end_index:]\n",
    "            if later_part == '=':\n",
    "                return first_aa\n",
    "            elif later_part == '*':\n",
    "                return 'STOP'\n",
    "            else:\n",
    "                return later_part\n",
    "    except TypeError:\n",
    "        import numpy as np\n",
    "        return np.nan\n",
    "\n",
    "testing_df['Amino_acids_before_3letters'] = testing_df['Protein_Patterns'].apply(extract_amino_acid, before = True)\n",
    "testing_df['Protein_positions'] = testing_df['Protein_Patterns'].apply(extract_amino_acid, pos = True)\n",
    "testing_df['Amino_acids_after_3letters'] = testing_df['Protein_Patterns'].apply(extract_amino_acid, after = True)\n",
    "testing_df['Protein_positions'] = testing_df['Protein_positions'].astype('Int64')\n",
    "testing_df"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "outputs": [
    {
     "data": {
      "text/plain": "                                               Name GeneSymbol Assembly  \\\n4      NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)     ZNF592   GRCh37   \n5      NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)     ZNF592   GRCh38   \n47999      NM_005633.4(SOS1):c.571G>A (p.Glu191Lys)       SOS1   GRCh38   \n\n      ChromosomeAccession ReferenceAlleleVCF AlternateAlleleVCF CDS_position  \\\n4             NC_000015.9                  G                  A         3136   \n5            NC_000015.10                  G                  A         3136   \n47999        NC_000002.12                  C                  T          571   \n\n      AlleleChange Protein_Patterns Amino_acids_before_3letters  \\\n4              G/A       Gly1046Arg                         Gly   \n5              G/A       Gly1046Arg                         Gly   \n47999          G/A        Glu191Lys                         Glu   \n\n       Protein_positions Amino_acids_after_3letters  \n4                   1046                        Arg  \n5                   1046                        Arg  \n47999                191                        Lys  ",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ChromosomeAccession</th>\n      <th>ReferenceAlleleVCF</th>\n      <th>AlternateAlleleVCF</th>\n      <th>CDS_position</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>4</th>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>ZNF592</td>\n      <td>GRCh37</td>\n      <td>NC_000015.9</td>\n      <td>G</td>\n      <td>A</td>\n      <td>3136</td>\n      <td>G/A</td>\n      <td>Gly1046Arg</td>\n      <td>Gly</td>\n      <td>1046</td>\n      <td>Arg</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>ZNF592</td>\n      <td>GRCh38</td>\n      <td>NC_000015.10</td>\n      <td>G</td>\n      <td>A</td>\n      <td>3136</td>\n      <td>G/A</td>\n      <td>Gly1046Arg</td>\n      <td>Gly</td>\n      <td>1046</td>\n      <td>Arg</td>\n    </tr>\n    <tr>\n      <th>47999</th>\n      <td>NM_005633.4(SOS1):c.571G&gt;A (p.Glu191Lys)</td>\n      <td>SOS1</td>\n      <td>GRCh38</td>\n      <td>NC_000002.12</td>\n      <td>C</td>\n      <td>T</td>\n      <td>571</td>\n      <td>G/A</td>\n      <td>Glu191Lys</td>\n      <td>Glu</td>\n      <td>191</td>\n      <td>Lys</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop introns and premature stop codon\n",
    "testing_df = testing_df[(testing_df['Protein_Patterns'].notnull()) &\n",
    "                        (testing_df['Amino_acids_before_3letters'] != 'Ter') &\n",
    "                        (testing_df['Amino_acids_after_3letters'] != 'Ter') &\n",
    "                        (testing_df['AlleleChange'].notnull())]\n",
    "testing_df"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "\"\"\"apply the above filtering scheme on the entire dataset\"\"\""
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "outputs": [
    {
     "data": {
      "text/plain": "         #AlleleID                       Type  \\\n4            15043  single nucleotide variant   \n5            15043  single nucleotide variant   \n6            15044  single nucleotide variant   \n7            15044  single nucleotide variant   \n8            15045  single nucleotide variant   \n...            ...                        ...   \n3251302    1863476  single nucleotide variant   \n3251305    1863478  single nucleotide variant   \n3251306    1863478  single nucleotide variant   \n3251307    1863479  single nucleotide variant   \n3251308    1863479  single nucleotide variant   \n\n                                                 Name  GeneID GeneSymbol  \\\n4        NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n5        NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n6         NM_017547.4(FOXRED1):c.694C>T (p.Gln232Ter)   55572    FOXRED1   \n7         NM_017547.4(FOXRED1):c.694C>T (p.Gln232Ter)   55572    FOXRED1   \n8        NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)   55572    FOXRED1   \n...                                               ...     ...        ...   \n3251302     NM_024616.3(C3orf52):c.34G>T (p.Glu12Ter)   79669    C3orf52   \n3251305      NM_004456.5(EZH2):c.395C>T (p.Pro132Leu)    2146       EZH2   \n3251306      NM_004456.5(EZH2):c.395C>T (p.Pro132Leu)    2146       EZH2   \n3251307     NM_001972.4(ELANE):c.640G>C (p.Gly214Arg)    1991      ELANE   \n3251308     NM_001972.4(ELANE):c.640G>C (p.Gly214Arg)    1991      ELANE   \n\n            HGNC_ID    ClinicalSignificance  ClinSigSimple LastEvaluated  \\\n4        HGNC:28986  Uncertain significance              0  Jun 29, 2015   \n5        HGNC:28986  Uncertain significance              0  Jun 29, 2015   \n6        HGNC:26927              Pathogenic              1  Dec 30, 2019   \n7        HGNC:26927              Pathogenic              1  Dec 30, 2019   \n8        HGNC:26927              Pathogenic              1  Oct 01, 2010   \n...             ...                     ...            ...           ...   \n3251302  HGNC:26255              Pathogenic              1  Dec 21, 2022   \n3251305   HGNC:3527            not provided              0             -   \n3251306   HGNC:3527            not provided              0             -   \n3251307   HGNC:3309            not provided              0             -   \n3251308   HGNC:3309            not provided              0             -   \n\n         RS# (dbSNP)  ... TestedInGTR  \\\n4          150829393  ...           N   \n5          150829393  ...           N   \n6          267606829  ...           N   \n7          267606829  ...           N   \n8          267606830  ...           N   \n...              ...  ...         ...   \n3251302           -1  ...           N   \n3251305           -1  ...           N   \n3251306           -1  ...           N   \n3251307           -1  ...           N   \n3251308           -1  ...           N   \n\n                                                  OtherIDs  \\\n4        OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...   \n5        OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...   \n6                        ClinGen:CA113792,OMIM:613622.0001   \n7                        ClinGen:CA113792,OMIM:613622.0001   \n8        ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...   \n...                                                    ...   \n3251302                                   OMIM:611956.0002   \n3251305                                                  -   \n3251306                                                  -   \n3251307                                                  -   \n3251308                                                  -   \n\n        SubmitterCategories VariationID PositionVCF ReferenceAlleleVCF  \\\n4                         1           4    85342440                  G   \n5                         1           4    84799209                  G   \n6                         3           5   126145284                  C   \n7                         3           5   126275389                  C   \n8                         1           6   126147412                  A   \n...                     ...         ...         ...                ...   \n3251302                   1     1806469   112086441                  G   \n3251305                   1     1806471   148526909                  G   \n3251306                   1     1806471   148829817                  G   \n3251307                   1     1806472      856000                  G   \n3251308                   1     1806472      856000                  G   \n\n        AlternateAlleleVCF CDS_position AlleleChange  Protein_Patterns  \n4                        A         3136          G/A        Gly1046Arg  \n5                        A         3136          G/A        Gly1046Arg  \n6                        T          694          C/T         Gln232Ter  \n7                        T          694          C/T         Gln232Ter  \n8                        G         1289          A/G         Asn430Ser  \n...                    ...          ...          ...               ...  \n3251302                  T           34          G/T          Glu12Ter  \n3251305                  A          395          C/T         Pro132Leu  \n3251306                  A          395          C/T         Pro132Leu  \n3251307                  C          640          G/C         Gly214Arg  \n3251308                  C          640          G/C         Gly214Arg  \n\n[2829012 rows x 37 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>#AlleleID</th>\n      <th>Type</th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>HGNC_ID</th>\n      <th>ClinicalSignificance</th>\n      <th>ClinSigSimple</th>\n      <th>LastEvaluated</th>\n      <th>RS# (dbSNP)</th>\n      <th>...</th>\n      <th>TestedInGTR</th>\n      <th>OtherIDs</th>\n      <th>SubmitterCategories</th>\n      <th>VariationID</th>\n      <th>PositionVCF</th>\n      <th>ReferenceAlleleVCF</th>\n      <th>AlternateAlleleVCF</th>\n      <th>CDS_position</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>4</th>\n      <td>15043</td>\n      <td>single nucleotide variant</td>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>9640</td>\n      <td>ZNF592</td>\n      <td>HGNC:28986</td>\n      <td>Uncertain significance</td>\n      <td>0</td>\n      <td>Jun 29, 2015</td>\n      <td>150829393</td>\n      <td>...</td>\n      <td>N</td>\n      <td>OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...</td>\n      <td>1</td>\n      <td>4</td>\n      <td>85342440</td>\n      <td>G</td>\n      <td>A</td>\n      <td>3136</td>\n      <td>G/A</td>\n      <td>Gly1046Arg</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>15043</td>\n      <td>single nucleotide variant</td>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>9640</td>\n      <td>ZNF592</td>\n      <td>HGNC:28986</td>\n      <td>Uncertain significance</td>\n      <td>0</td>\n      <td>Jun 29, 2015</td>\n      <td>150829393</td>\n      <td>...</td>\n      <td>N</td>\n      <td>OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...</td>\n      <td>1</td>\n      <td>4</td>\n      <td>84799209</td>\n      <td>G</td>\n      <td>A</td>\n      <td>3136</td>\n      <td>G/A</td>\n      <td>Gly1046Arg</td>\n    </tr>\n    <tr>\n      <th>6</th>\n      <td>15044</td>\n      <td>single nucleotide variant</td>\n      <td>NM_017547.4(FOXRED1):c.694C&gt;T (p.Gln232Ter)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>HGNC:26927</td>\n      <td>Pathogenic</td>\n      <td>1</td>\n      <td>Dec 30, 2019</td>\n      <td>267606829</td>\n      <td>...</td>\n      <td>N</td>\n      <td>ClinGen:CA113792,OMIM:613622.0001</td>\n      <td>3</td>\n      <td>5</td>\n      <td>126145284</td>\n      <td>C</td>\n      <td>T</td>\n      <td>694</td>\n      <td>C/T</td>\n      <td>Gln232Ter</td>\n    </tr>\n    <tr>\n      <th>7</th>\n      <td>15044</td>\n      <td>single nucleotide variant</td>\n      <td>NM_017547.4(FOXRED1):c.694C&gt;T (p.Gln232Ter)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>HGNC:26927</td>\n      <td>Pathogenic</td>\n      <td>1</td>\n      <td>Dec 30, 2019</td>\n      <td>267606829</td>\n      <td>...</td>\n      <td>N</td>\n      <td>ClinGen:CA113792,OMIM:613622.0001</td>\n      <td>3</td>\n      <td>5</td>\n      <td>126275389</td>\n      <td>C</td>\n      <td>T</td>\n      <td>694</td>\n      <td>C/T</td>\n      <td>Gln232Ter</td>\n    </tr>\n    <tr>\n      <th>8</th>\n      <td>15045</td>\n      <td>single nucleotide variant</td>\n      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>HGNC:26927</td>\n      <td>Pathogenic</td>\n      <td>1</td>\n      <td>Oct 01, 2010</td>\n      <td>267606830</td>\n      <td>...</td>\n      <td>N</td>\n      <td>ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...</td>\n      <td>1</td>\n      <td>6</td>\n      <td>126147412</td>\n      <td>A</td>\n      <td>G</td>\n      <td>1289</td>\n      <td>A/G</td>\n      <td>Asn430Ser</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>3251302</th>\n      <td>1863476</td>\n      <td>single nucleotide variant</td>\n      <td>NM_024616.3(C3orf52):c.34G&gt;T (p.Glu12Ter)</td>\n      <td>79669</td>\n      <td>C3orf52</td>\n      <td>HGNC:26255</td>\n      <td>Pathogenic</td>\n      <td>1</td>\n      <td>Dec 21, 2022</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>N</td>\n      <td>OMIM:611956.0002</td>\n      <td>1</td>\n      <td>1806469</td>\n      <td>112086441</td>\n      <td>G</td>\n      <td>T</td>\n      <td>34</td>\n      <td>G/T</td>\n      <td>Glu12Ter</td>\n    </tr>\n    <tr>\n      <th>3251305</th>\n      <td>1863478</td>\n      <td>single nucleotide variant</td>\n      <td>NM_004456.5(EZH2):c.395C&gt;T (p.Pro132Leu)</td>\n      <td>2146</td>\n      <td>EZH2</td>\n      <td>HGNC:3527</td>\n      <td>not provided</td>\n      <td>0</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>N</td>\n      <td>-</td>\n      <td>1</td>\n      <td>1806471</td>\n      <td>148526909</td>\n      <td>G</td>\n      <td>A</td>\n      <td>395</td>\n      <td>C/T</td>\n      <td>Pro132Leu</td>\n    </tr>\n    <tr>\n      <th>3251306</th>\n      <td>1863478</td>\n      <td>single nucleotide variant</td>\n      <td>NM_004456.5(EZH2):c.395C&gt;T (p.Pro132Leu)</td>\n      <td>2146</td>\n      <td>EZH2</td>\n      <td>HGNC:3527</td>\n      <td>not provided</td>\n      <td>0</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>N</td>\n      <td>-</td>\n      <td>1</td>\n      <td>1806471</td>\n      <td>148829817</td>\n      <td>G</td>\n      <td>A</td>\n      <td>395</td>\n      <td>C/T</td>\n      <td>Pro132Leu</td>\n    </tr>\n    <tr>\n      <th>3251307</th>\n      <td>1863479</td>\n      <td>single nucleotide variant</td>\n      <td>NM_001972.4(ELANE):c.640G&gt;C (p.Gly214Arg)</td>\n      <td>1991</td>\n      <td>ELANE</td>\n      <td>HGNC:3309</td>\n      <td>not provided</td>\n      <td>0</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>N</td>\n      <td>-</td>\n      <td>1</td>\n      <td>1806472</td>\n      <td>856000</td>\n      <td>G</td>\n      <td>C</td>\n      <td>640</td>\n      <td>G/C</td>\n      <td>Gly214Arg</td>\n    </tr>\n    <tr>\n      <th>3251308</th>\n      <td>1863479</td>\n      <td>single nucleotide variant</td>\n      <td>NM_001972.4(ELANE):c.640G&gt;C (p.Gly214Arg)</td>\n      <td>1991</td>\n      <td>ELANE</td>\n      <td>HGNC:3309</td>\n      <td>not provided</td>\n      <td>0</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>N</td>\n      <td>-</td>\n      <td>1</td>\n      <td>1806472</td>\n      <td>856000</td>\n      <td>G</td>\n      <td>C</td>\n      <td>640</td>\n      <td>G/C</td>\n      <td>Gly214Arg</td>\n    </tr>\n  </tbody>\n</table>\n<p>2829012 rows × 37 columns</p>\n</div>"
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# identify the amino acid changes\n",
    "import numpy as np\n",
    "df_meta['CDS_position'] = df_meta['Name'].str.extract(r'c\\.\\*?(\\d+)')\n",
    "df_meta['AlleleChange'] = df_meta['Name'].str.extract(r'c\\.\\d+([A-Z])>') + '/' + df_meta['Name'].str.extract(r'>([A-Z])')\n",
    "# put na for those entires of intron changes\n",
    "df_meta['Protein_Patterns'] = df_meta['Name'].str.extract(r'p\\.(\\w+\\d+[A-Z][a-z]+)', expand=False).fillna(np.nan)\n",
    "df_meta"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "outputs": [
    {
     "data": {
      "text/plain": "         #AlleleID                       Type  \\\n4            15043  single nucleotide variant   \n5            15043  single nucleotide variant   \n8            15045  single nucleotide variant   \n9            15045  single nucleotide variant   \n10           15046  single nucleotide variant   \n...            ...                        ...   \n3251294    1863472  single nucleotide variant   \n3251305    1863478  single nucleotide variant   \n3251306    1863478  single nucleotide variant   \n3251307    1863479  single nucleotide variant   \n3251308    1863479  single nucleotide variant   \n\n                                                 Name  GeneID GeneSymbol  \\\n4        NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n5        NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n8        NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)   55572    FOXRED1   \n9        NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)   55572    FOXRED1   \n10           NM_025152.3(NUBPL):c.166G>A (p.Gly56Arg)   80224      NUBPL   \n...                                               ...     ...        ...   \n3251294  NM_000089.4(COL1A2):c.3142G>A (p.Gly1048Ser)    1278     COL1A2   \n3251305      NM_004456.5(EZH2):c.395C>T (p.Pro132Leu)    2146       EZH2   \n3251306      NM_004456.5(EZH2):c.395C>T (p.Pro132Leu)    2146       EZH2   \n3251307     NM_001972.4(ELANE):c.640G>C (p.Gly214Arg)    1991      ELANE   \n3251308     NM_001972.4(ELANE):c.640G>C (p.Gly214Arg)    1991      ELANE   \n\n            HGNC_ID    ClinicalSignificance  ClinSigSimple LastEvaluated  \\\n4        HGNC:28986  Uncertain significance              0  Jun 29, 2015   \n5        HGNC:28986  Uncertain significance              0  Jun 29, 2015   \n8        HGNC:26927              Pathogenic              1  Oct 01, 2010   \n9        HGNC:26927              Pathogenic              1  Oct 01, 2010   \n10       HGNC:20278  Uncertain significance              0  Aug 24, 2020   \n...             ...                     ...            ...           ...   \n3251294   HGNC:2198       Likely pathogenic              1             -   \n3251305   HGNC:3527            not provided              0             -   \n3251306   HGNC:3527            not provided              0             -   \n3251307   HGNC:3309            not provided              0             -   \n3251308   HGNC:3309            not provided              0             -   \n\n         RS# (dbSNP)  ... VariationID PositionVCF ReferenceAlleleVCF  \\\n4          150829393  ...           4    85342440                  G   \n5          150829393  ...           4    84799209                  G   \n8          267606830  ...           6   126147412                  A   \n9          267606830  ...           6   126277517                  A   \n10         200401432  ...      214885    32031331                  G   \n...              ...  ...         ...         ...                ...   \n3251294           -1  ...     1806465    94427044                  G   \n3251305           -1  ...     1806471   148526909                  G   \n3251306           -1  ...     1806471   148829817                  G   \n3251307           -1  ...     1806472      856000                  G   \n3251308           -1  ...     1806472      856000                  G   \n\n        AlternateAlleleVCF CDS_position AlleleChange Protein_Patterns  \\\n4                        A         3136          G/A       Gly1046Arg   \n5                        A         3136          G/A       Gly1046Arg   \n8                        G         1289          A/G        Asn430Ser   \n9                        G         1289          A/G        Asn430Ser   \n10                       A          166          G/A         Gly56Arg   \n...                    ...          ...          ...              ...   \n3251294                  A         3142          G/A       Gly1048Ser   \n3251305                  A          395          C/T        Pro132Leu   \n3251306                  A          395          C/T        Pro132Leu   \n3251307                  C          640          G/C        Gly214Arg   \n3251308                  C          640          G/C        Gly214Arg   \n\n        Amino_acids_before_3letters Protein_positions  \\\n4                               Gly              1046   \n5                               Gly              1046   \n8                               Asn               430   \n9                               Asn               430   \n10                              Gly                56   \n...                             ...               ...   \n3251294                         Gly              1048   \n3251305                         Pro               132   \n3251306                         Pro               132   \n3251307                         Gly               214   \n3251308                         Gly               214   \n\n         Amino_acids_after_3letters  \n4                               Arg  \n5                               Arg  \n8                               Ser  \n9                               Ser  \n10                              Arg  \n...                             ...  \n3251294                         Ser  \n3251305                         Leu  \n3251306                         Leu  \n3251307                         Arg  \n3251308                         Arg  \n\n[1295746 rows x 40 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>#AlleleID</th>\n      <th>Type</th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>HGNC_ID</th>\n      <th>ClinicalSignificance</th>\n      <th>ClinSigSimple</th>\n      <th>LastEvaluated</th>\n      <th>RS# (dbSNP)</th>\n      <th>...</th>\n      <th>VariationID</th>\n      <th>PositionVCF</th>\n      <th>ReferenceAlleleVCF</th>\n      <th>AlternateAlleleVCF</th>\n      <th>CDS_position</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>4</th>\n      <td>15043</td>\n      <td>single nucleotide variant</td>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>9640</td>\n      <td>ZNF592</td>\n      <td>HGNC:28986</td>\n      <td>Uncertain significance</td>\n      <td>0</td>\n      <td>Jun 29, 2015</td>\n      <td>150829393</td>\n      <td>...</td>\n      <td>4</td>\n      <td>85342440</td>\n      <td>G</td>\n      <td>A</td>\n      <td>3136</td>\n      <td>G/A</td>\n      <td>Gly1046Arg</td>\n      <td>Gly</td>\n      <td>1046</td>\n      <td>Arg</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>15043</td>\n      <td>single nucleotide variant</td>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>9640</td>\n      <td>ZNF592</td>\n      <td>HGNC:28986</td>\n      <td>Uncertain significance</td>\n      <td>0</td>\n      <td>Jun 29, 2015</td>\n      <td>150829393</td>\n      <td>...</td>\n      <td>4</td>\n      <td>84799209</td>\n      <td>G</td>\n      <td>A</td>\n      <td>3136</td>\n      <td>G/A</td>\n      <td>Gly1046Arg</td>\n      <td>Gly</td>\n      <td>1046</td>\n      <td>Arg</td>\n    </tr>\n    <tr>\n      <th>8</th>\n      <td>15045</td>\n      <td>single nucleotide variant</td>\n      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>HGNC:26927</td>\n      <td>Pathogenic</td>\n      <td>1</td>\n      <td>Oct 01, 2010</td>\n      <td>267606830</td>\n      <td>...</td>\n      <td>6</td>\n      <td>126147412</td>\n      <td>A</td>\n      <td>G</td>\n      <td>1289</td>\n      <td>A/G</td>\n      <td>Asn430Ser</td>\n      <td>Asn</td>\n      <td>430</td>\n      <td>Ser</td>\n    </tr>\n    <tr>\n      <th>9</th>\n      <td>15045</td>\n      <td>single nucleotide variant</td>\n      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>HGNC:26927</td>\n      <td>Pathogenic</td>\n      <td>1</td>\n      <td>Oct 01, 2010</td>\n      <td>267606830</td>\n      <td>...</td>\n      <td>6</td>\n      <td>126277517</td>\n      <td>A</td>\n      <td>G</td>\n      <td>1289</td>\n      <td>A/G</td>\n      <td>Asn430Ser</td>\n      <td>Asn</td>\n      <td>430</td>\n      <td>Ser</td>\n    </tr>\n    <tr>\n      <th>10</th>\n      <td>15046</td>\n      <td>single nucleotide variant</td>\n      <td>NM_025152.3(NUBPL):c.166G&gt;A (p.Gly56Arg)</td>\n      <td>80224</td>\n      <td>NUBPL</td>\n      <td>HGNC:20278</td>\n      <td>Uncertain significance</td>\n      <td>0</td>\n      <td>Aug 24, 2020</td>\n      <td>200401432</td>\n      <td>...</td>\n      <td>214885</td>\n      <td>32031331</td>\n      <td>G</td>\n      <td>A</td>\n      <td>166</td>\n      <td>G/A</td>\n      <td>Gly56Arg</td>\n      <td>Gly</td>\n      <td>56</td>\n      <td>Arg</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>3251294</th>\n      <td>1863472</td>\n      <td>single nucleotide variant</td>\n      <td>NM_000089.4(COL1A2):c.3142G&gt;A (p.Gly1048Ser)</td>\n      <td>1278</td>\n      <td>COL1A2</td>\n      <td>HGNC:2198</td>\n      <td>Likely pathogenic</td>\n      <td>1</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>1806465</td>\n      <td>94427044</td>\n      <td>G</td>\n      <td>A</td>\n      <td>3142</td>\n      <td>G/A</td>\n      <td>Gly1048Ser</td>\n      <td>Gly</td>\n      <td>1048</td>\n      <td>Ser</td>\n    </tr>\n    <tr>\n      <th>3251305</th>\n      <td>1863478</td>\n      <td>single nucleotide variant</td>\n      <td>NM_004456.5(EZH2):c.395C&gt;T (p.Pro132Leu)</td>\n      <td>2146</td>\n      <td>EZH2</td>\n      <td>HGNC:3527</td>\n      <td>not provided</td>\n      <td>0</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>1806471</td>\n      <td>148526909</td>\n      <td>G</td>\n      <td>A</td>\n      <td>395</td>\n      <td>C/T</td>\n      <td>Pro132Leu</td>\n      <td>Pro</td>\n      <td>132</td>\n      <td>Leu</td>\n    </tr>\n    <tr>\n      <th>3251306</th>\n      <td>1863478</td>\n      <td>single nucleotide variant</td>\n      <td>NM_004456.5(EZH2):c.395C&gt;T (p.Pro132Leu)</td>\n      <td>2146</td>\n      <td>EZH2</td>\n      <td>HGNC:3527</td>\n      <td>not provided</td>\n      <td>0</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>1806471</td>\n      <td>148829817</td>\n      <td>G</td>\n      <td>A</td>\n      <td>395</td>\n      <td>C/T</td>\n      <td>Pro132Leu</td>\n      <td>Pro</td>\n      <td>132</td>\n      <td>Leu</td>\n    </tr>\n    <tr>\n      <th>3251307</th>\n      <td>1863479</td>\n      <td>single nucleotide variant</td>\n      <td>NM_001972.4(ELANE):c.640G&gt;C (p.Gly214Arg)</td>\n      <td>1991</td>\n      <td>ELANE</td>\n      <td>HGNC:3309</td>\n      <td>not provided</td>\n      <td>0</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>1806472</td>\n      <td>856000</td>\n      <td>G</td>\n      <td>C</td>\n      <td>640</td>\n      <td>G/C</td>\n      <td>Gly214Arg</td>\n      <td>Gly</td>\n      <td>214</td>\n      <td>Arg</td>\n    </tr>\n    <tr>\n      <th>3251308</th>\n      <td>1863479</td>\n      <td>single nucleotide variant</td>\n      <td>NM_001972.4(ELANE):c.640G&gt;C (p.Gly214Arg)</td>\n      <td>1991</td>\n      <td>ELANE</td>\n      <td>HGNC:3309</td>\n      <td>not provided</td>\n      <td>0</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>1806472</td>\n      <td>856000</td>\n      <td>G</td>\n      <td>C</td>\n      <td>640</td>\n      <td>G/C</td>\n      <td>Gly214Arg</td>\n      <td>Gly</td>\n      <td>214</td>\n      <td>Arg</td>\n    </tr>\n  </tbody>\n</table>\n<p>1295746 rows × 40 columns</p>\n</div>"
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def extract_amino_acid(row, before = False, pos = False, after = False):\n",
    "    import re\n",
    "    try:\n",
    "        if before:\n",
    "            first_aa = row[:3]\n",
    "            return first_aa\n",
    "        if pos:\n",
    "            pos = int(re.search(r'\\d+', row).group())\n",
    "            return pos\n",
    "        # the following command extract the index within the string where the position number ends\n",
    "        if after:\n",
    "            first_aa = row[:3]\n",
    "            position_end_index = re.search(r'\\d+', row).span()[-1]\n",
    "            later_part = row[position_end_index:]\n",
    "            if later_part == '=':\n",
    "                return first_aa\n",
    "            elif later_part == '*':\n",
    "                return 'STOP'\n",
    "            else:\n",
    "                return later_part\n",
    "    except TypeError:\n",
    "        import numpy as np\n",
    "        return np.nan\n",
    "\n",
    "df_meta['Amino_acids_before_3letters'] = df_meta['Protein_Patterns'].apply(extract_amino_acid, before = True)\n",
    "df_meta['Protein_positions'] = df_meta['Protein_Patterns'].apply(extract_amino_acid, pos = True)\n",
    "df_meta['Amino_acids_after_3letters'] = df_meta['Protein_Patterns'].apply(extract_amino_acid, after = True)\n",
    "df_meta['Protein_positions'] = df_meta['Protein_positions'].astype('Int64')\n",
    "# drop introns and premature stop codon\n",
    "df_meta = df_meta[(df_meta['Protein_Patterns'].notnull()) &\n",
    "                        (df_meta['Amino_acids_before_3letters'] != 'Ter') &\n",
    "                        (df_meta['Amino_acids_after_3letters'] != 'Ter') &\n",
    "                        (df_meta['AlleleChange'].notnull())]\n",
    "df_meta"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "outputs": [],
   "source": [
    "# save\n",
    "df_meta.to_csv('./data/metas/clinvar_nonsyn_proteins.csv', index=False)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/pbs.487569.hn-10-03/ipykernel_54108/1544089525.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_meta['NM_ID'] = df_meta['Name'].str.extract(nm_id_regex).fillna(np.nan)\n"
     ]
    },
    {
     "data": {
      "text/plain": "         #AlleleID                       Type  \\\n4            15043  single nucleotide variant   \n5            15043  single nucleotide variant   \n8            15045  single nucleotide variant   \n9            15045  single nucleotide variant   \n10           15046  single nucleotide variant   \n...            ...                        ...   \n3251294    1863472  single nucleotide variant   \n3251305    1863478  single nucleotide variant   \n3251306    1863478  single nucleotide variant   \n3251307    1863479  single nucleotide variant   \n3251308    1863479  single nucleotide variant   \n\n                                                 Name  GeneID GeneSymbol  \\\n4        NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n5        NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n8        NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)   55572    FOXRED1   \n9        NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)   55572    FOXRED1   \n10           NM_025152.3(NUBPL):c.166G>A (p.Gly56Arg)   80224      NUBPL   \n...                                               ...     ...        ...   \n3251294  NM_000089.4(COL1A2):c.3142G>A (p.Gly1048Ser)    1278     COL1A2   \n3251305      NM_004456.5(EZH2):c.395C>T (p.Pro132Leu)    2146       EZH2   \n3251306      NM_004456.5(EZH2):c.395C>T (p.Pro132Leu)    2146       EZH2   \n3251307     NM_001972.4(ELANE):c.640G>C (p.Gly214Arg)    1991      ELANE   \n3251308     NM_001972.4(ELANE):c.640G>C (p.Gly214Arg)    1991      ELANE   \n\n            HGNC_ID    ClinicalSignificance  ClinSigSimple LastEvaluated  \\\n4        HGNC:28986  Uncertain significance              0  Jun 29, 2015   \n5        HGNC:28986  Uncertain significance              0  Jun 29, 2015   \n8        HGNC:26927              Pathogenic              1  Oct 01, 2010   \n9        HGNC:26927              Pathogenic              1  Oct 01, 2010   \n10       HGNC:20278  Uncertain significance              0  Aug 24, 2020   \n...             ...                     ...            ...           ...   \n3251294   HGNC:2198       Likely pathogenic              1             -   \n3251305   HGNC:3527            not provided              0             -   \n3251306   HGNC:3527            not provided              0             -   \n3251307   HGNC:3309            not provided              0             -   \n3251308   HGNC:3309            not provided              0             -   \n\n         RS# (dbSNP)  ... PositionVCF ReferenceAlleleVCF AlternateAlleleVCF  \\\n4          150829393  ...    85342440                  G                  A   \n5          150829393  ...    84799209                  G                  A   \n8          267606830  ...   126147412                  A                  G   \n9          267606830  ...   126277517                  A                  G   \n10         200401432  ...    32031331                  G                  A   \n...              ...  ...         ...                ...                ...   \n3251294           -1  ...    94427044                  G                  A   \n3251305           -1  ...   148526909                  G                  A   \n3251306           -1  ...   148829817                  G                  A   \n3251307           -1  ...      856000                  G                  C   \n3251308           -1  ...      856000                  G                  C   \n\n        CDS_position AlleleChange Protein_Patterns  \\\n4               3136          G/A       Gly1046Arg   \n5               3136          G/A       Gly1046Arg   \n8               1289          A/G        Asn430Ser   \n9               1289          A/G        Asn430Ser   \n10               166          G/A         Gly56Arg   \n...              ...          ...              ...   \n3251294         3142          G/A       Gly1048Ser   \n3251305          395          C/T        Pro132Leu   \n3251306          395          C/T        Pro132Leu   \n3251307          640          G/C        Gly214Arg   \n3251308          640          G/C        Gly214Arg   \n\n        Amino_acids_before_3letters Protein_positions  \\\n4                               Gly              1046   \n5                               Gly              1046   \n8                               Asn               430   \n9                               Asn               430   \n10                              Gly                56   \n...                             ...               ...   \n3251294                         Gly              1048   \n3251305                         Pro               132   \n3251306                         Pro               132   \n3251307                         Gly               214   \n3251308                         Gly               214   \n\n        Amino_acids_after_3letters        NM_ID  \n4                              Arg  NM_014630.3  \n5                              Arg  NM_014630.3  \n8                              Ser  NM_017547.4  \n9                              Ser  NM_017547.4  \n10                             Arg  NM_025152.3  \n...                            ...          ...  \n3251294                        Ser  NM_000089.4  \n3251305                        Leu  NM_004456.5  \n3251306                        Leu  NM_004456.5  \n3251307                        Arg  NM_001972.4  \n3251308                        Arg  NM_001972.4  \n\n[1295744 rows x 41 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>#AlleleID</th>\n      <th>Type</th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>HGNC_ID</th>\n      <th>ClinicalSignificance</th>\n      <th>ClinSigSimple</th>\n      <th>LastEvaluated</th>\n      <th>RS# (dbSNP)</th>\n      <th>...</th>\n      <th>PositionVCF</th>\n      <th>ReferenceAlleleVCF</th>\n      <th>AlternateAlleleVCF</th>\n      <th>CDS_position</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>4</th>\n      <td>15043</td>\n      <td>single nucleotide variant</td>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>9640</td>\n      <td>ZNF592</td>\n      <td>HGNC:28986</td>\n      <td>Uncertain significance</td>\n      <td>0</td>\n      <td>Jun 29, 2015</td>\n      <td>150829393</td>\n      <td>...</td>\n      <td>85342440</td>\n      <td>G</td>\n      <td>A</td>\n      <td>3136</td>\n      <td>G/A</td>\n      <td>Gly1046Arg</td>\n      <td>Gly</td>\n      <td>1046</td>\n      <td>Arg</td>\n      <td>NM_014630.3</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>15043</td>\n      <td>single nucleotide variant</td>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>9640</td>\n      <td>ZNF592</td>\n      <td>HGNC:28986</td>\n      <td>Uncertain significance</td>\n      <td>0</td>\n      <td>Jun 29, 2015</td>\n      <td>150829393</td>\n      <td>...</td>\n      <td>84799209</td>\n      <td>G</td>\n      <td>A</td>\n      <td>3136</td>\n      <td>G/A</td>\n      <td>Gly1046Arg</td>\n      <td>Gly</td>\n      <td>1046</td>\n      <td>Arg</td>\n      <td>NM_014630.3</td>\n    </tr>\n    <tr>\n      <th>8</th>\n      <td>15045</td>\n      <td>single nucleotide variant</td>\n      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>HGNC:26927</td>\n      <td>Pathogenic</td>\n      <td>1</td>\n      <td>Oct 01, 2010</td>\n      <td>267606830</td>\n      <td>...</td>\n      <td>126147412</td>\n      <td>A</td>\n      <td>G</td>\n      <td>1289</td>\n      <td>A/G</td>\n      <td>Asn430Ser</td>\n      <td>Asn</td>\n      <td>430</td>\n      <td>Ser</td>\n      <td>NM_017547.4</td>\n    </tr>\n    <tr>\n      <th>9</th>\n      <td>15045</td>\n      <td>single nucleotide variant</td>\n      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>HGNC:26927</td>\n      <td>Pathogenic</td>\n      <td>1</td>\n      <td>Oct 01, 2010</td>\n      <td>267606830</td>\n      <td>...</td>\n      <td>126277517</td>\n      <td>A</td>\n      <td>G</td>\n      <td>1289</td>\n      <td>A/G</td>\n      <td>Asn430Ser</td>\n      <td>Asn</td>\n      <td>430</td>\n      <td>Ser</td>\n      <td>NM_017547.4</td>\n    </tr>\n    <tr>\n      <th>10</th>\n      <td>15046</td>\n      <td>single nucleotide variant</td>\n      <td>NM_025152.3(NUBPL):c.166G&gt;A (p.Gly56Arg)</td>\n      <td>80224</td>\n      <td>NUBPL</td>\n      <td>HGNC:20278</td>\n      <td>Uncertain significance</td>\n      <td>0</td>\n      <td>Aug 24, 2020</td>\n      <td>200401432</td>\n      <td>...</td>\n      <td>32031331</td>\n      <td>G</td>\n      <td>A</td>\n      <td>166</td>\n      <td>G/A</td>\n      <td>Gly56Arg</td>\n      <td>Gly</td>\n      <td>56</td>\n      <td>Arg</td>\n      <td>NM_025152.3</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>3251294</th>\n      <td>1863472</td>\n      <td>single nucleotide variant</td>\n      <td>NM_000089.4(COL1A2):c.3142G&gt;A (p.Gly1048Ser)</td>\n      <td>1278</td>\n      <td>COL1A2</td>\n      <td>HGNC:2198</td>\n      <td>Likely pathogenic</td>\n      <td>1</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>94427044</td>\n      <td>G</td>\n      <td>A</td>\n      <td>3142</td>\n      <td>G/A</td>\n      <td>Gly1048Ser</td>\n      <td>Gly</td>\n      <td>1048</td>\n      <td>Ser</td>\n      <td>NM_000089.4</td>\n    </tr>\n    <tr>\n      <th>3251305</th>\n      <td>1863478</td>\n      <td>single nucleotide variant</td>\n      <td>NM_004456.5(EZH2):c.395C&gt;T (p.Pro132Leu)</td>\n      <td>2146</td>\n      <td>EZH2</td>\n      <td>HGNC:3527</td>\n      <td>not provided</td>\n      <td>0</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>148526909</td>\n      <td>G</td>\n      <td>A</td>\n      <td>395</td>\n      <td>C/T</td>\n      <td>Pro132Leu</td>\n      <td>Pro</td>\n      <td>132</td>\n      <td>Leu</td>\n      <td>NM_004456.5</td>\n    </tr>\n    <tr>\n      <th>3251306</th>\n      <td>1863478</td>\n      <td>single nucleotide variant</td>\n      <td>NM_004456.5(EZH2):c.395C&gt;T (p.Pro132Leu)</td>\n      <td>2146</td>\n      <td>EZH2</td>\n      <td>HGNC:3527</td>\n      <td>not provided</td>\n      <td>0</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>148829817</td>\n      <td>G</td>\n      <td>A</td>\n      <td>395</td>\n      <td>C/T</td>\n      <td>Pro132Leu</td>\n      <td>Pro</td>\n      <td>132</td>\n      <td>Leu</td>\n      <td>NM_004456.5</td>\n    </tr>\n    <tr>\n      <th>3251307</th>\n      <td>1863479</td>\n      <td>single nucleotide variant</td>\n      <td>NM_001972.4(ELANE):c.640G&gt;C (p.Gly214Arg)</td>\n      <td>1991</td>\n      <td>ELANE</td>\n      <td>HGNC:3309</td>\n      <td>not provided</td>\n      <td>0</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>856000</td>\n      <td>G</td>\n      <td>C</td>\n      <td>640</td>\n      <td>G/C</td>\n      <td>Gly214Arg</td>\n      <td>Gly</td>\n      <td>214</td>\n      <td>Arg</td>\n      <td>NM_001972.4</td>\n    </tr>\n    <tr>\n      <th>3251308</th>\n      <td>1863479</td>\n      <td>single nucleotide variant</td>\n      <td>NM_001972.4(ELANE):c.640G&gt;C (p.Gly214Arg)</td>\n      <td>1991</td>\n      <td>ELANE</td>\n      <td>HGNC:3309</td>\n      <td>not provided</td>\n      <td>0</td>\n      <td>-</td>\n      <td>-1</td>\n      <td>...</td>\n      <td>856000</td>\n      <td>G</td>\n      <td>C</td>\n      <td>640</td>\n      <td>G/C</td>\n      <td>Gly214Arg</td>\n      <td>Gly</td>\n      <td>214</td>\n      <td>Arg</td>\n      <td>NM_001972.4</td>\n    </tr>\n  </tbody>\n</table>\n<p>1295744 rows × 41 columns</p>\n</div>"
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# extract NM_ refseq id\n",
    "# *: 0 or more repetitions\n",
    "nm_id_regex = r'(NM_\\d+\\.*\\d*)'\n",
    "df_meta['NM_ID'] = df_meta['Name'].str.extract(nm_id_regex).fillna(np.nan)\n",
    "df_meta = df_meta[(df_meta['NM_ID'].notnull())]\n",
    "df_meta"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "outputs": [
    {
     "data": {
      "text/plain": "                                                 Name  GeneID GeneSymbol  \\\n4        NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n5        NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n8        NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)   55572    FOXRED1   \n9        NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)   55572    FOXRED1   \n10           NM_025152.3(NUBPL):c.166G>A (p.Gly56Arg)   80224      NUBPL   \n...                                               ...     ...        ...   \n3251294  NM_000089.4(COL1A2):c.3142G>A (p.Gly1048Ser)    1278     COL1A2   \n3251305      NM_004456.5(EZH2):c.395C>T (p.Pro132Leu)    2146       EZH2   \n3251306      NM_004456.5(EZH2):c.395C>T (p.Pro132Leu)    2146       EZH2   \n3251307     NM_001972.4(ELANE):c.640G>C (p.Gly214Arg)    1991      ELANE   \n3251308     NM_001972.4(ELANE):c.640G>C (p.Gly214Arg)    1991      ELANE   \n\n        Assembly    ClinicalSignificance  \\\n4         GRCh37  Uncertain significance   \n5         GRCh38  Uncertain significance   \n8         GRCh37              Pathogenic   \n9         GRCh38              Pathogenic   \n10        GRCh37  Uncertain significance   \n...          ...                     ...   \n3251294   GRCh38       Likely pathogenic   \n3251305   GRCh37            not provided   \n3251306   GRCh38            not provided   \n3251307   GRCh37            not provided   \n3251308   GRCh38            not provided   \n\n                                              ReviewStatus CDS_position  \\\n4                           no assertion criteria provided         3136   \n5                           no assertion criteria provided         3136   \n8                           no assertion criteria provided         1289   \n9                           no assertion criteria provided         1289   \n10       criteria provided, multiple submitters, no con...          166   \n...                                                    ...          ...   \n3251294                     no assertion criteria provided         3142   \n3251305                              no assertion provided          395   \n3251306                              no assertion provided          395   \n3251307                              no assertion provided          640   \n3251308                              no assertion provided          640   \n\n                                                  OtherIDs AlleleChange  \\\n4        OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...          G/A   \n5        OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...          G/A   \n8        ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...          A/G   \n9        ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...          A/G   \n10       ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...          G/A   \n...                                                    ...          ...   \n3251294                                                  -          G/A   \n3251305                                                  -          C/T   \n3251306                                                  -          C/T   \n3251307                                                  -          G/C   \n3251308                                                  -          G/C   \n\n        Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n4             Gly1046Arg                         Gly               1046   \n5             Gly1046Arg                         Gly               1046   \n8              Asn430Ser                         Asn                430   \n9              Asn430Ser                         Asn                430   \n10              Gly56Arg                         Gly                 56   \n...                  ...                         ...                ...   \n3251294       Gly1048Ser                         Gly               1048   \n3251305        Pro132Leu                         Pro                132   \n3251306        Pro132Leu                         Pro                132   \n3251307        Gly214Arg                         Gly                214   \n3251308        Gly214Arg                         Gly                214   \n\n        Amino_acids_after_3letters        NM_ID  \n4                              Arg  NM_014630.3  \n5                              Arg  NM_014630.3  \n8                              Ser  NM_017547.4  \n9                              Ser  NM_017547.4  \n10                             Arg  NM_025152.3  \n...                            ...          ...  \n3251294                        Ser  NM_000089.4  \n3251305                        Leu  NM_004456.5  \n3251306                        Leu  NM_004456.5  \n3251307                        Arg  NM_001972.4  \n3251308                        Arg  NM_001972.4  \n\n[1295744 rows x 14 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ClinicalSignificance</th>\n      <th>ReviewStatus</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>4</th>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>9640</td>\n      <td>ZNF592</td>\n      <td>GRCh37</td>\n      <td>Uncertain significance</td>\n      <td>no assertion criteria provided</td>\n      <td>3136</td>\n      <td>OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...</td>\n      <td>G/A</td>\n      <td>Gly1046Arg</td>\n      <td>Gly</td>\n      <td>1046</td>\n      <td>Arg</td>\n      <td>NM_014630.3</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>9640</td>\n      <td>ZNF592</td>\n      <td>GRCh38</td>\n      <td>Uncertain significance</td>\n      <td>no assertion criteria provided</td>\n      <td>3136</td>\n      <td>OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...</td>\n      <td>G/A</td>\n      <td>Gly1046Arg</td>\n      <td>Gly</td>\n      <td>1046</td>\n      <td>Arg</td>\n      <td>NM_014630.3</td>\n    </tr>\n    <tr>\n      <th>8</th>\n      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>no assertion criteria provided</td>\n      <td>1289</td>\n      <td>ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...</td>\n      <td>A/G</td>\n      <td>Asn430Ser</td>\n      <td>Asn</td>\n      <td>430</td>\n      <td>Ser</td>\n      <td>NM_017547.4</td>\n    </tr>\n    <tr>\n      <th>9</th>\n      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>no assertion criteria provided</td>\n      <td>1289</td>\n      <td>ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...</td>\n      <td>A/G</td>\n      <td>Asn430Ser</td>\n      <td>Asn</td>\n      <td>430</td>\n      <td>Ser</td>\n      <td>NM_017547.4</td>\n    </tr>\n    <tr>\n      <th>10</th>\n      <td>NM_025152.3(NUBPL):c.166G&gt;A (p.Gly56Arg)</td>\n      <td>80224</td>\n      <td>NUBPL</td>\n      <td>GRCh37</td>\n      <td>Uncertain significance</td>\n      <td>criteria provided, multiple submitters, no con...</td>\n      <td>166</td>\n      <td>ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...</td>\n      <td>G/A</td>\n      <td>Gly56Arg</td>\n      <td>Gly</td>\n      <td>56</td>\n      <td>Arg</td>\n      <td>NM_025152.3</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>3251294</th>\n      <td>NM_000089.4(COL1A2):c.3142G&gt;A (p.Gly1048Ser)</td>\n      <td>1278</td>\n      <td>COL1A2</td>\n      <td>GRCh38</td>\n      <td>Likely pathogenic</td>\n      <td>no assertion criteria provided</td>\n      <td>3142</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Gly1048Ser</td>\n      <td>Gly</td>\n      <td>1048</td>\n      <td>Ser</td>\n      <td>NM_000089.4</td>\n    </tr>\n    <tr>\n      <th>3251305</th>\n      <td>NM_004456.5(EZH2):c.395C&gt;T (p.Pro132Leu)</td>\n      <td>2146</td>\n      <td>EZH2</td>\n      <td>GRCh37</td>\n      <td>not provided</td>\n      <td>no assertion provided</td>\n      <td>395</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Pro132Leu</td>\n      <td>Pro</td>\n      <td>132</td>\n      <td>Leu</td>\n      <td>NM_004456.5</td>\n    </tr>\n    <tr>\n      <th>3251306</th>\n      <td>NM_004456.5(EZH2):c.395C&gt;T (p.Pro132Leu)</td>\n      <td>2146</td>\n      <td>EZH2</td>\n      <td>GRCh38</td>\n      <td>not provided</td>\n      <td>no assertion provided</td>\n      <td>395</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Pro132Leu</td>\n      <td>Pro</td>\n      <td>132</td>\n      <td>Leu</td>\n      <td>NM_004456.5</td>\n    </tr>\n    <tr>\n      <th>3251307</th>\n      <td>NM_001972.4(ELANE):c.640G&gt;C (p.Gly214Arg)</td>\n      <td>1991</td>\n      <td>ELANE</td>\n      <td>GRCh37</td>\n      <td>not provided</td>\n      <td>no assertion provided</td>\n      <td>640</td>\n      <td>-</td>\n      <td>G/C</td>\n      <td>Gly214Arg</td>\n      <td>Gly</td>\n      <td>214</td>\n      <td>Arg</td>\n      <td>NM_001972.4</td>\n    </tr>\n    <tr>\n      <th>3251308</th>\n      <td>NM_001972.4(ELANE):c.640G&gt;C (p.Gly214Arg)</td>\n      <td>1991</td>\n      <td>ELANE</td>\n      <td>GRCh38</td>\n      <td>not provided</td>\n      <td>no assertion provided</td>\n      <td>640</td>\n      <td>-</td>\n      <td>G/C</td>\n      <td>Gly214Arg</td>\n      <td>Gly</td>\n      <td>214</td>\n      <td>Arg</td>\n      <td>NM_001972.4</td>\n    </tr>\n  </tbody>\n</table>\n<p>1295744 rows × 14 columns</p>\n</div>"
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta = df_meta[['Name', 'GeneID', 'GeneSymbol','Assembly', 'ClinicalSignificance', 'ReviewStatus', 'CDS_position', 'OtherIDs', 'AlleleChange', 'Protein_Patterns', 'Amino_acids_before_3letters', 'Protein_positions', 'Amino_acids_after_3letters', 'NM_ID']]\n",
    "df_meta"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "outputs": [],
   "source": [
    "#save\n",
    "df_meta.to_csv('./data/metas/clinvar_reduced.csv', index=False)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}