{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "\"\"\"This script find overlappings between EVE published variant scores and clinvar annotated variants\"\"\"\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "outputs": [
    {
     "data": {
      "text/plain": "                                                 Name  GeneID GeneSymbol  \\\n0        NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n1        NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n2        NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)   55572    FOXRED1   \n3        NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)   55572    FOXRED1   \n4            NM_025152.3(NUBPL):c.166G>A (p.Gly56Arg)   80224      NUBPL   \n...                                               ...     ...        ...   \n1213329        NM_000029.4(AGT):c.127C>T (p.Leu43Phe)     183        AGT   \n1213330         NM_000029.4(AGT):c.58A>C (p.Thr20Pro)     183        AGT   \n1213331         NM_000029.4(AGT):c.58A>C (p.Thr20Pro)     183        AGT   \n1213332         NM_000029.4(AGT):c.28A>T (p.Met10Leu)     183        AGT   \n1213333         NM_000029.4(AGT):c.28A>T (p.Met10Leu)     183        AGT   \n\n        Assembly    ClinicalSignificance  \\\n0         GRCh37  Uncertain significance   \n1         GRCh38  Uncertain significance   \n2         GRCh37              Pathogenic   \n3         GRCh38              Pathogenic   \n4         GRCh37  Uncertain significance   \n...          ...                     ...   \n1213329   GRCh38  Uncertain significance   \n1213330   GRCh37  Uncertain significance   \n1213331   GRCh38  Uncertain significance   \n1213332   GRCh37           Likely benign   \n1213333   GRCh38           Likely benign   \n\n                                              ReviewStatus  CDS_position  \\\n0                           no assertion criteria provided          3136   \n1                           no assertion criteria provided          3136   \n2                           no assertion criteria provided          1289   \n3                           no assertion criteria provided          1289   \n4        criteria provided, multiple submitters, no con...           166   \n...                                                    ...           ...   \n1213329  criteria provided, multiple submitters, no con...           127   \n1213330                criteria provided, single submitter            58   \n1213331                criteria provided, single submitter            58   \n1213332                criteria provided, single submitter            28   \n1213333                criteria provided, single submitter            28   \n\n                                                  OtherIDs AlleleChange  \\\n0        OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...          G/A   \n1        OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...          G/A   \n2        ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...          A/G   \n3        ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...          A/G   \n4        ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...          G/A   \n...                                                    ...          ...   \n1213329                                                  -          C/T   \n1213330                                                  -          A/C   \n1213331                                                  -          A/C   \n1213332                                                  -          A/T   \n1213333                                                  -          A/T   \n\n        Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n0             Gly1046Arg                         Gly               1046   \n1             Gly1046Arg                         Gly               1046   \n2              Asn430Ser                         Asn                430   \n3              Asn430Ser                         Asn                430   \n4               Gly56Arg                         Gly                 56   \n...                  ...                         ...                ...   \n1213329         Leu43Phe                         Leu                 43   \n1213330         Thr20Pro                         Thr                 20   \n1213331         Thr20Pro                         Thr                 20   \n1213332         Met10Leu                         Met                 10   \n1213333         Met10Leu                         Met                 10   \n\n        Amino_acids_after_3letters        NM_ID   protein_id UniProtKB-AC  \n0                              Arg  NM_014630.3  NP_055445.2       Q92610  \n1                              Arg  NM_014630.3  NP_055445.2       Q92610  \n2                              Ser  NM_017547.4  NP_060017.1       Q96CU9  \n3                              Ser  NM_017547.4  NP_060017.1       Q96CU9  \n4                              Arg  NM_025152.3  NP_079428.2       Q8TB37  \n...                            ...          ...          ...          ...  \n1213329                        Phe  NM_000029.4          NaN       B0ZBE2  \n1213330                        Pro  NM_000029.4          NaN       B0ZBE2  \n1213331                        Pro  NM_000029.4          NaN       B0ZBE2  \n1213332                        Leu  NM_000029.4          NaN       B0ZBE2  \n1213333                        Leu  NM_000029.4          NaN       B0ZBE2  \n\n[1213334 rows x 16 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ClinicalSignificance</th>\n      <th>ReviewStatus</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>protein_id</th>\n      <th>UniProtKB-AC</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>9640</td>\n      <td>ZNF592</td>\n      <td>GRCh37</td>\n      <td>Uncertain significance</td>\n      <td>no assertion criteria provided</td>\n      <td>3136</td>\n      <td>OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...</td>\n      <td>G/A</td>\n      <td>Gly1046Arg</td>\n      <td>Gly</td>\n      <td>1046</td>\n      <td>Arg</td>\n      <td>NM_014630.3</td>\n      <td>NP_055445.2</td>\n      <td>Q92610</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n      <td>9640</td>\n      <td>ZNF592</td>\n      <td>GRCh38</td>\n      <td>Uncertain significance</td>\n      <td>no assertion criteria provided</td>\n      <td>3136</td>\n      <td>OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...</td>\n      <td>G/A</td>\n      <td>Gly1046Arg</td>\n      <td>Gly</td>\n      <td>1046</td>\n      <td>Arg</td>\n      <td>NM_014630.3</td>\n      <td>NP_055445.2</td>\n      <td>Q92610</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>no assertion criteria provided</td>\n      <td>1289</td>\n      <td>ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...</td>\n      <td>A/G</td>\n      <td>Asn430Ser</td>\n      <td>Asn</td>\n      <td>430</td>\n      <td>Ser</td>\n      <td>NM_017547.4</td>\n      <td>NP_060017.1</td>\n      <td>Q96CU9</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>no assertion criteria provided</td>\n      <td>1289</td>\n      <td>ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...</td>\n      <td>A/G</td>\n      <td>Asn430Ser</td>\n      <td>Asn</td>\n      <td>430</td>\n      <td>Ser</td>\n      <td>NM_017547.4</td>\n      <td>NP_060017.1</td>\n      <td>Q96CU9</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NM_025152.3(NUBPL):c.166G&gt;A (p.Gly56Arg)</td>\n      <td>80224</td>\n      <td>NUBPL</td>\n      <td>GRCh37</td>\n      <td>Uncertain significance</td>\n      <td>criteria provided, multiple submitters, no con...</td>\n      <td>166</td>\n      <td>ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...</td>\n      <td>G/A</td>\n      <td>Gly56Arg</td>\n      <td>Gly</td>\n      <td>56</td>\n      <td>Arg</td>\n      <td>NM_025152.3</td>\n      <td>NP_079428.2</td>\n      <td>Q8TB37</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>1213329</th>\n      <td>NM_000029.4(AGT):c.127C&gt;T (p.Leu43Phe)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Uncertain significance</td>\n      <td>criteria provided, multiple submitters, no con...</td>\n      <td>127</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Leu43Phe</td>\n      <td>Leu</td>\n      <td>43</td>\n      <td>Phe</td>\n      <td>NM_000029.4</td>\n      <td>NaN</td>\n      <td>B0ZBE2</td>\n    </tr>\n    <tr>\n      <th>1213330</th>\n      <td>NM_000029.4(AGT):c.58A&gt;C (p.Thr20Pro)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Uncertain significance</td>\n      <td>criteria provided, single submitter</td>\n      <td>58</td>\n      <td>-</td>\n      <td>A/C</td>\n      <td>Thr20Pro</td>\n      <td>Thr</td>\n      <td>20</td>\n      <td>Pro</td>\n      <td>NM_000029.4</td>\n      <td>NaN</td>\n      <td>B0ZBE2</td>\n    </tr>\n    <tr>\n      <th>1213331</th>\n      <td>NM_000029.4(AGT):c.58A&gt;C (p.Thr20Pro)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Uncertain significance</td>\n      <td>criteria provided, single submitter</td>\n      <td>58</td>\n      <td>-</td>\n      <td>A/C</td>\n      <td>Thr20Pro</td>\n      <td>Thr</td>\n      <td>20</td>\n      <td>Pro</td>\n      <td>NM_000029.4</td>\n      <td>NaN</td>\n      <td>B0ZBE2</td>\n    </tr>\n    <tr>\n      <th>1213332</th>\n      <td>NM_000029.4(AGT):c.28A&gt;T (p.Met10Leu)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Likely benign</td>\n      <td>criteria provided, single submitter</td>\n      <td>28</td>\n      <td>-</td>\n      <td>A/T</td>\n      <td>Met10Leu</td>\n      <td>Met</td>\n      <td>10</td>\n      <td>Leu</td>\n      <td>NM_000029.4</td>\n      <td>NaN</td>\n      <td>B0ZBE2</td>\n    </tr>\n    <tr>\n      <th>1213333</th>\n      <td>NM_000029.4(AGT):c.28A&gt;T (p.Met10Leu)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Likely benign</td>\n      <td>criteria provided, single submitter</td>\n      <td>28</td>\n      <td>-</td>\n      <td>A/T</td>\n      <td>Met10Leu</td>\n      <td>Met</td>\n      <td>10</td>\n      <td>Leu</td>\n      <td>NM_000029.4</td>\n      <td>NaN</td>\n      <td>B0ZBE2</td>\n    </tr>\n  </tbody>\n</table>\n<p>1213334 rows × 16 columns</p>\n</div>"
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# filter clinvar dataset for one stars\n",
    "df_clinvar = pd.read_csv('./data/metas/final_clinvar_variant_uniprot_id_mapping.csv')\n",
    "df_clinvar"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "outputs": [
    {
     "data": {
      "text/plain": "                                                 Name  GeneID GeneSymbol  \\\n2        NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)   55572    FOXRED1   \n3        NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)   55572    FOXRED1   \n4            NM_025152.3(NUBPL):c.166G>A (p.Gly56Arg)   80224      NUBPL   \n5            NM_025152.3(NUBPL):c.166G>A (p.Gly56Arg)   80224      NUBPL   \n6             NM_000410.4(HFE):c.845G>A (p.Cys282Tyr)    3077        HFE   \n...                                               ...     ...        ...   \n1213329        NM_000029.4(AGT):c.127C>T (p.Leu43Phe)     183        AGT   \n1213330         NM_000029.4(AGT):c.58A>C (p.Thr20Pro)     183        AGT   \n1213331         NM_000029.4(AGT):c.58A>C (p.Thr20Pro)     183        AGT   \n1213332         NM_000029.4(AGT):c.28A>T (p.Met10Leu)     183        AGT   \n1213333         NM_000029.4(AGT):c.28A>T (p.Met10Leu)     183        AGT   \n\n        Assembly                               ClinicalSignificance  \\\n2         GRCh37                                         Pathogenic   \n3         GRCh38                                         Pathogenic   \n4         GRCh37                             Uncertain significance   \n5         GRCh38                             Uncertain significance   \n6         GRCh37  Conflicting interpretations of pathogenicity; ...   \n...          ...                                                ...   \n1213329   GRCh38                             Uncertain significance   \n1213330   GRCh37                             Uncertain significance   \n1213331   GRCh38                             Uncertain significance   \n1213332   GRCh37                                      Likely benign   \n1213333   GRCh38                                      Likely benign   \n\n                                              ReviewStatus  CDS_position  \\\n2                           no assertion criteria provided          1289   \n3                           no assertion criteria provided          1289   \n4        criteria provided, multiple submitters, no con...           166   \n5        criteria provided, multiple submitters, no con...           166   \n6           criteria provided, conflicting interpretations           845   \n...                                                    ...           ...   \n1213329  criteria provided, multiple submitters, no con...           127   \n1213330                criteria provided, single submitter            58   \n1213331                criteria provided, single submitter            58   \n1213332                criteria provided, single submitter            28   \n1213333                criteria provided, single submitter            28   \n\n                                                  OtherIDs AlleleChange  \\\n2        ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...          A/G   \n3        ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...          A/G   \n4        ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...          G/A   \n5        ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...          G/A   \n6        UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...          G/A   \n...                                                    ...          ...   \n1213329                                                  -          C/T   \n1213330                                                  -          A/C   \n1213331                                                  -          A/C   \n1213332                                                  -          A/T   \n1213333                                                  -          A/T   \n\n        Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n2              Asn430Ser                         Asn                430   \n3              Asn430Ser                         Asn                430   \n4               Gly56Arg                         Gly                 56   \n5               Gly56Arg                         Gly                 56   \n6              Cys282Tyr                         Cys                282   \n...                  ...                         ...                ...   \n1213329         Leu43Phe                         Leu                 43   \n1213330         Thr20Pro                         Thr                 20   \n1213331         Thr20Pro                         Thr                 20   \n1213332         Met10Leu                         Met                 10   \n1213333         Met10Leu                         Met                 10   \n\n        Amino_acids_after_3letters        NM_ID   protein_id UniProtKB-AC  \\\n2                              Ser  NM_017547.4  NP_060017.1       Q96CU9   \n3                              Ser  NM_017547.4  NP_060017.1       Q96CU9   \n4                              Arg  NM_025152.3  NP_079428.2       Q8TB37   \n5                              Arg  NM_025152.3  NP_079428.2       Q8TB37   \n6                              Tyr  NM_000410.4  NP_000401.1       Q30201   \n...                            ...          ...          ...          ...   \n1213329                        Phe  NM_000029.4          NaN       B0ZBE2   \n1213330                        Pro  NM_000029.4          NaN       B0ZBE2   \n1213331                        Pro  NM_000029.4          NaN       B0ZBE2   \n1213332                        Leu  NM_000029.4          NaN       B0ZBE2   \n1213333                        Leu  NM_000029.4          NaN       B0ZBE2   \n\n        UniProtKB-ID  \n2        FXRD1_HUMAN  \n3        FXRD1_HUMAN  \n4        NUBPL_HUMAN  \n5        NUBPL_HUMAN  \n6          HFE_HUMAN  \n...              ...  \n1213329   ANGT_HUMAN  \n1213330   ANGT_HUMAN  \n1213331   ANGT_HUMAN  \n1213332   ANGT_HUMAN  \n1213333   ANGT_HUMAN  \n\n[855875 rows x 17 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ClinicalSignificance</th>\n      <th>ReviewStatus</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>protein_id</th>\n      <th>UniProtKB-AC</th>\n      <th>UniProtKB-ID</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>2</th>\n      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>no assertion criteria provided</td>\n      <td>1289</td>\n      <td>ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...</td>\n      <td>A/G</td>\n      <td>Asn430Ser</td>\n      <td>Asn</td>\n      <td>430</td>\n      <td>Ser</td>\n      <td>NM_017547.4</td>\n      <td>NP_060017.1</td>\n      <td>Q96CU9</td>\n      <td>FXRD1_HUMAN</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n      <td>55572</td>\n      <td>FOXRED1</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>no assertion criteria provided</td>\n      <td>1289</td>\n      <td>ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...</td>\n      <td>A/G</td>\n      <td>Asn430Ser</td>\n      <td>Asn</td>\n      <td>430</td>\n      <td>Ser</td>\n      <td>NM_017547.4</td>\n      <td>NP_060017.1</td>\n      <td>Q96CU9</td>\n      <td>FXRD1_HUMAN</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NM_025152.3(NUBPL):c.166G&gt;A (p.Gly56Arg)</td>\n      <td>80224</td>\n      <td>NUBPL</td>\n      <td>GRCh37</td>\n      <td>Uncertain significance</td>\n      <td>criteria provided, multiple submitters, no con...</td>\n      <td>166</td>\n      <td>ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...</td>\n      <td>G/A</td>\n      <td>Gly56Arg</td>\n      <td>Gly</td>\n      <td>56</td>\n      <td>Arg</td>\n      <td>NM_025152.3</td>\n      <td>NP_079428.2</td>\n      <td>Q8TB37</td>\n      <td>NUBPL_HUMAN</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>NM_025152.3(NUBPL):c.166G&gt;A (p.Gly56Arg)</td>\n      <td>80224</td>\n      <td>NUBPL</td>\n      <td>GRCh38</td>\n      <td>Uncertain significance</td>\n      <td>criteria provided, multiple submitters, no con...</td>\n      <td>166</td>\n      <td>ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...</td>\n      <td>G/A</td>\n      <td>Gly56Arg</td>\n      <td>Gly</td>\n      <td>56</td>\n      <td>Arg</td>\n      <td>NM_025152.3</td>\n      <td>NP_079428.2</td>\n      <td>Q8TB37</td>\n      <td>NUBPL_HUMAN</td>\n    </tr>\n    <tr>\n      <th>6</th>\n      <td>NM_000410.4(HFE):c.845G&gt;A (p.Cys282Tyr)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh37</td>\n      <td>Conflicting interpretations of pathogenicity; ...</td>\n      <td>criteria provided, conflicting interpretations</td>\n      <td>845</td>\n      <td>UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...</td>\n      <td>G/A</td>\n      <td>Cys282Tyr</td>\n      <td>Cys</td>\n      <td>282</td>\n      <td>Tyr</td>\n      <td>NM_000410.4</td>\n      <td>NP_000401.1</td>\n      <td>Q30201</td>\n      <td>HFE_HUMAN</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>1213329</th>\n      <td>NM_000029.4(AGT):c.127C&gt;T (p.Leu43Phe)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Uncertain significance</td>\n      <td>criteria provided, multiple submitters, no con...</td>\n      <td>127</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Leu43Phe</td>\n      <td>Leu</td>\n      <td>43</td>\n      <td>Phe</td>\n      <td>NM_000029.4</td>\n      <td>NaN</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n    </tr>\n    <tr>\n      <th>1213330</th>\n      <td>NM_000029.4(AGT):c.58A&gt;C (p.Thr20Pro)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Uncertain significance</td>\n      <td>criteria provided, single submitter</td>\n      <td>58</td>\n      <td>-</td>\n      <td>A/C</td>\n      <td>Thr20Pro</td>\n      <td>Thr</td>\n      <td>20</td>\n      <td>Pro</td>\n      <td>NM_000029.4</td>\n      <td>NaN</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n    </tr>\n    <tr>\n      <th>1213331</th>\n      <td>NM_000029.4(AGT):c.58A&gt;C (p.Thr20Pro)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Uncertain significance</td>\n      <td>criteria provided, single submitter</td>\n      <td>58</td>\n      <td>-</td>\n      <td>A/C</td>\n      <td>Thr20Pro</td>\n      <td>Thr</td>\n      <td>20</td>\n      <td>Pro</td>\n      <td>NM_000029.4</td>\n      <td>NaN</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n    </tr>\n    <tr>\n      <th>1213332</th>\n      <td>NM_000029.4(AGT):c.28A&gt;T (p.Met10Leu)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Likely benign</td>\n      <td>criteria provided, single submitter</td>\n      <td>28</td>\n      <td>-</td>\n      <td>A/T</td>\n      <td>Met10Leu</td>\n      <td>Met</td>\n      <td>10</td>\n      <td>Leu</td>\n      <td>NM_000029.4</td>\n      <td>NaN</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n    </tr>\n    <tr>\n      <th>1213333</th>\n      <td>NM_000029.4(AGT):c.28A&gt;T (p.Met10Leu)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Likely benign</td>\n      <td>criteria provided, single submitter</td>\n      <td>28</td>\n      <td>-</td>\n      <td>A/T</td>\n      <td>Met10Leu</td>\n      <td>Met</td>\n      <td>10</td>\n      <td>Leu</td>\n      <td>NM_000029.4</td>\n      <td>NaN</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n    </tr>\n  </tbody>\n</table>\n<p>855875 rows × 17 columns</p>\n</div>"
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_uniprotid_uniprotkb_mappings = pd.read_csv('./data/metas/clinvar_variant_uniprotid_uniprotkb_mapping.csv')\n",
    "df_eve_covered_uniprotkbs = pd.merge(df_clinvar, df_uniprotid_uniprotkb_mappings, how='left', on='UniProtKB-AC')\n",
    "df_eve_covered_uniprotkbs.dropna(subset=['UniProtKB-ID'], inplace=True)\n",
    "df_eve_covered_uniprotkbs"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2865\n",
      "3166\n",
      "2872\n"
     ]
    }
   ],
   "source": [
    "# double check numbers of unique entries\n",
    "print(len(df_eve_covered_uniprotkbs['UniProtKB-AC'].unique()))\n",
    "print(len(df_eve_covered_uniprotkbs['GeneSymbol'].unique()))\n",
    "print(len(df_eve_covered_uniprotkbs['GeneID'].unique()))"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'no interpretation for the single variant'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/pbs.499107.hn-10-03/ipykernel_138049/1105321620.py:20: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_eve_covered_uniprotkbs.drop(['ReviewStatus', 'protein_id'], axis=1, inplace=True)\n"
     ]
    },
    {
     "data": {
      "text/plain": "                                             Name  GeneID GeneSymbol Assembly  \\\n4        NM_025152.3(NUBPL):c.166G>A (p.Gly56Arg)   80224      NUBPL   GRCh37   \n5        NM_025152.3(NUBPL):c.166G>A (p.Gly56Arg)   80224      NUBPL   GRCh38   \n6         NM_000410.4(HFE):c.845G>A (p.Cys282Tyr)    3077        HFE   GRCh37   \n7         NM_000410.4(HFE):c.845G>A (p.Cys282Tyr)    3077        HFE   GRCh38   \n8          NM_000410.4(HFE):c.187C>G (p.His63Asp)    3077        HFE   GRCh37   \n...                                           ...     ...        ...      ...   \n1213329    NM_000029.4(AGT):c.127C>T (p.Leu43Phe)     183        AGT   GRCh38   \n1213330     NM_000029.4(AGT):c.58A>C (p.Thr20Pro)     183        AGT   GRCh37   \n1213331     NM_000029.4(AGT):c.58A>C (p.Thr20Pro)     183        AGT   GRCh38   \n1213332     NM_000029.4(AGT):c.28A>T (p.Met10Leu)     183        AGT   GRCh37   \n1213333     NM_000029.4(AGT):c.28A>T (p.Met10Leu)     183        AGT   GRCh38   \n\n                                      ClinicalSignificance  CDS_position  \\\n4                                   Uncertain significance           166   \n5                                   Uncertain significance           166   \n6        Conflicting interpretations of pathogenicity; ...           845   \n7        Conflicting interpretations of pathogenicity; ...           845   \n8        Conflicting interpretations of pathogenicity; ...           187   \n...                                                    ...           ...   \n1213329                             Uncertain significance           127   \n1213330                             Uncertain significance            58   \n1213331                             Uncertain significance            58   \n1213332                                      Likely benign            28   \n1213333                                      Likely benign            28   \n\n                                                  OtherIDs AlleleChange  \\\n4        ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...          G/A   \n5        ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...          G/A   \n6        UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...          G/A   \n7        UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...          G/A   \n8        OMIM:613609.0002,ClinGen:CA113797,UniProtKB:Q3...          C/G   \n...                                                    ...          ...   \n1213329                                                  -          C/T   \n1213330                                                  -          A/C   \n1213331                                                  -          A/C   \n1213332                                                  -          A/T   \n1213333                                                  -          A/T   \n\n        Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n4               Gly56Arg                         Gly                 56   \n5               Gly56Arg                         Gly                 56   \n6              Cys282Tyr                         Cys                282   \n7              Cys282Tyr                         Cys                282   \n8               His63Asp                         His                 63   \n...                  ...                         ...                ...   \n1213329         Leu43Phe                         Leu                 43   \n1213330         Thr20Pro                         Thr                 20   \n1213331         Thr20Pro                         Thr                 20   \n1213332         Met10Leu                         Met                 10   \n1213333         Met10Leu                         Met                 10   \n\n        Amino_acids_after_3letters        NM_ID UniProtKB-AC UniProtKB-ID  \\\n4                              Arg  NM_025152.3       Q8TB37  NUBPL_HUMAN   \n5                              Arg  NM_025152.3       Q8TB37  NUBPL_HUMAN   \n6                              Tyr  NM_000410.4       Q30201    HFE_HUMAN   \n7                              Tyr  NM_000410.4       Q30201    HFE_HUMAN   \n8                              Asp  NM_000410.4       Q30201    HFE_HUMAN   \n...                            ...          ...          ...          ...   \n1213329                        Phe  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n1213330                        Pro  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n1213331                        Pro  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n1213332                        Leu  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n1213333                        Leu  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n\n         stars  \n4            2  \n5            2  \n6            1  \n7            1  \n8            1  \n...        ...  \n1213329      2  \n1213330      1  \n1213331      1  \n1213332      1  \n1213333      1  \n\n[811934 rows x 16 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>UniProtKB-ID</th>\n      <th>stars</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>4</th>\n      <td>NM_025152.3(NUBPL):c.166G&gt;A (p.Gly56Arg)</td>\n      <td>80224</td>\n      <td>NUBPL</td>\n      <td>GRCh37</td>\n      <td>Uncertain significance</td>\n      <td>166</td>\n      <td>ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...</td>\n      <td>G/A</td>\n      <td>Gly56Arg</td>\n      <td>Gly</td>\n      <td>56</td>\n      <td>Arg</td>\n      <td>NM_025152.3</td>\n      <td>Q8TB37</td>\n      <td>NUBPL_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>NM_025152.3(NUBPL):c.166G&gt;A (p.Gly56Arg)</td>\n      <td>80224</td>\n      <td>NUBPL</td>\n      <td>GRCh38</td>\n      <td>Uncertain significance</td>\n      <td>166</td>\n      <td>ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...</td>\n      <td>G/A</td>\n      <td>Gly56Arg</td>\n      <td>Gly</td>\n      <td>56</td>\n      <td>Arg</td>\n      <td>NM_025152.3</td>\n      <td>Q8TB37</td>\n      <td>NUBPL_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>6</th>\n      <td>NM_000410.4(HFE):c.845G&gt;A (p.Cys282Tyr)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh37</td>\n      <td>Conflicting interpretations of pathogenicity; ...</td>\n      <td>845</td>\n      <td>UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...</td>\n      <td>G/A</td>\n      <td>Cys282Tyr</td>\n      <td>Cys</td>\n      <td>282</td>\n      <td>Tyr</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>HFE_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>7</th>\n      <td>NM_000410.4(HFE):c.845G&gt;A (p.Cys282Tyr)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh38</td>\n      <td>Conflicting interpretations of pathogenicity; ...</td>\n      <td>845</td>\n      <td>UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...</td>\n      <td>G/A</td>\n      <td>Cys282Tyr</td>\n      <td>Cys</td>\n      <td>282</td>\n      <td>Tyr</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>HFE_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>8</th>\n      <td>NM_000410.4(HFE):c.187C&gt;G (p.His63Asp)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh37</td>\n      <td>Conflicting interpretations of pathogenicity; ...</td>\n      <td>187</td>\n      <td>OMIM:613609.0002,ClinGen:CA113797,UniProtKB:Q3...</td>\n      <td>C/G</td>\n      <td>His63Asp</td>\n      <td>His</td>\n      <td>63</td>\n      <td>Asp</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>HFE_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>1213329</th>\n      <td>NM_000029.4(AGT):c.127C&gt;T (p.Leu43Phe)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Uncertain significance</td>\n      <td>127</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Leu43Phe</td>\n      <td>Leu</td>\n      <td>43</td>\n      <td>Phe</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213330</th>\n      <td>NM_000029.4(AGT):c.58A&gt;C (p.Thr20Pro)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Uncertain significance</td>\n      <td>58</td>\n      <td>-</td>\n      <td>A/C</td>\n      <td>Thr20Pro</td>\n      <td>Thr</td>\n      <td>20</td>\n      <td>Pro</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213331</th>\n      <td>NM_000029.4(AGT):c.58A&gt;C (p.Thr20Pro)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Uncertain significance</td>\n      <td>58</td>\n      <td>-</td>\n      <td>A/C</td>\n      <td>Thr20Pro</td>\n      <td>Thr</td>\n      <td>20</td>\n      <td>Pro</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213332</th>\n      <td>NM_000029.4(AGT):c.28A&gt;T (p.Met10Leu)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Likely benign</td>\n      <td>28</td>\n      <td>-</td>\n      <td>A/T</td>\n      <td>Met10Leu</td>\n      <td>Met</td>\n      <td>10</td>\n      <td>Leu</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213333</th>\n      <td>NM_000029.4(AGT):c.28A&gt;T (p.Met10Leu)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Likely benign</td>\n      <td>28</td>\n      <td>-</td>\n      <td>A/T</td>\n      <td>Met10Leu</td>\n      <td>Met</td>\n      <td>10</td>\n      <td>Leu</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n  </tbody>\n</table>\n<p>811934 rows × 16 columns</p>\n</div>"
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "other_descriptions = set()\n",
    "def assign_stars(row):\n",
    "    if 'criteria provided, single submitter' in row or 'criteria provided, conflicting interpretations' in row:\n",
    "        return 1\n",
    "    elif 'criteria provided, multiple submitters' in row:\n",
    "        return 2\n",
    "    elif 'expert panel' in row:\n",
    "        return 3\n",
    "    elif 'guideline' in row:\n",
    "        return 4\n",
    "    elif 'no assertion' in row:\n",
    "        return None\n",
    "    else:\n",
    "        other_descriptions.add(row)\n",
    "        return None\n",
    "df_eve_covered_uniprotkbs['stars'] = df_eve_covered_uniprotkbs['ReviewStatus'].apply(assign_stars)\n",
    "# pandas can handle NaNs as integers now\n",
    "df_eve_covered_uniprotkbs['stars'] = df_eve_covered_uniprotkbs['stars'].astype('Int64')\n",
    "df_eve_covered_uniprotkbs = df_eve_covered_uniprotkbs[df_eve_covered_uniprotkbs['stars'] >= 1]\n",
    "df_eve_covered_uniprotkbs.drop(['ReviewStatus', 'protein_id'], axis=1, inplace=True)\n",
    "print(other_descriptions)\n",
    "df_eve_covered_uniprotkbs"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2808\n",
      "3075\n",
      "2813\n"
     ]
    }
   ],
   "source": [
    "# double check numbers of unique entries\n",
    "print(len(df_eve_covered_uniprotkbs['UniProtKB-AC'].unique()))\n",
    "print(len(df_eve_covered_uniprotkbs['GeneSymbol'].unique()))\n",
    "print(len(df_eve_covered_uniprotkbs['GeneID'].unique()))"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "outputs": [
    {
     "data": {
      "text/plain": "                                            Name  GeneID GeneSymbol Assembly  \\\n6        NM_000410.4(HFE):c.845G>A (p.Cys282Tyr)    3077        HFE   GRCh37   \n7        NM_000410.4(HFE):c.845G>A (p.Cys282Tyr)    3077        HFE   GRCh38   \n8         NM_000410.4(HFE):c.187C>G (p.His63Asp)    3077        HFE   GRCh37   \n9         NM_000410.4(HFE):c.187C>G (p.His63Asp)    3077        HFE   GRCh38   \n10        NM_000410.4(HFE):c.193A>T (p.Ser65Cys)    3077        HFE   GRCh37   \n...                                          ...     ...        ...      ...   \n1213285  NM_000029.4(AGT):c.410C>T (p.Thr137Met)     183        AGT   GRCh38   \n1213326   NM_000029.4(AGT):c.148A>T (p.Thr50Ser)     183        AGT   GRCh37   \n1213327   NM_000029.4(AGT):c.148A>T (p.Thr50Ser)     183        AGT   GRCh38   \n1213332    NM_000029.4(AGT):c.28A>T (p.Met10Leu)     183        AGT   GRCh37   \n1213333    NM_000029.4(AGT):c.28A>T (p.Met10Leu)     183        AGT   GRCh38   \n\n                                      ClinicalSignificance  CDS_position  \\\n6        Conflicting interpretations of pathogenicity; ...           845   \n7        Conflicting interpretations of pathogenicity; ...           845   \n8        Conflicting interpretations of pathogenicity; ...           187   \n9        Conflicting interpretations of pathogenicity; ...           187   \n10            Conflicting interpretations of pathogenicity           193   \n...                                                    ...           ...   \n1213285                               Benign/Likely benign           410   \n1213326       Conflicting interpretations of pathogenicity           148   \n1213327       Conflicting interpretations of pathogenicity           148   \n1213332                                      Likely benign            28   \n1213333                                      Likely benign            28   \n\n                                                  OtherIDs AlleleChange  \\\n6        UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...          G/A   \n7        UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...          G/A   \n8        OMIM:613609.0002,ClinGen:CA113797,UniProtKB:Q3...          C/G   \n9        OMIM:613609.0002,ClinGen:CA113797,UniProtKB:Q3...          C/G   \n10       ClinGen:CA339778,UniProtKB:Q30201#VAR_004397,O...          A/T   \n...                                                    ...          ...   \n1213285                                                  -          C/T   \n1213326                                                  -          A/T   \n1213327                                                  -          A/T   \n1213332                                                  -          A/T   \n1213333                                                  -          A/T   \n\n        Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n6              Cys282Tyr                         Cys                282   \n7              Cys282Tyr                         Cys                282   \n8               His63Asp                         His                 63   \n9               His63Asp                         His                 63   \n10              Ser65Cys                         Ser                 65   \n...                  ...                         ...                ...   \n1213285        Thr137Met                         Thr                137   \n1213326         Thr50Ser                         Thr                 50   \n1213327         Thr50Ser                         Thr                 50   \n1213332         Met10Leu                         Met                 10   \n1213333         Met10Leu                         Met                 10   \n\n        Amino_acids_after_3letters        NM_ID UniProtKB-AC UniProtKB-ID  \\\n6                              Tyr  NM_000410.4       Q30201    HFE_HUMAN   \n7                              Tyr  NM_000410.4       Q30201    HFE_HUMAN   \n8                              Asp  NM_000410.4       Q30201    HFE_HUMAN   \n9                              Asp  NM_000410.4       Q30201    HFE_HUMAN   \n10                             Cys  NM_000410.4       Q30201    HFE_HUMAN   \n...                            ...          ...          ...          ...   \n1213285                        Met  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n1213326                        Ser  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n1213327                        Ser  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n1213332                        Leu  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n1213333                        Leu  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n\n         stars  \n6            1  \n7            1  \n8            1  \n9            1  \n10           1  \n...        ...  \n1213285      2  \n1213326      1  \n1213327      1  \n1213332      1  \n1213333      1  \n\n[165308 rows x 16 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>UniProtKB-ID</th>\n      <th>stars</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>6</th>\n      <td>NM_000410.4(HFE):c.845G&gt;A (p.Cys282Tyr)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh37</td>\n      <td>Conflicting interpretations of pathogenicity; ...</td>\n      <td>845</td>\n      <td>UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...</td>\n      <td>G/A</td>\n      <td>Cys282Tyr</td>\n      <td>Cys</td>\n      <td>282</td>\n      <td>Tyr</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>HFE_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>7</th>\n      <td>NM_000410.4(HFE):c.845G&gt;A (p.Cys282Tyr)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh38</td>\n      <td>Conflicting interpretations of pathogenicity; ...</td>\n      <td>845</td>\n      <td>UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...</td>\n      <td>G/A</td>\n      <td>Cys282Tyr</td>\n      <td>Cys</td>\n      <td>282</td>\n      <td>Tyr</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>HFE_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>8</th>\n      <td>NM_000410.4(HFE):c.187C&gt;G (p.His63Asp)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh37</td>\n      <td>Conflicting interpretations of pathogenicity; ...</td>\n      <td>187</td>\n      <td>OMIM:613609.0002,ClinGen:CA113797,UniProtKB:Q3...</td>\n      <td>C/G</td>\n      <td>His63Asp</td>\n      <td>His</td>\n      <td>63</td>\n      <td>Asp</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>HFE_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>9</th>\n      <td>NM_000410.4(HFE):c.187C&gt;G (p.His63Asp)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh38</td>\n      <td>Conflicting interpretations of pathogenicity; ...</td>\n      <td>187</td>\n      <td>OMIM:613609.0002,ClinGen:CA113797,UniProtKB:Q3...</td>\n      <td>C/G</td>\n      <td>His63Asp</td>\n      <td>His</td>\n      <td>63</td>\n      <td>Asp</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>HFE_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>10</th>\n      <td>NM_000410.4(HFE):c.193A&gt;T (p.Ser65Cys)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh37</td>\n      <td>Conflicting interpretations of pathogenicity</td>\n      <td>193</td>\n      <td>ClinGen:CA339778,UniProtKB:Q30201#VAR_004397,O...</td>\n      <td>A/T</td>\n      <td>Ser65Cys</td>\n      <td>Ser</td>\n      <td>65</td>\n      <td>Cys</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>HFE_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>1213285</th>\n      <td>NM_000029.4(AGT):c.410C&gt;T (p.Thr137Met)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Benign/Likely benign</td>\n      <td>410</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Thr137Met</td>\n      <td>Thr</td>\n      <td>137</td>\n      <td>Met</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213326</th>\n      <td>NM_000029.4(AGT):c.148A&gt;T (p.Thr50Ser)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Conflicting interpretations of pathogenicity</td>\n      <td>148</td>\n      <td>-</td>\n      <td>A/T</td>\n      <td>Thr50Ser</td>\n      <td>Thr</td>\n      <td>50</td>\n      <td>Ser</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213327</th>\n      <td>NM_000029.4(AGT):c.148A&gt;T (p.Thr50Ser)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Conflicting interpretations of pathogenicity</td>\n      <td>148</td>\n      <td>-</td>\n      <td>A/T</td>\n      <td>Thr50Ser</td>\n      <td>Thr</td>\n      <td>50</td>\n      <td>Ser</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213332</th>\n      <td>NM_000029.4(AGT):c.28A&gt;T (p.Met10Leu)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Likely benign</td>\n      <td>28</td>\n      <td>-</td>\n      <td>A/T</td>\n      <td>Met10Leu</td>\n      <td>Met</td>\n      <td>10</td>\n      <td>Leu</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213333</th>\n      <td>NM_000029.4(AGT):c.28A&gt;T (p.Met10Leu)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Likely benign</td>\n      <td>28</td>\n      <td>-</td>\n      <td>A/T</td>\n      <td>Met10Leu</td>\n      <td>Met</td>\n      <td>10</td>\n      <td>Leu</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n  </tbody>\n</table>\n<p>165308 rows × 16 columns</p>\n</div>"
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_clinvar_not_vus = df_eve_covered_uniprotkbs[~df_eve_covered_uniprotkbs['ClinicalSignificance'].str.contains('Uncertain significance')]\n",
    "df_clinvar_not_vus"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2741\n",
      "2869\n",
      "2744\n"
     ]
    }
   ],
   "source": [
    "print(len(df_clinvar_not_vus['UniProtKB-AC'].unique()))\n",
    "print(len(df_clinvar_not_vus['GeneSymbol'].unique()))\n",
    "print(len(df_clinvar_not_vus['GeneID'].unique()))"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "outputs": [
    {
     "data": {
      "text/plain": "                                                   Name  GeneID GeneSymbol  \\\n24              NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n25              NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n36             NM_138413.4(HOGA1):c.209G>C (p.Arg70Pro)  112817      HOGA1   \n37             NM_138413.4(HOGA1):c.209G>C (p.Arg70Pro)  112817      HOGA1   \n48       NM_001199107.2(TBC1D24):c.439G>C (p.Asp147His)   57465    TBC1D24   \n...                                                 ...     ...        ...   \n1213199        NM_021870.2(FGG):c.1007T>C (p.Met336Thr)    2266        FGG   \n1213224     NM_000342.3(SLC4A1):c.1462G>A (p.Val488Met)    6521     SLC4A1   \n1213225     NM_000342.3(SLC4A1):c.1462G>A (p.Val488Met)    6521     SLC4A1   \n1213242        NM_000029.4(AGT):c.1124G>A (p.Arg375Gln)     183        AGT   \n1213243        NM_000029.4(AGT):c.1124G>A (p.Arg375Gln)     183        AGT   \n\n        Assembly ClinicalSignificance  CDS_position  \\\n24        GRCh37           Pathogenic           848   \n25        GRCh38           Pathogenic           848   \n36        GRCh37           Pathogenic           209   \n37        GRCh38           Pathogenic           209   \n48        GRCh37           Pathogenic           439   \n...          ...                  ...           ...   \n1213199   GRCh38           Pathogenic          1007   \n1213224   GRCh37           Pathogenic          1462   \n1213225   GRCh38           Pathogenic          1462   \n1213242   GRCh37           Pathogenic          1124   \n1213243   GRCh38           Pathogenic          1124   \n\n                                                  OtherIDs AlleleChange  \\\n24       ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...          A/C   \n25       ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...          A/C   \n36                       ClinGen:CA113816,OMIM:613597.0005          G/C   \n37                       ClinGen:CA113816,OMIM:613597.0005          G/C   \n48       ClinGen:CA339785,UniProtKB:Q9ULP9#VAR_064365,O...          G/C   \n...                                                    ...          ...   \n1213199  ClinGen:CA126399,UniProtKB:P02679#VAR_002414,O...          T/C   \n1213224  ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...          G/A   \n1213225  ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...          G/A   \n1213242  ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...          G/A   \n1213243  ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...          G/A   \n\n        Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n24             Gln283Pro                         Gln                283   \n25             Gln283Pro                         Gln                283   \n36              Arg70Pro                         Arg                 70   \n37              Arg70Pro                         Arg                 70   \n48             Asp147His                         Asp                147   \n...                  ...                         ...                ...   \n1213199        Met336Thr                         Met                336   \n1213224        Val488Met                         Val                488   \n1213225        Val488Met                         Val                488   \n1213242        Arg375Gln                         Arg                375   \n1213243        Arg375Gln                         Arg                375   \n\n        Amino_acids_after_3letters           NM_ID UniProtKB-AC UniProtKB-ID  \\\n24                             Pro     NM_000410.4       Q30201    HFE_HUMAN   \n25                             Pro     NM_000410.4       Q30201    HFE_HUMAN   \n36                             Pro     NM_138413.4       Q86XE5  HOGA1_HUMAN   \n37                             Pro     NM_138413.4       Q86XE5  HOGA1_HUMAN   \n48                             His  NM_001199107.2       Q9ULP9  TBC24_HUMAN   \n...                            ...             ...          ...          ...   \n1213199                        Thr     NM_021870.2       P02679   FIBG_HUMAN   \n1213224                        Met     NM_000342.3       P02730   B3AT_HUMAN   \n1213225                        Met     NM_000342.3       P02730   B3AT_HUMAN   \n1213242                        Gln     NM_000029.4       B0ZBE2   ANGT_HUMAN   \n1213243                        Gln     NM_000029.4       B0ZBE2   ANGT_HUMAN   \n\n         stars  \n24           1  \n25           1  \n36           1  \n37           1  \n48           1  \n...        ...  \n1213199      1  \n1213224      2  \n1213225      2  \n1213242      1  \n1213243      1  \n\n[20257 rows x 16 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>UniProtKB-ID</th>\n      <th>stars</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>24</th>\n      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>848</td>\n      <td>ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...</td>\n      <td>A/C</td>\n      <td>Gln283Pro</td>\n      <td>Gln</td>\n      <td>283</td>\n      <td>Pro</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>HFE_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>25</th>\n      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>848</td>\n      <td>ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...</td>\n      <td>A/C</td>\n      <td>Gln283Pro</td>\n      <td>Gln</td>\n      <td>283</td>\n      <td>Pro</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>HFE_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>36</th>\n      <td>NM_138413.4(HOGA1):c.209G&gt;C (p.Arg70Pro)</td>\n      <td>112817</td>\n      <td>HOGA1</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>209</td>\n      <td>ClinGen:CA113816,OMIM:613597.0005</td>\n      <td>G/C</td>\n      <td>Arg70Pro</td>\n      <td>Arg</td>\n      <td>70</td>\n      <td>Pro</td>\n      <td>NM_138413.4</td>\n      <td>Q86XE5</td>\n      <td>HOGA1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>37</th>\n      <td>NM_138413.4(HOGA1):c.209G&gt;C (p.Arg70Pro)</td>\n      <td>112817</td>\n      <td>HOGA1</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>209</td>\n      <td>ClinGen:CA113816,OMIM:613597.0005</td>\n      <td>G/C</td>\n      <td>Arg70Pro</td>\n      <td>Arg</td>\n      <td>70</td>\n      <td>Pro</td>\n      <td>NM_138413.4</td>\n      <td>Q86XE5</td>\n      <td>HOGA1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>48</th>\n      <td>NM_001199107.2(TBC1D24):c.439G&gt;C (p.Asp147His)</td>\n      <td>57465</td>\n      <td>TBC1D24</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>439</td>\n      <td>ClinGen:CA339785,UniProtKB:Q9ULP9#VAR_064365,O...</td>\n      <td>G/C</td>\n      <td>Asp147His</td>\n      <td>Asp</td>\n      <td>147</td>\n      <td>His</td>\n      <td>NM_001199107.2</td>\n      <td>Q9ULP9</td>\n      <td>TBC24_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>1213199</th>\n      <td>NM_021870.2(FGG):c.1007T&gt;C (p.Met336Thr)</td>\n      <td>2266</td>\n      <td>FGG</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>1007</td>\n      <td>ClinGen:CA126399,UniProtKB:P02679#VAR_002414,O...</td>\n      <td>T/C</td>\n      <td>Met336Thr</td>\n      <td>Met</td>\n      <td>336</td>\n      <td>Thr</td>\n      <td>NM_021870.2</td>\n      <td>P02679</td>\n      <td>FIBG_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213224</th>\n      <td>NM_000342.3(SLC4A1):c.1462G&gt;A (p.Val488Met)</td>\n      <td>6521</td>\n      <td>SLC4A1</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>1462</td>\n      <td>ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...</td>\n      <td>G/A</td>\n      <td>Val488Met</td>\n      <td>Val</td>\n      <td>488</td>\n      <td>Met</td>\n      <td>NM_000342.3</td>\n      <td>P02730</td>\n      <td>B3AT_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213225</th>\n      <td>NM_000342.3(SLC4A1):c.1462G&gt;A (p.Val488Met)</td>\n      <td>6521</td>\n      <td>SLC4A1</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>1462</td>\n      <td>ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...</td>\n      <td>G/A</td>\n      <td>Val488Met</td>\n      <td>Val</td>\n      <td>488</td>\n      <td>Met</td>\n      <td>NM_000342.3</td>\n      <td>P02730</td>\n      <td>B3AT_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213242</th>\n      <td>NM_000029.4(AGT):c.1124G&gt;A (p.Arg375Gln)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>1124</td>\n      <td>ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...</td>\n      <td>G/A</td>\n      <td>Arg375Gln</td>\n      <td>Arg</td>\n      <td>375</td>\n      <td>Gln</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213243</th>\n      <td>NM_000029.4(AGT):c.1124G&gt;A (p.Arg375Gln)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>1124</td>\n      <td>ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...</td>\n      <td>G/A</td>\n      <td>Arg375Gln</td>\n      <td>Arg</td>\n      <td>375</td>\n      <td>Gln</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n  </tbody>\n</table>\n<p>20257 rows × 16 columns</p>\n</div>"
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance']=='Pathogenic')]"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/pbs.499107.hn-10-03/ipykernel_138049/3210608625.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_pathogenic.drop_duplicates(inplace=True)\n"
     ]
    },
    {
     "data": {
      "text/plain": "                                                   Name  GeneID GeneSymbol  \\\n24              NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n25              NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n36             NM_138413.4(HOGA1):c.209G>C (p.Arg70Pro)  112817      HOGA1   \n37             NM_138413.4(HOGA1):c.209G>C (p.Arg70Pro)  112817      HOGA1   \n48       NM_001199107.2(TBC1D24):c.439G>C (p.Asp147His)   57465    TBC1D24   \n...                                                 ...     ...        ...   \n1213199        NM_021870.2(FGG):c.1007T>C (p.Met336Thr)    2266        FGG   \n1213224     NM_000342.3(SLC4A1):c.1462G>A (p.Val488Met)    6521     SLC4A1   \n1213225     NM_000342.3(SLC4A1):c.1462G>A (p.Val488Met)    6521     SLC4A1   \n1213242        NM_000029.4(AGT):c.1124G>A (p.Arg375Gln)     183        AGT   \n1213243        NM_000029.4(AGT):c.1124G>A (p.Arg375Gln)     183        AGT   \n\n        Assembly ClinicalSignificance  CDS_position  \\\n24        GRCh37           Pathogenic           848   \n25        GRCh38           Pathogenic           848   \n36        GRCh37           Pathogenic           209   \n37        GRCh38           Pathogenic           209   \n48        GRCh37           Pathogenic           439   \n...          ...                  ...           ...   \n1213199   GRCh38           Pathogenic          1007   \n1213224   GRCh37           Pathogenic          1462   \n1213225   GRCh38           Pathogenic          1462   \n1213242   GRCh37           Pathogenic          1124   \n1213243   GRCh38           Pathogenic          1124   \n\n                                                  OtherIDs AlleleChange  \\\n24       ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...          A/C   \n25       ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...          A/C   \n36                       ClinGen:CA113816,OMIM:613597.0005          G/C   \n37                       ClinGen:CA113816,OMIM:613597.0005          G/C   \n48       ClinGen:CA339785,UniProtKB:Q9ULP9#VAR_064365,O...          G/C   \n...                                                    ...          ...   \n1213199  ClinGen:CA126399,UniProtKB:P02679#VAR_002414,O...          T/C   \n1213224  ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...          G/A   \n1213225  ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...          G/A   \n1213242  ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...          G/A   \n1213243  ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...          G/A   \n\n        Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n24             Gln283Pro                         Gln                283   \n25             Gln283Pro                         Gln                283   \n36              Arg70Pro                         Arg                 70   \n37              Arg70Pro                         Arg                 70   \n48             Asp147His                         Asp                147   \n...                  ...                         ...                ...   \n1213199        Met336Thr                         Met                336   \n1213224        Val488Met                         Val                488   \n1213225        Val488Met                         Val                488   \n1213242        Arg375Gln                         Arg                375   \n1213243        Arg375Gln                         Arg                375   \n\n        Amino_acids_after_3letters           NM_ID UniProtKB-AC UniProtKB-ID  \\\n24                             Pro     NM_000410.4       Q30201    HFE_HUMAN   \n25                             Pro     NM_000410.4       Q30201    HFE_HUMAN   \n36                             Pro     NM_138413.4       Q86XE5  HOGA1_HUMAN   \n37                             Pro     NM_138413.4       Q86XE5  HOGA1_HUMAN   \n48                             His  NM_001199107.2       Q9ULP9  TBC24_HUMAN   \n...                            ...             ...          ...          ...   \n1213199                        Thr     NM_021870.2       P02679   FIBG_HUMAN   \n1213224                        Met     NM_000342.3       P02730   B3AT_HUMAN   \n1213225                        Met     NM_000342.3       P02730   B3AT_HUMAN   \n1213242                        Gln     NM_000029.4       B0ZBE2   ANGT_HUMAN   \n1213243                        Gln     NM_000029.4       B0ZBE2   ANGT_HUMAN   \n\n         stars  \n24           1  \n25           1  \n36           1  \n37           1  \n48           1  \n...        ...  \n1213199      1  \n1213224      2  \n1213225      2  \n1213242      1  \n1213243      1  \n\n[20243 rows x 16 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>UniProtKB-ID</th>\n      <th>stars</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>24</th>\n      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>848</td>\n      <td>ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...</td>\n      <td>A/C</td>\n      <td>Gln283Pro</td>\n      <td>Gln</td>\n      <td>283</td>\n      <td>Pro</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>HFE_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>25</th>\n      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>848</td>\n      <td>ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...</td>\n      <td>A/C</td>\n      <td>Gln283Pro</td>\n      <td>Gln</td>\n      <td>283</td>\n      <td>Pro</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>HFE_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>36</th>\n      <td>NM_138413.4(HOGA1):c.209G&gt;C (p.Arg70Pro)</td>\n      <td>112817</td>\n      <td>HOGA1</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>209</td>\n      <td>ClinGen:CA113816,OMIM:613597.0005</td>\n      <td>G/C</td>\n      <td>Arg70Pro</td>\n      <td>Arg</td>\n      <td>70</td>\n      <td>Pro</td>\n      <td>NM_138413.4</td>\n      <td>Q86XE5</td>\n      <td>HOGA1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>37</th>\n      <td>NM_138413.4(HOGA1):c.209G&gt;C (p.Arg70Pro)</td>\n      <td>112817</td>\n      <td>HOGA1</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>209</td>\n      <td>ClinGen:CA113816,OMIM:613597.0005</td>\n      <td>G/C</td>\n      <td>Arg70Pro</td>\n      <td>Arg</td>\n      <td>70</td>\n      <td>Pro</td>\n      <td>NM_138413.4</td>\n      <td>Q86XE5</td>\n      <td>HOGA1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>48</th>\n      <td>NM_001199107.2(TBC1D24):c.439G&gt;C (p.Asp147His)</td>\n      <td>57465</td>\n      <td>TBC1D24</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>439</td>\n      <td>ClinGen:CA339785,UniProtKB:Q9ULP9#VAR_064365,O...</td>\n      <td>G/C</td>\n      <td>Asp147His</td>\n      <td>Asp</td>\n      <td>147</td>\n      <td>His</td>\n      <td>NM_001199107.2</td>\n      <td>Q9ULP9</td>\n      <td>TBC24_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>1213199</th>\n      <td>NM_021870.2(FGG):c.1007T&gt;C (p.Met336Thr)</td>\n      <td>2266</td>\n      <td>FGG</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>1007</td>\n      <td>ClinGen:CA126399,UniProtKB:P02679#VAR_002414,O...</td>\n      <td>T/C</td>\n      <td>Met336Thr</td>\n      <td>Met</td>\n      <td>336</td>\n      <td>Thr</td>\n      <td>NM_021870.2</td>\n      <td>P02679</td>\n      <td>FIBG_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213224</th>\n      <td>NM_000342.3(SLC4A1):c.1462G&gt;A (p.Val488Met)</td>\n      <td>6521</td>\n      <td>SLC4A1</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>1462</td>\n      <td>ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...</td>\n      <td>G/A</td>\n      <td>Val488Met</td>\n      <td>Val</td>\n      <td>488</td>\n      <td>Met</td>\n      <td>NM_000342.3</td>\n      <td>P02730</td>\n      <td>B3AT_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213225</th>\n      <td>NM_000342.3(SLC4A1):c.1462G&gt;A (p.Val488Met)</td>\n      <td>6521</td>\n      <td>SLC4A1</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>1462</td>\n      <td>ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...</td>\n      <td>G/A</td>\n      <td>Val488Met</td>\n      <td>Val</td>\n      <td>488</td>\n      <td>Met</td>\n      <td>NM_000342.3</td>\n      <td>P02730</td>\n      <td>B3AT_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213242</th>\n      <td>NM_000029.4(AGT):c.1124G&gt;A (p.Arg375Gln)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>1124</td>\n      <td>ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...</td>\n      <td>G/A</td>\n      <td>Arg375Gln</td>\n      <td>Arg</td>\n      <td>375</td>\n      <td>Gln</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213243</th>\n      <td>NM_000029.4(AGT):c.1124G&gt;A (p.Arg375Gln)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>1124</td>\n      <td>ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...</td>\n      <td>G/A</td>\n      <td>Arg375Gln</td>\n      <td>Arg</td>\n      <td>375</td>\n      <td>Gln</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n  </tbody>\n</table>\n<p>20243 rows × 16 columns</p>\n</div>"
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_pathogenic = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance']=='Pathogenic')]\n",
    "df_pathogenic.drop_duplicates(inplace=True)\n",
    "df_pathogenic"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "outputs": [
    {
     "data": {
      "text/plain": "                                                 Name  GeneID GeneSymbol  \\\n10661           NM_000504.4(F10):c.61G>A (p.Gly21Arg)    2159        F10   \n761427     NM_000138.5(FBN1):c.4120T>C (p.Cys1374Arg)    2200       FBN1   \n761428     NM_000138.5(FBN1):c.4120T>G (p.Cys1374Gly)    2200       FBN1   \n761429     NM_000138.5(FBN1):c.4204T>G (p.Cys1402Gly)    2200       FBN1   \n761433     NM_000138.5(FBN1):c.5714G>C (p.Cys1905Ser)    2200       FBN1   \n761434     NM_000138.5(FBN1):c.5800T>C (p.Cys1934Arg)    2200       FBN1   \n761436     NM_000138.5(FBN1):c.5992T>A (p.Cys1998Ser)    2200       FBN1   \n761441     NM_000138.5(FBN1):c.6391T>C (p.Cys2131Arg)    2200       FBN1   \n761442     NM_000138.5(FBN1):c.6409T>G (p.Cys2137Gly)    2200       FBN1   \n761444     NM_000138.5(FBN1):c.6694T>A (p.Cys2232Ser)    2200       FBN1   \n761445     NM_000138.5(FBN1):c.7286G>T (p.Cys2429Phe)    2200       FBN1   \n761446     NM_000138.5(FBN1):c.7525T>A (p.Cys2509Ser)    2200       FBN1   \n761449      NM_000138.5(FBN1):c.1884C>G (p.Cys628Trp)    2200       FBN1   \n761450      NM_000138.5(FBN1):c.2307T>G (p.Cys769Trp)    2200       FBN1   \n761453      NM_000138.5(FBN1):c.2777G>C (p.Cys926Ser)    2200       FBN1   \n761455     NM_000138.5(FBN1):c.3130T>G (p.Cys1044Gly)    2200       FBN1   \n761456     NM_000138.5(FBN1):c.3370T>G (p.Cys1124Gly)    2200       FBN1   \n1019927  NM_000094.4(COL7A1):c.6602G>A (p.Gly2201Glu)    1294     COL7A1   \n1020037  NM_000094.4(COL7A1):c.6073G>A (p.Gly2025Ser)    1294     COL7A1   \n1020038  NM_000094.4(COL7A1):c.6647G>A (p.Gly2216Glu)    1294     COL7A1   \n1020039  NM_000094.4(COL7A1):c.6733G>A (p.Gly2245Ser)    1294     COL7A1   \n1020046  NM_000094.4(COL7A1):c.5318G>A (p.Gly1773Asp)    1294     COL7A1   \n1020048  NM_000094.4(COL7A1):c.6008G>A (p.Gly2003Glu)    1294     COL7A1   \n1020049  NM_000094.4(COL7A1):c.6016G>A (p.Gly2006Ser)    1294     COL7A1   \n1020050  NM_000094.4(COL7A1):c.6025G>A (p.Gly2009Arg)    1294     COL7A1   \n1020051  NM_000094.4(COL7A1):c.6035G>A (p.Gly2012Asp)    1294     COL7A1   \n1020052  NM_000094.4(COL7A1):c.6136G>A (p.Gly2046Ser)    1294     COL7A1   \n1020053  NM_000094.4(COL7A1):c.6191G>A (p.Gly2064Glu)    1294     COL7A1   \n1020054  NM_000094.4(COL7A1):c.6209G>A (p.Gly2070Glu)    1294     COL7A1   \n1020055  NM_000094.4(COL7A1):c.6697G>A (p.Gly2233Ser)    1294     COL7A1   \n1020057  NM_000094.4(COL7A1):c.7877G>A (p.Gly2626Asp)    1294     COL7A1   \n\n        Assembly ClinicalSignificance  CDS_position          OtherIDs  \\\n10661         na           Pathogenic            61  OMIM:613872.0005   \n761427        na           Pathogenic          4120                 -   \n761428        na           Pathogenic          4120                 -   \n761429        na           Pathogenic          4204                 -   \n761433        na           Pathogenic          5714                 -   \n761434        na           Pathogenic          5800                 -   \n761436        na           Pathogenic          5992                 -   \n761441        na           Pathogenic          6391                 -   \n761442        na           Pathogenic          6409                 -   \n761444        na           Pathogenic          6694                 -   \n761445        na           Pathogenic          7286                 -   \n761446        na           Pathogenic          7525                 -   \n761449        na           Pathogenic          1884                 -   \n761450        na           Pathogenic          2307                 -   \n761453        na           Pathogenic          2777                 -   \n761455        na           Pathogenic          3130                 -   \n761456        na           Pathogenic          3370                 -   \n1019927       na           Pathogenic          6602                 -   \n1020037       na           Pathogenic          6073                 -   \n1020038       na           Pathogenic          6647                 -   \n1020039       na           Pathogenic          6733                 -   \n1020046       na           Pathogenic          5318                 -   \n1020048       na           Pathogenic          6008                 -   \n1020049       na           Pathogenic          6016                 -   \n1020050       na           Pathogenic          6025                 -   \n1020051       na           Pathogenic          6035                 -   \n1020052       na           Pathogenic          6136                 -   \n1020053       na           Pathogenic          6191                 -   \n1020054       na           Pathogenic          6209                 -   \n1020055       na           Pathogenic          6697                 -   \n1020057       na           Pathogenic          7877                 -   \n\n        AlleleChange Protein_Patterns Amino_acids_before_3letters  \\\n10661            G/A         Gly21Arg                         Gly   \n761427           T/C       Cys1374Arg                         Cys   \n761428           T/G       Cys1374Gly                         Cys   \n761429           T/G       Cys1402Gly                         Cys   \n761433           G/C       Cys1905Ser                         Cys   \n761434           T/C       Cys1934Arg                         Cys   \n761436           T/A       Cys1998Ser                         Cys   \n761441           T/C       Cys2131Arg                         Cys   \n761442           T/G       Cys2137Gly                         Cys   \n761444           T/A       Cys2232Ser                         Cys   \n761445           G/T       Cys2429Phe                         Cys   \n761446           T/A       Cys2509Ser                         Cys   \n761449           C/G        Cys628Trp                         Cys   \n761450           T/G        Cys769Trp                         Cys   \n761453           G/C        Cys926Ser                         Cys   \n761455           T/G       Cys1044Gly                         Cys   \n761456           T/G       Cys1124Gly                         Cys   \n1019927          G/A       Gly2201Glu                         Gly   \n1020037          G/A       Gly2025Ser                         Gly   \n1020038          G/A       Gly2216Glu                         Gly   \n1020039          G/A       Gly2245Ser                         Gly   \n1020046          G/A       Gly1773Asp                         Gly   \n1020048          G/A       Gly2003Glu                         Gly   \n1020049          G/A       Gly2006Ser                         Gly   \n1020050          G/A       Gly2009Arg                         Gly   \n1020051          G/A       Gly2012Asp                         Gly   \n1020052          G/A       Gly2046Ser                         Gly   \n1020053          G/A       Gly2064Glu                         Gly   \n1020054          G/A       Gly2070Glu                         Gly   \n1020055          G/A       Gly2233Ser                         Gly   \n1020057          G/A       Gly2626Asp                         Gly   \n\n         Protein_positions Amino_acids_after_3letters        NM_ID  \\\n10661                   21                        Arg  NM_000504.4   \n761427                1374                        Arg  NM_000138.5   \n761428                1374                        Gly  NM_000138.5   \n761429                1402                        Gly  NM_000138.5   \n761433                1905                        Ser  NM_000138.5   \n761434                1934                        Arg  NM_000138.5   \n761436                1998                        Ser  NM_000138.5   \n761441                2131                        Arg  NM_000138.5   \n761442                2137                        Gly  NM_000138.5   \n761444                2232                        Ser  NM_000138.5   \n761445                2429                        Phe  NM_000138.5   \n761446                2509                        Ser  NM_000138.5   \n761449                 628                        Trp  NM_000138.5   \n761450                 769                        Trp  NM_000138.5   \n761453                 926                        Ser  NM_000138.5   \n761455                1044                        Gly  NM_000138.5   \n761456                1124                        Gly  NM_000138.5   \n1019927               2201                        Glu  NM_000094.4   \n1020037               2025                        Ser  NM_000094.4   \n1020038               2216                        Glu  NM_000094.4   \n1020039               2245                        Ser  NM_000094.4   \n1020046               1773                        Asp  NM_000094.4   \n1020048               2003                        Glu  NM_000094.4   \n1020049               2006                        Ser  NM_000094.4   \n1020050               2009                        Arg  NM_000094.4   \n1020051               2012                        Asp  NM_000094.4   \n1020052               2046                        Ser  NM_000094.4   \n1020053               2064                        Glu  NM_000094.4   \n1020054               2070                        Glu  NM_000094.4   \n1020055               2233                        Ser  NM_000094.4   \n1020057               2626                        Asp  NM_000094.4   \n\n        UniProtKB-AC UniProtKB-ID  stars  \n10661         P00742   FA10_HUMAN      1  \n761427        P35555   FBN1_HUMAN      1  \n761428        P35555   FBN1_HUMAN      2  \n761429        P35555   FBN1_HUMAN      1  \n761433        P35555   FBN1_HUMAN      1  \n761434        P35555   FBN1_HUMAN      1  \n761436        P35555   FBN1_HUMAN      1  \n761441        P35555   FBN1_HUMAN      1  \n761442        P35555   FBN1_HUMAN      1  \n761444        P35555   FBN1_HUMAN      1  \n761445        P35555   FBN1_HUMAN      1  \n761446        P35555   FBN1_HUMAN      1  \n761449        P35555   FBN1_HUMAN      1  \n761450        P35555   FBN1_HUMAN      1  \n761453        P35555   FBN1_HUMAN      1  \n761455        P35555   FBN1_HUMAN      1  \n761456        P35555   FBN1_HUMAN      1  \n1019927       Q02388  CO7A1_HUMAN      1  \n1020037       Q02388  CO7A1_HUMAN      1  \n1020038       Q02388  CO7A1_HUMAN      1  \n1020039       Q02388  CO7A1_HUMAN      1  \n1020046       Q02388  CO7A1_HUMAN      1  \n1020048       Q02388  CO7A1_HUMAN      2  \n1020049       Q02388  CO7A1_HUMAN      1  \n1020050       Q02388  CO7A1_HUMAN      1  \n1020051       Q02388  CO7A1_HUMAN      1  \n1020052       Q02388  CO7A1_HUMAN      1  \n1020053       Q02388  CO7A1_HUMAN      1  \n1020054       Q02388  CO7A1_HUMAN      1  \n1020055       Q02388  CO7A1_HUMAN      1  \n1020057       Q02388  CO7A1_HUMAN      1  ",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>UniProtKB-ID</th>\n      <th>stars</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>10661</th>\n      <td>NM_000504.4(F10):c.61G&gt;A (p.Gly21Arg)</td>\n      <td>2159</td>\n      <td>F10</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>61</td>\n      <td>OMIM:613872.0005</td>\n      <td>G/A</td>\n      <td>Gly21Arg</td>\n      <td>Gly</td>\n      <td>21</td>\n      <td>Arg</td>\n      <td>NM_000504.4</td>\n      <td>P00742</td>\n      <td>FA10_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>761427</th>\n      <td>NM_000138.5(FBN1):c.4120T&gt;C (p.Cys1374Arg)</td>\n      <td>2200</td>\n      <td>FBN1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>4120</td>\n      <td>-</td>\n      <td>T/C</td>\n      <td>Cys1374Arg</td>\n      <td>Cys</td>\n      <td>1374</td>\n      <td>Arg</td>\n      <td>NM_000138.5</td>\n      <td>P35555</td>\n      <td>FBN1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>761428</th>\n      <td>NM_000138.5(FBN1):c.4120T&gt;G (p.Cys1374Gly)</td>\n      <td>2200</td>\n      <td>FBN1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>4120</td>\n      <td>-</td>\n      <td>T/G</td>\n      <td>Cys1374Gly</td>\n      <td>Cys</td>\n      <td>1374</td>\n      <td>Gly</td>\n      <td>NM_000138.5</td>\n      <td>P35555</td>\n      <td>FBN1_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>761429</th>\n      <td>NM_000138.5(FBN1):c.4204T&gt;G (p.Cys1402Gly)</td>\n      <td>2200</td>\n      <td>FBN1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>4204</td>\n      <td>-</td>\n      <td>T/G</td>\n      <td>Cys1402Gly</td>\n      <td>Cys</td>\n      <td>1402</td>\n      <td>Gly</td>\n      <td>NM_000138.5</td>\n      <td>P35555</td>\n      <td>FBN1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>761433</th>\n      <td>NM_000138.5(FBN1):c.5714G&gt;C (p.Cys1905Ser)</td>\n      <td>2200</td>\n      <td>FBN1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>5714</td>\n      <td>-</td>\n      <td>G/C</td>\n      <td>Cys1905Ser</td>\n      <td>Cys</td>\n      <td>1905</td>\n      <td>Ser</td>\n      <td>NM_000138.5</td>\n      <td>P35555</td>\n      <td>FBN1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>761434</th>\n      <td>NM_000138.5(FBN1):c.5800T&gt;C (p.Cys1934Arg)</td>\n      <td>2200</td>\n      <td>FBN1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>5800</td>\n      <td>-</td>\n      <td>T/C</td>\n      <td>Cys1934Arg</td>\n      <td>Cys</td>\n      <td>1934</td>\n      <td>Arg</td>\n      <td>NM_000138.5</td>\n      <td>P35555</td>\n      <td>FBN1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>761436</th>\n      <td>NM_000138.5(FBN1):c.5992T&gt;A (p.Cys1998Ser)</td>\n      <td>2200</td>\n      <td>FBN1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>5992</td>\n      <td>-</td>\n      <td>T/A</td>\n      <td>Cys1998Ser</td>\n      <td>Cys</td>\n      <td>1998</td>\n      <td>Ser</td>\n      <td>NM_000138.5</td>\n      <td>P35555</td>\n      <td>FBN1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>761441</th>\n      <td>NM_000138.5(FBN1):c.6391T&gt;C (p.Cys2131Arg)</td>\n      <td>2200</td>\n      <td>FBN1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>6391</td>\n      <td>-</td>\n      <td>T/C</td>\n      <td>Cys2131Arg</td>\n      <td>Cys</td>\n      <td>2131</td>\n      <td>Arg</td>\n      <td>NM_000138.5</td>\n      <td>P35555</td>\n      <td>FBN1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>761442</th>\n      <td>NM_000138.5(FBN1):c.6409T&gt;G (p.Cys2137Gly)</td>\n      <td>2200</td>\n      <td>FBN1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>6409</td>\n      <td>-</td>\n      <td>T/G</td>\n      <td>Cys2137Gly</td>\n      <td>Cys</td>\n      <td>2137</td>\n      <td>Gly</td>\n      <td>NM_000138.5</td>\n      <td>P35555</td>\n      <td>FBN1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>761444</th>\n      <td>NM_000138.5(FBN1):c.6694T&gt;A (p.Cys2232Ser)</td>\n      <td>2200</td>\n      <td>FBN1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>6694</td>\n      <td>-</td>\n      <td>T/A</td>\n      <td>Cys2232Ser</td>\n      <td>Cys</td>\n      <td>2232</td>\n      <td>Ser</td>\n      <td>NM_000138.5</td>\n      <td>P35555</td>\n      <td>FBN1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>761445</th>\n      <td>NM_000138.5(FBN1):c.7286G&gt;T (p.Cys2429Phe)</td>\n      <td>2200</td>\n      <td>FBN1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>7286</td>\n      <td>-</td>\n      <td>G/T</td>\n      <td>Cys2429Phe</td>\n      <td>Cys</td>\n      <td>2429</td>\n      <td>Phe</td>\n      <td>NM_000138.5</td>\n      <td>P35555</td>\n      <td>FBN1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>761446</th>\n      <td>NM_000138.5(FBN1):c.7525T&gt;A (p.Cys2509Ser)</td>\n      <td>2200</td>\n      <td>FBN1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>7525</td>\n      <td>-</td>\n      <td>T/A</td>\n      <td>Cys2509Ser</td>\n      <td>Cys</td>\n      <td>2509</td>\n      <td>Ser</td>\n      <td>NM_000138.5</td>\n      <td>P35555</td>\n      <td>FBN1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>761449</th>\n      <td>NM_000138.5(FBN1):c.1884C&gt;G (p.Cys628Trp)</td>\n      <td>2200</td>\n      <td>FBN1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>1884</td>\n      <td>-</td>\n      <td>C/G</td>\n      <td>Cys628Trp</td>\n      <td>Cys</td>\n      <td>628</td>\n      <td>Trp</td>\n      <td>NM_000138.5</td>\n      <td>P35555</td>\n      <td>FBN1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>761450</th>\n      <td>NM_000138.5(FBN1):c.2307T&gt;G (p.Cys769Trp)</td>\n      <td>2200</td>\n      <td>FBN1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>2307</td>\n      <td>-</td>\n      <td>T/G</td>\n      <td>Cys769Trp</td>\n      <td>Cys</td>\n      <td>769</td>\n      <td>Trp</td>\n      <td>NM_000138.5</td>\n      <td>P35555</td>\n      <td>FBN1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>761453</th>\n      <td>NM_000138.5(FBN1):c.2777G&gt;C (p.Cys926Ser)</td>\n      <td>2200</td>\n      <td>FBN1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>2777</td>\n      <td>-</td>\n      <td>G/C</td>\n      <td>Cys926Ser</td>\n      <td>Cys</td>\n      <td>926</td>\n      <td>Ser</td>\n      <td>NM_000138.5</td>\n      <td>P35555</td>\n      <td>FBN1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>761455</th>\n      <td>NM_000138.5(FBN1):c.3130T&gt;G (p.Cys1044Gly)</td>\n      <td>2200</td>\n      <td>FBN1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>3130</td>\n      <td>-</td>\n      <td>T/G</td>\n      <td>Cys1044Gly</td>\n      <td>Cys</td>\n      <td>1044</td>\n      <td>Gly</td>\n      <td>NM_000138.5</td>\n      <td>P35555</td>\n      <td>FBN1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>761456</th>\n      <td>NM_000138.5(FBN1):c.3370T&gt;G (p.Cys1124Gly)</td>\n      <td>2200</td>\n      <td>FBN1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>3370</td>\n      <td>-</td>\n      <td>T/G</td>\n      <td>Cys1124Gly</td>\n      <td>Cys</td>\n      <td>1124</td>\n      <td>Gly</td>\n      <td>NM_000138.5</td>\n      <td>P35555</td>\n      <td>FBN1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1019927</th>\n      <td>NM_000094.4(COL7A1):c.6602G&gt;A (p.Gly2201Glu)</td>\n      <td>1294</td>\n      <td>COL7A1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>6602</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Gly2201Glu</td>\n      <td>Gly</td>\n      <td>2201</td>\n      <td>Glu</td>\n      <td>NM_000094.4</td>\n      <td>Q02388</td>\n      <td>CO7A1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1020037</th>\n      <td>NM_000094.4(COL7A1):c.6073G&gt;A (p.Gly2025Ser)</td>\n      <td>1294</td>\n      <td>COL7A1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>6073</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Gly2025Ser</td>\n      <td>Gly</td>\n      <td>2025</td>\n      <td>Ser</td>\n      <td>NM_000094.4</td>\n      <td>Q02388</td>\n      <td>CO7A1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1020038</th>\n      <td>NM_000094.4(COL7A1):c.6647G&gt;A (p.Gly2216Glu)</td>\n      <td>1294</td>\n      <td>COL7A1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>6647</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Gly2216Glu</td>\n      <td>Gly</td>\n      <td>2216</td>\n      <td>Glu</td>\n      <td>NM_000094.4</td>\n      <td>Q02388</td>\n      <td>CO7A1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1020039</th>\n      <td>NM_000094.4(COL7A1):c.6733G&gt;A (p.Gly2245Ser)</td>\n      <td>1294</td>\n      <td>COL7A1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>6733</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Gly2245Ser</td>\n      <td>Gly</td>\n      <td>2245</td>\n      <td>Ser</td>\n      <td>NM_000094.4</td>\n      <td>Q02388</td>\n      <td>CO7A1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1020046</th>\n      <td>NM_000094.4(COL7A1):c.5318G&gt;A (p.Gly1773Asp)</td>\n      <td>1294</td>\n      <td>COL7A1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>5318</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Gly1773Asp</td>\n      <td>Gly</td>\n      <td>1773</td>\n      <td>Asp</td>\n      <td>NM_000094.4</td>\n      <td>Q02388</td>\n      <td>CO7A1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1020048</th>\n      <td>NM_000094.4(COL7A1):c.6008G&gt;A (p.Gly2003Glu)</td>\n      <td>1294</td>\n      <td>COL7A1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>6008</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Gly2003Glu</td>\n      <td>Gly</td>\n      <td>2003</td>\n      <td>Glu</td>\n      <td>NM_000094.4</td>\n      <td>Q02388</td>\n      <td>CO7A1_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1020049</th>\n      <td>NM_000094.4(COL7A1):c.6016G&gt;A (p.Gly2006Ser)</td>\n      <td>1294</td>\n      <td>COL7A1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>6016</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Gly2006Ser</td>\n      <td>Gly</td>\n      <td>2006</td>\n      <td>Ser</td>\n      <td>NM_000094.4</td>\n      <td>Q02388</td>\n      <td>CO7A1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1020050</th>\n      <td>NM_000094.4(COL7A1):c.6025G&gt;A (p.Gly2009Arg)</td>\n      <td>1294</td>\n      <td>COL7A1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>6025</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Gly2009Arg</td>\n      <td>Gly</td>\n      <td>2009</td>\n      <td>Arg</td>\n      <td>NM_000094.4</td>\n      <td>Q02388</td>\n      <td>CO7A1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1020051</th>\n      <td>NM_000094.4(COL7A1):c.6035G&gt;A (p.Gly2012Asp)</td>\n      <td>1294</td>\n      <td>COL7A1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>6035</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Gly2012Asp</td>\n      <td>Gly</td>\n      <td>2012</td>\n      <td>Asp</td>\n      <td>NM_000094.4</td>\n      <td>Q02388</td>\n      <td>CO7A1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1020052</th>\n      <td>NM_000094.4(COL7A1):c.6136G&gt;A (p.Gly2046Ser)</td>\n      <td>1294</td>\n      <td>COL7A1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>6136</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Gly2046Ser</td>\n      <td>Gly</td>\n      <td>2046</td>\n      <td>Ser</td>\n      <td>NM_000094.4</td>\n      <td>Q02388</td>\n      <td>CO7A1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1020053</th>\n      <td>NM_000094.4(COL7A1):c.6191G&gt;A (p.Gly2064Glu)</td>\n      <td>1294</td>\n      <td>COL7A1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>6191</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Gly2064Glu</td>\n      <td>Gly</td>\n      <td>2064</td>\n      <td>Glu</td>\n      <td>NM_000094.4</td>\n      <td>Q02388</td>\n      <td>CO7A1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1020054</th>\n      <td>NM_000094.4(COL7A1):c.6209G&gt;A (p.Gly2070Glu)</td>\n      <td>1294</td>\n      <td>COL7A1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>6209</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Gly2070Glu</td>\n      <td>Gly</td>\n      <td>2070</td>\n      <td>Glu</td>\n      <td>NM_000094.4</td>\n      <td>Q02388</td>\n      <td>CO7A1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1020055</th>\n      <td>NM_000094.4(COL7A1):c.6697G&gt;A (p.Gly2233Ser)</td>\n      <td>1294</td>\n      <td>COL7A1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>6697</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Gly2233Ser</td>\n      <td>Gly</td>\n      <td>2233</td>\n      <td>Ser</td>\n      <td>NM_000094.4</td>\n      <td>Q02388</td>\n      <td>CO7A1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1020057</th>\n      <td>NM_000094.4(COL7A1):c.7877G&gt;A (p.Gly2626Asp)</td>\n      <td>1294</td>\n      <td>COL7A1</td>\n      <td>na</td>\n      <td>Pathogenic</td>\n      <td>7877</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Gly2626Asp</td>\n      <td>Gly</td>\n      <td>2626</td>\n      <td>Asp</td>\n      <td>NM_000094.4</td>\n      <td>Q02388</td>\n      <td>CO7A1_HUMAN</td>\n      <td>1</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_pathogenic[df_pathogenic['Assembly']=='na']"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "outputs": [
    {
     "data": {
      "text/plain": "                                             Name  GeneID GeneSymbol  \\\n0      NM_000016.6(ACADM):c.1034A>T (p.Asp345Val)      34      ACADM   \n1      NM_000016.6(ACADM):c.1042C>T (p.Arg348Cys)      34      ACADM   \n2      NM_000016.6(ACADM):c.1045C>G (p.Arg349Gly)      34      ACADM   \n3      NM_000016.6(ACADM):c.1046G>A (p.Arg349Gln)      34      ACADM   \n4      NM_000016.6(ACADM):c.1067T>C (p.Ile356Thr)      34      ACADM   \n...                                           ...     ...        ...   \n10132      NM_213653.4(HJV):c.238T>C (p.Cys80Arg)  148738        HJV   \n10133     NM_213653.4(HJV):c.302T>C (p.Leu101Pro)  148738        HJV   \n10134     NM_213653.4(HJV):c.842T>C (p.Ile281Thr)  148738        HJV   \n10135     NM_213653.4(HJV):c.959G>T (p.Gly320Val)  148738        HJV   \n10136   NM_213720.3(CHCHD10):c.44G>T (p.Arg15Leu)  400916    CHCHD10   \n\n      ClinicalSignificance  CDS_position  \\\n0               Pathogenic          1034   \n1               Pathogenic          1042   \n2               Pathogenic          1045   \n3               Pathogenic          1046   \n4               Pathogenic          1067   \n...                    ...           ...   \n10132           Pathogenic           238   \n10133           Pathogenic           302   \n10134           Pathogenic           842   \n10135           Pathogenic           959   \n10136           Pathogenic            44   \n\n                                                OtherIDs AlleleChange  \\\n0                                       ClinGen:CA913267          A/T   \n1                                                      -          C/T   \n2                                                      -          C/G   \n3                                                      -          G/A   \n4                                                      -          T/C   \n...                                                  ...          ...   \n10132  ClinGen:CA252255,UniProtKB:Q6ZVN8#VAR_019617,O...          T/C   \n10133  ClinGen:CA252256,UniProtKB:Q6ZVN8#VAR_019620,O...          T/C   \n10134                  ClinGen:CA252253,OMIM:608374.0004          T/C   \n10135  ClinGen:CA252249,UniProtKB:Q6ZVN8#VAR_019629,O...          G/T   \n10136                  ClinGen:CA235291,OMIM:615903.0002          G/T   \n\n      Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n0            Asp345Val                         Asp                345   \n1            Arg348Cys                         Arg                348   \n2            Arg349Gly                         Arg                349   \n3            Arg349Gln                         Arg                349   \n4            Ile356Thr                         Ile                356   \n...                ...                         ...                ...   \n10132         Cys80Arg                         Cys                 80   \n10133        Leu101Pro                         Leu                101   \n10134        Ile281Thr                         Ile                281   \n10135        Gly320Val                         Gly                320   \n10136         Arg15Leu                         Arg                 15   \n\n      Amino_acids_after_3letters        NM_ID UniProtKB-AC UniProtKB-ID  \\\n0                            Val  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n1                            Cys  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n2                            Gly  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n3                            Gln  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n4                            Thr  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n...                          ...          ...          ...          ...   \n10132                        Arg  NM_213653.4   A0A024R4F5   RGMC_HUMAN   \n10133                        Pro  NM_213653.4   A0A024R4F5   RGMC_HUMAN   \n10134                        Thr  NM_213653.4   A0A024R4F5   RGMC_HUMAN   \n10135                        Val  NM_213653.4   A0A024R4F5   RGMC_HUMAN   \n10136                        Leu  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n\n       stars          Assembly  \n0          1  [GRCh37, GRCh38]  \n1          1  [GRCh37, GRCh38]  \n2          1  [GRCh37, GRCh38]  \n3          1  [GRCh37, GRCh38]  \n4          1  [GRCh37, GRCh38]  \n...      ...               ...  \n10132      1  [GRCh37, GRCh38]  \n10133      1  [GRCh37, GRCh38]  \n10134      1  [GRCh37, GRCh38]  \n10135      2  [GRCh37, GRCh38]  \n10136      2  [GRCh37, GRCh38]  \n\n[10137 rows x 16 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>UniProtKB-ID</th>\n      <th>stars</th>\n      <th>Assembly</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NM_000016.6(ACADM):c.1034A&gt;T (p.Asp345Val)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1034</td>\n      <td>ClinGen:CA913267</td>\n      <td>A/T</td>\n      <td>Asp345Val</td>\n      <td>Asp</td>\n      <td>345</td>\n      <td>Val</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NM_000016.6(ACADM):c.1042C&gt;T (p.Arg348Cys)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1042</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Arg348Cys</td>\n      <td>Arg</td>\n      <td>348</td>\n      <td>Cys</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NM_000016.6(ACADM):c.1045C&gt;G (p.Arg349Gly)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1045</td>\n      <td>-</td>\n      <td>C/G</td>\n      <td>Arg349Gly</td>\n      <td>Arg</td>\n      <td>349</td>\n      <td>Gly</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NM_000016.6(ACADM):c.1046G&gt;A (p.Arg349Gln)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1046</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Arg349Gln</td>\n      <td>Arg</td>\n      <td>349</td>\n      <td>Gln</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NM_000016.6(ACADM):c.1067T&gt;C (p.Ile356Thr)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1067</td>\n      <td>-</td>\n      <td>T/C</td>\n      <td>Ile356Thr</td>\n      <td>Ile</td>\n      <td>356</td>\n      <td>Thr</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>10132</th>\n      <td>NM_213653.4(HJV):c.238T&gt;C (p.Cys80Arg)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>238</td>\n      <td>ClinGen:CA252255,UniProtKB:Q6ZVN8#VAR_019617,O...</td>\n      <td>T/C</td>\n      <td>Cys80Arg</td>\n      <td>Cys</td>\n      <td>80</td>\n      <td>Arg</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>RGMC_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>10133</th>\n      <td>NM_213653.4(HJV):c.302T&gt;C (p.Leu101Pro)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>302</td>\n      <td>ClinGen:CA252256,UniProtKB:Q6ZVN8#VAR_019620,O...</td>\n      <td>T/C</td>\n      <td>Leu101Pro</td>\n      <td>Leu</td>\n      <td>101</td>\n      <td>Pro</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>RGMC_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>10134</th>\n      <td>NM_213653.4(HJV):c.842T&gt;C (p.Ile281Thr)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>842</td>\n      <td>ClinGen:CA252253,OMIM:608374.0004</td>\n      <td>T/C</td>\n      <td>Ile281Thr</td>\n      <td>Ile</td>\n      <td>281</td>\n      <td>Thr</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>RGMC_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>10135</th>\n      <td>NM_213653.4(HJV):c.959G&gt;T (p.Gly320Val)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>959</td>\n      <td>ClinGen:CA252249,UniProtKB:Q6ZVN8#VAR_019629,O...</td>\n      <td>G/T</td>\n      <td>Gly320Val</td>\n      <td>Gly</td>\n      <td>320</td>\n      <td>Val</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>RGMC_HUMAN</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>10136</th>\n      <td>NM_213720.3(CHCHD10):c.44G&gt;T (p.Arg15Leu)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Pathogenic</td>\n      <td>44</td>\n      <td>ClinGen:CA235291,OMIM:615903.0002</td>\n      <td>G/T</td>\n      <td>Arg15Leu</td>\n      <td>Arg</td>\n      <td>15</td>\n      <td>Leu</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n  </tbody>\n</table>\n<p>10137 rows × 16 columns</p>\n</div>"
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# remove duplicates but maintain same order of column names\n",
    "col_list = df_pathogenic.columns.tolist()\n",
    "col_list.remove('Assembly')\n",
    "d = df_pathogenic.groupby(col_list).agg({'Assembly': pd.Series.unique}).reset_index()\n",
    "d"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/pbs.499107.hn-10-03/ipykernel_138049/143085840.py:2: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.\n",
      "  df_pathogenic_new = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance'].str.contains('(Pathogenic|Likely pathogenic)'))]\n",
      "/var/tmp/pbs.499107.hn-10-03/ipykernel_138049/143085840.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_pathogenic_new.drop_duplicates(inplace=True)\n"
     ]
    },
    {
     "data": {
      "text/plain": "                                                Name  GeneID GeneSymbol  \\\n24           NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n25           NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n32         NM_138413.4(HOGA1):c.860G>T (p.Gly287Val)  112817      HOGA1   \n33         NM_138413.4(HOGA1):c.860G>T (p.Gly287Val)  112817      HOGA1   \n36          NM_138413.4(HOGA1):c.209G>C (p.Arg70Pro)  112817      HOGA1   \n...                                              ...     ...        ...   \n1213229  NM_000342.3(SLC4A1):c.2608C>T (p.Arg870Trp)    6521     SLC4A1   \n1213230  NM_000342.3(SLC4A1):c.2279G>A (p.Arg760Gln)    6521     SLC4A1   \n1213231  NM_000342.3(SLC4A1):c.2279G>A (p.Arg760Gln)    6521     SLC4A1   \n1213242     NM_000029.4(AGT):c.1124G>A (p.Arg375Gln)     183        AGT   \n1213243     NM_000029.4(AGT):c.1124G>A (p.Arg375Gln)     183        AGT   \n\n        Assembly          ClinicalSignificance  CDS_position  \\\n24        GRCh37                    Pathogenic           848   \n25        GRCh38                    Pathogenic           848   \n32        GRCh37  Pathogenic/Likely pathogenic           860   \n33        GRCh38  Pathogenic/Likely pathogenic           860   \n36        GRCh37                    Pathogenic           209   \n...          ...                           ...           ...   \n1213229   GRCh38             Likely pathogenic          2608   \n1213230   GRCh37             Likely pathogenic          2279   \n1213231   GRCh38             Likely pathogenic          2279   \n1213242   GRCh37                    Pathogenic          1124   \n1213243   GRCh38                    Pathogenic          1124   \n\n                                                  OtherIDs AlleleChange  \\\n24       ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...          A/C   \n25       ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...          A/C   \n32       ClinGen:CA113813,UniProtKB:Q86XE5#VAR_064036,O...          G/T   \n33       ClinGen:CA113813,UniProtKB:Q86XE5#VAR_064036,O...          G/T   \n36                       ClinGen:CA113816,OMIM:613597.0005          G/C   \n...                                                    ...          ...   \n1213229  ClinGen:CA127403,UniProtKB:P02730#VAR_013816,O...          C/T   \n1213230  ClinGen:CA210834,UniProtKB:P02730#VAR_013806,O...          G/A   \n1213231  ClinGen:CA210834,UniProtKB:P02730#VAR_013806,O...          G/A   \n1213242  ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...          G/A   \n1213243  ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...          G/A   \n\n        Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n24             Gln283Pro                         Gln                283   \n25             Gln283Pro                         Gln                283   \n32             Gly287Val                         Gly                287   \n33             Gly287Val                         Gly                287   \n36              Arg70Pro                         Arg                 70   \n...                  ...                         ...                ...   \n1213229        Arg870Trp                         Arg                870   \n1213230        Arg760Gln                         Arg                760   \n1213231        Arg760Gln                         Arg                760   \n1213242        Arg375Gln                         Arg                375   \n1213243        Arg375Gln                         Arg                375   \n\n        Amino_acids_after_3letters        NM_ID UniProtKB-AC UniProtKB-ID  \\\n24                             Pro  NM_000410.4       Q30201    HFE_HUMAN   \n25                             Pro  NM_000410.4       Q30201    HFE_HUMAN   \n32                             Val  NM_138413.4       Q86XE5  HOGA1_HUMAN   \n33                             Val  NM_138413.4       Q86XE5  HOGA1_HUMAN   \n36                             Pro  NM_138413.4       Q86XE5  HOGA1_HUMAN   \n...                            ...          ...          ...          ...   \n1213229                        Trp  NM_000342.3       P02730   B3AT_HUMAN   \n1213230                        Gln  NM_000342.3       P02730   B3AT_HUMAN   \n1213231                        Gln  NM_000342.3       P02730   B3AT_HUMAN   \n1213242                        Gln  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n1213243                        Gln  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n\n         stars  \n24           1  \n25           1  \n32           2  \n33           2  \n36           1  \n...        ...  \n1213229      2  \n1213230      1  \n1213231      1  \n1213242      1  \n1213243      1  \n\n[59471 rows x 16 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>UniProtKB-ID</th>\n      <th>stars</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>24</th>\n      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>848</td>\n      <td>ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...</td>\n      <td>A/C</td>\n      <td>Gln283Pro</td>\n      <td>Gln</td>\n      <td>283</td>\n      <td>Pro</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>HFE_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>25</th>\n      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>848</td>\n      <td>ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...</td>\n      <td>A/C</td>\n      <td>Gln283Pro</td>\n      <td>Gln</td>\n      <td>283</td>\n      <td>Pro</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>HFE_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>32</th>\n      <td>NM_138413.4(HOGA1):c.860G&gt;T (p.Gly287Val)</td>\n      <td>112817</td>\n      <td>HOGA1</td>\n      <td>GRCh37</td>\n      <td>Pathogenic/Likely pathogenic</td>\n      <td>860</td>\n      <td>ClinGen:CA113813,UniProtKB:Q86XE5#VAR_064036,O...</td>\n      <td>G/T</td>\n      <td>Gly287Val</td>\n      <td>Gly</td>\n      <td>287</td>\n      <td>Val</td>\n      <td>NM_138413.4</td>\n      <td>Q86XE5</td>\n      <td>HOGA1_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>33</th>\n      <td>NM_138413.4(HOGA1):c.860G&gt;T (p.Gly287Val)</td>\n      <td>112817</td>\n      <td>HOGA1</td>\n      <td>GRCh38</td>\n      <td>Pathogenic/Likely pathogenic</td>\n      <td>860</td>\n      <td>ClinGen:CA113813,UniProtKB:Q86XE5#VAR_064036,O...</td>\n      <td>G/T</td>\n      <td>Gly287Val</td>\n      <td>Gly</td>\n      <td>287</td>\n      <td>Val</td>\n      <td>NM_138413.4</td>\n      <td>Q86XE5</td>\n      <td>HOGA1_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>36</th>\n      <td>NM_138413.4(HOGA1):c.209G&gt;C (p.Arg70Pro)</td>\n      <td>112817</td>\n      <td>HOGA1</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>209</td>\n      <td>ClinGen:CA113816,OMIM:613597.0005</td>\n      <td>G/C</td>\n      <td>Arg70Pro</td>\n      <td>Arg</td>\n      <td>70</td>\n      <td>Pro</td>\n      <td>NM_138413.4</td>\n      <td>Q86XE5</td>\n      <td>HOGA1_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>1213229</th>\n      <td>NM_000342.3(SLC4A1):c.2608C&gt;T (p.Arg870Trp)</td>\n      <td>6521</td>\n      <td>SLC4A1</td>\n      <td>GRCh38</td>\n      <td>Likely pathogenic</td>\n      <td>2608</td>\n      <td>ClinGen:CA127403,UniProtKB:P02730#VAR_013816,O...</td>\n      <td>C/T</td>\n      <td>Arg870Trp</td>\n      <td>Arg</td>\n      <td>870</td>\n      <td>Trp</td>\n      <td>NM_000342.3</td>\n      <td>P02730</td>\n      <td>B3AT_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213230</th>\n      <td>NM_000342.3(SLC4A1):c.2279G&gt;A (p.Arg760Gln)</td>\n      <td>6521</td>\n      <td>SLC4A1</td>\n      <td>GRCh37</td>\n      <td>Likely pathogenic</td>\n      <td>2279</td>\n      <td>ClinGen:CA210834,UniProtKB:P02730#VAR_013806,O...</td>\n      <td>G/A</td>\n      <td>Arg760Gln</td>\n      <td>Arg</td>\n      <td>760</td>\n      <td>Gln</td>\n      <td>NM_000342.3</td>\n      <td>P02730</td>\n      <td>B3AT_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213231</th>\n      <td>NM_000342.3(SLC4A1):c.2279G&gt;A (p.Arg760Gln)</td>\n      <td>6521</td>\n      <td>SLC4A1</td>\n      <td>GRCh38</td>\n      <td>Likely pathogenic</td>\n      <td>2279</td>\n      <td>ClinGen:CA210834,UniProtKB:P02730#VAR_013806,O...</td>\n      <td>G/A</td>\n      <td>Arg760Gln</td>\n      <td>Arg</td>\n      <td>760</td>\n      <td>Gln</td>\n      <td>NM_000342.3</td>\n      <td>P02730</td>\n      <td>B3AT_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213242</th>\n      <td>NM_000029.4(AGT):c.1124G&gt;A (p.Arg375Gln)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>1124</td>\n      <td>ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...</td>\n      <td>G/A</td>\n      <td>Arg375Gln</td>\n      <td>Arg</td>\n      <td>375</td>\n      <td>Gln</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213243</th>\n      <td>NM_000029.4(AGT):c.1124G&gt;A (p.Arg375Gln)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>1124</td>\n      <td>ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...</td>\n      <td>G/A</td>\n      <td>Arg375Gln</td>\n      <td>Arg</td>\n      <td>375</td>\n      <td>Gln</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n  </tbody>\n</table>\n<p>59471 rows × 16 columns</p>\n</div>"
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# include more pathogenic variant\n",
    "df_pathogenic_new = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance'].str.contains('(Pathogenic|Likely pathogenic)'))]\n",
    "#df_pathogenic_new = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance'].isin(['Pathogenic','Likely pathogenic']))]\n",
    "#df_pathogenic_new = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance'].str.contains('Pathogenic'))]\n",
    "#df_pathogenic_new = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance'] == 'Pathogenic')]\n",
    "df_pathogenic_new.drop_duplicates(inplace=True)\n",
    "df_pathogenic_new"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "outputs": [
    {
     "data": {
      "text/plain": "                                             Name  GeneID GeneSymbol  \\\n0      NM_000016.6(ACADM):c.1010A>C (p.Tyr337Ser)      34      ACADM   \n1      NM_000016.6(ACADM):c.1034A>T (p.Asp345Val)      34      ACADM   \n2      NM_000016.6(ACADM):c.1042C>T (p.Arg348Cys)      34      ACADM   \n3      NM_000016.6(ACADM):c.1045C>G (p.Arg349Gly)      34      ACADM   \n4      NM_000016.6(ACADM):c.1046G>A (p.Arg349Gln)      34      ACADM   \n...                                           ...     ...        ...   \n29757     NM_213653.4(HJV):c.572G>C (p.Trp191Ser)  148738        HJV   \n29758     NM_213653.4(HJV):c.842T>C (p.Ile281Thr)  148738        HJV   \n29759     NM_213653.4(HJV):c.959G>T (p.Gly320Val)  148738        HJV   \n29760  NM_213720.3(CHCHD10):c.176C>T (p.Ser59Leu)  400916    CHCHD10   \n29761   NM_213720.3(CHCHD10):c.44G>T (p.Arg15Leu)  400916    CHCHD10   \n\n               ClinicalSignificance  CDS_position  \\\n0                 Likely pathogenic          1010   \n1                        Pathogenic          1034   \n2                        Pathogenic          1042   \n3                        Pathogenic          1045   \n4                        Pathogenic          1046   \n...                             ...           ...   \n29757             Likely pathogenic           572   \n29758                    Pathogenic           842   \n29759                    Pathogenic           959   \n29760  Pathogenic/Likely pathogenic           176   \n29761                    Pathogenic            44   \n\n                                                OtherIDs AlleleChange  \\\n0                                                      -          A/C   \n1                                       ClinGen:CA913267          A/T   \n2                                                      -          C/T   \n3                                                      -          C/G   \n4                                                      -          G/A   \n...                                                  ...          ...   \n29757                                                  -          G/C   \n29758                  ClinGen:CA252253,OMIM:608374.0004          T/C   \n29759  ClinGen:CA252249,UniProtKB:Q6ZVN8#VAR_019629,O...          G/T   \n29760  ClinGen:CA163486,UniProtKB:Q8WYQ3#VAR_071806,O...          C/T   \n29761                  ClinGen:CA235291,OMIM:615903.0002          G/T   \n\n      Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n0            Tyr337Ser                         Tyr                337   \n1            Asp345Val                         Asp                345   \n2            Arg348Cys                         Arg                348   \n3            Arg349Gly                         Arg                349   \n4            Arg349Gln                         Arg                349   \n...                ...                         ...                ...   \n29757        Trp191Ser                         Trp                191   \n29758        Ile281Thr                         Ile                281   \n29759        Gly320Val                         Gly                320   \n29760         Ser59Leu                         Ser                 59   \n29761         Arg15Leu                         Arg                 15   \n\n      Amino_acids_after_3letters        NM_ID UniProtKB-AC UniProtKB-ID  \\\n0                            Ser  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n1                            Val  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n2                            Cys  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n3                            Gly  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n4                            Gln  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n...                          ...          ...          ...          ...   \n29757                        Ser  NM_213653.4   A0A024R4F5   RGMC_HUMAN   \n29758                        Thr  NM_213653.4   A0A024R4F5   RGMC_HUMAN   \n29759                        Val  NM_213653.4   A0A024R4F5   RGMC_HUMAN   \n29760                        Leu  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n29761                        Leu  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n\n       stars          Assembly  \n0          2  [GRCh37, GRCh38]  \n1          1  [GRCh37, GRCh38]  \n2          1  [GRCh37, GRCh38]  \n3          1  [GRCh37, GRCh38]  \n4          1  [GRCh37, GRCh38]  \n...      ...               ...  \n29757      1  [GRCh37, GRCh38]  \n29758      1  [GRCh37, GRCh38]  \n29759      2  [GRCh37, GRCh38]  \n29760      2  [GRCh37, GRCh38]  \n29761      2  [GRCh37, GRCh38]  \n\n[29762 rows x 16 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>UniProtKB-ID</th>\n      <th>stars</th>\n      <th>Assembly</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NM_000016.6(ACADM):c.1010A&gt;C (p.Tyr337Ser)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Likely pathogenic</td>\n      <td>1010</td>\n      <td>-</td>\n      <td>A/C</td>\n      <td>Tyr337Ser</td>\n      <td>Tyr</td>\n      <td>337</td>\n      <td>Ser</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NM_000016.6(ACADM):c.1034A&gt;T (p.Asp345Val)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1034</td>\n      <td>ClinGen:CA913267</td>\n      <td>A/T</td>\n      <td>Asp345Val</td>\n      <td>Asp</td>\n      <td>345</td>\n      <td>Val</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NM_000016.6(ACADM):c.1042C&gt;T (p.Arg348Cys)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1042</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Arg348Cys</td>\n      <td>Arg</td>\n      <td>348</td>\n      <td>Cys</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NM_000016.6(ACADM):c.1045C&gt;G (p.Arg349Gly)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1045</td>\n      <td>-</td>\n      <td>C/G</td>\n      <td>Arg349Gly</td>\n      <td>Arg</td>\n      <td>349</td>\n      <td>Gly</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NM_000016.6(ACADM):c.1046G&gt;A (p.Arg349Gln)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1046</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Arg349Gln</td>\n      <td>Arg</td>\n      <td>349</td>\n      <td>Gln</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>29757</th>\n      <td>NM_213653.4(HJV):c.572G&gt;C (p.Trp191Ser)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Likely pathogenic</td>\n      <td>572</td>\n      <td>-</td>\n      <td>G/C</td>\n      <td>Trp191Ser</td>\n      <td>Trp</td>\n      <td>191</td>\n      <td>Ser</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>RGMC_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>29758</th>\n      <td>NM_213653.4(HJV):c.842T&gt;C (p.Ile281Thr)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>842</td>\n      <td>ClinGen:CA252253,OMIM:608374.0004</td>\n      <td>T/C</td>\n      <td>Ile281Thr</td>\n      <td>Ile</td>\n      <td>281</td>\n      <td>Thr</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>RGMC_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>29759</th>\n      <td>NM_213653.4(HJV):c.959G&gt;T (p.Gly320Val)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>959</td>\n      <td>ClinGen:CA252249,UniProtKB:Q6ZVN8#VAR_019629,O...</td>\n      <td>G/T</td>\n      <td>Gly320Val</td>\n      <td>Gly</td>\n      <td>320</td>\n      <td>Val</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>RGMC_HUMAN</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>29760</th>\n      <td>NM_213720.3(CHCHD10):c.176C&gt;T (p.Ser59Leu)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Pathogenic/Likely pathogenic</td>\n      <td>176</td>\n      <td>ClinGen:CA163486,UniProtKB:Q8WYQ3#VAR_071806,O...</td>\n      <td>C/T</td>\n      <td>Ser59Leu</td>\n      <td>Ser</td>\n      <td>59</td>\n      <td>Leu</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>29761</th>\n      <td>NM_213720.3(CHCHD10):c.44G&gt;T (p.Arg15Leu)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Pathogenic</td>\n      <td>44</td>\n      <td>ClinGen:CA235291,OMIM:615903.0002</td>\n      <td>G/T</td>\n      <td>Arg15Leu</td>\n      <td>Arg</td>\n      <td>15</td>\n      <td>Leu</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n  </tbody>\n</table>\n<p>29762 rows × 16 columns</p>\n</div>"
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d_new = df_pathogenic_new.groupby(col_list).agg({'Assembly': pd.Series.unique}).reset_index()\n",
    "d_new"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2095\n",
      "2158\n",
      "2102\n"
     ]
    }
   ],
   "source": [
    "print(len(d_new['UniProtKB-AC'].unique()))\n",
    "print(len(d_new['GeneSymbol'].unique()))\n",
    "print(len(d_new['GeneID'].unique()))"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/pbs.499107.hn-10-03/ipykernel_138049/1331444648.py:4: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.\n",
      "  df_benign_new = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance'].str.contains('(Benign|Likely benign)'))]\n",
      "/var/tmp/pbs.499107.hn-10-03/ipykernel_138049/1331444648.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_benign_new.drop_duplicates(inplace=True)\n"
     ]
    },
    {
     "data": {
      "text/plain": "                                               Name  GeneID GeneSymbol  \\\n90         NM_000355.4(TCN2):c.776G>C (p.Arg259Pro)    6948       TCN2   \n91         NM_000355.4(TCN2):c.776G>C (p.Arg259Pro)    6948       TCN2   \n244      NM_000398.7(CYB5R3):c.350C>G (p.Thr117Ser)    1727     CYB5R3   \n245      NM_000398.7(CYB5R3):c.350C>G (p.Thr117Ser)    1727     CYB5R3   \n592       NM_000147.5(FUCA1):c.857A>G (p.Gln286Arg)    2517      FUCA1   \n...                                             ...     ...        ...   \n1213283      NM_000029.4(AGT):c.292G>A (p.Glu98Lys)     183        AGT   \n1213284     NM_000029.4(AGT):c.410C>T (p.Thr137Met)     183        AGT   \n1213285     NM_000029.4(AGT):c.410C>T (p.Thr137Met)     183        AGT   \n1213332       NM_000029.4(AGT):c.28A>T (p.Met10Leu)     183        AGT   \n1213333       NM_000029.4(AGT):c.28A>T (p.Met10Leu)     183        AGT   \n\n        Assembly  ClinicalSignificance  CDS_position  \\\n90        GRCh37                Benign           776   \n91        GRCh38                Benign           776   \n244       GRCh37                Benign           350   \n245       GRCh38                Benign           350   \n592       GRCh37                Benign           857   \n...          ...                   ...           ...   \n1213283   GRCh38                Benign           292   \n1213284   GRCh37  Benign/Likely benign           410   \n1213285   GRCh38  Benign/Likely benign           410   \n1213332   GRCh37         Likely benign            28   \n1213333   GRCh38         Likely benign            28   \n\n                                                  OtherIDs AlleleChange  \\\n90       ClinGen:CA113848,UniProtKB:P20062#VAR_001640,O...          G/C   \n91       ClinGen:CA113848,UniProtKB:P20062#VAR_001640,O...          G/C   \n244      ClinGen:CA114081,UniProtKB:P00387#VAR_010751,O...          C/G   \n245      ClinGen:CA114081,UniProtKB:P00387#VAR_010751,O...          C/G   \n592      ClinGen:CA114444,UniProtKB:P04066#VAR_002444,O...          A/G   \n...                                                    ...          ...   \n1213283                                                  -          G/A   \n1213284                                                  -          C/T   \n1213285                                                  -          C/T   \n1213332                                                  -          A/T   \n1213333                                                  -          A/T   \n\n        Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n90             Arg259Pro                         Arg                259   \n91             Arg259Pro                         Arg                259   \n244            Thr117Ser                         Thr                117   \n245            Thr117Ser                         Thr                117   \n592            Gln286Arg                         Gln                286   \n...                  ...                         ...                ...   \n1213283         Glu98Lys                         Glu                 98   \n1213284        Thr137Met                         Thr                137   \n1213285        Thr137Met                         Thr                137   \n1213332         Met10Leu                         Met                 10   \n1213333         Met10Leu                         Met                 10   \n\n        Amino_acids_after_3letters        NM_ID UniProtKB-AC UniProtKB-ID  \\\n90                             Pro  NM_000355.4       P20062   TCO2_HUMAN   \n91                             Pro  NM_000355.4       P20062   TCO2_HUMAN   \n244                            Ser  NM_000398.7       P00387  NB5R3_HUMAN   \n245                            Ser  NM_000398.7       P00387  NB5R3_HUMAN   \n592                            Arg  NM_000147.5       P04066   FUCO_HUMAN   \n...                            ...          ...          ...          ...   \n1213283                        Lys  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n1213284                        Met  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n1213285                        Met  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n1213332                        Leu  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n1213333                        Leu  NM_000029.4       B0ZBE2   ANGT_HUMAN   \n\n         stars  \n90           2  \n91           2  \n244          2  \n245          2  \n592          2  \n...        ...  \n1213283      1  \n1213284      2  \n1213285      2  \n1213332      1  \n1213333      1  \n\n[53504 rows x 16 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>UniProtKB-ID</th>\n      <th>stars</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>90</th>\n      <td>NM_000355.4(TCN2):c.776G&gt;C (p.Arg259Pro)</td>\n      <td>6948</td>\n      <td>TCN2</td>\n      <td>GRCh37</td>\n      <td>Benign</td>\n      <td>776</td>\n      <td>ClinGen:CA113848,UniProtKB:P20062#VAR_001640,O...</td>\n      <td>G/C</td>\n      <td>Arg259Pro</td>\n      <td>Arg</td>\n      <td>259</td>\n      <td>Pro</td>\n      <td>NM_000355.4</td>\n      <td>P20062</td>\n      <td>TCO2_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>91</th>\n      <td>NM_000355.4(TCN2):c.776G&gt;C (p.Arg259Pro)</td>\n      <td>6948</td>\n      <td>TCN2</td>\n      <td>GRCh38</td>\n      <td>Benign</td>\n      <td>776</td>\n      <td>ClinGen:CA113848,UniProtKB:P20062#VAR_001640,O...</td>\n      <td>G/C</td>\n      <td>Arg259Pro</td>\n      <td>Arg</td>\n      <td>259</td>\n      <td>Pro</td>\n      <td>NM_000355.4</td>\n      <td>P20062</td>\n      <td>TCO2_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>244</th>\n      <td>NM_000398.7(CYB5R3):c.350C&gt;G (p.Thr117Ser)</td>\n      <td>1727</td>\n      <td>CYB5R3</td>\n      <td>GRCh37</td>\n      <td>Benign</td>\n      <td>350</td>\n      <td>ClinGen:CA114081,UniProtKB:P00387#VAR_010751,O...</td>\n      <td>C/G</td>\n      <td>Thr117Ser</td>\n      <td>Thr</td>\n      <td>117</td>\n      <td>Ser</td>\n      <td>NM_000398.7</td>\n      <td>P00387</td>\n      <td>NB5R3_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>245</th>\n      <td>NM_000398.7(CYB5R3):c.350C&gt;G (p.Thr117Ser)</td>\n      <td>1727</td>\n      <td>CYB5R3</td>\n      <td>GRCh38</td>\n      <td>Benign</td>\n      <td>350</td>\n      <td>ClinGen:CA114081,UniProtKB:P00387#VAR_010751,O...</td>\n      <td>C/G</td>\n      <td>Thr117Ser</td>\n      <td>Thr</td>\n      <td>117</td>\n      <td>Ser</td>\n      <td>NM_000398.7</td>\n      <td>P00387</td>\n      <td>NB5R3_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>592</th>\n      <td>NM_000147.5(FUCA1):c.857A&gt;G (p.Gln286Arg)</td>\n      <td>2517</td>\n      <td>FUCA1</td>\n      <td>GRCh37</td>\n      <td>Benign</td>\n      <td>857</td>\n      <td>ClinGen:CA114444,UniProtKB:P04066#VAR_002444,O...</td>\n      <td>A/G</td>\n      <td>Gln286Arg</td>\n      <td>Gln</td>\n      <td>286</td>\n      <td>Arg</td>\n      <td>NM_000147.5</td>\n      <td>P04066</td>\n      <td>FUCO_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>1213283</th>\n      <td>NM_000029.4(AGT):c.292G&gt;A (p.Glu98Lys)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Benign</td>\n      <td>292</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Glu98Lys</td>\n      <td>Glu</td>\n      <td>98</td>\n      <td>Lys</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213284</th>\n      <td>NM_000029.4(AGT):c.410C&gt;T (p.Thr137Met)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Benign/Likely benign</td>\n      <td>410</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Thr137Met</td>\n      <td>Thr</td>\n      <td>137</td>\n      <td>Met</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213285</th>\n      <td>NM_000029.4(AGT):c.410C&gt;T (p.Thr137Met)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Benign/Likely benign</td>\n      <td>410</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Thr137Met</td>\n      <td>Thr</td>\n      <td>137</td>\n      <td>Met</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213332</th>\n      <td>NM_000029.4(AGT):c.28A&gt;T (p.Met10Leu)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Likely benign</td>\n      <td>28</td>\n      <td>-</td>\n      <td>A/T</td>\n      <td>Met10Leu</td>\n      <td>Met</td>\n      <td>10</td>\n      <td>Leu</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213333</th>\n      <td>NM_000029.4(AGT):c.28A&gt;T (p.Met10Leu)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Likely benign</td>\n      <td>28</td>\n      <td>-</td>\n      <td>A/T</td>\n      <td>Met10Leu</td>\n      <td>Met</td>\n      <td>10</td>\n      <td>Leu</td>\n      <td>NM_000029.4</td>\n      <td>B0ZBE2</td>\n      <td>ANGT_HUMAN</td>\n      <td>1</td>\n    </tr>\n  </tbody>\n</table>\n<p>53504 rows × 16 columns</p>\n</div>"
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# now benign ones\n",
    "# find benign\n",
    "#df_benign_new = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance'].str.contains('Benign'))]\n",
    "df_benign_new = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance'].str.contains('(Benign|Likely benign)'))]\n",
    "df_benign_new.drop_duplicates(inplace=True)\n",
    "df_benign_new"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "outputs": [
    {
     "data": {
      "text/plain": "                                              Name  GeneID GeneSymbol  \\\n0         NM_000016.6(ACADM):c.261G>A (p.Met87Ile)      34      ACADM   \n1        NM_000016.6(ACADM):c.477T>G (p.Cys159Trp)      34      ACADM   \n2       NM_000017.4(ACADS):c.1198C>G (p.Leu400Val)      35      ACADS   \n3       NM_000017.4(ACADS):c.1210G>A (p.Gly404Arg)      35      ACADS   \n4        NM_000017.4(ACADS):c.625G>A (p.Gly209Ser)      35      ACADS   \n...                                            ...     ...        ...   \n26747   NM_213720.3(CHCHD10):c.268A>C (p.Thr90Pro)  400916    CHCHD10   \n26748   NM_213720.3(CHCHD10):c.274G>C (p.Ala92Pro)  400916    CHCHD10   \n26749   NM_213720.3(CHCHD10):c.284A>C (p.Gln95Pro)  400916    CHCHD10   \n26750   NM_213720.3(CHCHD10):c.286C>A (p.Pro96Thr)  400916    CHCHD10   \n26751  NM_213720.3(CHCHD10):c.403T>C (p.Tyr135His)  400916    CHCHD10   \n\n       ClinicalSignificance  CDS_position  \\\n0             Likely benign           261   \n1                    Benign           477   \n2             Likely benign          1198   \n3             Likely benign          1210   \n4      Benign/Likely benign           625   \n...                     ...           ...   \n26747  Benign/Likely benign           268   \n26748  Benign/Likely benign           274   \n26749         Likely benign           284   \n26750                Benign           286   \n26751         Likely benign           403   \n\n                                                OtherIDs AlleleChange  \\\n0                                     ClinGen:CA16603755          G/A   \n1                                     ClinGen:CA10586744          T/G   \n2                                      ClinGen:CA6831239          C/G   \n3                                                      -          G/A   \n4      ClinGen:CA145599,UniProtKB:P16219#VAR_000315,O...          G/A   \n...                                                  ...          ...   \n26747                                                  -          A/C   \n26748                                                  -          G/C   \n26749                                                  -          A/C   \n26750                                 ClinGen:CA10145267          C/A   \n26751                                 ClinGen:CA10145244          T/C   \n\n      Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n0             Met87Ile                         Met                 87   \n1            Cys159Trp                         Cys                159   \n2            Leu400Val                         Leu                400   \n3            Gly404Arg                         Gly                404   \n4            Gly209Ser                         Gly                209   \n...                ...                         ...                ...   \n26747         Thr90Pro                         Thr                 90   \n26748         Ala92Pro                         Ala                 92   \n26749         Gln95Pro                         Gln                 95   \n26750         Pro96Thr                         Pro                 96   \n26751        Tyr135His                         Tyr                135   \n\n      Amino_acids_after_3letters        NM_ID UniProtKB-AC UniProtKB-ID  \\\n0                            Ile  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n1                            Trp  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n2                            Val  NM_000017.4       E5KSD5  ACADS_HUMAN   \n3                            Arg  NM_000017.4       E5KSD5  ACADS_HUMAN   \n4                            Ser  NM_000017.4       E5KSD5  ACADS_HUMAN   \n...                          ...          ...          ...          ...   \n26747                        Pro  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n26748                        Pro  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n26749                        Pro  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n26750                        Thr  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n26751                        His  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n\n       stars          Assembly  \n0          1  [GRCh37, GRCh38]  \n1          1  [GRCh37, GRCh38]  \n2          1  [GRCh37, GRCh38]  \n3          1  [GRCh37, GRCh38]  \n4          2  [GRCh37, GRCh38]  \n...      ...               ...  \n26747      2  [GRCh37, GRCh38]  \n26748      2  [GRCh37, GRCh38]  \n26749      1  [GRCh37, GRCh38]  \n26750      2  [GRCh37, GRCh38]  \n26751      2  [GRCh37, GRCh38]  \n\n[26752 rows x 16 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>UniProtKB-ID</th>\n      <th>stars</th>\n      <th>Assembly</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NM_000016.6(ACADM):c.261G&gt;A (p.Met87Ile)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Likely benign</td>\n      <td>261</td>\n      <td>ClinGen:CA16603755</td>\n      <td>G/A</td>\n      <td>Met87Ile</td>\n      <td>Met</td>\n      <td>87</td>\n      <td>Ile</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NM_000016.6(ACADM):c.477T&gt;G (p.Cys159Trp)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Benign</td>\n      <td>477</td>\n      <td>ClinGen:CA10586744</td>\n      <td>T/G</td>\n      <td>Cys159Trp</td>\n      <td>Cys</td>\n      <td>159</td>\n      <td>Trp</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NM_000017.4(ACADS):c.1198C&gt;G (p.Leu400Val)</td>\n      <td>35</td>\n      <td>ACADS</td>\n      <td>Likely benign</td>\n      <td>1198</td>\n      <td>ClinGen:CA6831239</td>\n      <td>C/G</td>\n      <td>Leu400Val</td>\n      <td>Leu</td>\n      <td>400</td>\n      <td>Val</td>\n      <td>NM_000017.4</td>\n      <td>E5KSD5</td>\n      <td>ACADS_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NM_000017.4(ACADS):c.1210G&gt;A (p.Gly404Arg)</td>\n      <td>35</td>\n      <td>ACADS</td>\n      <td>Likely benign</td>\n      <td>1210</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Gly404Arg</td>\n      <td>Gly</td>\n      <td>404</td>\n      <td>Arg</td>\n      <td>NM_000017.4</td>\n      <td>E5KSD5</td>\n      <td>ACADS_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NM_000017.4(ACADS):c.625G&gt;A (p.Gly209Ser)</td>\n      <td>35</td>\n      <td>ACADS</td>\n      <td>Benign/Likely benign</td>\n      <td>625</td>\n      <td>ClinGen:CA145599,UniProtKB:P16219#VAR_000315,O...</td>\n      <td>G/A</td>\n      <td>Gly209Ser</td>\n      <td>Gly</td>\n      <td>209</td>\n      <td>Ser</td>\n      <td>NM_000017.4</td>\n      <td>E5KSD5</td>\n      <td>ACADS_HUMAN</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>26747</th>\n      <td>NM_213720.3(CHCHD10):c.268A&gt;C (p.Thr90Pro)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Benign/Likely benign</td>\n      <td>268</td>\n      <td>-</td>\n      <td>A/C</td>\n      <td>Thr90Pro</td>\n      <td>Thr</td>\n      <td>90</td>\n      <td>Pro</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>26748</th>\n      <td>NM_213720.3(CHCHD10):c.274G&gt;C (p.Ala92Pro)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Benign/Likely benign</td>\n      <td>274</td>\n      <td>-</td>\n      <td>G/C</td>\n      <td>Ala92Pro</td>\n      <td>Ala</td>\n      <td>92</td>\n      <td>Pro</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>26749</th>\n      <td>NM_213720.3(CHCHD10):c.284A&gt;C (p.Gln95Pro)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Likely benign</td>\n      <td>284</td>\n      <td>-</td>\n      <td>A/C</td>\n      <td>Gln95Pro</td>\n      <td>Gln</td>\n      <td>95</td>\n      <td>Pro</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>26750</th>\n      <td>NM_213720.3(CHCHD10):c.286C&gt;A (p.Pro96Thr)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Benign</td>\n      <td>286</td>\n      <td>ClinGen:CA10145267</td>\n      <td>C/A</td>\n      <td>Pro96Thr</td>\n      <td>Pro</td>\n      <td>96</td>\n      <td>Thr</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>26751</th>\n      <td>NM_213720.3(CHCHD10):c.403T&gt;C (p.Tyr135His)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Likely benign</td>\n      <td>403</td>\n      <td>ClinGen:CA10145244</td>\n      <td>T/C</td>\n      <td>Tyr135His</td>\n      <td>Tyr</td>\n      <td>135</td>\n      <td>His</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n  </tbody>\n</table>\n<p>26752 rows × 16 columns</p>\n</div>"
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b_new = df_benign_new.groupby(col_list).agg({'Assembly': pd.Series.unique}).reset_index()\n",
    "b_new"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "outputs": [
    {
     "data": {
      "text/plain": "array(['Likely benign', 'Benign', 'Benign/Likely benign',\n       'Benign/Likely benign; other', 'Benign; other',\n       'Benign/Likely benign; risk factor',\n       'Benign/Likely benign; other; risk factor',\n       'Benign; drug response', 'Benign/Likely benign; association'],\n      dtype=object)"
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b_new['ClinicalSignificance'].unique()"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2517\n",
      "2576\n",
      "2511\n"
     ]
    }
   ],
   "source": [
    "print(len(b_new['UniProtKB-AC'].unique()))\n",
    "print(len(b_new['GeneSymbol'].unique()))\n",
    "print(len(b_new['GeneID'].unique()))"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "outputs": [
    {
     "data": {
      "text/plain": "                                              Name  GeneID GeneSymbol  \\\n0       NM_000016.6(ACADM):c.1010A>C (p.Tyr337Ser)      34      ACADM   \n1       NM_000016.6(ACADM):c.1034A>T (p.Asp345Val)      34      ACADM   \n2       NM_000016.6(ACADM):c.1042C>T (p.Arg348Cys)      34      ACADM   \n3       NM_000016.6(ACADM):c.1045C>G (p.Arg349Gly)      34      ACADM   \n4       NM_000016.6(ACADM):c.1046G>A (p.Arg349Gln)      34      ACADM   \n...                                            ...     ...        ...   \n26747   NM_213720.3(CHCHD10):c.268A>C (p.Thr90Pro)  400916    CHCHD10   \n26748   NM_213720.3(CHCHD10):c.274G>C (p.Ala92Pro)  400916    CHCHD10   \n26749   NM_213720.3(CHCHD10):c.284A>C (p.Gln95Pro)  400916    CHCHD10   \n26750   NM_213720.3(CHCHD10):c.286C>A (p.Pro96Thr)  400916    CHCHD10   \n26751  NM_213720.3(CHCHD10):c.403T>C (p.Tyr135His)  400916    CHCHD10   \n\n       ClinicalSignificance  CDS_position            OtherIDs AlleleChange  \\\n0         Likely pathogenic          1010                   -          A/C   \n1                Pathogenic          1034    ClinGen:CA913267          A/T   \n2                Pathogenic          1042                   -          C/T   \n3                Pathogenic          1045                   -          C/G   \n4                Pathogenic          1046                   -          G/A   \n...                     ...           ...                 ...          ...   \n26747  Benign/Likely benign           268                   -          A/C   \n26748  Benign/Likely benign           274                   -          G/C   \n26749         Likely benign           284                   -          A/C   \n26750                Benign           286  ClinGen:CA10145267          C/A   \n26751         Likely benign           403  ClinGen:CA10145244          T/C   \n\n      Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n0            Tyr337Ser                         Tyr                337   \n1            Asp345Val                         Asp                345   \n2            Arg348Cys                         Arg                348   \n3            Arg349Gly                         Arg                349   \n4            Arg349Gln                         Arg                349   \n...                ...                         ...                ...   \n26747         Thr90Pro                         Thr                 90   \n26748         Ala92Pro                         Ala                 92   \n26749         Gln95Pro                         Gln                 95   \n26750         Pro96Thr                         Pro                 96   \n26751        Tyr135His                         Tyr                135   \n\n      Amino_acids_after_3letters        NM_ID UniProtKB-AC UniProtKB-ID  \\\n0                            Ser  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n1                            Val  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n2                            Cys  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n3                            Gly  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n4                            Gln  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n...                          ...          ...          ...          ...   \n26747                        Pro  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n26748                        Pro  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n26749                        Pro  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n26750                        Thr  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n26751                        His  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n\n       stars          Assembly  \n0          2  [GRCh37, GRCh38]  \n1          1  [GRCh37, GRCh38]  \n2          1  [GRCh37, GRCh38]  \n3          1  [GRCh37, GRCh38]  \n4          1  [GRCh37, GRCh38]  \n...      ...               ...  \n26747      2  [GRCh37, GRCh38]  \n26748      2  [GRCh37, GRCh38]  \n26749      1  [GRCh37, GRCh38]  \n26750      2  [GRCh37, GRCh38]  \n26751      2  [GRCh37, GRCh38]  \n\n[56514 rows x 16 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>UniProtKB-ID</th>\n      <th>stars</th>\n      <th>Assembly</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NM_000016.6(ACADM):c.1010A&gt;C (p.Tyr337Ser)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Likely pathogenic</td>\n      <td>1010</td>\n      <td>-</td>\n      <td>A/C</td>\n      <td>Tyr337Ser</td>\n      <td>Tyr</td>\n      <td>337</td>\n      <td>Ser</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NM_000016.6(ACADM):c.1034A&gt;T (p.Asp345Val)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1034</td>\n      <td>ClinGen:CA913267</td>\n      <td>A/T</td>\n      <td>Asp345Val</td>\n      <td>Asp</td>\n      <td>345</td>\n      <td>Val</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NM_000016.6(ACADM):c.1042C&gt;T (p.Arg348Cys)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1042</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Arg348Cys</td>\n      <td>Arg</td>\n      <td>348</td>\n      <td>Cys</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NM_000016.6(ACADM):c.1045C&gt;G (p.Arg349Gly)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1045</td>\n      <td>-</td>\n      <td>C/G</td>\n      <td>Arg349Gly</td>\n      <td>Arg</td>\n      <td>349</td>\n      <td>Gly</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NM_000016.6(ACADM):c.1046G&gt;A (p.Arg349Gln)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1046</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Arg349Gln</td>\n      <td>Arg</td>\n      <td>349</td>\n      <td>Gln</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>26747</th>\n      <td>NM_213720.3(CHCHD10):c.268A&gt;C (p.Thr90Pro)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Benign/Likely benign</td>\n      <td>268</td>\n      <td>-</td>\n      <td>A/C</td>\n      <td>Thr90Pro</td>\n      <td>Thr</td>\n      <td>90</td>\n      <td>Pro</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>26748</th>\n      <td>NM_213720.3(CHCHD10):c.274G&gt;C (p.Ala92Pro)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Benign/Likely benign</td>\n      <td>274</td>\n      <td>-</td>\n      <td>G/C</td>\n      <td>Ala92Pro</td>\n      <td>Ala</td>\n      <td>92</td>\n      <td>Pro</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>26749</th>\n      <td>NM_213720.3(CHCHD10):c.284A&gt;C (p.Gln95Pro)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Likely benign</td>\n      <td>284</td>\n      <td>-</td>\n      <td>A/C</td>\n      <td>Gln95Pro</td>\n      <td>Gln</td>\n      <td>95</td>\n      <td>Pro</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>26750</th>\n      <td>NM_213720.3(CHCHD10):c.286C&gt;A (p.Pro96Thr)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Benign</td>\n      <td>286</td>\n      <td>ClinGen:CA10145267</td>\n      <td>C/A</td>\n      <td>Pro96Thr</td>\n      <td>Pro</td>\n      <td>96</td>\n      <td>Thr</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>26751</th>\n      <td>NM_213720.3(CHCHD10):c.403T&gt;C (p.Tyr135His)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Likely benign</td>\n      <td>403</td>\n      <td>ClinGen:CA10145244</td>\n      <td>T/C</td>\n      <td>Tyr135His</td>\n      <td>Tyr</td>\n      <td>135</td>\n      <td>His</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n  </tbody>\n</table>\n<p>56514 rows × 16 columns</p>\n</div>"
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variant_selected = pd.concat([d_new, b_new])\n",
    "variant_selected"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "outputs": [
    {
     "data": {
      "text/plain": "                                              Name  GeneID GeneSymbol  \\\n0       NM_000016.6(ACADM):c.1010A>C (p.Tyr337Ser)      34      ACADM   \n1       NM_000016.6(ACADM):c.1034A>T (p.Asp345Val)      34      ACADM   \n2       NM_000016.6(ACADM):c.1042C>T (p.Arg348Cys)      34      ACADM   \n3       NM_000016.6(ACADM):c.1045C>G (p.Arg349Gly)      34      ACADM   \n4       NM_000016.6(ACADM):c.1046G>A (p.Arg349Gln)      34      ACADM   \n...                                            ...     ...        ...   \n26747   NM_213720.3(CHCHD10):c.268A>C (p.Thr90Pro)  400916    CHCHD10   \n26748   NM_213720.3(CHCHD10):c.274G>C (p.Ala92Pro)  400916    CHCHD10   \n26749   NM_213720.3(CHCHD10):c.284A>C (p.Gln95Pro)  400916    CHCHD10   \n26750   NM_213720.3(CHCHD10):c.286C>A (p.Pro96Thr)  400916    CHCHD10   \n26751  NM_213720.3(CHCHD10):c.403T>C (p.Tyr135His)  400916    CHCHD10   \n\n       ClinicalSignificance  CDS_position            OtherIDs AlleleChange  \\\n0         Likely pathogenic          1010                   -          A/C   \n1                Pathogenic          1034    ClinGen:CA913267          A/T   \n2                Pathogenic          1042                   -          C/T   \n3                Pathogenic          1045                   -          C/G   \n4                Pathogenic          1046                   -          G/A   \n...                     ...           ...                 ...          ...   \n26747  Benign/Likely benign           268                   -          A/C   \n26748  Benign/Likely benign           274                   -          G/C   \n26749         Likely benign           284                   -          A/C   \n26750                Benign           286  ClinGen:CA10145267          C/A   \n26751         Likely benign           403  ClinGen:CA10145244          T/C   \n\n      Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n0            Tyr337Ser                         Tyr                337   \n1            Asp345Val                         Asp                345   \n2            Arg348Cys                         Arg                348   \n3            Arg349Gly                         Arg                349   \n4            Arg349Gln                         Arg                349   \n...                ...                         ...                ...   \n26747         Thr90Pro                         Thr                 90   \n26748         Ala92Pro                         Ala                 92   \n26749         Gln95Pro                         Gln                 95   \n26750         Pro96Thr                         Pro                 96   \n26751        Tyr135His                         Tyr                135   \n\n      Amino_acids_after_3letters        NM_ID UniProtKB-AC UniProtKB-ID  \\\n0                            Ser  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n1                            Val  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n2                            Cys  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n3                            Gly  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n4                            Gln  NM_000016.6   A0A0S2Z366  ACADM_HUMAN   \n...                          ...          ...          ...          ...   \n26747                        Pro  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n26748                        Pro  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n26749                        Pro  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n26750                        Thr  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n26751                        His  NM_213720.3       Q8WYQ3  CHC10_HUMAN   \n\n       stars     classes  \n0          2  Pathogenic  \n1          1  Pathogenic  \n2          1  Pathogenic  \n3          1  Pathogenic  \n4          1  Pathogenic  \n...      ...         ...  \n26747      2      Benign  \n26748      2      Benign  \n26749      1      Benign  \n26750      2      Benign  \n26751      2      Benign  \n\n[56514 rows x 16 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>UniProtKB-ID</th>\n      <th>stars</th>\n      <th>classes</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NM_000016.6(ACADM):c.1010A&gt;C (p.Tyr337Ser)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Likely pathogenic</td>\n      <td>1010</td>\n      <td>-</td>\n      <td>A/C</td>\n      <td>Tyr337Ser</td>\n      <td>Tyr</td>\n      <td>337</td>\n      <td>Ser</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>2</td>\n      <td>Pathogenic</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NM_000016.6(ACADM):c.1034A&gt;T (p.Asp345Val)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1034</td>\n      <td>ClinGen:CA913267</td>\n      <td>A/T</td>\n      <td>Asp345Val</td>\n      <td>Asp</td>\n      <td>345</td>\n      <td>Val</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>Pathogenic</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NM_000016.6(ACADM):c.1042C&gt;T (p.Arg348Cys)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1042</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Arg348Cys</td>\n      <td>Arg</td>\n      <td>348</td>\n      <td>Cys</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>Pathogenic</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NM_000016.6(ACADM):c.1045C&gt;G (p.Arg349Gly)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1045</td>\n      <td>-</td>\n      <td>C/G</td>\n      <td>Arg349Gly</td>\n      <td>Arg</td>\n      <td>349</td>\n      <td>Gly</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>Pathogenic</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NM_000016.6(ACADM):c.1046G&gt;A (p.Arg349Gln)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1046</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Arg349Gln</td>\n      <td>Arg</td>\n      <td>349</td>\n      <td>Gln</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>ACADM_HUMAN</td>\n      <td>1</td>\n      <td>Pathogenic</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>26747</th>\n      <td>NM_213720.3(CHCHD10):c.268A&gt;C (p.Thr90Pro)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Benign/Likely benign</td>\n      <td>268</td>\n      <td>-</td>\n      <td>A/C</td>\n      <td>Thr90Pro</td>\n      <td>Thr</td>\n      <td>90</td>\n      <td>Pro</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>2</td>\n      <td>Benign</td>\n    </tr>\n    <tr>\n      <th>26748</th>\n      <td>NM_213720.3(CHCHD10):c.274G&gt;C (p.Ala92Pro)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Benign/Likely benign</td>\n      <td>274</td>\n      <td>-</td>\n      <td>G/C</td>\n      <td>Ala92Pro</td>\n      <td>Ala</td>\n      <td>92</td>\n      <td>Pro</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>2</td>\n      <td>Benign</td>\n    </tr>\n    <tr>\n      <th>26749</th>\n      <td>NM_213720.3(CHCHD10):c.284A&gt;C (p.Gln95Pro)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Likely benign</td>\n      <td>284</td>\n      <td>-</td>\n      <td>A/C</td>\n      <td>Gln95Pro</td>\n      <td>Gln</td>\n      <td>95</td>\n      <td>Pro</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>1</td>\n      <td>Benign</td>\n    </tr>\n    <tr>\n      <th>26750</th>\n      <td>NM_213720.3(CHCHD10):c.286C&gt;A (p.Pro96Thr)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Benign</td>\n      <td>286</td>\n      <td>ClinGen:CA10145267</td>\n      <td>C/A</td>\n      <td>Pro96Thr</td>\n      <td>Pro</td>\n      <td>96</td>\n      <td>Thr</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>2</td>\n      <td>Benign</td>\n    </tr>\n    <tr>\n      <th>26751</th>\n      <td>NM_213720.3(CHCHD10):c.403T&gt;C (p.Tyr135His)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Likely benign</td>\n      <td>403</td>\n      <td>ClinGen:CA10145244</td>\n      <td>T/C</td>\n      <td>Tyr135His</td>\n      <td>Tyr</td>\n      <td>135</td>\n      <td>His</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>CHC10_HUMAN</td>\n      <td>2</td>\n      <td>Benign</td>\n    </tr>\n  </tbody>\n</table>\n<p>56514 rows × 16 columns</p>\n</div>"
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Reassign class\n",
    "def assign_class(row):\n",
    "    if 'Benign/Likely benign' in row:\n",
    "        return 'Benign'\n",
    "    elif 'Pathogenic/Likely pathogenic' in row:\n",
    "        return 'Pathogenic'\n",
    "    elif 'Benign' in row:\n",
    "        return 'Benign'\n",
    "    elif 'Likely benign' in row:\n",
    "        return 'Benign'\n",
    "    elif 'Pathogenic' in row:\n",
    "        return 'Pathogenic'\n",
    "    elif 'Likely pathogenic' in row:\n",
    "        return 'Pathogenic'\n",
    "    else:\n",
    "        raise KeyError(\"You don't belong here\")\n",
    "variant_selected['classes'] = variant_selected['ClinicalSignificance'].apply(assign_class)\n",
    "# pandas can handle NaNs as integers now\n",
    "variant_selected.drop(['Assembly'], axis=1, inplace=True)\n",
    "#variant_selected = variant_selected[variant_selected['classes'].notnull()]\n",
    "variant_selected"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "outputs": [],
   "source": [
    "# make sure only two classes are kept\n",
    "assert len(variant_selected['classes'].unique()) == 2"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2715\n",
      "2828\n",
      "2717\n"
     ]
    }
   ],
   "source": [
    "print(len(variant_selected['UniProtKB-AC'].unique()))\n",
    "print(len(variant_selected['GeneSymbol'].unique()))\n",
    "print(len(variant_selected['GeneID'].unique()))"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "outputs": [],
   "source": [
    "variant_selected.to_csv('./data/metas/clinvar_eve_overlapped_genes.csv', index=False)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "\"\"\"The method below didn't work\"\"\""
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/pbs.498551.hn-10-03/ipykernel_42587/3210608625.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_pathogenic.drop_duplicates(inplace=True)\n"
     ]
    },
    {
     "data": {
      "text/plain": "                                                   Name  GeneID GeneSymbol  \\\n24              NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n25              NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n36             NM_138413.4(HOGA1):c.209G>C (p.Arg70Pro)  112817      HOGA1   \n37             NM_138413.4(HOGA1):c.209G>C (p.Arg70Pro)  112817      HOGA1   \n48       NM_001199107.2(TBC1D24):c.439G>C (p.Asp147His)   57465    TBC1D24   \n...                                                 ...     ...        ...   \n1213199        NM_021870.2(FGG):c.1007T>C (p.Met336Thr)    2266        FGG   \n1213224     NM_000342.3(SLC4A1):c.1462G>A (p.Val488Met)    6521     SLC4A1   \n1213225     NM_000342.3(SLC4A1):c.1462G>A (p.Val488Met)    6521     SLC4A1   \n1213242        NM_000029.4(AGT):c.1124G>A (p.Arg375Gln)     183        AGT   \n1213243        NM_000029.4(AGT):c.1124G>A (p.Arg375Gln)     183        AGT   \n\n        Assembly ClinicalSignificance  CDS_position  \\\n24        GRCh37           Pathogenic           848   \n25        GRCh38           Pathogenic           848   \n36        GRCh37           Pathogenic           209   \n37        GRCh38           Pathogenic           209   \n48        GRCh37           Pathogenic           439   \n...          ...                  ...           ...   \n1213199   GRCh38           Pathogenic          1007   \n1213224   GRCh37           Pathogenic          1462   \n1213225   GRCh38           Pathogenic          1462   \n1213242   GRCh37           Pathogenic          1124   \n1213243   GRCh38           Pathogenic          1124   \n\n                                                  OtherIDs AlleleChange  \\\n24       ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...          A/C   \n25       ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...          A/C   \n36                       ClinGen:CA113816,OMIM:613597.0005          G/C   \n37                       ClinGen:CA113816,OMIM:613597.0005          G/C   \n48       ClinGen:CA339785,UniProtKB:Q9ULP9#VAR_064365,O...          G/C   \n...                                                    ...          ...   \n1213199  ClinGen:CA126399,UniProtKB:P02679#VAR_002414,O...          T/C   \n1213224  ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...          G/A   \n1213225  ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...          G/A   \n1213242  ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...          G/A   \n1213243  ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...          G/A   \n\n        Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n24             Gln283Pro                         Gln                283   \n25             Gln283Pro                         Gln                283   \n36              Arg70Pro                         Arg                 70   \n37              Arg70Pro                         Arg                 70   \n48             Asp147His                         Asp                147   \n...                  ...                         ...                ...   \n1213199        Met336Thr                         Met                336   \n1213224        Val488Met                         Val                488   \n1213225        Val488Met                         Val                488   \n1213242        Arg375Gln                         Arg                375   \n1213243        Arg375Gln                         Arg                375   \n\n        Amino_acids_after_3letters           NM_ID UniProtKB-AC  stars  \n24                             Pro     NM_000410.4       Q30201      1  \n25                             Pro     NM_000410.4       Q30201      1  \n36                             Pro     NM_138413.4       Q86XE5      1  \n37                             Pro     NM_138413.4       Q86XE5      1  \n48                             His  NM_001199107.2       Q9ULP9      1  \n...                            ...             ...          ...    ...  \n1213199                        Thr     NM_021870.2       P02679      1  \n1213224                        Met     NM_000342.3       P02730      2  \n1213225                        Met     NM_000342.3       P02730      2  \n1213242                        Gln     NM_000029.4       P01019      1  \n1213243                        Gln     NM_000029.4       P01019      1  \n\n[22951 rows x 15 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>stars</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>24</th>\n      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>848</td>\n      <td>ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...</td>\n      <td>A/C</td>\n      <td>Gln283Pro</td>\n      <td>Gln</td>\n      <td>283</td>\n      <td>Pro</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>25</th>\n      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>848</td>\n      <td>ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...</td>\n      <td>A/C</td>\n      <td>Gln283Pro</td>\n      <td>Gln</td>\n      <td>283</td>\n      <td>Pro</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>36</th>\n      <td>NM_138413.4(HOGA1):c.209G&gt;C (p.Arg70Pro)</td>\n      <td>112817</td>\n      <td>HOGA1</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>209</td>\n      <td>ClinGen:CA113816,OMIM:613597.0005</td>\n      <td>G/C</td>\n      <td>Arg70Pro</td>\n      <td>Arg</td>\n      <td>70</td>\n      <td>Pro</td>\n      <td>NM_138413.4</td>\n      <td>Q86XE5</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>37</th>\n      <td>NM_138413.4(HOGA1):c.209G&gt;C (p.Arg70Pro)</td>\n      <td>112817</td>\n      <td>HOGA1</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>209</td>\n      <td>ClinGen:CA113816,OMIM:613597.0005</td>\n      <td>G/C</td>\n      <td>Arg70Pro</td>\n      <td>Arg</td>\n      <td>70</td>\n      <td>Pro</td>\n      <td>NM_138413.4</td>\n      <td>Q86XE5</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>48</th>\n      <td>NM_001199107.2(TBC1D24):c.439G&gt;C (p.Asp147His)</td>\n      <td>57465</td>\n      <td>TBC1D24</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>439</td>\n      <td>ClinGen:CA339785,UniProtKB:Q9ULP9#VAR_064365,O...</td>\n      <td>G/C</td>\n      <td>Asp147His</td>\n      <td>Asp</td>\n      <td>147</td>\n      <td>His</td>\n      <td>NM_001199107.2</td>\n      <td>Q9ULP9</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>1213199</th>\n      <td>NM_021870.2(FGG):c.1007T&gt;C (p.Met336Thr)</td>\n      <td>2266</td>\n      <td>FGG</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>1007</td>\n      <td>ClinGen:CA126399,UniProtKB:P02679#VAR_002414,O...</td>\n      <td>T/C</td>\n      <td>Met336Thr</td>\n      <td>Met</td>\n      <td>336</td>\n      <td>Thr</td>\n      <td>NM_021870.2</td>\n      <td>P02679</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213224</th>\n      <td>NM_000342.3(SLC4A1):c.1462G&gt;A (p.Val488Met)</td>\n      <td>6521</td>\n      <td>SLC4A1</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>1462</td>\n      <td>ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...</td>\n      <td>G/A</td>\n      <td>Val488Met</td>\n      <td>Val</td>\n      <td>488</td>\n      <td>Met</td>\n      <td>NM_000342.3</td>\n      <td>P02730</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213225</th>\n      <td>NM_000342.3(SLC4A1):c.1462G&gt;A (p.Val488Met)</td>\n      <td>6521</td>\n      <td>SLC4A1</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>1462</td>\n      <td>ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...</td>\n      <td>G/A</td>\n      <td>Val488Met</td>\n      <td>Val</td>\n      <td>488</td>\n      <td>Met</td>\n      <td>NM_000342.3</td>\n      <td>P02730</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213242</th>\n      <td>NM_000029.4(AGT):c.1124G&gt;A (p.Arg375Gln)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>1124</td>\n      <td>ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...</td>\n      <td>G/A</td>\n      <td>Arg375Gln</td>\n      <td>Arg</td>\n      <td>375</td>\n      <td>Gln</td>\n      <td>NM_000029.4</td>\n      <td>P01019</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213243</th>\n      <td>NM_000029.4(AGT):c.1124G&gt;A (p.Arg375Gln)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>1124</td>\n      <td>ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...</td>\n      <td>G/A</td>\n      <td>Arg375Gln</td>\n      <td>Arg</td>\n      <td>375</td>\n      <td>Gln</td>\n      <td>NM_000029.4</td>\n      <td>P01019</td>\n      <td>1</td>\n    </tr>\n  </tbody>\n</table>\n<p>22951 rows × 15 columns</p>\n</div>"
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_pathogenic = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance']=='Pathogenic')]\n",
    "df_pathogenic.drop_duplicates(inplace=True)\n",
    "df_pathogenic"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "outputs": [
    {
     "data": {
      "text/plain": "                                             Name  GeneID GeneSymbol  \\\n0      NM_000016.6(ACADM):c.1034A>T (p.Asp345Val)      34      ACADM   \n1      NM_000016.6(ACADM):c.1042C>T (p.Arg348Cys)      34      ACADM   \n2      NM_000016.6(ACADM):c.1045C>G (p.Arg349Gly)      34      ACADM   \n3      NM_000016.6(ACADM):c.1046G>A (p.Arg349Gln)      34      ACADM   \n4      NM_000016.6(ACADM):c.1067T>C (p.Ile356Thr)      34      ACADM   \n...                                           ...     ...        ...   \n11486      NM_213653.4(HJV):c.238T>C (p.Cys80Arg)  148738        HJV   \n11487     NM_213653.4(HJV):c.302T>C (p.Leu101Pro)  148738        HJV   \n11488     NM_213653.4(HJV):c.842T>C (p.Ile281Thr)  148738        HJV   \n11489     NM_213653.4(HJV):c.959G>T (p.Gly320Val)  148738        HJV   \n11490   NM_213720.3(CHCHD10):c.44G>T (p.Arg15Leu)  400916    CHCHD10   \n\n      ClinicalSignificance  CDS_position  \\\n0               Pathogenic          1034   \n1               Pathogenic          1042   \n2               Pathogenic          1045   \n3               Pathogenic          1046   \n4               Pathogenic          1067   \n...                    ...           ...   \n11486           Pathogenic           238   \n11487           Pathogenic           302   \n11488           Pathogenic           842   \n11489           Pathogenic           959   \n11490           Pathogenic            44   \n\n                                                OtherIDs AlleleChange  \\\n0                                       ClinGen:CA913267          A/T   \n1                                                      -          C/T   \n2                                                      -          C/G   \n3                                                      -          G/A   \n4                                                      -          T/C   \n...                                                  ...          ...   \n11486  ClinGen:CA252255,UniProtKB:Q6ZVN8#VAR_019617,O...          T/C   \n11487  ClinGen:CA252256,UniProtKB:Q6ZVN8#VAR_019620,O...          T/C   \n11488                  ClinGen:CA252253,OMIM:608374.0004          T/C   \n11489  ClinGen:CA252249,UniProtKB:Q6ZVN8#VAR_019629,O...          G/T   \n11490                  ClinGen:CA235291,OMIM:615903.0002          G/T   \n\n      Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n0            Asp345Val                         Asp                345   \n1            Arg348Cys                         Arg                348   \n2            Arg349Gly                         Arg                349   \n3            Arg349Gln                         Arg                349   \n4            Ile356Thr                         Ile                356   \n...                ...                         ...                ...   \n11486         Cys80Arg                         Cys                 80   \n11487        Leu101Pro                         Leu                101   \n11488        Ile281Thr                         Ile                281   \n11489        Gly320Val                         Gly                320   \n11490         Arg15Leu                         Arg                 15   \n\n      Amino_acids_after_3letters        NM_ID UniProtKB-AC  stars  \\\n0                            Val  NM_000016.6   A0A0S2Z366      1   \n1                            Cys  NM_000016.6   A0A0S2Z366      1   \n2                            Gly  NM_000016.6   A0A0S2Z366      1   \n3                            Gln  NM_000016.6   A0A0S2Z366      1   \n4                            Thr  NM_000016.6   A0A0S2Z366      1   \n...                          ...          ...          ...    ...   \n11486                        Arg  NM_213653.4   A0A024R4F5      1   \n11487                        Pro  NM_213653.4   A0A024R4F5      1   \n11488                        Thr  NM_213653.4   A0A024R4F5      1   \n11489                        Val  NM_213653.4   A0A024R4F5      2   \n11490                        Leu  NM_213720.3       Q8WYQ3      2   \n\n               Assembly  \n0      [GRCh37, GRCh38]  \n1      [GRCh37, GRCh38]  \n2      [GRCh37, GRCh38]  \n3      [GRCh37, GRCh38]  \n4      [GRCh37, GRCh38]  \n...                 ...  \n11486  [GRCh37, GRCh38]  \n11487  [GRCh37, GRCh38]  \n11488  [GRCh37, GRCh38]  \n11489  [GRCh37, GRCh38]  \n11490  [GRCh37, GRCh38]  \n\n[11491 rows x 15 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>stars</th>\n      <th>Assembly</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NM_000016.6(ACADM):c.1034A&gt;T (p.Asp345Val)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1034</td>\n      <td>ClinGen:CA913267</td>\n      <td>A/T</td>\n      <td>Asp345Val</td>\n      <td>Asp</td>\n      <td>345</td>\n      <td>Val</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NM_000016.6(ACADM):c.1042C&gt;T (p.Arg348Cys)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1042</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Arg348Cys</td>\n      <td>Arg</td>\n      <td>348</td>\n      <td>Cys</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NM_000016.6(ACADM):c.1045C&gt;G (p.Arg349Gly)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1045</td>\n      <td>-</td>\n      <td>C/G</td>\n      <td>Arg349Gly</td>\n      <td>Arg</td>\n      <td>349</td>\n      <td>Gly</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NM_000016.6(ACADM):c.1046G&gt;A (p.Arg349Gln)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1046</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Arg349Gln</td>\n      <td>Arg</td>\n      <td>349</td>\n      <td>Gln</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NM_000016.6(ACADM):c.1067T&gt;C (p.Ile356Thr)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1067</td>\n      <td>-</td>\n      <td>T/C</td>\n      <td>Ile356Thr</td>\n      <td>Ile</td>\n      <td>356</td>\n      <td>Thr</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>11486</th>\n      <td>NM_213653.4(HJV):c.238T&gt;C (p.Cys80Arg)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>238</td>\n      <td>ClinGen:CA252255,UniProtKB:Q6ZVN8#VAR_019617,O...</td>\n      <td>T/C</td>\n      <td>Cys80Arg</td>\n      <td>Cys</td>\n      <td>80</td>\n      <td>Arg</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>11487</th>\n      <td>NM_213653.4(HJV):c.302T&gt;C (p.Leu101Pro)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>302</td>\n      <td>ClinGen:CA252256,UniProtKB:Q6ZVN8#VAR_019620,O...</td>\n      <td>T/C</td>\n      <td>Leu101Pro</td>\n      <td>Leu</td>\n      <td>101</td>\n      <td>Pro</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>11488</th>\n      <td>NM_213653.4(HJV):c.842T&gt;C (p.Ile281Thr)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>842</td>\n      <td>ClinGen:CA252253,OMIM:608374.0004</td>\n      <td>T/C</td>\n      <td>Ile281Thr</td>\n      <td>Ile</td>\n      <td>281</td>\n      <td>Thr</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>11489</th>\n      <td>NM_213653.4(HJV):c.959G&gt;T (p.Gly320Val)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>959</td>\n      <td>ClinGen:CA252249,UniProtKB:Q6ZVN8#VAR_019629,O...</td>\n      <td>G/T</td>\n      <td>Gly320Val</td>\n      <td>Gly</td>\n      <td>320</td>\n      <td>Val</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>11490</th>\n      <td>NM_213720.3(CHCHD10):c.44G&gt;T (p.Arg15Leu)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Pathogenic</td>\n      <td>44</td>\n      <td>ClinGen:CA235291,OMIM:615903.0002</td>\n      <td>G/T</td>\n      <td>Arg15Leu</td>\n      <td>Arg</td>\n      <td>15</td>\n      <td>Leu</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n  </tbody>\n</table>\n<p>11491 rows × 15 columns</p>\n</div>"
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# remove duplicates but maintain same order of column names\n",
    "col_list = df_pathogenic.columns.tolist()\n",
    "col_list.remove('Assembly')\n",
    "d = df_pathogenic.groupby(col_list).agg({'Assembly': pd.Series.unique}).reset_index()\n",
    "d"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "outputs": [
    {
     "data": {
      "text/plain": "[['GRCh37', 'GRCh38'], ['na']]"
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# convert the 'versions' column to a list of tuples\n",
    "versions_tuples = [tuple(x) for x in d['Assembly']]\n",
    "\n",
    "# get the unique tuples in the 'versions' column\n",
    "unique_tuples = pd.unique(versions_tuples)\n",
    "\n",
    "# convert the unique tuples back to lists\n",
    "unique_lists = [list(x) for x in unique_tuples]\n",
    "unique_lists"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/pbs.498551.hn-10-03/ipykernel_42587/2947019198.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_pathogenic_new.drop_duplicates(inplace=True)\n"
     ]
    },
    {
     "data": {
      "text/plain": "                                                Name  GeneID GeneSymbol  \\\n24           NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n25           NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n32         NM_138413.4(HOGA1):c.860G>T (p.Gly287Val)  112817      HOGA1   \n33         NM_138413.4(HOGA1):c.860G>T (p.Gly287Val)  112817      HOGA1   \n36          NM_138413.4(HOGA1):c.209G>C (p.Arg70Pro)  112817      HOGA1   \n...                                              ...     ...        ...   \n1213199     NM_021870.2(FGG):c.1007T>C (p.Met336Thr)    2266        FGG   \n1213224  NM_000342.3(SLC4A1):c.1462G>A (p.Val488Met)    6521     SLC4A1   \n1213225  NM_000342.3(SLC4A1):c.1462G>A (p.Val488Met)    6521     SLC4A1   \n1213242     NM_000029.4(AGT):c.1124G>A (p.Arg375Gln)     183        AGT   \n1213243     NM_000029.4(AGT):c.1124G>A (p.Arg375Gln)     183        AGT   \n\n        Assembly          ClinicalSignificance  CDS_position  \\\n24        GRCh37                    Pathogenic           848   \n25        GRCh38                    Pathogenic           848   \n32        GRCh37  Pathogenic/Likely pathogenic           860   \n33        GRCh38  Pathogenic/Likely pathogenic           860   \n36        GRCh37                    Pathogenic           209   \n...          ...                           ...           ...   \n1213199   GRCh38                    Pathogenic          1007   \n1213224   GRCh37                    Pathogenic          1462   \n1213225   GRCh38                    Pathogenic          1462   \n1213242   GRCh37                    Pathogenic          1124   \n1213243   GRCh38                    Pathogenic          1124   \n\n                                                  OtherIDs AlleleChange  \\\n24       ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...          A/C   \n25       ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...          A/C   \n32       ClinGen:CA113813,UniProtKB:Q86XE5#VAR_064036,O...          G/T   \n33       ClinGen:CA113813,UniProtKB:Q86XE5#VAR_064036,O...          G/T   \n36                       ClinGen:CA113816,OMIM:613597.0005          G/C   \n...                                                    ...          ...   \n1213199  ClinGen:CA126399,UniProtKB:P02679#VAR_002414,O...          T/C   \n1213224  ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...          G/A   \n1213225  ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...          G/A   \n1213242  ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...          G/A   \n1213243  ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...          G/A   \n\n        Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n24             Gln283Pro                         Gln                283   \n25             Gln283Pro                         Gln                283   \n32             Gly287Val                         Gly                287   \n33             Gly287Val                         Gly                287   \n36              Arg70Pro                         Arg                 70   \n...                  ...                         ...                ...   \n1213199        Met336Thr                         Met                336   \n1213224        Val488Met                         Val                488   \n1213225        Val488Met                         Val                488   \n1213242        Arg375Gln                         Arg                375   \n1213243        Arg375Gln                         Arg                375   \n\n        Amino_acids_after_3letters        NM_ID UniProtKB-AC  stars  \n24                             Pro  NM_000410.4       Q30201      1  \n25                             Pro  NM_000410.4       Q30201      1  \n32                             Val  NM_138413.4       Q86XE5      2  \n33                             Val  NM_138413.4       Q86XE5      2  \n36                             Pro  NM_138413.4       Q86XE5      1  \n...                            ...          ...          ...    ...  \n1213199                        Thr  NM_021870.2       P02679      1  \n1213224                        Met  NM_000342.3       P02730      2  \n1213225                        Met  NM_000342.3       P02730      2  \n1213242                        Gln  NM_000029.4       P01019      1  \n1213243                        Gln  NM_000029.4       P01019      1  \n\n[33403 rows x 15 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>stars</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>24</th>\n      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>848</td>\n      <td>ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...</td>\n      <td>A/C</td>\n      <td>Gln283Pro</td>\n      <td>Gln</td>\n      <td>283</td>\n      <td>Pro</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>25</th>\n      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>848</td>\n      <td>ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...</td>\n      <td>A/C</td>\n      <td>Gln283Pro</td>\n      <td>Gln</td>\n      <td>283</td>\n      <td>Pro</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>32</th>\n      <td>NM_138413.4(HOGA1):c.860G&gt;T (p.Gly287Val)</td>\n      <td>112817</td>\n      <td>HOGA1</td>\n      <td>GRCh37</td>\n      <td>Pathogenic/Likely pathogenic</td>\n      <td>860</td>\n      <td>ClinGen:CA113813,UniProtKB:Q86XE5#VAR_064036,O...</td>\n      <td>G/T</td>\n      <td>Gly287Val</td>\n      <td>Gly</td>\n      <td>287</td>\n      <td>Val</td>\n      <td>NM_138413.4</td>\n      <td>Q86XE5</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>33</th>\n      <td>NM_138413.4(HOGA1):c.860G&gt;T (p.Gly287Val)</td>\n      <td>112817</td>\n      <td>HOGA1</td>\n      <td>GRCh38</td>\n      <td>Pathogenic/Likely pathogenic</td>\n      <td>860</td>\n      <td>ClinGen:CA113813,UniProtKB:Q86XE5#VAR_064036,O...</td>\n      <td>G/T</td>\n      <td>Gly287Val</td>\n      <td>Gly</td>\n      <td>287</td>\n      <td>Val</td>\n      <td>NM_138413.4</td>\n      <td>Q86XE5</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>36</th>\n      <td>NM_138413.4(HOGA1):c.209G&gt;C (p.Arg70Pro)</td>\n      <td>112817</td>\n      <td>HOGA1</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>209</td>\n      <td>ClinGen:CA113816,OMIM:613597.0005</td>\n      <td>G/C</td>\n      <td>Arg70Pro</td>\n      <td>Arg</td>\n      <td>70</td>\n      <td>Pro</td>\n      <td>NM_138413.4</td>\n      <td>Q86XE5</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>1213199</th>\n      <td>NM_021870.2(FGG):c.1007T&gt;C (p.Met336Thr)</td>\n      <td>2266</td>\n      <td>FGG</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>1007</td>\n      <td>ClinGen:CA126399,UniProtKB:P02679#VAR_002414,O...</td>\n      <td>T/C</td>\n      <td>Met336Thr</td>\n      <td>Met</td>\n      <td>336</td>\n      <td>Thr</td>\n      <td>NM_021870.2</td>\n      <td>P02679</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213224</th>\n      <td>NM_000342.3(SLC4A1):c.1462G&gt;A (p.Val488Met)</td>\n      <td>6521</td>\n      <td>SLC4A1</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>1462</td>\n      <td>ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...</td>\n      <td>G/A</td>\n      <td>Val488Met</td>\n      <td>Val</td>\n      <td>488</td>\n      <td>Met</td>\n      <td>NM_000342.3</td>\n      <td>P02730</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213225</th>\n      <td>NM_000342.3(SLC4A1):c.1462G&gt;A (p.Val488Met)</td>\n      <td>6521</td>\n      <td>SLC4A1</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>1462</td>\n      <td>ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...</td>\n      <td>G/A</td>\n      <td>Val488Met</td>\n      <td>Val</td>\n      <td>488</td>\n      <td>Met</td>\n      <td>NM_000342.3</td>\n      <td>P02730</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213242</th>\n      <td>NM_000029.4(AGT):c.1124G&gt;A (p.Arg375Gln)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>1124</td>\n      <td>ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...</td>\n      <td>G/A</td>\n      <td>Arg375Gln</td>\n      <td>Arg</td>\n      <td>375</td>\n      <td>Gln</td>\n      <td>NM_000029.4</td>\n      <td>P01019</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213243</th>\n      <td>NM_000029.4(AGT):c.1124G&gt;A (p.Arg375Gln)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>1124</td>\n      <td>ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...</td>\n      <td>G/A</td>\n      <td>Arg375Gln</td>\n      <td>Arg</td>\n      <td>375</td>\n      <td>Gln</td>\n      <td>NM_000029.4</td>\n      <td>P01019</td>\n      <td>1</td>\n    </tr>\n  </tbody>\n</table>\n<p>33403 rows × 15 columns</p>\n</div>"
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# include more pathogenic variant\n",
    "df_pathogenic_new = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance'].str.contains('Pathogenic'))]\n",
    "df_pathogenic_new.drop_duplicates(inplace=True)\n",
    "df_pathogenic_new"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "outputs": [
    {
     "data": {
      "text/plain": "                                             Name  GeneID GeneSymbol  \\\n0      NM_000016.6(ACADM):c.1034A>T (p.Asp345Val)      34      ACADM   \n1      NM_000016.6(ACADM):c.1042C>T (p.Arg348Cys)      34      ACADM   \n2      NM_000016.6(ACADM):c.1045C>G (p.Arg349Gly)      34      ACADM   \n3      NM_000016.6(ACADM):c.1046G>A (p.Arg349Gln)      34      ACADM   \n4      NM_000016.6(ACADM):c.1052C>T (p.Thr351Ile)      34      ACADM   \n...                                           ...     ...        ...   \n16713     NM_213653.4(HJV):c.302T>C (p.Leu101Pro)  148738        HJV   \n16714     NM_213653.4(HJV):c.842T>C (p.Ile281Thr)  148738        HJV   \n16715     NM_213653.4(HJV):c.959G>T (p.Gly320Val)  148738        HJV   \n16716  NM_213720.3(CHCHD10):c.176C>T (p.Ser59Leu)  400916    CHCHD10   \n16717   NM_213720.3(CHCHD10):c.44G>T (p.Arg15Leu)  400916    CHCHD10   \n\n               ClinicalSignificance  CDS_position  \\\n0                        Pathogenic          1034   \n1                        Pathogenic          1042   \n2                        Pathogenic          1045   \n3                        Pathogenic          1046   \n4      Pathogenic/Likely pathogenic          1052   \n...                             ...           ...   \n16713                    Pathogenic           302   \n16714                    Pathogenic           842   \n16715                    Pathogenic           959   \n16716  Pathogenic/Likely pathogenic           176   \n16717                    Pathogenic            44   \n\n                                                OtherIDs AlleleChange  \\\n0                                       ClinGen:CA913267          A/T   \n1                                                      -          C/T   \n2                                                      -          C/G   \n3                                                      -          G/A   \n4                                       ClinGen:CA312184          C/T   \n...                                                  ...          ...   \n16713  ClinGen:CA252256,UniProtKB:Q6ZVN8#VAR_019620,O...          T/C   \n16714                  ClinGen:CA252253,OMIM:608374.0004          T/C   \n16715  ClinGen:CA252249,UniProtKB:Q6ZVN8#VAR_019629,O...          G/T   \n16716  ClinGen:CA163486,UniProtKB:Q8WYQ3#VAR_071806,O...          C/T   \n16717                  ClinGen:CA235291,OMIM:615903.0002          G/T   \n\n      Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n0            Asp345Val                         Asp                345   \n1            Arg348Cys                         Arg                348   \n2            Arg349Gly                         Arg                349   \n3            Arg349Gln                         Arg                349   \n4            Thr351Ile                         Thr                351   \n...                ...                         ...                ...   \n16713        Leu101Pro                         Leu                101   \n16714        Ile281Thr                         Ile                281   \n16715        Gly320Val                         Gly                320   \n16716         Ser59Leu                         Ser                 59   \n16717         Arg15Leu                         Arg                 15   \n\n      Amino_acids_after_3letters        NM_ID UniProtKB-AC  stars  \\\n0                            Val  NM_000016.6   A0A0S2Z366      1   \n1                            Cys  NM_000016.6   A0A0S2Z366      1   \n2                            Gly  NM_000016.6   A0A0S2Z366      1   \n3                            Gln  NM_000016.6   A0A0S2Z366      1   \n4                            Ile  NM_000016.6   A0A0S2Z366      2   \n...                          ...          ...          ...    ...   \n16713                        Pro  NM_213653.4   A0A024R4F5      1   \n16714                        Thr  NM_213653.4   A0A024R4F5      1   \n16715                        Val  NM_213653.4   A0A024R4F5      2   \n16716                        Leu  NM_213720.3       Q8WYQ3      2   \n16717                        Leu  NM_213720.3       Q8WYQ3      2   \n\n               Assembly  \n0      [GRCh37, GRCh38]  \n1      [GRCh37, GRCh38]  \n2      [GRCh37, GRCh38]  \n3      [GRCh37, GRCh38]  \n4      [GRCh37, GRCh38]  \n...                 ...  \n16713  [GRCh37, GRCh38]  \n16714  [GRCh37, GRCh38]  \n16715  [GRCh37, GRCh38]  \n16716  [GRCh37, GRCh38]  \n16717  [GRCh37, GRCh38]  \n\n[16718 rows x 15 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>stars</th>\n      <th>Assembly</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NM_000016.6(ACADM):c.1034A&gt;T (p.Asp345Val)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1034</td>\n      <td>ClinGen:CA913267</td>\n      <td>A/T</td>\n      <td>Asp345Val</td>\n      <td>Asp</td>\n      <td>345</td>\n      <td>Val</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NM_000016.6(ACADM):c.1042C&gt;T (p.Arg348Cys)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1042</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Arg348Cys</td>\n      <td>Arg</td>\n      <td>348</td>\n      <td>Cys</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NM_000016.6(ACADM):c.1045C&gt;G (p.Arg349Gly)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1045</td>\n      <td>-</td>\n      <td>C/G</td>\n      <td>Arg349Gly</td>\n      <td>Arg</td>\n      <td>349</td>\n      <td>Gly</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NM_000016.6(ACADM):c.1046G&gt;A (p.Arg349Gln)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1046</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Arg349Gln</td>\n      <td>Arg</td>\n      <td>349</td>\n      <td>Gln</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NM_000016.6(ACADM):c.1052C&gt;T (p.Thr351Ile)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic/Likely pathogenic</td>\n      <td>1052</td>\n      <td>ClinGen:CA312184</td>\n      <td>C/T</td>\n      <td>Thr351Ile</td>\n      <td>Thr</td>\n      <td>351</td>\n      <td>Ile</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>16713</th>\n      <td>NM_213653.4(HJV):c.302T&gt;C (p.Leu101Pro)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>302</td>\n      <td>ClinGen:CA252256,UniProtKB:Q6ZVN8#VAR_019620,O...</td>\n      <td>T/C</td>\n      <td>Leu101Pro</td>\n      <td>Leu</td>\n      <td>101</td>\n      <td>Pro</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>16714</th>\n      <td>NM_213653.4(HJV):c.842T&gt;C (p.Ile281Thr)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>842</td>\n      <td>ClinGen:CA252253,OMIM:608374.0004</td>\n      <td>T/C</td>\n      <td>Ile281Thr</td>\n      <td>Ile</td>\n      <td>281</td>\n      <td>Thr</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>16715</th>\n      <td>NM_213653.4(HJV):c.959G&gt;T (p.Gly320Val)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>959</td>\n      <td>ClinGen:CA252249,UniProtKB:Q6ZVN8#VAR_019629,O...</td>\n      <td>G/T</td>\n      <td>Gly320Val</td>\n      <td>Gly</td>\n      <td>320</td>\n      <td>Val</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>16716</th>\n      <td>NM_213720.3(CHCHD10):c.176C&gt;T (p.Ser59Leu)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Pathogenic/Likely pathogenic</td>\n      <td>176</td>\n      <td>ClinGen:CA163486,UniProtKB:Q8WYQ3#VAR_071806,O...</td>\n      <td>C/T</td>\n      <td>Ser59Leu</td>\n      <td>Ser</td>\n      <td>59</td>\n      <td>Leu</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>16717</th>\n      <td>NM_213720.3(CHCHD10):c.44G&gt;T (p.Arg15Leu)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Pathogenic</td>\n      <td>44</td>\n      <td>ClinGen:CA235291,OMIM:615903.0002</td>\n      <td>G/T</td>\n      <td>Arg15Leu</td>\n      <td>Arg</td>\n      <td>15</td>\n      <td>Leu</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n  </tbody>\n</table>\n<p>16718 rows × 15 columns</p>\n</div>"
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d_new = df_pathogenic_new.groupby(col_list).agg({'Assembly': pd.Series.unique}).reset_index()\n",
    "d_new"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/pbs.498551.hn-10-03/ipykernel_42587/628818179.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_pathogenic_new.drop_duplicates(inplace=True)\n"
     ]
    },
    {
     "data": {
      "text/plain": "                                                   Name  GeneID GeneSymbol  \\\n24              NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n25              NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n36             NM_138413.4(HOGA1):c.209G>C (p.Arg70Pro)  112817      HOGA1   \n37             NM_138413.4(HOGA1):c.209G>C (p.Arg70Pro)  112817      HOGA1   \n48       NM_001199107.2(TBC1D24):c.439G>C (p.Asp147His)   57465    TBC1D24   \n...                                                 ...     ...        ...   \n1213199        NM_021870.2(FGG):c.1007T>C (p.Met336Thr)    2266        FGG   \n1213224     NM_000342.3(SLC4A1):c.1462G>A (p.Val488Met)    6521     SLC4A1   \n1213225     NM_000342.3(SLC4A1):c.1462G>A (p.Val488Met)    6521     SLC4A1   \n1213242        NM_000029.4(AGT):c.1124G>A (p.Arg375Gln)     183        AGT   \n1213243        NM_000029.4(AGT):c.1124G>A (p.Arg375Gln)     183        AGT   \n\n        Assembly ClinicalSignificance  CDS_position  \\\n24        GRCh37           Pathogenic           848   \n25        GRCh38           Pathogenic           848   \n36        GRCh37           Pathogenic           209   \n37        GRCh38           Pathogenic           209   \n48        GRCh37           Pathogenic           439   \n...          ...                  ...           ...   \n1213199   GRCh38           Pathogenic          1007   \n1213224   GRCh37           Pathogenic          1462   \n1213225   GRCh38           Pathogenic          1462   \n1213242   GRCh37           Pathogenic          1124   \n1213243   GRCh38           Pathogenic          1124   \n\n                                                  OtherIDs AlleleChange  \\\n24       ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...          A/C   \n25       ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...          A/C   \n36                       ClinGen:CA113816,OMIM:613597.0005          G/C   \n37                       ClinGen:CA113816,OMIM:613597.0005          G/C   \n48       ClinGen:CA339785,UniProtKB:Q9ULP9#VAR_064365,O...          G/C   \n...                                                    ...          ...   \n1213199  ClinGen:CA126399,UniProtKB:P02679#VAR_002414,O...          T/C   \n1213224  ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...          G/A   \n1213225  ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...          G/A   \n1213242  ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...          G/A   \n1213243  ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...          G/A   \n\n        Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n24             Gln283Pro                         Gln                283   \n25             Gln283Pro                         Gln                283   \n36              Arg70Pro                         Arg                 70   \n37              Arg70Pro                         Arg                 70   \n48             Asp147His                         Asp                147   \n...                  ...                         ...                ...   \n1213199        Met336Thr                         Met                336   \n1213224        Val488Met                         Val                488   \n1213225        Val488Met                         Val                488   \n1213242        Arg375Gln                         Arg                375   \n1213243        Arg375Gln                         Arg                375   \n\n        Amino_acids_after_3letters           NM_ID UniProtKB-AC  stars  \n24                             Pro     NM_000410.4       Q30201      1  \n25                             Pro     NM_000410.4       Q30201      1  \n36                             Pro     NM_138413.4       Q86XE5      1  \n37                             Pro     NM_138413.4       Q86XE5      1  \n48                             His  NM_001199107.2       Q9ULP9      1  \n...                            ...             ...          ...    ...  \n1213199                        Thr     NM_021870.2       P02679      1  \n1213224                        Met     NM_000342.3       P02730      2  \n1213225                        Met     NM_000342.3       P02730      2  \n1213242                        Gln     NM_000029.4       P01019      1  \n1213243                        Gln     NM_000029.4       P01019      1  \n\n[22951 rows x 15 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>stars</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>24</th>\n      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>848</td>\n      <td>ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...</td>\n      <td>A/C</td>\n      <td>Gln283Pro</td>\n      <td>Gln</td>\n      <td>283</td>\n      <td>Pro</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>25</th>\n      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>848</td>\n      <td>ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...</td>\n      <td>A/C</td>\n      <td>Gln283Pro</td>\n      <td>Gln</td>\n      <td>283</td>\n      <td>Pro</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>36</th>\n      <td>NM_138413.4(HOGA1):c.209G&gt;C (p.Arg70Pro)</td>\n      <td>112817</td>\n      <td>HOGA1</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>209</td>\n      <td>ClinGen:CA113816,OMIM:613597.0005</td>\n      <td>G/C</td>\n      <td>Arg70Pro</td>\n      <td>Arg</td>\n      <td>70</td>\n      <td>Pro</td>\n      <td>NM_138413.4</td>\n      <td>Q86XE5</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>37</th>\n      <td>NM_138413.4(HOGA1):c.209G&gt;C (p.Arg70Pro)</td>\n      <td>112817</td>\n      <td>HOGA1</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>209</td>\n      <td>ClinGen:CA113816,OMIM:613597.0005</td>\n      <td>G/C</td>\n      <td>Arg70Pro</td>\n      <td>Arg</td>\n      <td>70</td>\n      <td>Pro</td>\n      <td>NM_138413.4</td>\n      <td>Q86XE5</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>48</th>\n      <td>NM_001199107.2(TBC1D24):c.439G&gt;C (p.Asp147His)</td>\n      <td>57465</td>\n      <td>TBC1D24</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>439</td>\n      <td>ClinGen:CA339785,UniProtKB:Q9ULP9#VAR_064365,O...</td>\n      <td>G/C</td>\n      <td>Asp147His</td>\n      <td>Asp</td>\n      <td>147</td>\n      <td>His</td>\n      <td>NM_001199107.2</td>\n      <td>Q9ULP9</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>1213199</th>\n      <td>NM_021870.2(FGG):c.1007T&gt;C (p.Met336Thr)</td>\n      <td>2266</td>\n      <td>FGG</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>1007</td>\n      <td>ClinGen:CA126399,UniProtKB:P02679#VAR_002414,O...</td>\n      <td>T/C</td>\n      <td>Met336Thr</td>\n      <td>Met</td>\n      <td>336</td>\n      <td>Thr</td>\n      <td>NM_021870.2</td>\n      <td>P02679</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213224</th>\n      <td>NM_000342.3(SLC4A1):c.1462G&gt;A (p.Val488Met)</td>\n      <td>6521</td>\n      <td>SLC4A1</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>1462</td>\n      <td>ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...</td>\n      <td>G/A</td>\n      <td>Val488Met</td>\n      <td>Val</td>\n      <td>488</td>\n      <td>Met</td>\n      <td>NM_000342.3</td>\n      <td>P02730</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213225</th>\n      <td>NM_000342.3(SLC4A1):c.1462G&gt;A (p.Val488Met)</td>\n      <td>6521</td>\n      <td>SLC4A1</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>1462</td>\n      <td>ClinGen:CA127399,UniProtKB:P02730#VAR_013791,O...</td>\n      <td>G/A</td>\n      <td>Val488Met</td>\n      <td>Val</td>\n      <td>488</td>\n      <td>Met</td>\n      <td>NM_000342.3</td>\n      <td>P02730</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213242</th>\n      <td>NM_000029.4(AGT):c.1124G&gt;A (p.Arg375Gln)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Pathogenic</td>\n      <td>1124</td>\n      <td>ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...</td>\n      <td>G/A</td>\n      <td>Arg375Gln</td>\n      <td>Arg</td>\n      <td>375</td>\n      <td>Gln</td>\n      <td>NM_000029.4</td>\n      <td>P01019</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213243</th>\n      <td>NM_000029.4(AGT):c.1124G&gt;A (p.Arg375Gln)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Pathogenic</td>\n      <td>1124</td>\n      <td>ClinGen:CA127786,UniProtKB:P01019#VAR_035433,O...</td>\n      <td>G/A</td>\n      <td>Arg375Gln</td>\n      <td>Arg</td>\n      <td>375</td>\n      <td>Gln</td>\n      <td>NM_000029.4</td>\n      <td>P01019</td>\n      <td>1</td>\n    </tr>\n  </tbody>\n</table>\n<p>22951 rows × 15 columns</p>\n</div>"
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# include more pathogenic variant\n",
    "#df_pathogenic_new = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance'].str.contains('(Pathogenic|Likely pathogenic)'))]\n",
    "#df_pathogenic_new = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance'].isin(['Pathogenic','Likely pathogenic']))]\n",
    "#df_pathogenic_new = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance'].str.contains('Pathogenic'))]\n",
    "df_pathogenic_new = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance'] == 'Pathogenic')]\n",
    "df_pathogenic_new.drop_duplicates(inplace=True)\n",
    "df_pathogenic_new"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "outputs": [
    {
     "data": {
      "text/plain": "                                             Name  GeneID GeneSymbol  \\\n0      NM_000016.6(ACADM):c.1034A>T (p.Asp345Val)      34      ACADM   \n1      NM_000016.6(ACADM):c.1042C>T (p.Arg348Cys)      34      ACADM   \n2      NM_000016.6(ACADM):c.1045C>G (p.Arg349Gly)      34      ACADM   \n3      NM_000016.6(ACADM):c.1046G>A (p.Arg349Gln)      34      ACADM   \n4      NM_000016.6(ACADM):c.1067T>C (p.Ile356Thr)      34      ACADM   \n...                                           ...     ...        ...   \n11486      NM_213653.4(HJV):c.238T>C (p.Cys80Arg)  148738        HJV   \n11487     NM_213653.4(HJV):c.302T>C (p.Leu101Pro)  148738        HJV   \n11488     NM_213653.4(HJV):c.842T>C (p.Ile281Thr)  148738        HJV   \n11489     NM_213653.4(HJV):c.959G>T (p.Gly320Val)  148738        HJV   \n11490   NM_213720.3(CHCHD10):c.44G>T (p.Arg15Leu)  400916    CHCHD10   \n\n      ClinicalSignificance  CDS_position  \\\n0               Pathogenic          1034   \n1               Pathogenic          1042   \n2               Pathogenic          1045   \n3               Pathogenic          1046   \n4               Pathogenic          1067   \n...                    ...           ...   \n11486           Pathogenic           238   \n11487           Pathogenic           302   \n11488           Pathogenic           842   \n11489           Pathogenic           959   \n11490           Pathogenic            44   \n\n                                                OtherIDs AlleleChange  \\\n0                                       ClinGen:CA913267          A/T   \n1                                                      -          C/T   \n2                                                      -          C/G   \n3                                                      -          G/A   \n4                                                      -          T/C   \n...                                                  ...          ...   \n11486  ClinGen:CA252255,UniProtKB:Q6ZVN8#VAR_019617,O...          T/C   \n11487  ClinGen:CA252256,UniProtKB:Q6ZVN8#VAR_019620,O...          T/C   \n11488                  ClinGen:CA252253,OMIM:608374.0004          T/C   \n11489  ClinGen:CA252249,UniProtKB:Q6ZVN8#VAR_019629,O...          G/T   \n11490                  ClinGen:CA235291,OMIM:615903.0002          G/T   \n\n      Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n0            Asp345Val                         Asp                345   \n1            Arg348Cys                         Arg                348   \n2            Arg349Gly                         Arg                349   \n3            Arg349Gln                         Arg                349   \n4            Ile356Thr                         Ile                356   \n...                ...                         ...                ...   \n11486         Cys80Arg                         Cys                 80   \n11487        Leu101Pro                         Leu                101   \n11488        Ile281Thr                         Ile                281   \n11489        Gly320Val                         Gly                320   \n11490         Arg15Leu                         Arg                 15   \n\n      Amino_acids_after_3letters        NM_ID UniProtKB-AC  stars  \\\n0                            Val  NM_000016.6   A0A0S2Z366      1   \n1                            Cys  NM_000016.6   A0A0S2Z366      1   \n2                            Gly  NM_000016.6   A0A0S2Z366      1   \n3                            Gln  NM_000016.6   A0A0S2Z366      1   \n4                            Thr  NM_000016.6   A0A0S2Z366      1   \n...                          ...          ...          ...    ...   \n11486                        Arg  NM_213653.4   A0A024R4F5      1   \n11487                        Pro  NM_213653.4   A0A024R4F5      1   \n11488                        Thr  NM_213653.4   A0A024R4F5      1   \n11489                        Val  NM_213653.4   A0A024R4F5      2   \n11490                        Leu  NM_213720.3       Q8WYQ3      2   \n\n               Assembly  \n0      [GRCh37, GRCh38]  \n1      [GRCh37, GRCh38]  \n2      [GRCh37, GRCh38]  \n3      [GRCh37, GRCh38]  \n4      [GRCh37, GRCh38]  \n...                 ...  \n11486  [GRCh37, GRCh38]  \n11487  [GRCh37, GRCh38]  \n11488  [GRCh37, GRCh38]  \n11489  [GRCh37, GRCh38]  \n11490  [GRCh37, GRCh38]  \n\n[11491 rows x 15 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>stars</th>\n      <th>Assembly</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NM_000016.6(ACADM):c.1034A&gt;T (p.Asp345Val)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1034</td>\n      <td>ClinGen:CA913267</td>\n      <td>A/T</td>\n      <td>Asp345Val</td>\n      <td>Asp</td>\n      <td>345</td>\n      <td>Val</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NM_000016.6(ACADM):c.1042C&gt;T (p.Arg348Cys)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1042</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Arg348Cys</td>\n      <td>Arg</td>\n      <td>348</td>\n      <td>Cys</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NM_000016.6(ACADM):c.1045C&gt;G (p.Arg349Gly)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1045</td>\n      <td>-</td>\n      <td>C/G</td>\n      <td>Arg349Gly</td>\n      <td>Arg</td>\n      <td>349</td>\n      <td>Gly</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NM_000016.6(ACADM):c.1046G&gt;A (p.Arg349Gln)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1046</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Arg349Gln</td>\n      <td>Arg</td>\n      <td>349</td>\n      <td>Gln</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NM_000016.6(ACADM):c.1067T&gt;C (p.Ile356Thr)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1067</td>\n      <td>-</td>\n      <td>T/C</td>\n      <td>Ile356Thr</td>\n      <td>Ile</td>\n      <td>356</td>\n      <td>Thr</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>11486</th>\n      <td>NM_213653.4(HJV):c.238T&gt;C (p.Cys80Arg)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>238</td>\n      <td>ClinGen:CA252255,UniProtKB:Q6ZVN8#VAR_019617,O...</td>\n      <td>T/C</td>\n      <td>Cys80Arg</td>\n      <td>Cys</td>\n      <td>80</td>\n      <td>Arg</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>11487</th>\n      <td>NM_213653.4(HJV):c.302T&gt;C (p.Leu101Pro)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>302</td>\n      <td>ClinGen:CA252256,UniProtKB:Q6ZVN8#VAR_019620,O...</td>\n      <td>T/C</td>\n      <td>Leu101Pro</td>\n      <td>Leu</td>\n      <td>101</td>\n      <td>Pro</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>11488</th>\n      <td>NM_213653.4(HJV):c.842T&gt;C (p.Ile281Thr)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>842</td>\n      <td>ClinGen:CA252253,OMIM:608374.0004</td>\n      <td>T/C</td>\n      <td>Ile281Thr</td>\n      <td>Ile</td>\n      <td>281</td>\n      <td>Thr</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>11489</th>\n      <td>NM_213653.4(HJV):c.959G&gt;T (p.Gly320Val)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Pathogenic</td>\n      <td>959</td>\n      <td>ClinGen:CA252249,UniProtKB:Q6ZVN8#VAR_019629,O...</td>\n      <td>G/T</td>\n      <td>Gly320Val</td>\n      <td>Gly</td>\n      <td>320</td>\n      <td>Val</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>11490</th>\n      <td>NM_213720.3(CHCHD10):c.44G&gt;T (p.Arg15Leu)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Pathogenic</td>\n      <td>44</td>\n      <td>ClinGen:CA235291,OMIM:615903.0002</td>\n      <td>G/T</td>\n      <td>Arg15Leu</td>\n      <td>Arg</td>\n      <td>15</td>\n      <td>Leu</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n  </tbody>\n</table>\n<p>11491 rows × 15 columns</p>\n</div>"
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p_new = df_pathogenic_new.groupby(col_list).agg({'Assembly': pd.Series.unique}).reset_index()\n",
    "p_new"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "outputs": [
    {
     "data": {
      "text/plain": "array(['Pathogenic'], dtype=object)"
     },
     "execution_count": 159,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p_new['ClinicalSignificance'].unique()"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/pbs.498551.hn-10-03/ipykernel_42587/1341507667.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_benign_new.drop_duplicates(inplace=True)\n"
     ]
    },
    {
     "data": {
      "text/plain": "                                               Name  GeneID GeneSymbol  \\\n90         NM_000355.4(TCN2):c.776G>C (p.Arg259Pro)    6948       TCN2   \n91         NM_000355.4(TCN2):c.776G>C (p.Arg259Pro)    6948       TCN2   \n244      NM_000398.7(CYB5R3):c.350C>G (p.Thr117Ser)    1727     CYB5R3   \n245      NM_000398.7(CYB5R3):c.350C>G (p.Thr117Ser)    1727     CYB5R3   \n592       NM_000147.5(FUCA1):c.857A>G (p.Gln286Arg)    2517      FUCA1   \n...                                             ...     ...        ...   \n1213241     NM_000029.4(AGT):c.803T>C (p.Met268Thr)     183        AGT   \n1213262     NM_000029.4(AGT):c.620C>T (p.Thr207Met)     183        AGT   \n1213263     NM_000029.4(AGT):c.620C>T (p.Thr207Met)     183        AGT   \n1213282      NM_000029.4(AGT):c.292G>A (p.Glu98Lys)     183        AGT   \n1213283      NM_000029.4(AGT):c.292G>A (p.Glu98Lys)     183        AGT   \n\n        Assembly ClinicalSignificance  CDS_position  \\\n90        GRCh37               Benign           776   \n91        GRCh38               Benign           776   \n244       GRCh37               Benign           350   \n245       GRCh38               Benign           350   \n592       GRCh37               Benign           857   \n...          ...                  ...           ...   \n1213241   GRCh38               Benign           803   \n1213262   GRCh37               Benign           620   \n1213263   GRCh38               Benign           620   \n1213282   GRCh37               Benign           292   \n1213283   GRCh38               Benign           292   \n\n                                                  OtherIDs AlleleChange  \\\n90       ClinGen:CA113848,UniProtKB:P20062#VAR_001640,O...          G/C   \n91       ClinGen:CA113848,UniProtKB:P20062#VAR_001640,O...          G/C   \n244      ClinGen:CA114081,UniProtKB:P00387#VAR_010751,O...          C/G   \n245      ClinGen:CA114081,UniProtKB:P00387#VAR_010751,O...          C/G   \n592      ClinGen:CA114444,UniProtKB:P04066#VAR_002444,O...          A/G   \n...                                                    ...          ...   \n1213241  ClinGen:CA127784,UniProtKB:P01019#VAR_007096,O...          T/C   \n1213262      ClinGen:CA1448272,UniProtKB:P01019#VAR_007093          C/T   \n1213263      ClinGen:CA1448272,UniProtKB:P01019#VAR_007093          C/T   \n1213282                                                  -          G/A   \n1213283                                                  -          G/A   \n\n        Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n90             Arg259Pro                         Arg                259   \n91             Arg259Pro                         Arg                259   \n244            Thr117Ser                         Thr                117   \n245            Thr117Ser                         Thr                117   \n592            Gln286Arg                         Gln                286   \n...                  ...                         ...                ...   \n1213241        Met268Thr                         Met                268   \n1213262        Thr207Met                         Thr                207   \n1213263        Thr207Met                         Thr                207   \n1213282         Glu98Lys                         Glu                 98   \n1213283         Glu98Lys                         Glu                 98   \n\n        Amino_acids_after_3letters        NM_ID UniProtKB-AC  stars  \n90                             Pro  NM_000355.4       P20062      2  \n91                             Pro  NM_000355.4       P20062      2  \n244                            Ser  NM_000398.7       P00387      2  \n245                            Ser  NM_000398.7       P00387      2  \n592                            Arg  NM_000147.5       P04066      2  \n...                            ...          ...          ...    ...  \n1213241                        Thr  NM_000029.4       P01019      2  \n1213262                        Met  NM_000029.4       P01019      2  \n1213263                        Met  NM_000029.4       P01019      2  \n1213282                        Lys  NM_000029.4       P01019      1  \n1213283                        Lys  NM_000029.4       P01019      1  \n\n[45009 rows x 15 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>stars</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>90</th>\n      <td>NM_000355.4(TCN2):c.776G&gt;C (p.Arg259Pro)</td>\n      <td>6948</td>\n      <td>TCN2</td>\n      <td>GRCh37</td>\n      <td>Benign</td>\n      <td>776</td>\n      <td>ClinGen:CA113848,UniProtKB:P20062#VAR_001640,O...</td>\n      <td>G/C</td>\n      <td>Arg259Pro</td>\n      <td>Arg</td>\n      <td>259</td>\n      <td>Pro</td>\n      <td>NM_000355.4</td>\n      <td>P20062</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>91</th>\n      <td>NM_000355.4(TCN2):c.776G&gt;C (p.Arg259Pro)</td>\n      <td>6948</td>\n      <td>TCN2</td>\n      <td>GRCh38</td>\n      <td>Benign</td>\n      <td>776</td>\n      <td>ClinGen:CA113848,UniProtKB:P20062#VAR_001640,O...</td>\n      <td>G/C</td>\n      <td>Arg259Pro</td>\n      <td>Arg</td>\n      <td>259</td>\n      <td>Pro</td>\n      <td>NM_000355.4</td>\n      <td>P20062</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>244</th>\n      <td>NM_000398.7(CYB5R3):c.350C&gt;G (p.Thr117Ser)</td>\n      <td>1727</td>\n      <td>CYB5R3</td>\n      <td>GRCh37</td>\n      <td>Benign</td>\n      <td>350</td>\n      <td>ClinGen:CA114081,UniProtKB:P00387#VAR_010751,O...</td>\n      <td>C/G</td>\n      <td>Thr117Ser</td>\n      <td>Thr</td>\n      <td>117</td>\n      <td>Ser</td>\n      <td>NM_000398.7</td>\n      <td>P00387</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>245</th>\n      <td>NM_000398.7(CYB5R3):c.350C&gt;G (p.Thr117Ser)</td>\n      <td>1727</td>\n      <td>CYB5R3</td>\n      <td>GRCh38</td>\n      <td>Benign</td>\n      <td>350</td>\n      <td>ClinGen:CA114081,UniProtKB:P00387#VAR_010751,O...</td>\n      <td>C/G</td>\n      <td>Thr117Ser</td>\n      <td>Thr</td>\n      <td>117</td>\n      <td>Ser</td>\n      <td>NM_000398.7</td>\n      <td>P00387</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>592</th>\n      <td>NM_000147.5(FUCA1):c.857A&gt;G (p.Gln286Arg)</td>\n      <td>2517</td>\n      <td>FUCA1</td>\n      <td>GRCh37</td>\n      <td>Benign</td>\n      <td>857</td>\n      <td>ClinGen:CA114444,UniProtKB:P04066#VAR_002444,O...</td>\n      <td>A/G</td>\n      <td>Gln286Arg</td>\n      <td>Gln</td>\n      <td>286</td>\n      <td>Arg</td>\n      <td>NM_000147.5</td>\n      <td>P04066</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>1213241</th>\n      <td>NM_000029.4(AGT):c.803T&gt;C (p.Met268Thr)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Benign</td>\n      <td>803</td>\n      <td>ClinGen:CA127784,UniProtKB:P01019#VAR_007096,O...</td>\n      <td>T/C</td>\n      <td>Met268Thr</td>\n      <td>Met</td>\n      <td>268</td>\n      <td>Thr</td>\n      <td>NM_000029.4</td>\n      <td>P01019</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213262</th>\n      <td>NM_000029.4(AGT):c.620C&gt;T (p.Thr207Met)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Benign</td>\n      <td>620</td>\n      <td>ClinGen:CA1448272,UniProtKB:P01019#VAR_007093</td>\n      <td>C/T</td>\n      <td>Thr207Met</td>\n      <td>Thr</td>\n      <td>207</td>\n      <td>Met</td>\n      <td>NM_000029.4</td>\n      <td>P01019</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213263</th>\n      <td>NM_000029.4(AGT):c.620C&gt;T (p.Thr207Met)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Benign</td>\n      <td>620</td>\n      <td>ClinGen:CA1448272,UniProtKB:P01019#VAR_007093</td>\n      <td>C/T</td>\n      <td>Thr207Met</td>\n      <td>Thr</td>\n      <td>207</td>\n      <td>Met</td>\n      <td>NM_000029.4</td>\n      <td>P01019</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213282</th>\n      <td>NM_000029.4(AGT):c.292G&gt;A (p.Glu98Lys)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh37</td>\n      <td>Benign</td>\n      <td>292</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Glu98Lys</td>\n      <td>Glu</td>\n      <td>98</td>\n      <td>Lys</td>\n      <td>NM_000029.4</td>\n      <td>P01019</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213283</th>\n      <td>NM_000029.4(AGT):c.292G&gt;A (p.Glu98Lys)</td>\n      <td>183</td>\n      <td>AGT</td>\n      <td>GRCh38</td>\n      <td>Benign</td>\n      <td>292</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Glu98Lys</td>\n      <td>Glu</td>\n      <td>98</td>\n      <td>Lys</td>\n      <td>NM_000029.4</td>\n      <td>P01019</td>\n      <td>1</td>\n    </tr>\n  </tbody>\n</table>\n<p>45009 rows × 15 columns</p>\n</div>"
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# find benign\n",
    "#df_benign_new = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance'].str.contains('Benign'))]\n",
    "df_benign_new = df_clinvar_not_vus[(df_clinvar_not_vus['ClinicalSignificance'] == 'Benign')]\n",
    "df_benign_new.drop_duplicates(inplace=True)\n",
    "df_benign_new"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "outputs": [
    {
     "data": {
      "text/plain": "                                             Name  GeneID GeneSymbol  \\\n0       NM_000014.6(A2M):c.2998A>G (p.Ile1000Val)       2        A2M   \n1         NM_000014.6(A2M):c.782C>T (p.Pro261Leu)       2        A2M   \n2         NM_000014.6(A2M):c.829G>A (p.Asp277Asn)       2        A2M   \n3        NM_000015.3(NAT2):c.609G>T (p.Glu203Asp)      10       NAT2   \n4        NM_000015.3(NAT2):c.838G>A (p.Val280Met)      10       NAT2   \n...                                           ...     ...        ...   \n22500  NM_213647.3(FGFR4):c.1772C>T (p.Ser591Phe)    2264      FGFR4   \n22501     NM_213647.3(FGFR4):c.28G>A (p.Val10Ile)    2264      FGFR4   \n22502         NM_213653.4(HJV):c.9G>C (p.Glu3Asp)  148738        HJV   \n22503   NM_213655.5(WNK1):c.2190G>C (p.Leu730Phe)   65125       WNK1   \n22504  NM_213720.3(CHCHD10):c.286C>A (p.Pro96Thr)  400916    CHCHD10   \n\n      ClinicalSignificance  CDS_position  \\\n0                   Benign          2998   \n1                   Benign           782   \n2                   Benign           829   \n3                   Benign           609   \n4                   Benign           838   \n...                    ...           ...   \n22500               Benign          1772   \n22501               Benign            28   \n22502               Benign             9   \n22503               Benign          2190   \n22504               Benign           286   \n\n                                                OtherIDs AlleleChange  \\\n0      ClinGen:CA127858,UniProtKB:P01023#VAR_000014,O...          A/G   \n1                                                      -          C/T   \n2                                                      -          G/A   \n3                                                      -          G/T   \n4                                                      -          G/A   \n...                                                  ...          ...   \n22500                                                  -          C/T   \n22501                                                  -          G/A   \n22502                                   ClinGen:CA349439          G/C   \n22503                                                  -          G/C   \n22504                                 ClinGen:CA10145267          C/A   \n\n      Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n0           Ile1000Val                         Ile               1000   \n1            Pro261Leu                         Pro                261   \n2            Asp277Asn                         Asp                277   \n3            Glu203Asp                         Glu                203   \n4            Val280Met                         Val                280   \n...                ...                         ...                ...   \n22500        Ser591Phe                         Ser                591   \n22501         Val10Ile                         Val                 10   \n22502          Glu3Asp                         Glu                  3   \n22503        Leu730Phe                         Leu                730   \n22504         Pro96Thr                         Pro                 96   \n\n      Amino_acids_after_3letters        NM_ID UniProtKB-AC  stars  \\\n0                            Val  NM_000014.6       P01023      1   \n1                            Leu  NM_000014.6       P01023      1   \n2                            Asn  NM_000014.6       P01023      1   \n3                            Asp  NM_000015.3       A4Z6T7      1   \n4                            Met  NM_000015.3       A4Z6T7      1   \n...                          ...          ...          ...    ...   \n22500                        Phe  NM_213647.3       P22455      1   \n22501                        Ile  NM_213647.3       P22455      1   \n22502                        Asp  NM_213653.4   A0A024R4F5      2   \n22503                        Phe  NM_213655.5     Q9H4A3-5      2   \n22504                        Thr  NM_213720.3       Q8WYQ3      2   \n\n               Assembly  \n0      [GRCh37, GRCh38]  \n1      [GRCh37, GRCh38]  \n2      [GRCh37, GRCh38]  \n3      [GRCh37, GRCh38]  \n4      [GRCh37, GRCh38]  \n...                 ...  \n22500  [GRCh37, GRCh38]  \n22501  [GRCh37, GRCh38]  \n22502  [GRCh37, GRCh38]  \n22503  [GRCh37, GRCh38]  \n22504  [GRCh37, GRCh38]  \n\n[22505 rows x 15 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>stars</th>\n      <th>Assembly</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NM_000014.6(A2M):c.2998A&gt;G (p.Ile1000Val)</td>\n      <td>2</td>\n      <td>A2M</td>\n      <td>Benign</td>\n      <td>2998</td>\n      <td>ClinGen:CA127858,UniProtKB:P01023#VAR_000014,O...</td>\n      <td>A/G</td>\n      <td>Ile1000Val</td>\n      <td>Ile</td>\n      <td>1000</td>\n      <td>Val</td>\n      <td>NM_000014.6</td>\n      <td>P01023</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NM_000014.6(A2M):c.782C&gt;T (p.Pro261Leu)</td>\n      <td>2</td>\n      <td>A2M</td>\n      <td>Benign</td>\n      <td>782</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Pro261Leu</td>\n      <td>Pro</td>\n      <td>261</td>\n      <td>Leu</td>\n      <td>NM_000014.6</td>\n      <td>P01023</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NM_000014.6(A2M):c.829G&gt;A (p.Asp277Asn)</td>\n      <td>2</td>\n      <td>A2M</td>\n      <td>Benign</td>\n      <td>829</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Asp277Asn</td>\n      <td>Asp</td>\n      <td>277</td>\n      <td>Asn</td>\n      <td>NM_000014.6</td>\n      <td>P01023</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NM_000015.3(NAT2):c.609G&gt;T (p.Glu203Asp)</td>\n      <td>10</td>\n      <td>NAT2</td>\n      <td>Benign</td>\n      <td>609</td>\n      <td>-</td>\n      <td>G/T</td>\n      <td>Glu203Asp</td>\n      <td>Glu</td>\n      <td>203</td>\n      <td>Asp</td>\n      <td>NM_000015.3</td>\n      <td>A4Z6T7</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NM_000015.3(NAT2):c.838G&gt;A (p.Val280Met)</td>\n      <td>10</td>\n      <td>NAT2</td>\n      <td>Benign</td>\n      <td>838</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Val280Met</td>\n      <td>Val</td>\n      <td>280</td>\n      <td>Met</td>\n      <td>NM_000015.3</td>\n      <td>A4Z6T7</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>22500</th>\n      <td>NM_213647.3(FGFR4):c.1772C&gt;T (p.Ser591Phe)</td>\n      <td>2264</td>\n      <td>FGFR4</td>\n      <td>Benign</td>\n      <td>1772</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Ser591Phe</td>\n      <td>Ser</td>\n      <td>591</td>\n      <td>Phe</td>\n      <td>NM_213647.3</td>\n      <td>P22455</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>22501</th>\n      <td>NM_213647.3(FGFR4):c.28G&gt;A (p.Val10Ile)</td>\n      <td>2264</td>\n      <td>FGFR4</td>\n      <td>Benign</td>\n      <td>28</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Val10Ile</td>\n      <td>Val</td>\n      <td>10</td>\n      <td>Ile</td>\n      <td>NM_213647.3</td>\n      <td>P22455</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>22502</th>\n      <td>NM_213653.4(HJV):c.9G&gt;C (p.Glu3Asp)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Benign</td>\n      <td>9</td>\n      <td>ClinGen:CA349439</td>\n      <td>G/C</td>\n      <td>Glu3Asp</td>\n      <td>Glu</td>\n      <td>3</td>\n      <td>Asp</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>22503</th>\n      <td>NM_213655.5(WNK1):c.2190G&gt;C (p.Leu730Phe)</td>\n      <td>65125</td>\n      <td>WNK1</td>\n      <td>Benign</td>\n      <td>2190</td>\n      <td>-</td>\n      <td>G/C</td>\n      <td>Leu730Phe</td>\n      <td>Leu</td>\n      <td>730</td>\n      <td>Phe</td>\n      <td>NM_213655.5</td>\n      <td>Q9H4A3-5</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>22504</th>\n      <td>NM_213720.3(CHCHD10):c.286C&gt;A (p.Pro96Thr)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Benign</td>\n      <td>286</td>\n      <td>ClinGen:CA10145267</td>\n      <td>C/A</td>\n      <td>Pro96Thr</td>\n      <td>Pro</td>\n      <td>96</td>\n      <td>Thr</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n  </tbody>\n</table>\n<p>22505 rows × 15 columns</p>\n</div>"
     },
     "execution_count": 161,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b_new = df_benign_new.groupby(col_list).agg({'Assembly': pd.Series.unique}).reset_index()\n",
    "b_new"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "outputs": [
    {
     "data": {
      "text/plain": "array(['Benign'], dtype=object)"
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b_new['ClinicalSignificance'].unique()"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "outputs": [
    {
     "data": {
      "text/plain": "                                             Name  GeneID GeneSymbol  \\\n0      NM_000016.6(ACADM):c.1034A>T (p.Asp345Val)      34      ACADM   \n1      NM_000016.6(ACADM):c.1042C>T (p.Arg348Cys)      34      ACADM   \n2      NM_000016.6(ACADM):c.1045C>G (p.Arg349Gly)      34      ACADM   \n3      NM_000016.6(ACADM):c.1046G>A (p.Arg349Gln)      34      ACADM   \n4      NM_000016.6(ACADM):c.1067T>C (p.Ile356Thr)      34      ACADM   \n...                                           ...     ...        ...   \n22500  NM_213647.3(FGFR4):c.1772C>T (p.Ser591Phe)    2264      FGFR4   \n22501     NM_213647.3(FGFR4):c.28G>A (p.Val10Ile)    2264      FGFR4   \n22502         NM_213653.4(HJV):c.9G>C (p.Glu3Asp)  148738        HJV   \n22503   NM_213655.5(WNK1):c.2190G>C (p.Leu730Phe)   65125       WNK1   \n22504  NM_213720.3(CHCHD10):c.286C>A (p.Pro96Thr)  400916    CHCHD10   \n\n      ClinicalSignificance  CDS_position            OtherIDs AlleleChange  \\\n0               Pathogenic          1034    ClinGen:CA913267          A/T   \n1               Pathogenic          1042                   -          C/T   \n2               Pathogenic          1045                   -          C/G   \n3               Pathogenic          1046                   -          G/A   \n4               Pathogenic          1067                   -          T/C   \n...                    ...           ...                 ...          ...   \n22500               Benign          1772                   -          C/T   \n22501               Benign            28                   -          G/A   \n22502               Benign             9    ClinGen:CA349439          G/C   \n22503               Benign          2190                   -          G/C   \n22504               Benign           286  ClinGen:CA10145267          C/A   \n\n      Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n0            Asp345Val                         Asp                345   \n1            Arg348Cys                         Arg                348   \n2            Arg349Gly                         Arg                349   \n3            Arg349Gln                         Arg                349   \n4            Ile356Thr                         Ile                356   \n...                ...                         ...                ...   \n22500        Ser591Phe                         Ser                591   \n22501         Val10Ile                         Val                 10   \n22502          Glu3Asp                         Glu                  3   \n22503        Leu730Phe                         Leu                730   \n22504         Pro96Thr                         Pro                 96   \n\n      Amino_acids_after_3letters        NM_ID UniProtKB-AC  stars  \\\n0                            Val  NM_000016.6   A0A0S2Z366      1   \n1                            Cys  NM_000016.6   A0A0S2Z366      1   \n2                            Gly  NM_000016.6   A0A0S2Z366      1   \n3                            Gln  NM_000016.6   A0A0S2Z366      1   \n4                            Thr  NM_000016.6   A0A0S2Z366      1   \n...                          ...          ...          ...    ...   \n22500                        Phe  NM_213647.3       P22455      1   \n22501                        Ile  NM_213647.3       P22455      1   \n22502                        Asp  NM_213653.4   A0A024R4F5      2   \n22503                        Phe  NM_213655.5     Q9H4A3-5      2   \n22504                        Thr  NM_213720.3       Q8WYQ3      2   \n\n               Assembly  \n0      [GRCh37, GRCh38]  \n1      [GRCh37, GRCh38]  \n2      [GRCh37, GRCh38]  \n3      [GRCh37, GRCh38]  \n4      [GRCh37, GRCh38]  \n...                 ...  \n22500  [GRCh37, GRCh38]  \n22501  [GRCh37, GRCh38]  \n22502  [GRCh37, GRCh38]  \n22503  [GRCh37, GRCh38]  \n22504  [GRCh37, GRCh38]  \n\n[33996 rows x 15 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>stars</th>\n      <th>Assembly</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NM_000016.6(ACADM):c.1034A&gt;T (p.Asp345Val)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1034</td>\n      <td>ClinGen:CA913267</td>\n      <td>A/T</td>\n      <td>Asp345Val</td>\n      <td>Asp</td>\n      <td>345</td>\n      <td>Val</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NM_000016.6(ACADM):c.1042C&gt;T (p.Arg348Cys)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1042</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Arg348Cys</td>\n      <td>Arg</td>\n      <td>348</td>\n      <td>Cys</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NM_000016.6(ACADM):c.1045C&gt;G (p.Arg349Gly)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1045</td>\n      <td>-</td>\n      <td>C/G</td>\n      <td>Arg349Gly</td>\n      <td>Arg</td>\n      <td>349</td>\n      <td>Gly</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NM_000016.6(ACADM):c.1046G&gt;A (p.Arg349Gln)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1046</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Arg349Gln</td>\n      <td>Arg</td>\n      <td>349</td>\n      <td>Gln</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NM_000016.6(ACADM):c.1067T&gt;C (p.Ile356Thr)</td>\n      <td>34</td>\n      <td>ACADM</td>\n      <td>Pathogenic</td>\n      <td>1067</td>\n      <td>-</td>\n      <td>T/C</td>\n      <td>Ile356Thr</td>\n      <td>Ile</td>\n      <td>356</td>\n      <td>Thr</td>\n      <td>NM_000016.6</td>\n      <td>A0A0S2Z366</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>22500</th>\n      <td>NM_213647.3(FGFR4):c.1772C&gt;T (p.Ser591Phe)</td>\n      <td>2264</td>\n      <td>FGFR4</td>\n      <td>Benign</td>\n      <td>1772</td>\n      <td>-</td>\n      <td>C/T</td>\n      <td>Ser591Phe</td>\n      <td>Ser</td>\n      <td>591</td>\n      <td>Phe</td>\n      <td>NM_213647.3</td>\n      <td>P22455</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>22501</th>\n      <td>NM_213647.3(FGFR4):c.28G&gt;A (p.Val10Ile)</td>\n      <td>2264</td>\n      <td>FGFR4</td>\n      <td>Benign</td>\n      <td>28</td>\n      <td>-</td>\n      <td>G/A</td>\n      <td>Val10Ile</td>\n      <td>Val</td>\n      <td>10</td>\n      <td>Ile</td>\n      <td>NM_213647.3</td>\n      <td>P22455</td>\n      <td>1</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>22502</th>\n      <td>NM_213653.4(HJV):c.9G&gt;C (p.Glu3Asp)</td>\n      <td>148738</td>\n      <td>HJV</td>\n      <td>Benign</td>\n      <td>9</td>\n      <td>ClinGen:CA349439</td>\n      <td>G/C</td>\n      <td>Glu3Asp</td>\n      <td>Glu</td>\n      <td>3</td>\n      <td>Asp</td>\n      <td>NM_213653.4</td>\n      <td>A0A024R4F5</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>22503</th>\n      <td>NM_213655.5(WNK1):c.2190G&gt;C (p.Leu730Phe)</td>\n      <td>65125</td>\n      <td>WNK1</td>\n      <td>Benign</td>\n      <td>2190</td>\n      <td>-</td>\n      <td>G/C</td>\n      <td>Leu730Phe</td>\n      <td>Leu</td>\n      <td>730</td>\n      <td>Phe</td>\n      <td>NM_213655.5</td>\n      <td>Q9H4A3-5</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n    <tr>\n      <th>22504</th>\n      <td>NM_213720.3(CHCHD10):c.286C&gt;A (p.Pro96Thr)</td>\n      <td>400916</td>\n      <td>CHCHD10</td>\n      <td>Benign</td>\n      <td>286</td>\n      <td>ClinGen:CA10145267</td>\n      <td>C/A</td>\n      <td>Pro96Thr</td>\n      <td>Pro</td>\n      <td>96</td>\n      <td>Thr</td>\n      <td>NM_213720.3</td>\n      <td>Q8WYQ3</td>\n      <td>2</td>\n      <td>[GRCh37, GRCh38]</td>\n    </tr>\n  </tbody>\n</table>\n<p>33996 rows × 15 columns</p>\n</div>"
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_p_b_combined = pd.concat([p_new, b_new])\n",
    "df_p_b_combined"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "outputs": [
    {
     "data": {
      "text/plain": "array([    34,     35,     37, ..., 387700,   6764,   2264])"
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_p_b_combined['GeneID'].unique()"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3212\n"
     ]
    }
   ],
   "source": [
    "# doesn't match number of unique genes\n",
    "# retrieve gene names from file names\n",
    "import os, sys\n",
    "eve_gene_name_list = []\n",
    "path = './data/eve_download/variant_files/'\n",
    "for file in os.listdir(path):\n",
    "    gene_name = file.strip().split('_')[0]\n",
    "    eve_gene_name_list.append(gene_name)\n",
    "print(len(eve_gene_name_list))"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "outputs": [
    {
     "data": {
      "text/plain": "                                             Name  GeneID GeneSymbol Assembly  \\\n6         NM_000410.4(HFE):c.845G>A (p.Cys282Tyr)    3077        HFE   GRCh37   \n7         NM_000410.4(HFE):c.845G>A (p.Cys282Tyr)    3077        HFE   GRCh38   \n8          NM_000410.4(HFE):c.187C>G (p.His63Asp)    3077        HFE   GRCh37   \n9          NM_000410.4(HFE):c.187C>G (p.His63Asp)    3077        HFE   GRCh38   \n10         NM_000410.4(HFE):c.193A>T (p.Ser65Cys)    3077        HFE   GRCh37   \n...                                           ...     ...        ...      ...   \n1213011   NM_005912.2(MC4R):c.110A>T (p.Asp37Val)    4160       MC4R   GRCh38   \n1213234  NM_000041.2(APOE):c.526C>T (p.Arg176Cys)     348       APOE   GRCh37   \n1213235  NM_000041.2(APOE):c.526C>T (p.Arg176Cys)     348       APOE   GRCh38   \n1213252  NM_000520.4(HEXA):c.745C>T (p.Arg249Trp)    3073       HEXA   GRCh37   \n1213253  NM_000520.4(HEXA):c.745C>T (p.Arg249Trp)    3073       HEXA   GRCh38   \n\n                                      ClinicalSignificance  CDS_position  \\\n6        Conflicting interpretations of pathogenicity; ...           845   \n7        Conflicting interpretations of pathogenicity; ...           845   \n8        Conflicting interpretations of pathogenicity; ...           187   \n9        Conflicting interpretations of pathogenicity; ...           187   \n10            Conflicting interpretations of pathogenicity           193   \n...                                                    ...           ...   \n1213011       Conflicting interpretations of pathogenicity           110   \n1213234                                      drug response           526   \n1213235                                      drug response           526   \n1213252                                      Benign; other           745   \n1213253                                      Benign; other           745   \n\n                                                  OtherIDs AlleleChange  \\\n6        UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...          G/A   \n7        UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...          G/A   \n8        OMIM:613609.0002,ClinGen:CA113797,UniProtKB:Q3...          C/G   \n9        OMIM:613609.0002,ClinGen:CA113797,UniProtKB:Q3...          C/G   \n10       ClinGen:CA339778,UniProtKB:Q30201#VAR_004397,O...          A/T   \n...                                                    ...          ...   \n1213011  ClinGen:CA213428,UniProtKB:P32245#VAR_010705,O...          A/T   \n1213234  ClinVar:441265,ClinVar:441266,ClinVar:666796,C...          C/T   \n1213235  ClinVar:441265,ClinVar:441266,ClinVar:666796,C...          C/T   \n1213252       ClinGen:CA276005,UniProtKB:P06865#VAR_003220          C/T   \n1213253       ClinGen:CA276005,UniProtKB:P06865#VAR_003220          C/T   \n\n        Protein_Patterns Amino_acids_before_3letters  Protein_positions  \\\n6              Cys282Tyr                         Cys                282   \n7              Cys282Tyr                         Cys                282   \n8               His63Asp                         His                 63   \n9               His63Asp                         His                 63   \n10              Ser65Cys                         Ser                 65   \n...                  ...                         ...                ...   \n1213011         Asp37Val                         Asp                 37   \n1213234        Arg176Cys                         Arg                176   \n1213235        Arg176Cys                         Arg                176   \n1213252        Arg249Trp                         Arg                249   \n1213253        Arg249Trp                         Arg                249   \n\n        Amino_acids_after_3letters        NM_ID UniProtKB-AC  stars  \n6                              Tyr  NM_000410.4       Q30201      1  \n7                              Tyr  NM_000410.4       Q30201      1  \n8                              Asp  NM_000410.4       Q30201      1  \n9                              Asp  NM_000410.4       Q30201      1  \n10                             Cys  NM_000410.4       Q30201      1  \n...                            ...          ...          ...    ...  \n1213011                        Val  NM_005912.2       P32245      1  \n1213234                        Cys  NM_000041.2       P02649      3  \n1213235                        Cys  NM_000041.2       P02649      3  \n1213252                        Trp  NM_000520.4       P06865      2  \n1213253                        Trp  NM_000520.4       P06865      2  \n\n[100305 rows x 15 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Name</th>\n      <th>GeneID</th>\n      <th>GeneSymbol</th>\n      <th>Assembly</th>\n      <th>ClinicalSignificance</th>\n      <th>CDS_position</th>\n      <th>OtherIDs</th>\n      <th>AlleleChange</th>\n      <th>Protein_Patterns</th>\n      <th>Amino_acids_before_3letters</th>\n      <th>Protein_positions</th>\n      <th>Amino_acids_after_3letters</th>\n      <th>NM_ID</th>\n      <th>UniProtKB-AC</th>\n      <th>stars</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>6</th>\n      <td>NM_000410.4(HFE):c.845G&gt;A (p.Cys282Tyr)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh37</td>\n      <td>Conflicting interpretations of pathogenicity; ...</td>\n      <td>845</td>\n      <td>UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...</td>\n      <td>G/A</td>\n      <td>Cys282Tyr</td>\n      <td>Cys</td>\n      <td>282</td>\n      <td>Tyr</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>7</th>\n      <td>NM_000410.4(HFE):c.845G&gt;A (p.Cys282Tyr)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh38</td>\n      <td>Conflicting interpretations of pathogenicity; ...</td>\n      <td>845</td>\n      <td>UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...</td>\n      <td>G/A</td>\n      <td>Cys282Tyr</td>\n      <td>Cys</td>\n      <td>282</td>\n      <td>Tyr</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>8</th>\n      <td>NM_000410.4(HFE):c.187C&gt;G (p.His63Asp)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh37</td>\n      <td>Conflicting interpretations of pathogenicity; ...</td>\n      <td>187</td>\n      <td>OMIM:613609.0002,ClinGen:CA113797,UniProtKB:Q3...</td>\n      <td>C/G</td>\n      <td>His63Asp</td>\n      <td>His</td>\n      <td>63</td>\n      <td>Asp</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>9</th>\n      <td>NM_000410.4(HFE):c.187C&gt;G (p.His63Asp)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh38</td>\n      <td>Conflicting interpretations of pathogenicity; ...</td>\n      <td>187</td>\n      <td>OMIM:613609.0002,ClinGen:CA113797,UniProtKB:Q3...</td>\n      <td>C/G</td>\n      <td>His63Asp</td>\n      <td>His</td>\n      <td>63</td>\n      <td>Asp</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>10</th>\n      <td>NM_000410.4(HFE):c.193A&gt;T (p.Ser65Cys)</td>\n      <td>3077</td>\n      <td>HFE</td>\n      <td>GRCh37</td>\n      <td>Conflicting interpretations of pathogenicity</td>\n      <td>193</td>\n      <td>ClinGen:CA339778,UniProtKB:Q30201#VAR_004397,O...</td>\n      <td>A/T</td>\n      <td>Ser65Cys</td>\n      <td>Ser</td>\n      <td>65</td>\n      <td>Cys</td>\n      <td>NM_000410.4</td>\n      <td>Q30201</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>1213011</th>\n      <td>NM_005912.2(MC4R):c.110A&gt;T (p.Asp37Val)</td>\n      <td>4160</td>\n      <td>MC4R</td>\n      <td>GRCh38</td>\n      <td>Conflicting interpretations of pathogenicity</td>\n      <td>110</td>\n      <td>ClinGen:CA213428,UniProtKB:P32245#VAR_010705,O...</td>\n      <td>A/T</td>\n      <td>Asp37Val</td>\n      <td>Asp</td>\n      <td>37</td>\n      <td>Val</td>\n      <td>NM_005912.2</td>\n      <td>P32245</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>1213234</th>\n      <td>NM_000041.2(APOE):c.526C&gt;T (p.Arg176Cys)</td>\n      <td>348</td>\n      <td>APOE</td>\n      <td>GRCh37</td>\n      <td>drug response</td>\n      <td>526</td>\n      <td>ClinVar:441265,ClinVar:441266,ClinVar:666796,C...</td>\n      <td>C/T</td>\n      <td>Arg176Cys</td>\n      <td>Arg</td>\n      <td>176</td>\n      <td>Cys</td>\n      <td>NM_000041.2</td>\n      <td>P02649</td>\n      <td>3</td>\n    </tr>\n    <tr>\n      <th>1213235</th>\n      <td>NM_000041.2(APOE):c.526C&gt;T (p.Arg176Cys)</td>\n      <td>348</td>\n      <td>APOE</td>\n      <td>GRCh38</td>\n      <td>drug response</td>\n      <td>526</td>\n      <td>ClinVar:441265,ClinVar:441266,ClinVar:666796,C...</td>\n      <td>C/T</td>\n      <td>Arg176Cys</td>\n      <td>Arg</td>\n      <td>176</td>\n      <td>Cys</td>\n      <td>NM_000041.2</td>\n      <td>P02649</td>\n      <td>3</td>\n    </tr>\n    <tr>\n      <th>1213252</th>\n      <td>NM_000520.4(HEXA):c.745C&gt;T (p.Arg249Trp)</td>\n      <td>3073</td>\n      <td>HEXA</td>\n      <td>GRCh37</td>\n      <td>Benign; other</td>\n      <td>745</td>\n      <td>ClinGen:CA276005,UniProtKB:P06865#VAR_003220</td>\n      <td>C/T</td>\n      <td>Arg249Trp</td>\n      <td>Arg</td>\n      <td>249</td>\n      <td>Trp</td>\n      <td>NM_000520.4</td>\n      <td>P06865</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1213253</th>\n      <td>NM_000520.4(HEXA):c.745C&gt;T (p.Arg249Trp)</td>\n      <td>3073</td>\n      <td>HEXA</td>\n      <td>GRCh38</td>\n      <td>Benign; other</td>\n      <td>745</td>\n      <td>ClinGen:CA276005,UniProtKB:P06865#VAR_003220</td>\n      <td>C/T</td>\n      <td>Arg249Trp</td>\n      <td>Arg</td>\n      <td>249</td>\n      <td>Trp</td>\n      <td>NM_000520.4</td>\n      <td>P06865</td>\n      <td>2</td>\n    </tr>\n  </tbody>\n</table>\n<p>100305 rows × 15 columns</p>\n</div>"
     },
     "execution_count": 176,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# keep only rows that match the gene name list\n",
    "df_filtered_with_eve = df_clinvar_not_vus[df_clinvar_not_vus['GeneSymbol'].isin(eve_gene_name_list)]\n",
    "df_filtered_with_eve"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "outputs": [
    {
     "data": {
      "text/plain": "array(['HFE', 'WDR35', 'HOGA1', ..., 'TAF2', 'DPH1', 'MOCOS'],\n      dtype=object)"
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_filtered_with_eve['GeneSymbol'].unique()"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}