{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "'''filter out clinvar variants to retrieve protein data'''\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/apps/anaconda3/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3444: DtypeWarning: Columns (18) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv('./variant_summary.txt', sep='\\t')\n",
    "# 3,251,310 rows, not show"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>...</th>\n",
       "      <th>ReviewStatus</th>\n",
       "      <th>NumberSubmitters</th>\n",
       "      <th>Guidelines</th>\n",
       "      <th>TestedInGTR</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>SubmitterCategories</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PositionVCF</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>15043</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n",
       "      <td>9640</td>\n",
       "      <td>ZNF592</td>\n",
       "      <td>HGNC:28986</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>Jun 29, 2015</td>\n",
       "      <td>150829393</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>84799209</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>15044</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_017547.4(FOXRED1):c.694C&gt;T (p.Gln232Ter)</td>\n",
       "      <td>55572</td>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>HGNC:26927</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 30, 2019</td>\n",
       "      <td>267606829</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>3</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA113792,OMIM:613622.0001</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>126275389</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>15045</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n",
       "      <td>55572</td>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>HGNC:26927</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Oct 01, 2010</td>\n",
       "      <td>267606830</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>126277517</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>15046</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_025152.3(NUBPL):c.166G&gt;A (p.Gly56Arg)</td>\n",
       "      <td>80224</td>\n",
       "      <td>NUBPL</td>\n",
       "      <td>HGNC:20278</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>Aug 24, 2020</td>\n",
       "      <td>200401432</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>3</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...</td>\n",
       "      <td>2</td>\n",
       "      <td>214885</td>\n",
       "      <td>31562125</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>15048</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000410.4(HFE):c.845G&gt;A (p.Cys282Tyr)</td>\n",
       "      <td>3077</td>\n",
       "      <td>HFE</td>\n",
       "      <td>HGNC:4886</td>\n",
       "      <td>Conflicting interpretations of pathogenicity; ...</td>\n",
       "      <td>1</td>\n",
       "      <td>Jun 30, 2022</td>\n",
       "      <td>1800562</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria provided, conflicting interpretations</td>\n",
       "      <td>35</td>\n",
       "      <td>ACMG2021,ACMG2022</td>\n",
       "      <td>Y</td>\n",
       "      <td>UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>26092913</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251296</th>\n",
       "      <td>1863473</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000089.4(COL1A2):c.3159G&gt;A (p.Arg1053_Gly10...</td>\n",
       "      <td>1278</td>\n",
       "      <td>COL1A2</td>\n",
       "      <td>HGNC:2198</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>1806466</td>\n",
       "      <td>94427061</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251300</th>\n",
       "      <td>1863475</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_024616.3(C3orf52):c.492T&gt;A (p.Tyr164Ter)</td>\n",
       "      <td>79669</td>\n",
       "      <td>C3orf52</td>\n",
       "      <td>HGNC:26255</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 21, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM:611956.0001</td>\n",
       "      <td>1</td>\n",
       "      <td>1806468</td>\n",
       "      <td>112112988</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251302</th>\n",
       "      <td>1863476</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_024616.3(C3orf52):c.34G&gt;T (p.Glu12Ter)</td>\n",
       "      <td>79669</td>\n",
       "      <td>C3orf52</td>\n",
       "      <td>HGNC:26255</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 21, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM:611956.0002</td>\n",
       "      <td>1</td>\n",
       "      <td>1806469</td>\n",
       "      <td>112086441</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251306</th>\n",
       "      <td>1863478</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_004456.5(EZH2):c.395C&gt;T (p.Pro132Leu)</td>\n",
       "      <td>2146</td>\n",
       "      <td>EZH2</td>\n",
       "      <td>HGNC:3527</td>\n",
       "      <td>not provided</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion provided</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>1806471</td>\n",
       "      <td>148829817</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251308</th>\n",
       "      <td>1863479</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_001972.4(ELANE):c.640G&gt;C (p.Gly214Arg)</td>\n",
       "      <td>1991</td>\n",
       "      <td>ELANE</td>\n",
       "      <td>HGNC:3309</td>\n",
       "      <td>not provided</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion provided</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>1806472</td>\n",
       "      <td>856000</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1414257 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #AlleleID                       Type  \\\n",
       "5            15043  single nucleotide variant   \n",
       "7            15044  single nucleotide variant   \n",
       "9            15045  single nucleotide variant   \n",
       "11           15046  single nucleotide variant   \n",
       "13           15048  single nucleotide variant   \n",
       "...            ...                        ...   \n",
       "3251296    1863473  single nucleotide variant   \n",
       "3251300    1863475  single nucleotide variant   \n",
       "3251302    1863476  single nucleotide variant   \n",
       "3251306    1863478  single nucleotide variant   \n",
       "3251308    1863479  single nucleotide variant   \n",
       "\n",
       "                                                      Name  GeneID GeneSymbol  \\\n",
       "5             NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n",
       "7              NM_017547.4(FOXRED1):c.694C>T (p.Gln232Ter)   55572    FOXRED1   \n",
       "9             NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)   55572    FOXRED1   \n",
       "11                NM_025152.3(NUBPL):c.166G>A (p.Gly56Arg)   80224      NUBPL   \n",
       "13                 NM_000410.4(HFE):c.845G>A (p.Cys282Tyr)    3077        HFE   \n",
       "...                                                    ...     ...        ...   \n",
       "3251296  NM_000089.4(COL1A2):c.3159G>A (p.Arg1053_Gly10...    1278     COL1A2   \n",
       "3251300        NM_024616.3(C3orf52):c.492T>A (p.Tyr164Ter)   79669    C3orf52   \n",
       "3251302          NM_024616.3(C3orf52):c.34G>T (p.Glu12Ter)   79669    C3orf52   \n",
       "3251306           NM_004456.5(EZH2):c.395C>T (p.Pro132Leu)    2146       EZH2   \n",
       "3251308          NM_001972.4(ELANE):c.640G>C (p.Gly214Arg)    1991      ELANE   \n",
       "\n",
       "            HGNC_ID                               ClinicalSignificance  \\\n",
       "5        HGNC:28986                             Uncertain significance   \n",
       "7        HGNC:26927                                         Pathogenic   \n",
       "9        HGNC:26927                                         Pathogenic   \n",
       "11       HGNC:20278                             Uncertain significance   \n",
       "13        HGNC:4886  Conflicting interpretations of pathogenicity; ...   \n",
       "...             ...                                                ...   \n",
       "3251296   HGNC:2198                                  Likely pathogenic   \n",
       "3251300  HGNC:26255                                         Pathogenic   \n",
       "3251302  HGNC:26255                                         Pathogenic   \n",
       "3251306   HGNC:3527                                       not provided   \n",
       "3251308   HGNC:3309                                       not provided   \n",
       "\n",
       "         ClinSigSimple LastEvaluated  RS# (dbSNP)  ...  \\\n",
       "5                    0  Jun 29, 2015    150829393  ...   \n",
       "7                    1  Dec 30, 2019    267606829  ...   \n",
       "9                    1  Oct 01, 2010    267606830  ...   \n",
       "11                   0  Aug 24, 2020    200401432  ...   \n",
       "13                   1  Jun 30, 2022      1800562  ...   \n",
       "...                ...           ...          ...  ...   \n",
       "3251296              1             -           -1  ...   \n",
       "3251300              1  Dec 21, 2022           -1  ...   \n",
       "3251302              1  Dec 21, 2022           -1  ...   \n",
       "3251306              0             -           -1  ...   \n",
       "3251308              0             -           -1  ...   \n",
       "\n",
       "                                              ReviewStatus NumberSubmitters  \\\n",
       "5                           no assertion criteria provided                1   \n",
       "7        criteria provided, multiple submitters, no con...                3   \n",
       "9                           no assertion criteria provided                1   \n",
       "11       criteria provided, multiple submitters, no con...                3   \n",
       "13          criteria provided, conflicting interpretations               35   \n",
       "...                                                    ...              ...   \n",
       "3251296                     no assertion criteria provided                1   \n",
       "3251300                     no assertion criteria provided                1   \n",
       "3251302                     no assertion criteria provided                1   \n",
       "3251306                              no assertion provided                1   \n",
       "3251308                              no assertion provided                1   \n",
       "\n",
       "                Guidelines TestedInGTR  \\\n",
       "5                        -           N   \n",
       "7                        -           N   \n",
       "9                        -           N   \n",
       "11                       -           N   \n",
       "13       ACMG2021,ACMG2022           Y   \n",
       "...                    ...         ...   \n",
       "3251296                  -           N   \n",
       "3251300                  -           N   \n",
       "3251302                  -           N   \n",
       "3251306                  -           N   \n",
       "3251308                  -           N   \n",
       "\n",
       "                                                  OtherIDs  \\\n",
       "5        OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...   \n",
       "7                        ClinGen:CA113792,OMIM:613622.0001   \n",
       "9        ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...   \n",
       "11       ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...   \n",
       "13       UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...   \n",
       "...                                                    ...   \n",
       "3251296                                                  -   \n",
       "3251300                                   OMIM:611956.0001   \n",
       "3251302                                   OMIM:611956.0002   \n",
       "3251306                                                  -   \n",
       "3251308                                                  -   \n",
       "\n",
       "        SubmitterCategories VariationID PositionVCF ReferenceAlleleVCF  \\\n",
       "5                         1           4    84799209                  G   \n",
       "7                         3           5   126275389                  C   \n",
       "9                         1           6   126277517                  A   \n",
       "11                        2      214885    31562125                  G   \n",
       "13                        3           9    26092913                  G   \n",
       "...                     ...         ...         ...                ...   \n",
       "3251296                   2     1806466    94427061                  G   \n",
       "3251300                   1     1806468   112112988                  T   \n",
       "3251302                   1     1806469   112086441                  G   \n",
       "3251306                   1     1806471   148829817                  G   \n",
       "3251308                   1     1806472      856000                  G   \n",
       "\n",
       "         AlternateAlleleVCF  \n",
       "5                         A  \n",
       "7                         T  \n",
       "9                         G  \n",
       "11                        A  \n",
       "13                        A  \n",
       "...                     ...  \n",
       "3251296                   A  \n",
       "3251300                   A  \n",
       "3251302                   T  \n",
       "3251306                   A  \n",
       "3251308                   C  \n",
       "\n",
       "[1414257 rows x 34 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df[(df['Type'] == 'single nucleotide variant')]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "df_meta = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n",
      "something is wrong\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>...</th>\n",
       "      <th>NumberSubmitters</th>\n",
       "      <th>Guidelines</th>\n",
       "      <th>TestedInGTR</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>SubmitterCategories</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PositionVCF</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>stars</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>15044</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_017547.4(FOXRED1):c.694C&gt;T (p.Gln232Ter)</td>\n",
       "      <td>55572</td>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>HGNC:26927</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 30, 2019</td>\n",
       "      <td>267606829</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA113792,OMIM:613622.0001</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>126275389</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>15046</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_025152.3(NUBPL):c.166G&gt;A (p.Gly56Arg)</td>\n",
       "      <td>80224</td>\n",
       "      <td>NUBPL</td>\n",
       "      <td>HGNC:20278</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>Aug 24, 2020</td>\n",
       "      <td>200401432</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...</td>\n",
       "      <td>2</td>\n",
       "      <td>214885</td>\n",
       "      <td>31562125</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>15048</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000410.4(HFE):c.845G&gt;A (p.Cys282Tyr)</td>\n",
       "      <td>3077</td>\n",
       "      <td>HFE</td>\n",
       "      <td>HGNC:4886</td>\n",
       "      <td>Conflicting interpretations of pathogenicity; ...</td>\n",
       "      <td>1</td>\n",
       "      <td>Jun 30, 2022</td>\n",
       "      <td>1800562</td>\n",
       "      <td>...</td>\n",
       "      <td>35</td>\n",
       "      <td>ACMG2021,ACMG2022</td>\n",
       "      <td>Y</td>\n",
       "      <td>UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>26092913</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>15049</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000410.4(HFE):c.187C&gt;G (p.His63Asp)</td>\n",
       "      <td>3077</td>\n",
       "      <td>HFE</td>\n",
       "      <td>HGNC:4886</td>\n",
       "      <td>Conflicting interpretations of pathogenicity; ...</td>\n",
       "      <td>1</td>\n",
       "      <td>May 01, 2022</td>\n",
       "      <td>1799945</td>\n",
       "      <td>...</td>\n",
       "      <td>41</td>\n",
       "      <td>ACMG2021,ACMG2022</td>\n",
       "      <td>Y</td>\n",
       "      <td>OMIM:613609.0002,ClinGen:CA113797,UniProtKB:Q3...</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>26090951</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>15050</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000410.4(HFE):c.193A&gt;T (p.Ser65Cys)</td>\n",
       "      <td>3077</td>\n",
       "      <td>HFE</td>\n",
       "      <td>HGNC:4886</td>\n",
       "      <td>Conflicting interpretations of pathogenicity</td>\n",
       "      <td>1</td>\n",
       "      <td>Jul 01, 2022</td>\n",
       "      <td>1800730</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>ACMG2021,ACMG2022</td>\n",
       "      <td>Y</td>\n",
       "      <td>ClinGen:CA339778,UniProtKB:Q30201#VAR_004397,O...</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>26090957</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251260</th>\n",
       "      <td>1863453</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_004006.3(DMD):c.9974+1G&gt;C</td>\n",
       "      <td>1756</td>\n",
       "      <td>DMD</td>\n",
       "      <td>HGNC:2928</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 16, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>1806444</td>\n",
       "      <td>31182737</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251262</th>\n",
       "      <td>1863454</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_004006.3(DMD):c.5449-2A&gt;T</td>\n",
       "      <td>1756</td>\n",
       "      <td>DMD</td>\n",
       "      <td>HGNC:2928</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 16, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>1806445</td>\n",
       "      <td>32346082</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251264</th>\n",
       "      <td>1863455</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_001142966.3(GREB1L):c.1426G&gt;A (p.Glu476Lys)</td>\n",
       "      <td>80000</td>\n",
       "      <td>GREB1L</td>\n",
       "      <td>HGNC:31042</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>Dec 22, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>1806446</td>\n",
       "      <td>21449542</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251266</th>\n",
       "      <td>1863457</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_032436.4(CHAMP1):c.67G&gt;A (p.Gly23Ser)</td>\n",
       "      <td>283489</td>\n",
       "      <td>CHAMP1</td>\n",
       "      <td>HGNC:20311</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>Dec 21, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>1806448</td>\n",
       "      <td>114323909</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251268</th>\n",
       "      <td>1863458</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_021072.4(HCN1):c.1460T&gt;C (p.Met487Thr)</td>\n",
       "      <td>348980</td>\n",
       "      <td>HCN1</td>\n",
       "      <td>HGNC:4845</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>Jun 07, 2021</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>1806449</td>\n",
       "      <td>45303757</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1361940 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #AlleleID                       Type  \\\n",
       "7            15044  single nucleotide variant   \n",
       "11           15046  single nucleotide variant   \n",
       "13           15048  single nucleotide variant   \n",
       "15           15049  single nucleotide variant   \n",
       "17           15050  single nucleotide variant   \n",
       "...            ...                        ...   \n",
       "3251260    1863453  single nucleotide variant   \n",
       "3251262    1863454  single nucleotide variant   \n",
       "3251264    1863455  single nucleotide variant   \n",
       "3251266    1863457  single nucleotide variant   \n",
       "3251268    1863458  single nucleotide variant   \n",
       "\n",
       "                                                   Name  GeneID GeneSymbol  \\\n",
       "7           NM_017547.4(FOXRED1):c.694C>T (p.Gln232Ter)   55572    FOXRED1   \n",
       "11             NM_025152.3(NUBPL):c.166G>A (p.Gly56Arg)   80224      NUBPL   \n",
       "13              NM_000410.4(HFE):c.845G>A (p.Cys282Tyr)    3077        HFE   \n",
       "15               NM_000410.4(HFE):c.187C>G (p.His63Asp)    3077        HFE   \n",
       "17               NM_000410.4(HFE):c.193A>T (p.Ser65Cys)    3077        HFE   \n",
       "...                                                 ...     ...        ...   \n",
       "3251260                    NM_004006.3(DMD):c.9974+1G>C    1756        DMD   \n",
       "3251262                    NM_004006.3(DMD):c.5449-2A>T    1756        DMD   \n",
       "3251264  NM_001142966.3(GREB1L):c.1426G>A (p.Glu476Lys)   80000     GREB1L   \n",
       "3251266        NM_032436.4(CHAMP1):c.67G>A (p.Gly23Ser)  283489     CHAMP1   \n",
       "3251268       NM_021072.4(HCN1):c.1460T>C (p.Met487Thr)  348980       HCN1   \n",
       "\n",
       "            HGNC_ID                               ClinicalSignificance  \\\n",
       "7        HGNC:26927                                         Pathogenic   \n",
       "11       HGNC:20278                             Uncertain significance   \n",
       "13        HGNC:4886  Conflicting interpretations of pathogenicity; ...   \n",
       "15        HGNC:4886  Conflicting interpretations of pathogenicity; ...   \n",
       "17        HGNC:4886       Conflicting interpretations of pathogenicity   \n",
       "...             ...                                                ...   \n",
       "3251260   HGNC:2928                                  Likely pathogenic   \n",
       "3251262   HGNC:2928                                  Likely pathogenic   \n",
       "3251264  HGNC:31042                             Uncertain significance   \n",
       "3251266  HGNC:20311                             Uncertain significance   \n",
       "3251268   HGNC:4845                             Uncertain significance   \n",
       "\n",
       "         ClinSigSimple LastEvaluated  RS# (dbSNP)  ... NumberSubmitters  \\\n",
       "7                    1  Dec 30, 2019    267606829  ...                3   \n",
       "11                   0  Aug 24, 2020    200401432  ...                3   \n",
       "13                   1  Jun 30, 2022      1800562  ...               35   \n",
       "15                   1  May 01, 2022      1799945  ...               41   \n",
       "17                   1  Jul 01, 2022      1800730  ...               12   \n",
       "...                ...           ...          ...  ...              ...   \n",
       "3251260              1  Dec 16, 2022           -1  ...                1   \n",
       "3251262              1  Dec 16, 2022           -1  ...                1   \n",
       "3251264              0  Dec 22, 2022           -1  ...                1   \n",
       "3251266              0  Dec 21, 2022           -1  ...                1   \n",
       "3251268              0  Jun 07, 2021           -1  ...                1   \n",
       "\n",
       "                Guidelines TestedInGTR  \\\n",
       "7                        -           N   \n",
       "11                       -           N   \n",
       "13       ACMG2021,ACMG2022           Y   \n",
       "15       ACMG2021,ACMG2022           Y   \n",
       "17       ACMG2021,ACMG2022           Y   \n",
       "...                    ...         ...   \n",
       "3251260                  -           N   \n",
       "3251262                  -           N   \n",
       "3251264                  -           N   \n",
       "3251266                  -           N   \n",
       "3251268                  -           N   \n",
       "\n",
       "                                                  OtherIDs  \\\n",
       "7                        ClinGen:CA113792,OMIM:613622.0001   \n",
       "11       ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...   \n",
       "13       UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...   \n",
       "15       OMIM:613609.0002,ClinGen:CA113797,UniProtKB:Q3...   \n",
       "17       ClinGen:CA339778,UniProtKB:Q30201#VAR_004397,O...   \n",
       "...                                                    ...   \n",
       "3251260                                                  -   \n",
       "3251262                                                  -   \n",
       "3251264                                                  -   \n",
       "3251266                                                  -   \n",
       "3251268                                                  -   \n",
       "\n",
       "        SubmitterCategories VariationID PositionVCF ReferenceAlleleVCF  \\\n",
       "7                         3           5   126275389                  C   \n",
       "11                        2      214885    31562125                  G   \n",
       "13                        3           9    26092913                  G   \n",
       "15                        3          10    26090951                  C   \n",
       "17                        3          11    26090957                  A   \n",
       "...                     ...         ...         ...                ...   \n",
       "3251260                   2     1806444    31182737                  C   \n",
       "3251262                   2     1806445    32346082                  T   \n",
       "3251264                   2     1806446    21449542                  G   \n",
       "3251266                   2     1806448   114323909                  G   \n",
       "3251268                   2     1806449    45303757                  A   \n",
       "\n",
       "        AlternateAlleleVCF  stars  \n",
       "7                        T      2  \n",
       "11                       A      2  \n",
       "13                       A      1  \n",
       "15                       G      1  \n",
       "17                       T      1  \n",
       "...                    ...    ...  \n",
       "3251260                  G      1  \n",
       "3251262                  A      1  \n",
       "3251264                  A      1  \n",
       "3251266                  A      1  \n",
       "3251268                  G      1  \n",
       "\n",
       "[1361940 rows x 34 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def assign_stars(row):\n",
    "    if 'criteria provided, single submitter' in row or 'criteria provided, conflicting interpretations' in row:\n",
    "        return 1\n",
    "    elif 'criteria provided, multiple submitters' in row:\n",
    "        return 2\n",
    "    elif 'expert panel' in row:\n",
    "        return 3\n",
    "    elif 'guideline' in row:\n",
    "        return 4\n",
    "    elif 'no assertion' in row:\n",
    "        return None\n",
    "    else:\n",
    "        #print(row)\n",
    "        # it's phrased as 'no interpretation for the single variant'\n",
    "        print('something is wrong')\n",
    "\n",
    "df_meta['stars'] = df_meta['ReviewStatus'].apply(assign_stars)\n",
    "# pandas can handle NaNs as integers now\n",
    "df_meta['stars'] = df_meta['stars'].astype('Int64')\n",
    "df_meta = df_meta[df_meta['stars'] >= 1]\n",
    "df_meta.drop(['ReviewStatus'], axis=1, inplace=True)\n",
    "df_meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Pathogenic', 'Uncertain significance',\n",
       "       'Conflicting interpretations of pathogenicity; other; risk factor',\n",
       "       'Conflicting interpretations of pathogenicity; other',\n",
       "       'Conflicting interpretations of pathogenicity', 'Benign',\n",
       "       'Likely pathogenic', 'Pathogenic/Likely pathogenic',\n",
       "       'Conflicting interpretations of pathogenicity; risk factor',\n",
       "       'drug response',\n",
       "       'Pathogenic/Pathogenic, low penetrance; risk factor',\n",
       "       'Conflicting interpretations of pathogenicity; association; risk factor',\n",
       "       'Benign/Likely benign', 'Likely benign',\n",
       "       'Benign/Likely benign; other',\n",
       "       'Pathogenic/Likely pathogenic; other',\n",
       "       'Pathogenic/Likely pathogenic; risk factor', 'Benign; other',\n",
       "       'Likely benign; other', 'Pathogenic; risk factor',\n",
       "       'Conflicting interpretations of pathogenicity; association',\n",
       "       'Benign/Likely benign; association',\n",
       "       'Benign/Likely benign; other; risk factor',\n",
       "       'Pathogenic; drug response', 'drug response; other',\n",
       "       'Likely benign; drug response; other', 'Likely risk allele',\n",
       "       'Benign/Likely benign; risk factor',\n",
       "       'Likely pathogenic; drug response',\n",
       "       'Conflicting interpretations of pathogenicity; drug response',\n",
       "       'Likely pathogenic; other', 'Benign; risk factor',\n",
       "       'Pathogenic; drug response; other',\n",
       "       'Uncertain significance; drug response',\n",
       "       'Likely pathogenic/Likely risk allele',\n",
       "       'Uncertain significance/Uncertain risk allele', 'other',\n",
       "       'not provided', 'Pathogenic; other', 'Benign; drug response',\n",
       "       'Pathogenic/Likely risk allele', 'risk factor', 'Affects',\n",
       "       'Uncertain significance; association', 'association',\n",
       "       'Benign/Likely benign; drug response',\n",
       "       'Benign/Likely benign; drug response; other',\n",
       "       'Uncertain significance; other', 'Benign; confers sensitivity',\n",
       "       'protective', 'Uncertain risk allele',\n",
       "       'Uncertain significance; Affects', 'association not found',\n",
       "       'Likely benign; risk factor',\n",
       "       'Pathogenic/Likely pathogenic/Likely risk allele',\n",
       "       'Likely pathogenic; risk factor', 'Benign; association',\n",
       "       'confers sensitivity', 'Uncertain significance; risk factor'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta['ClinicalSignificance'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>...</th>\n",
       "      <th>Guidelines</th>\n",
       "      <th>TestedInGTR</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>SubmitterCategories</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PositionVCF</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>stars</th>\n",
       "      <th>classes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>15044</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_017547.4(FOXRED1):c.694C&gt;T (p.Gln232Ter)</td>\n",
       "      <td>55572</td>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>HGNC:26927</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 30, 2019</td>\n",
       "      <td>267606829</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA113792,OMIM:613622.0001</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>126275389</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>15053</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000410.4(HFE):c.892+48G&gt;A</td>\n",
       "      <td>3077</td>\n",
       "      <td>HFE</td>\n",
       "      <td>HGNC:4886</td>\n",
       "      <td>Benign</td>\n",
       "      <td>0</td>\n",
       "      <td>Sep 11, 2018</td>\n",
       "      <td>1800758</td>\n",
       "      <td>...</td>\n",
       "      <td>ACMG2021,ACMG2022</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA113800,OMIM:613609.0004</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "      <td>26093008</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>15058</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n",
       "      <td>3077</td>\n",
       "      <td>HFE</td>\n",
       "      <td>HGNC:4886</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Sep 04, 2021</td>\n",
       "      <td>111033563</td>\n",
       "      <td>...</td>\n",
       "      <td>ACMG2021,ACMG2022</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>26092916</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>15062</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_020779.4(WDR35):c.2590G&gt;A (p.Ala864Thr)</td>\n",
       "      <td>57539</td>\n",
       "      <td>WDR35</td>\n",
       "      <td>HGNC:29250</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>May 25, 2017</td>\n",
       "      <td>267607175</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA339782,UniProtKB:Q9P2L0#VAR_064582,O...</td>\n",
       "      <td>3</td>\n",
       "      <td>23</td>\n",
       "      <td>19933469</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>15066</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_001042472.3(ABHD12):c.1054C&gt;T (p.Arg352Ter)</td>\n",
       "      <td>26090</td>\n",
       "      <td>ABHD12</td>\n",
       "      <td>HGNC:15868</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Jan 05, 2022</td>\n",
       "      <td>267606624</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA113811,OMIM:613599.0004</td>\n",
       "      <td>3</td>\n",
       "      <td>27</td>\n",
       "      <td>25302322</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251244</th>\n",
       "      <td>1863445</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_004006.3(DMD):c.9224+2T&gt;A</td>\n",
       "      <td>1756</td>\n",
       "      <td>DMD</td>\n",
       "      <td>HGNC:2928</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 16, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>1806436</td>\n",
       "      <td>31323596</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251246</th>\n",
       "      <td>1863446</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_004006.3(DMD):c.9361+1G&gt;T</td>\n",
       "      <td>1756</td>\n",
       "      <td>DMD</td>\n",
       "      <td>HGNC:2928</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 16, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>1806437</td>\n",
       "      <td>31223046</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251250</th>\n",
       "      <td>1863448</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_004006.3(DMD):c.1704+2T&gt;C</td>\n",
       "      <td>1756</td>\n",
       "      <td>DMD</td>\n",
       "      <td>HGNC:2928</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 16, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>1806439</td>\n",
       "      <td>32573743</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251260</th>\n",
       "      <td>1863453</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_004006.3(DMD):c.9974+1G&gt;C</td>\n",
       "      <td>1756</td>\n",
       "      <td>DMD</td>\n",
       "      <td>HGNC:2928</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 16, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>1806444</td>\n",
       "      <td>31182737</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251262</th>\n",
       "      <td>1863454</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_004006.3(DMD):c.5449-2A&gt;T</td>\n",
       "      <td>1756</td>\n",
       "      <td>DMD</td>\n",
       "      <td>HGNC:2928</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 16, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>1806445</td>\n",
       "      <td>32346082</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>661134 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #AlleleID                       Type  \\\n",
       "7            15044  single nucleotide variant   \n",
       "23           15053  single nucleotide variant   \n",
       "33           15058  single nucleotide variant   \n",
       "41           15062  single nucleotide variant   \n",
       "49           15066  single nucleotide variant   \n",
       "...            ...                        ...   \n",
       "3251244    1863445  single nucleotide variant   \n",
       "3251246    1863446  single nucleotide variant   \n",
       "3251250    1863448  single nucleotide variant   \n",
       "3251260    1863453  single nucleotide variant   \n",
       "3251262    1863454  single nucleotide variant   \n",
       "\n",
       "                                                   Name  GeneID GeneSymbol  \\\n",
       "7           NM_017547.4(FOXRED1):c.694C>T (p.Gln232Ter)   55572    FOXRED1   \n",
       "23                         NM_000410.4(HFE):c.892+48G>A    3077        HFE   \n",
       "33              NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n",
       "41           NM_020779.4(WDR35):c.2590G>A (p.Ala864Thr)   57539      WDR35   \n",
       "49       NM_001042472.3(ABHD12):c.1054C>T (p.Arg352Ter)   26090     ABHD12   \n",
       "...                                                 ...     ...        ...   \n",
       "3251244                    NM_004006.3(DMD):c.9224+2T>A    1756        DMD   \n",
       "3251246                    NM_004006.3(DMD):c.9361+1G>T    1756        DMD   \n",
       "3251250                    NM_004006.3(DMD):c.1704+2T>C    1756        DMD   \n",
       "3251260                    NM_004006.3(DMD):c.9974+1G>C    1756        DMD   \n",
       "3251262                    NM_004006.3(DMD):c.5449-2A>T    1756        DMD   \n",
       "\n",
       "            HGNC_ID ClinicalSignificance  ClinSigSimple LastEvaluated  \\\n",
       "7        HGNC:26927           Pathogenic              1  Dec 30, 2019   \n",
       "23        HGNC:4886               Benign              0  Sep 11, 2018   \n",
       "33        HGNC:4886           Pathogenic              1  Sep 04, 2021   \n",
       "41       HGNC:29250    Likely pathogenic              1  May 25, 2017   \n",
       "49       HGNC:15868           Pathogenic              1  Jan 05, 2022   \n",
       "...             ...                  ...            ...           ...   \n",
       "3251244   HGNC:2928    Likely pathogenic              1  Dec 16, 2022   \n",
       "3251246   HGNC:2928    Likely pathogenic              1  Dec 16, 2022   \n",
       "3251250   HGNC:2928    Likely pathogenic              1  Dec 16, 2022   \n",
       "3251260   HGNC:2928    Likely pathogenic              1  Dec 16, 2022   \n",
       "3251262   HGNC:2928    Likely pathogenic              1  Dec 16, 2022   \n",
       "\n",
       "         RS# (dbSNP)  ...         Guidelines TestedInGTR  \\\n",
       "7          267606829  ...                  -           N   \n",
       "23           1800758  ...  ACMG2021,ACMG2022           N   \n",
       "33         111033563  ...  ACMG2021,ACMG2022           N   \n",
       "41         267607175  ...                  -           N   \n",
       "49         267606624  ...                  -           N   \n",
       "...              ...  ...                ...         ...   \n",
       "3251244           -1  ...                  -           N   \n",
       "3251246           -1  ...                  -           N   \n",
       "3251250           -1  ...                  -           N   \n",
       "3251260           -1  ...                  -           N   \n",
       "3251262           -1  ...                  -           N   \n",
       "\n",
       "                                                  OtherIDs  \\\n",
       "7                        ClinGen:CA113792,OMIM:613622.0001   \n",
       "23                       ClinGen:CA113800,OMIM:613609.0004   \n",
       "33       ClinGen:CA280949,UniProtKB:Q30201#VAR_037304,O...   \n",
       "41       ClinGen:CA339782,UniProtKB:Q9P2L0#VAR_064582,O...   \n",
       "49                       ClinGen:CA113811,OMIM:613599.0004   \n",
       "...                                                    ...   \n",
       "3251244                                                  -   \n",
       "3251246                                                  -   \n",
       "3251250                                                  -   \n",
       "3251260                                                  -   \n",
       "3251262                                                  -   \n",
       "\n",
       "        SubmitterCategories VariationID PositionVCF ReferenceAlleleVCF  \\\n",
       "7                         3           5   126275389                  C   \n",
       "23                        3          14    26093008                  G   \n",
       "33                        3          19    26092916                  A   \n",
       "41                        3          23    19933469                  C   \n",
       "49                        3          27    25302322                  G   \n",
       "...                     ...         ...         ...                ...   \n",
       "3251244                   2     1806436    31323596                  A   \n",
       "3251246                   2     1806437    31223046                  C   \n",
       "3251250                   2     1806439    32573743                  A   \n",
       "3251260                   2     1806444    31182737                  C   \n",
       "3251262                   2     1806445    32346082                  T   \n",
       "\n",
       "        AlternateAlleleVCF stars            classes  \n",
       "7                        T     2         Pathogenic  \n",
       "23                       A     1             Benign  \n",
       "33                       C     1         Pathogenic  \n",
       "41                       T     1  Likely pathogenic  \n",
       "49                       A     2         Pathogenic  \n",
       "...                    ...   ...                ...  \n",
       "3251244                  T     1  Likely pathogenic  \n",
       "3251246                  A     1  Likely pathogenic  \n",
       "3251250                  G     1  Likely pathogenic  \n",
       "3251260                  G     1  Likely pathogenic  \n",
       "3251262                  A     1  Likely pathogenic  \n",
       "\n",
       "[661134 rows x 35 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def assign_class(row):\n",
    "    if 'Benign/Likely benign' in row or 'Pathogenic/Likely pathogenic' in row:\n",
    "        return None\n",
    "    elif 'Benign' in row:\n",
    "        return 'Benign'\n",
    "    elif 'Likely benign' in row:\n",
    "        return 'Likely benign'\n",
    "    elif 'Pathogenic' in row:\n",
    "        return 'Pathogenic'\n",
    "    elif 'Likely pathogenic' in row:\n",
    "        return 'Likely pathogenic'\n",
    "    else:\n",
    "        return None\n",
    "\n",
    "df_meta['classes'] = df_meta['ClinicalSignificance'].apply(assign_class)\n",
    "# pandas can handle NaNs as integers now\n",
    "df_meta = df_meta[df_meta['classes'].notnull()]\n",
    "df_meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>...</th>\n",
       "      <th>TestedInGTR</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>SubmitterCategories</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PositionVCF</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>AlleleChange</th>\n",
       "      <th>Protein_Patterns</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>15043</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n",
       "      <td>9640</td>\n",
       "      <td>ZNF592</td>\n",
       "      <td>HGNC:28986</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>Jun 29, 2015</td>\n",
       "      <td>150829393</td>\n",
       "      <td>...</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>84799209</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>3136</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Gly1046Arg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>15044</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_017547.4(FOXRED1):c.694C&gt;T (p.Gln232Ter)</td>\n",
       "      <td>55572</td>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>HGNC:26927</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 30, 2019</td>\n",
       "      <td>267606829</td>\n",
       "      <td>...</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA113792,OMIM:613622.0001</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>126275389</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>694</td>\n",
       "      <td>C/T</td>\n",
       "      <td>Gln232Ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>15045</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n",
       "      <td>55572</td>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>HGNC:26927</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Oct 01, 2010</td>\n",
       "      <td>267606830</td>\n",
       "      <td>...</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>126277517</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1289</td>\n",
       "      <td>A/G</td>\n",
       "      <td>Asn430Ser</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>15046</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_025152.3(NUBPL):c.166G&gt;A (p.Gly56Arg)</td>\n",
       "      <td>80224</td>\n",
       "      <td>NUBPL</td>\n",
       "      <td>HGNC:20278</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>Aug 24, 2020</td>\n",
       "      <td>200401432</td>\n",
       "      <td>...</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...</td>\n",
       "      <td>2</td>\n",
       "      <td>214885</td>\n",
       "      <td>31562125</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>166</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Gly56Arg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>15048</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000410.4(HFE):c.845G&gt;A (p.Cys282Tyr)</td>\n",
       "      <td>3077</td>\n",
       "      <td>HFE</td>\n",
       "      <td>HGNC:4886</td>\n",
       "      <td>Conflicting interpretations of pathogenicity; ...</td>\n",
       "      <td>1</td>\n",
       "      <td>Jun 30, 2022</td>\n",
       "      <td>1800562</td>\n",
       "      <td>...</td>\n",
       "      <td>Y</td>\n",
       "      <td>UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>26092913</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>845</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Cys282Tyr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251296</th>\n",
       "      <td>1863473</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000089.4(COL1A2):c.3159G&gt;A (p.Arg1053_Gly10...</td>\n",
       "      <td>1278</td>\n",
       "      <td>COL1A2</td>\n",
       "      <td>HGNC:2198</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>1806466</td>\n",
       "      <td>94427061</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>3159</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Arg1053_Gly1054=</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251300</th>\n",
       "      <td>1863475</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_024616.3(C3orf52):c.492T&gt;A (p.Tyr164Ter)</td>\n",
       "      <td>79669</td>\n",
       "      <td>C3orf52</td>\n",
       "      <td>HGNC:26255</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 21, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM:611956.0001</td>\n",
       "      <td>1</td>\n",
       "      <td>1806468</td>\n",
       "      <td>112112988</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>492</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Tyr164Ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251302</th>\n",
       "      <td>1863476</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_024616.3(C3orf52):c.34G&gt;T (p.Glu12Ter)</td>\n",
       "      <td>79669</td>\n",
       "      <td>C3orf52</td>\n",
       "      <td>HGNC:26255</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 21, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM:611956.0002</td>\n",
       "      <td>1</td>\n",
       "      <td>1806469</td>\n",
       "      <td>112086441</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>34</td>\n",
       "      <td>G/T</td>\n",
       "      <td>Glu12Ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251306</th>\n",
       "      <td>1863478</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_004456.5(EZH2):c.395C&gt;T (p.Pro132Leu)</td>\n",
       "      <td>2146</td>\n",
       "      <td>EZH2</td>\n",
       "      <td>HGNC:3527</td>\n",
       "      <td>not provided</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>1806471</td>\n",
       "      <td>148829817</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>395</td>\n",
       "      <td>C/T</td>\n",
       "      <td>Pro132Leu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251308</th>\n",
       "      <td>1863479</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_001972.4(ELANE):c.640G&gt;C (p.Gly214Arg)</td>\n",
       "      <td>1991</td>\n",
       "      <td>ELANE</td>\n",
       "      <td>HGNC:3309</td>\n",
       "      <td>not provided</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>1806472</td>\n",
       "      <td>856000</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>640</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Gly214Arg</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1414257 rows × 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #AlleleID                       Type  \\\n",
       "5            15043  single nucleotide variant   \n",
       "7            15044  single nucleotide variant   \n",
       "9            15045  single nucleotide variant   \n",
       "11           15046  single nucleotide variant   \n",
       "13           15048  single nucleotide variant   \n",
       "...            ...                        ...   \n",
       "3251296    1863473  single nucleotide variant   \n",
       "3251300    1863475  single nucleotide variant   \n",
       "3251302    1863476  single nucleotide variant   \n",
       "3251306    1863478  single nucleotide variant   \n",
       "3251308    1863479  single nucleotide variant   \n",
       "\n",
       "                                                      Name  GeneID GeneSymbol  \\\n",
       "5             NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n",
       "7              NM_017547.4(FOXRED1):c.694C>T (p.Gln232Ter)   55572    FOXRED1   \n",
       "9             NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)   55572    FOXRED1   \n",
       "11                NM_025152.3(NUBPL):c.166G>A (p.Gly56Arg)   80224      NUBPL   \n",
       "13                 NM_000410.4(HFE):c.845G>A (p.Cys282Tyr)    3077        HFE   \n",
       "...                                                    ...     ...        ...   \n",
       "3251296  NM_000089.4(COL1A2):c.3159G>A (p.Arg1053_Gly10...    1278     COL1A2   \n",
       "3251300        NM_024616.3(C3orf52):c.492T>A (p.Tyr164Ter)   79669    C3orf52   \n",
       "3251302          NM_024616.3(C3orf52):c.34G>T (p.Glu12Ter)   79669    C3orf52   \n",
       "3251306           NM_004456.5(EZH2):c.395C>T (p.Pro132Leu)    2146       EZH2   \n",
       "3251308          NM_001972.4(ELANE):c.640G>C (p.Gly214Arg)    1991      ELANE   \n",
       "\n",
       "            HGNC_ID                               ClinicalSignificance  \\\n",
       "5        HGNC:28986                             Uncertain significance   \n",
       "7        HGNC:26927                                         Pathogenic   \n",
       "9        HGNC:26927                                         Pathogenic   \n",
       "11       HGNC:20278                             Uncertain significance   \n",
       "13        HGNC:4886  Conflicting interpretations of pathogenicity; ...   \n",
       "...             ...                                                ...   \n",
       "3251296   HGNC:2198                                  Likely pathogenic   \n",
       "3251300  HGNC:26255                                         Pathogenic   \n",
       "3251302  HGNC:26255                                         Pathogenic   \n",
       "3251306   HGNC:3527                                       not provided   \n",
       "3251308   HGNC:3309                                       not provided   \n",
       "\n",
       "         ClinSigSimple LastEvaluated  RS# (dbSNP)  ... TestedInGTR  \\\n",
       "5                    0  Jun 29, 2015    150829393  ...           N   \n",
       "7                    1  Dec 30, 2019    267606829  ...           N   \n",
       "9                    1  Oct 01, 2010    267606830  ...           N   \n",
       "11                   0  Aug 24, 2020    200401432  ...           N   \n",
       "13                   1  Jun 30, 2022      1800562  ...           Y   \n",
       "...                ...           ...          ...  ...         ...   \n",
       "3251296              1             -           -1  ...           N   \n",
       "3251300              1  Dec 21, 2022           -1  ...           N   \n",
       "3251302              1  Dec 21, 2022           -1  ...           N   \n",
       "3251306              0             -           -1  ...           N   \n",
       "3251308              0             -           -1  ...           N   \n",
       "\n",
       "                                                  OtherIDs  \\\n",
       "5        OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...   \n",
       "7                        ClinGen:CA113792,OMIM:613622.0001   \n",
       "9        ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...   \n",
       "11       ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...   \n",
       "13       UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...   \n",
       "...                                                    ...   \n",
       "3251296                                                  -   \n",
       "3251300                                   OMIM:611956.0001   \n",
       "3251302                                   OMIM:611956.0002   \n",
       "3251306                                                  -   \n",
       "3251308                                                  -   \n",
       "\n",
       "        SubmitterCategories VariationID PositionVCF ReferenceAlleleVCF  \\\n",
       "5                         1           4    84799209                  G   \n",
       "7                         3           5   126275389                  C   \n",
       "9                         1           6   126277517                  A   \n",
       "11                        2      214885    31562125                  G   \n",
       "13                        3           9    26092913                  G   \n",
       "...                     ...         ...         ...                ...   \n",
       "3251296                   2     1806466    94427061                  G   \n",
       "3251300                   1     1806468   112112988                  T   \n",
       "3251302                   1     1806469   112086441                  G   \n",
       "3251306                   1     1806471   148829817                  G   \n",
       "3251308                   1     1806472      856000                  G   \n",
       "\n",
       "        AlternateAlleleVCF CDS_position AlleleChange  Protein_Patterns  \n",
       "5                        A         3136          G/A        Gly1046Arg  \n",
       "7                        T          694          C/T         Gln232Ter  \n",
       "9                        G         1289          A/G         Asn430Ser  \n",
       "11                       A          166          G/A          Gly56Arg  \n",
       "13                       A          845          G/A         Cys282Tyr  \n",
       "...                    ...          ...          ...               ...  \n",
       "3251296                  A         3159          G/A  Arg1053_Gly1054=  \n",
       "3251300                  A          492          T/A         Tyr164Ter  \n",
       "3251302                  T           34          G/T          Glu12Ter  \n",
       "3251306                  A          395          C/T         Pro132Leu  \n",
       "3251308                  C          640          G/C         Gly214Arg  \n",
       "\n",
       "[1414257 rows x 37 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# identify the amino acid changes\n",
    "import numpy as np\n",
    "df_meta['CDS_position'] = df_meta['Name'].str.extract(r'c\\.(\\d+)')\n",
    "df_meta['AlleleChange'] = df_meta['Name'].str.extract(r'c\\.\\d+([A-Z])>') + '/' + df_meta['Name'].str.extract(r'>([A-Z])')\n",
    "# put na for those entires of intron changes\n",
    "df_meta['Protein_Patterns'] = df_meta['Name'].str.extract(r'p\\.(\\w+\\**=*)', expand=False).fillna(np.nan)\n",
    "df_meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>AlleleChange</th>\n",
       "      <th>Protein_Patterns</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n",
       "      <td>ZNF592</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...</td>\n",
       "      <td>3136</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Gly1046Arg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NM_017547.4(FOXRED1):c.694C&gt;T (p.Gln232Ter)</td>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ClinGen:CA113792,OMIM:613622.0001</td>\n",
       "      <td>694</td>\n",
       "      <td>C/T</td>\n",
       "      <td>Gln232Ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...</td>\n",
       "      <td>1289</td>\n",
       "      <td>A/G</td>\n",
       "      <td>Asn430Ser</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NM_025152.3(NUBPL):c.166G&gt;A (p.Gly56Arg)</td>\n",
       "      <td>NUBPL</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...</td>\n",
       "      <td>166</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Gly56Arg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NM_000410.4(HFE):c.845G&gt;A (p.Cys282Tyr)</td>\n",
       "      <td>HFE</td>\n",
       "      <td>Conflicting interpretations of pathogenicity; ...</td>\n",
       "      <td>UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...</td>\n",
       "      <td>845</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Cys282Tyr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251296</th>\n",
       "      <td>NM_000089.4(COL1A2):c.3159G&gt;A (p.Arg1053_Gly10...</td>\n",
       "      <td>COL1A2</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>3159</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Arg1053_Gly1054=</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251300</th>\n",
       "      <td>NM_024616.3(C3orf52):c.492T&gt;A (p.Tyr164Ter)</td>\n",
       "      <td>C3orf52</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>OMIM:611956.0001</td>\n",
       "      <td>492</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Tyr164Ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251302</th>\n",
       "      <td>NM_024616.3(C3orf52):c.34G&gt;T (p.Glu12Ter)</td>\n",
       "      <td>C3orf52</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>OMIM:611956.0002</td>\n",
       "      <td>34</td>\n",
       "      <td>G/T</td>\n",
       "      <td>Glu12Ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251306</th>\n",
       "      <td>NM_004456.5(EZH2):c.395C&gt;T (p.Pro132Leu)</td>\n",
       "      <td>EZH2</td>\n",
       "      <td>not provided</td>\n",
       "      <td>-</td>\n",
       "      <td>395</td>\n",
       "      <td>C/T</td>\n",
       "      <td>Pro132Leu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251308</th>\n",
       "      <td>NM_001972.4(ELANE):c.640G&gt;C (p.Gly214Arg)</td>\n",
       "      <td>ELANE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>-</td>\n",
       "      <td>640</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Gly214Arg</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1414257 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      Name GeneSymbol  \\\n",
       "5             NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)     ZNF592   \n",
       "7              NM_017547.4(FOXRED1):c.694C>T (p.Gln232Ter)    FOXRED1   \n",
       "9             NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)    FOXRED1   \n",
       "11                NM_025152.3(NUBPL):c.166G>A (p.Gly56Arg)      NUBPL   \n",
       "13                 NM_000410.4(HFE):c.845G>A (p.Cys282Tyr)        HFE   \n",
       "...                                                    ...        ...   \n",
       "3251296  NM_000089.4(COL1A2):c.3159G>A (p.Arg1053_Gly10...     COL1A2   \n",
       "3251300        NM_024616.3(C3orf52):c.492T>A (p.Tyr164Ter)    C3orf52   \n",
       "3251302          NM_024616.3(C3orf52):c.34G>T (p.Glu12Ter)    C3orf52   \n",
       "3251306           NM_004456.5(EZH2):c.395C>T (p.Pro132Leu)       EZH2   \n",
       "3251308          NM_001972.4(ELANE):c.640G>C (p.Gly214Arg)      ELANE   \n",
       "\n",
       "                                      ClinicalSignificance  \\\n",
       "5                                   Uncertain significance   \n",
       "7                                               Pathogenic   \n",
       "9                                               Pathogenic   \n",
       "11                                  Uncertain significance   \n",
       "13       Conflicting interpretations of pathogenicity; ...   \n",
       "...                                                    ...   \n",
       "3251296                                  Likely pathogenic   \n",
       "3251300                                         Pathogenic   \n",
       "3251302                                         Pathogenic   \n",
       "3251306                                       not provided   \n",
       "3251308                                       not provided   \n",
       "\n",
       "                                                  OtherIDs CDS_position  \\\n",
       "5        OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...         3136   \n",
       "7                        ClinGen:CA113792,OMIM:613622.0001          694   \n",
       "9        ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...         1289   \n",
       "11       ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...          166   \n",
       "13       UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...          845   \n",
       "...                                                    ...          ...   \n",
       "3251296                                                  -         3159   \n",
       "3251300                                   OMIM:611956.0001          492   \n",
       "3251302                                   OMIM:611956.0002           34   \n",
       "3251306                                                  -          395   \n",
       "3251308                                                  -          640   \n",
       "\n",
       "        AlleleChange  Protein_Patterns  \n",
       "5                G/A        Gly1046Arg  \n",
       "7                C/T         Gln232Ter  \n",
       "9                A/G         Asn430Ser  \n",
       "11               G/A          Gly56Arg  \n",
       "13               G/A         Cys282Tyr  \n",
       "...              ...               ...  \n",
       "3251296          G/A  Arg1053_Gly1054=  \n",
       "3251300          T/A         Tyr164Ter  \n",
       "3251302          G/T          Glu12Ter  \n",
       "3251306          C/T         Pro132Leu  \n",
       "3251308          G/C         Gly214Arg  \n",
       "\n",
       "[1414257 rows x 7 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_need = df_meta[['Name', 'GeneSymbol', 'ClinicalSignificance', 'OtherIDs', 'CDS_position', 'AlleleChange', 'Protein_Patterns']]\n",
    "df_need"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/pbs.464684.hn-10-03/ipykernel_45625/2433982364.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_need['NM_ID'] = df_need['Name'].str.extract(nm_id_regex).fillna('NA')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>AlleleChange</th>\n",
       "      <th>Protein_Patterns</th>\n",
       "      <th>NM_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n",
       "      <td>ZNF592</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...</td>\n",
       "      <td>3136</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Gly1046Arg</td>\n",
       "      <td>NM_014630.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NM_017547.4(FOXRED1):c.694C&gt;T (p.Gln232Ter)</td>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ClinGen:CA113792,OMIM:613622.0001</td>\n",
       "      <td>694</td>\n",
       "      <td>C/T</td>\n",
       "      <td>Gln232Ter</td>\n",
       "      <td>NM_017547.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...</td>\n",
       "      <td>1289</td>\n",
       "      <td>A/G</td>\n",
       "      <td>Asn430Ser</td>\n",
       "      <td>NM_017547.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NM_025152.3(NUBPL):c.166G&gt;A (p.Gly56Arg)</td>\n",
       "      <td>NUBPL</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...</td>\n",
       "      <td>166</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Gly56Arg</td>\n",
       "      <td>NM_025152.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NM_000410.4(HFE):c.845G&gt;A (p.Cys282Tyr)</td>\n",
       "      <td>HFE</td>\n",
       "      <td>Conflicting interpretations of pathogenicity; ...</td>\n",
       "      <td>UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...</td>\n",
       "      <td>845</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Cys282Tyr</td>\n",
       "      <td>NM_000410.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251296</th>\n",
       "      <td>NM_000089.4(COL1A2):c.3159G&gt;A (p.Arg1053_Gly10...</td>\n",
       "      <td>COL1A2</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>3159</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Arg1053_Gly1054=</td>\n",
       "      <td>NM_000089.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251300</th>\n",
       "      <td>NM_024616.3(C3orf52):c.492T&gt;A (p.Tyr164Ter)</td>\n",
       "      <td>C3orf52</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>OMIM:611956.0001</td>\n",
       "      <td>492</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Tyr164Ter</td>\n",
       "      <td>NM_024616.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251302</th>\n",
       "      <td>NM_024616.3(C3orf52):c.34G&gt;T (p.Glu12Ter)</td>\n",
       "      <td>C3orf52</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>OMIM:611956.0002</td>\n",
       "      <td>34</td>\n",
       "      <td>G/T</td>\n",
       "      <td>Glu12Ter</td>\n",
       "      <td>NM_024616.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251306</th>\n",
       "      <td>NM_004456.5(EZH2):c.395C&gt;T (p.Pro132Leu)</td>\n",
       "      <td>EZH2</td>\n",
       "      <td>not provided</td>\n",
       "      <td>-</td>\n",
       "      <td>395</td>\n",
       "      <td>C/T</td>\n",
       "      <td>Pro132Leu</td>\n",
       "      <td>NM_004456.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251308</th>\n",
       "      <td>NM_001972.4(ELANE):c.640G&gt;C (p.Gly214Arg)</td>\n",
       "      <td>ELANE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>-</td>\n",
       "      <td>640</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Gly214Arg</td>\n",
       "      <td>NM_001972.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1414257 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      Name GeneSymbol  \\\n",
       "5             NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)     ZNF592   \n",
       "7              NM_017547.4(FOXRED1):c.694C>T (p.Gln232Ter)    FOXRED1   \n",
       "9             NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)    FOXRED1   \n",
       "11                NM_025152.3(NUBPL):c.166G>A (p.Gly56Arg)      NUBPL   \n",
       "13                 NM_000410.4(HFE):c.845G>A (p.Cys282Tyr)        HFE   \n",
       "...                                                    ...        ...   \n",
       "3251296  NM_000089.4(COL1A2):c.3159G>A (p.Arg1053_Gly10...     COL1A2   \n",
       "3251300        NM_024616.3(C3orf52):c.492T>A (p.Tyr164Ter)    C3orf52   \n",
       "3251302          NM_024616.3(C3orf52):c.34G>T (p.Glu12Ter)    C3orf52   \n",
       "3251306           NM_004456.5(EZH2):c.395C>T (p.Pro132Leu)       EZH2   \n",
       "3251308          NM_001972.4(ELANE):c.640G>C (p.Gly214Arg)      ELANE   \n",
       "\n",
       "                                      ClinicalSignificance  \\\n",
       "5                                   Uncertain significance   \n",
       "7                                               Pathogenic   \n",
       "9                                               Pathogenic   \n",
       "11                                  Uncertain significance   \n",
       "13       Conflicting interpretations of pathogenicity; ...   \n",
       "...                                                    ...   \n",
       "3251296                                  Likely pathogenic   \n",
       "3251300                                         Pathogenic   \n",
       "3251302                                         Pathogenic   \n",
       "3251306                                       not provided   \n",
       "3251308                                       not provided   \n",
       "\n",
       "                                                  OtherIDs CDS_position  \\\n",
       "5        OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...         3136   \n",
       "7                        ClinGen:CA113792,OMIM:613622.0001          694   \n",
       "9        ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...         1289   \n",
       "11       ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...          166   \n",
       "13       UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...          845   \n",
       "...                                                    ...          ...   \n",
       "3251296                                                  -         3159   \n",
       "3251300                                   OMIM:611956.0001          492   \n",
       "3251302                                   OMIM:611956.0002           34   \n",
       "3251306                                                  -          395   \n",
       "3251308                                                  -          640   \n",
       "\n",
       "        AlleleChange  Protein_Patterns        NM_ID  \n",
       "5                G/A        Gly1046Arg  NM_014630.3  \n",
       "7                C/T         Gln232Ter  NM_017547.4  \n",
       "9                A/G         Asn430Ser  NM_017547.4  \n",
       "11               G/A          Gly56Arg  NM_025152.3  \n",
       "13               G/A         Cys282Tyr  NM_000410.4  \n",
       "...              ...               ...          ...  \n",
       "3251296          G/A  Arg1053_Gly1054=  NM_000089.4  \n",
       "3251300          T/A         Tyr164Ter  NM_024616.3  \n",
       "3251302          G/T          Glu12Ter  NM_024616.3  \n",
       "3251306          C/T         Pro132Leu  NM_004456.5  \n",
       "3251308          G/C         Gly214Arg  NM_001972.4  \n",
       "\n",
       "[1414257 rows x 8 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nm_id_regex = r'(NM_\\d+\\.\\d+)'\n",
    "df_need['NM_ID'] = df_need['Name'].str.extract(nm_id_regex).fillna('NA')\n",
    "df_need"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>AlleleChange</th>\n",
       "      <th>Protein_Patterns</th>\n",
       "      <th>NM_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ZNF592</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...</td>\n",
       "      <td>3136</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Gly1046Arg</td>\n",
       "      <td>NM_014630.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ClinGen:CA113792,OMIM:613622.0001</td>\n",
       "      <td>694</td>\n",
       "      <td>C/T</td>\n",
       "      <td>Gln232Ter</td>\n",
       "      <td>NM_017547.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...</td>\n",
       "      <td>1289</td>\n",
       "      <td>A/G</td>\n",
       "      <td>Asn430Ser</td>\n",
       "      <td>NM_017547.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NUBPL</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...</td>\n",
       "      <td>166</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Gly56Arg</td>\n",
       "      <td>NM_025152.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HFE</td>\n",
       "      <td>Conflicting interpretations of pathogenicity; ...</td>\n",
       "      <td>UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...</td>\n",
       "      <td>845</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Cys282Tyr</td>\n",
       "      <td>NM_000410.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251296</th>\n",
       "      <td>COL1A2</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>3159</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Arg1053_Gly1054=</td>\n",
       "      <td>NM_000089.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251300</th>\n",
       "      <td>C3orf52</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>OMIM:611956.0001</td>\n",
       "      <td>492</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Tyr164Ter</td>\n",
       "      <td>NM_024616.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251302</th>\n",
       "      <td>C3orf52</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>OMIM:611956.0002</td>\n",
       "      <td>34</td>\n",
       "      <td>G/T</td>\n",
       "      <td>Glu12Ter</td>\n",
       "      <td>NM_024616.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251306</th>\n",
       "      <td>EZH2</td>\n",
       "      <td>not provided</td>\n",
       "      <td>-</td>\n",
       "      <td>395</td>\n",
       "      <td>C/T</td>\n",
       "      <td>Pro132Leu</td>\n",
       "      <td>NM_004456.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251308</th>\n",
       "      <td>ELANE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>-</td>\n",
       "      <td>640</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Gly214Arg</td>\n",
       "      <td>NM_001972.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1414257 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GeneSymbol                               ClinicalSignificance  \\\n",
       "5           ZNF592                             Uncertain significance   \n",
       "7          FOXRED1                                         Pathogenic   \n",
       "9          FOXRED1                                         Pathogenic   \n",
       "11           NUBPL                             Uncertain significance   \n",
       "13             HFE  Conflicting interpretations of pathogenicity; ...   \n",
       "...            ...                                                ...   \n",
       "3251296     COL1A2                                  Likely pathogenic   \n",
       "3251300    C3orf52                                         Pathogenic   \n",
       "3251302    C3orf52                                         Pathogenic   \n",
       "3251306       EZH2                                       not provided   \n",
       "3251308      ELANE                                       not provided   \n",
       "\n",
       "                                                  OtherIDs CDS_position  \\\n",
       "5        OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...         3136   \n",
       "7                        ClinGen:CA113792,OMIM:613622.0001          694   \n",
       "9        ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...         1289   \n",
       "11       ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...          166   \n",
       "13       UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...          845   \n",
       "...                                                    ...          ...   \n",
       "3251296                                                  -         3159   \n",
       "3251300                                   OMIM:611956.0001          492   \n",
       "3251302                                   OMIM:611956.0002           34   \n",
       "3251306                                                  -          395   \n",
       "3251308                                                  -          640   \n",
       "\n",
       "        AlleleChange  Protein_Patterns        NM_ID  \n",
       "5                G/A        Gly1046Arg  NM_014630.3  \n",
       "7                C/T         Gln232Ter  NM_017547.4  \n",
       "9                A/G         Asn430Ser  NM_017547.4  \n",
       "11               G/A          Gly56Arg  NM_025152.3  \n",
       "13               G/A         Cys282Tyr  NM_000410.4  \n",
       "...              ...               ...          ...  \n",
       "3251296          G/A  Arg1053_Gly1054=  NM_000089.4  \n",
       "3251300          T/A         Tyr164Ter  NM_024616.3  \n",
       "3251302          G/T          Glu12Ter  NM_024616.3  \n",
       "3251306          C/T         Pro132Leu  NM_004456.5  \n",
       "3251308          G/C         Gly214Arg  NM_001972.4  \n",
       "\n",
       "[1414257 rows x 7 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_need = df_need[['GeneSymbol','ClinicalSignificance','OtherIDs','CDS_position','AlleleChange','Protein_Patterns','NM_ID']]\n",
    "df_need"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NM_ID</th>\n",
       "      <th>NP_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NM_001318856.2</td>\n",
       "      <td>NP_001305785.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NM_001318857.2</td>\n",
       "      <td>NP_001305786.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NM_001040075.1</td>\n",
       "      <td>NP_001035164.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NM_017618.1</td>\n",
       "      <td>NP_060088.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NM_001318809.2</td>\n",
       "      <td>NP_001305738.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67102</th>\n",
       "      <td>NM_001376153.1</td>\n",
       "      <td>NP_001363082.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67103</th>\n",
       "      <td>NM_001376154.1</td>\n",
       "      <td>NP_001363083.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67104</th>\n",
       "      <td>NM_001376155.1</td>\n",
       "      <td>NP_001363084.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67105</th>\n",
       "      <td>NM_001376156.1</td>\n",
       "      <td>NP_001363085.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67106</th>\n",
       "      <td>NM_015534.6</td>\n",
       "      <td>NP_056349.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67107 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                NM_ID           NP_ID\n",
       "0      NM_001318856.2  NP_001305785.1\n",
       "1      NM_001318857.2  NP_001305786.1\n",
       "2      NM_001040075.1  NP_001035164.1\n",
       "3         NM_017618.1     NP_060088.1\n",
       "4      NM_001318809.2  NP_001305738.1\n",
       "...               ...             ...\n",
       "67102  NM_001376153.1  NP_001363082.1\n",
       "67103  NM_001376154.1  NP_001363083.1\n",
       "67104  NM_001376155.1  NP_001363084.1\n",
       "67105  NM_001376156.1  NP_001363085.1\n",
       "67106     NM_015534.6     NP_056349.1\n",
       "\n",
       "[67107 rows x 2 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_idmapping = pd.read_csv('./data/metas/nm_np_ids.csv')\n",
    "df_idmapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>AlleleChange</th>\n",
       "      <th>Protein_Patterns</th>\n",
       "      <th>NM_ID</th>\n",
       "      <th>NP_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZNF592</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...</td>\n",
       "      <td>3136</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Gly1046Arg</td>\n",
       "      <td>NM_014630.3</td>\n",
       "      <td>NP_055445.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ClinGen:CA113792,OMIM:613622.0001</td>\n",
       "      <td>694</td>\n",
       "      <td>C/T</td>\n",
       "      <td>Gln232Ter</td>\n",
       "      <td>NM_017547.4</td>\n",
       "      <td>NP_060017.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...</td>\n",
       "      <td>1289</td>\n",
       "      <td>A/G</td>\n",
       "      <td>Asn430Ser</td>\n",
       "      <td>NM_017547.4</td>\n",
       "      <td>NP_060017.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NUBPL</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...</td>\n",
       "      <td>166</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Gly56Arg</td>\n",
       "      <td>NM_025152.3</td>\n",
       "      <td>NP_079428.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HFE</td>\n",
       "      <td>Conflicting interpretations of pathogenicity; ...</td>\n",
       "      <td>UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...</td>\n",
       "      <td>845</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Cys282Tyr</td>\n",
       "      <td>NM_000410.4</td>\n",
       "      <td>NP_000401.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1414252</th>\n",
       "      <td>COL1A2</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>3159</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Arg1053_Gly1054=</td>\n",
       "      <td>NM_000089.4</td>\n",
       "      <td>NP_000080.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1414253</th>\n",
       "      <td>C3orf52</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>OMIM:611956.0001</td>\n",
       "      <td>492</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Tyr164Ter</td>\n",
       "      <td>NM_024616.3</td>\n",
       "      <td>NP_078892.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1414254</th>\n",
       "      <td>C3orf52</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>OMIM:611956.0002</td>\n",
       "      <td>34</td>\n",
       "      <td>G/T</td>\n",
       "      <td>Glu12Ter</td>\n",
       "      <td>NM_024616.3</td>\n",
       "      <td>NP_078892.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1414255</th>\n",
       "      <td>EZH2</td>\n",
       "      <td>not provided</td>\n",
       "      <td>-</td>\n",
       "      <td>395</td>\n",
       "      <td>C/T</td>\n",
       "      <td>Pro132Leu</td>\n",
       "      <td>NM_004456.5</td>\n",
       "      <td>NP_004447.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1414256</th>\n",
       "      <td>ELANE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>-</td>\n",
       "      <td>640</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Gly214Arg</td>\n",
       "      <td>NM_001972.4</td>\n",
       "      <td>NP_001963.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1414257 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GeneSymbol                               ClinicalSignificance  \\\n",
       "0           ZNF592                             Uncertain significance   \n",
       "1          FOXRED1                                         Pathogenic   \n",
       "2          FOXRED1                                         Pathogenic   \n",
       "3            NUBPL                             Uncertain significance   \n",
       "4              HFE  Conflicting interpretations of pathogenicity; ...   \n",
       "...            ...                                                ...   \n",
       "1414252     COL1A2                                  Likely pathogenic   \n",
       "1414253    C3orf52                                         Pathogenic   \n",
       "1414254    C3orf52                                         Pathogenic   \n",
       "1414255       EZH2                                       not provided   \n",
       "1414256      ELANE                                       not provided   \n",
       "\n",
       "                                                  OtherIDs CDS_position  \\\n",
       "0        OMIM:613624.0001,ClinGen:CA210674,UniProtKB:Q9...         3136   \n",
       "1                        ClinGen:CA113792,OMIM:613622.0001          694   \n",
       "2        ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,O...         1289   \n",
       "3        ClinVar:7,ClinGen:CA321015,UniProtKB:Q8TB37#VA...          166   \n",
       "4        UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,C...          845   \n",
       "...                                                    ...          ...   \n",
       "1414252                                                  -         3159   \n",
       "1414253                                   OMIM:611956.0001          492   \n",
       "1414254                                   OMIM:611956.0002           34   \n",
       "1414255                                                  -          395   \n",
       "1414256                                                  -          640   \n",
       "\n",
       "        AlleleChange  Protein_Patterns        NM_ID        NP_ID  \n",
       "0                G/A        Gly1046Arg  NM_014630.3  NP_055445.2  \n",
       "1                C/T         Gln232Ter  NM_017547.4  NP_060017.1  \n",
       "2                A/G         Asn430Ser  NM_017547.4  NP_060017.1  \n",
       "3                G/A          Gly56Arg  NM_025152.3  NP_079428.2  \n",
       "4                G/A         Cys282Tyr  NM_000410.4  NP_000401.1  \n",
       "...              ...               ...          ...          ...  \n",
       "1414252          G/A  Arg1053_Gly1054=  NM_000089.4  NP_000080.2  \n",
       "1414253          T/A         Tyr164Ter  NM_024616.3  NP_078892.3  \n",
       "1414254          G/T          Glu12Ter  NM_024616.3  NP_078892.3  \n",
       "1414255          C/T         Pro132Leu  NM_004456.5  NP_004447.2  \n",
       "1414256          G/C         Gly214Arg  NM_001972.4  NP_001963.1  \n",
       "\n",
       "[1414257 rows x 8 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merged_table = pd.merge(df_need, df_idmapping, on=['NM_ID'], how='left')\n",
    "df_merged_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>AlleleChange</th>\n",
       "      <th>Protein_Patterns</th>\n",
       "      <th>NM_ID</th>\n",
       "      <th>NP_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>OPN1SW</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>ClinGen:CA113829,UniProtKB:P03999#VAR_004838,O...</td>\n",
       "      <td>235</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Gly79Arg</td>\n",
       "      <td>NM_001708.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>OPN1SW</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ClinGen:CA113830,UniProtKB:P03999#VAR_004839,O...</td>\n",
       "      <td>640</td>\n",
       "      <td>T/C</td>\n",
       "      <td>Ser214Pro</td>\n",
       "      <td>NM_001708.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>OPN1SW</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ClinGen:CA113831,UniProtKB:P03999#VAR_004840,O...</td>\n",
       "      <td>790</td>\n",
       "      <td>C/T</td>\n",
       "      <td>Pro264Ser</td>\n",
       "      <td>NM_001708.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>MIR2861</td>\n",
       "      <td>association</td>\n",
       "      <td>ClinGen:CA113853,OMIM:613405.0001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>CTSA</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ClinGen:CA114221,UniProtKB:P10619#VAR_063020,O...</td>\n",
       "      <td>707</td>\n",
       "      <td>T/C</td>\n",
       "      <td>Leu236Pro</td>\n",
       "      <td>NM_001127695.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1412012</th>\n",
       "      <td>APRT</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1412754</th>\n",
       "      <td>CPA1</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1412783</th>\n",
       "      <td>ACD</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1412818</th>\n",
       "      <td>SYT14</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_153262.3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1412820</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_000314.6</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10242 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GeneSymbol    ClinicalSignificance  \\\n",
       "43          OPN1SW  Uncertain significance   \n",
       "44          OPN1SW              Pathogenic   \n",
       "45          OPN1SW              Pathogenic   \n",
       "76         MIR2861             association   \n",
       "285           CTSA              Pathogenic   \n",
       "...            ...                     ...   \n",
       "1412012       APRT           Likely benign   \n",
       "1412754       CPA1           Likely benign   \n",
       "1412783        ACD           Likely benign   \n",
       "1412818      SYT14           Likely benign   \n",
       "1412820       PTEN  Uncertain significance   \n",
       "\n",
       "                                                  OtherIDs CDS_position  \\\n",
       "43       ClinGen:CA113829,UniProtKB:P03999#VAR_004838,O...          235   \n",
       "44       ClinGen:CA113830,UniProtKB:P03999#VAR_004839,O...          640   \n",
       "45       ClinGen:CA113831,UniProtKB:P03999#VAR_004840,O...          790   \n",
       "76                       ClinGen:CA113853,OMIM:613405.0001          NaN   \n",
       "285      ClinGen:CA114221,UniProtKB:P10619#VAR_063020,O...          707   \n",
       "...                                                    ...          ...   \n",
       "1412012                                                  -          NaN   \n",
       "1412754                                                  -          NaN   \n",
       "1412783                                                  -          NaN   \n",
       "1412818                                                  -          NaN   \n",
       "1412820                                                  -          NaN   \n",
       "\n",
       "        AlleleChange Protein_Patterns           NM_ID NP_ID  \n",
       "43               G/A         Gly79Arg     NM_001708.2   NaN  \n",
       "44               T/C        Ser214Pro     NM_001708.2   NaN  \n",
       "45               C/T        Pro264Ser     NM_001708.2   NaN  \n",
       "76               NaN              NaN              NA   NaN  \n",
       "285              T/C        Leu236Pro  NM_001127695.1   NaN  \n",
       "...              ...              ...             ...   ...  \n",
       "1412012          NaN              NaN              NA   NaN  \n",
       "1412754          NaN              NaN              NA   NaN  \n",
       "1412783          NaN              NaN              NA   NaN  \n",
       "1412818          NaN              NaN     NM_153262.3   NaN  \n",
       "1412820          NaN              NaN     NM_000314.6   NaN  \n",
       "\n",
       "[10242 rows x 8 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_np_na = df_merged_table[~df_merged_table['NP_ID'].isna()]\n",
    "df_np_na"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/pbs.463587.hn-10-03/ipykernel_34616/2358675891.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_meta['Amino_acids_before_3letters'] = df_meta['Protein_Patterns'].apply(extract_amino_acid, before = True)\n",
      "/var/tmp/pbs.463587.hn-10-03/ipykernel_34616/2358675891.py:30: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_meta['Protein_positions'] = df_meta['Protein_Patterns'].apply(extract_amino_acid, pos = True)\n",
      "/var/tmp/pbs.463587.hn-10-03/ipykernel_34616/2358675891.py:31: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_meta['Amino_acids_after_3letters'] = df_meta['Protein_Patterns'].apply(extract_amino_acid, after = True)\n",
      "/var/tmp/pbs.463587.hn-10-03/ipykernel_34616/2358675891.py:32: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_meta['Protein_positions'] = df_meta['Protein_positions'].astype('Int64')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>...</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>stars</th>\n",
       "      <th>classes</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>AlleleChange</th>\n",
       "      <th>Protein_Patterns</th>\n",
       "      <th>Amino_acids_before_3letters</th>\n",
       "      <th>Protein_positions</th>\n",
       "      <th>Amino_acids_after_3letters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>15044</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_017547.4(FOXRED1):c.694C&gt;T (p.Gln232Ter)</td>\n",
       "      <td>55572</td>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>HGNC:26927</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 30, 2019</td>\n",
       "      <td>267606829</td>\n",
       "      <td>...</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>694</td>\n",
       "      <td>C/T</td>\n",
       "      <td>Gln232Ter</td>\n",
       "      <td>Gln</td>\n",
       "      <td>232</td>\n",
       "      <td>Ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>15053</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000410.4(HFE):c.892+48G&gt;A</td>\n",
       "      <td>3077</td>\n",
       "      <td>HFE</td>\n",
       "      <td>HGNC:4886</td>\n",
       "      <td>Benign</td>\n",
       "      <td>0</td>\n",
       "      <td>Sep 11, 2018</td>\n",
       "      <td>1800758</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>Benign</td>\n",
       "      <td>892</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>15058</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n",
       "      <td>3077</td>\n",
       "      <td>HFE</td>\n",
       "      <td>HGNC:4886</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Sep 04, 2021</td>\n",
       "      <td>111033563</td>\n",
       "      <td>...</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>848</td>\n",
       "      <td>A/C</td>\n",
       "      <td>Gln283Pro</td>\n",
       "      <td>Gln</td>\n",
       "      <td>283</td>\n",
       "      <td>Pro</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>15062</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_020779.4(WDR35):c.2590G&gt;A (p.Ala864Thr)</td>\n",
       "      <td>57539</td>\n",
       "      <td>WDR35</td>\n",
       "      <td>HGNC:29250</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>May 25, 2017</td>\n",
       "      <td>267607175</td>\n",
       "      <td>...</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>2590</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Ala864Thr</td>\n",
       "      <td>Ala</td>\n",
       "      <td>864</td>\n",
       "      <td>Thr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>15066</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_001042472.3(ABHD12):c.1054C&gt;T (p.Arg352Ter)</td>\n",
       "      <td>26090</td>\n",
       "      <td>ABHD12</td>\n",
       "      <td>HGNC:15868</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Jan 05, 2022</td>\n",
       "      <td>267606624</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1054</td>\n",
       "      <td>C/T</td>\n",
       "      <td>Arg352Ter</td>\n",
       "      <td>Arg</td>\n",
       "      <td>352</td>\n",
       "      <td>Ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251244</th>\n",
       "      <td>1863445</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_004006.3(DMD):c.9224+2T&gt;A</td>\n",
       "      <td>1756</td>\n",
       "      <td>DMD</td>\n",
       "      <td>HGNC:2928</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 16, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>9224</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251246</th>\n",
       "      <td>1863446</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_004006.3(DMD):c.9361+1G&gt;T</td>\n",
       "      <td>1756</td>\n",
       "      <td>DMD</td>\n",
       "      <td>HGNC:2928</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 16, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>9361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251250</th>\n",
       "      <td>1863448</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_004006.3(DMD):c.1704+2T&gt;C</td>\n",
       "      <td>1756</td>\n",
       "      <td>DMD</td>\n",
       "      <td>HGNC:2928</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 16, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1704</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251260</th>\n",
       "      <td>1863453</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_004006.3(DMD):c.9974+1G&gt;C</td>\n",
       "      <td>1756</td>\n",
       "      <td>DMD</td>\n",
       "      <td>HGNC:2928</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 16, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>9974</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251262</th>\n",
       "      <td>1863454</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_004006.3(DMD):c.5449-2A&gt;T</td>\n",
       "      <td>1756</td>\n",
       "      <td>DMD</td>\n",
       "      <td>HGNC:2928</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 16, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>5449</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>661134 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #AlleleID                       Type  \\\n",
       "7            15044  single nucleotide variant   \n",
       "23           15053  single nucleotide variant   \n",
       "33           15058  single nucleotide variant   \n",
       "41           15062  single nucleotide variant   \n",
       "49           15066  single nucleotide variant   \n",
       "...            ...                        ...   \n",
       "3251244    1863445  single nucleotide variant   \n",
       "3251246    1863446  single nucleotide variant   \n",
       "3251250    1863448  single nucleotide variant   \n",
       "3251260    1863453  single nucleotide variant   \n",
       "3251262    1863454  single nucleotide variant   \n",
       "\n",
       "                                                   Name  GeneID GeneSymbol  \\\n",
       "7           NM_017547.4(FOXRED1):c.694C>T (p.Gln232Ter)   55572    FOXRED1   \n",
       "23                         NM_000410.4(HFE):c.892+48G>A    3077        HFE   \n",
       "33              NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n",
       "41           NM_020779.4(WDR35):c.2590G>A (p.Ala864Thr)   57539      WDR35   \n",
       "49       NM_001042472.3(ABHD12):c.1054C>T (p.Arg352Ter)   26090     ABHD12   \n",
       "...                                                 ...     ...        ...   \n",
       "3251244                    NM_004006.3(DMD):c.9224+2T>A    1756        DMD   \n",
       "3251246                    NM_004006.3(DMD):c.9361+1G>T    1756        DMD   \n",
       "3251250                    NM_004006.3(DMD):c.1704+2T>C    1756        DMD   \n",
       "3251260                    NM_004006.3(DMD):c.9974+1G>C    1756        DMD   \n",
       "3251262                    NM_004006.3(DMD):c.5449-2A>T    1756        DMD   \n",
       "\n",
       "            HGNC_ID ClinicalSignificance  ClinSigSimple LastEvaluated  \\\n",
       "7        HGNC:26927           Pathogenic              1  Dec 30, 2019   \n",
       "23        HGNC:4886               Benign              0  Sep 11, 2018   \n",
       "33        HGNC:4886           Pathogenic              1  Sep 04, 2021   \n",
       "41       HGNC:29250    Likely pathogenic              1  May 25, 2017   \n",
       "49       HGNC:15868           Pathogenic              1  Jan 05, 2022   \n",
       "...             ...                  ...            ...           ...   \n",
       "3251244   HGNC:2928    Likely pathogenic              1  Dec 16, 2022   \n",
       "3251246   HGNC:2928    Likely pathogenic              1  Dec 16, 2022   \n",
       "3251250   HGNC:2928    Likely pathogenic              1  Dec 16, 2022   \n",
       "3251260   HGNC:2928    Likely pathogenic              1  Dec 16, 2022   \n",
       "3251262   HGNC:2928    Likely pathogenic              1  Dec 16, 2022   \n",
       "\n",
       "         RS# (dbSNP)  ... ReferenceAlleleVCF AlternateAlleleVCF stars  \\\n",
       "7          267606829  ...                  C                  T     2   \n",
       "23           1800758  ...                  G                  A     1   \n",
       "33         111033563  ...                  A                  C     1   \n",
       "41         267607175  ...                  C                  T     1   \n",
       "49         267606624  ...                  G                  A     2   \n",
       "...              ...  ...                ...                ...   ...   \n",
       "3251244           -1  ...                  A                  T     1   \n",
       "3251246           -1  ...                  C                  A     1   \n",
       "3251250           -1  ...                  A                  G     1   \n",
       "3251260           -1  ...                  C                  G     1   \n",
       "3251262           -1  ...                  T                  A     1   \n",
       "\n",
       "                   classes CDS_position AlleleChange Protein_Patterns  \\\n",
       "7               Pathogenic          694          C/T        Gln232Ter   \n",
       "23                  Benign          892          NaN              NaN   \n",
       "33              Pathogenic          848          A/C        Gln283Pro   \n",
       "41       Likely pathogenic         2590          G/A        Ala864Thr   \n",
       "49              Pathogenic         1054          C/T        Arg352Ter   \n",
       "...                    ...          ...          ...              ...   \n",
       "3251244  Likely pathogenic         9224          NaN              NaN   \n",
       "3251246  Likely pathogenic         9361          NaN              NaN   \n",
       "3251250  Likely pathogenic         1704          NaN              NaN   \n",
       "3251260  Likely pathogenic         9974          NaN              NaN   \n",
       "3251262  Likely pathogenic         5449          NaN              NaN   \n",
       "\n",
       "        Amino_acids_before_3letters Protein_positions  \\\n",
       "7                               Gln               232   \n",
       "23                              NaN              <NA>   \n",
       "33                              Gln               283   \n",
       "41                              Ala               864   \n",
       "49                              Arg               352   \n",
       "...                             ...               ...   \n",
       "3251244                         NaN              <NA>   \n",
       "3251246                         NaN              <NA>   \n",
       "3251250                         NaN              <NA>   \n",
       "3251260                         NaN              <NA>   \n",
       "3251262                         NaN              <NA>   \n",
       "\n",
       "         Amino_acids_after_3letters  \n",
       "7                               Ter  \n",
       "23                              NaN  \n",
       "33                              Pro  \n",
       "41                              Thr  \n",
       "49                              Ter  \n",
       "...                             ...  \n",
       "3251244                         NaN  \n",
       "3251246                         NaN  \n",
       "3251250                         NaN  \n",
       "3251260                         NaN  \n",
       "3251262                         NaN  \n",
       "\n",
       "[661134 rows x 41 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# need to consider conditions of:\n",
    "# p.Tyr5033=  synonymous mutation\n",
    "# p.Ter5039*  premature stop codon\n",
    "\n",
    "def extract_amino_acid(row, before = False, pos = False, after = False):\n",
    "    import re\n",
    "    try:\n",
    "        if before:\n",
    "            first_aa = row[:3]\n",
    "            return first_aa\n",
    "        if pos:\n",
    "            pos = int(re.search(r'\\d+', row).group())\n",
    "            return pos\n",
    "        # the following command extract the index within the string where the position number ends\n",
    "        if after:\n",
    "            first_aa = row[:3]\n",
    "            position_end_index = re.search(r'\\d+', row).span()[-1]\n",
    "            later_part = row[position_end_index:]\n",
    "            if later_part == '=':\n",
    "                return first_aa\n",
    "            elif later_part == '*':\n",
    "                return 'STOP'\n",
    "            else:\n",
    "                return later_part\n",
    "    except TypeError:\n",
    "        import numpy as np\n",
    "        return np.nan\n",
    "\n",
    "df_meta['Amino_acids_before_3letters'] = df_meta['Protein_Patterns'].apply(extract_amino_acid, before = True)\n",
    "df_meta['Protein_positions'] = df_meta['Protein_Patterns'].apply(extract_amino_acid, pos = True)\n",
    "df_meta['Amino_acids_after_3letters'] = df_meta['Protein_Patterns'].apply(extract_amino_acid, after = True)\n",
    "df_meta['Protein_positions'] = df_meta['Protein_positions'].astype('Int64')\n",
    "df_meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>...</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>stars</th>\n",
       "      <th>classes</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>AlleleChange</th>\n",
       "      <th>Protein_Patterns</th>\n",
       "      <th>Amino_acids_before_3letters</th>\n",
       "      <th>Protein_positions</th>\n",
       "      <th>Amino_acids_after_3letters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>15058</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n",
       "      <td>3077</td>\n",
       "      <td>HFE</td>\n",
       "      <td>HGNC:4886</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Sep 04, 2021</td>\n",
       "      <td>111033563</td>\n",
       "      <td>...</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>848</td>\n",
       "      <td>A/C</td>\n",
       "      <td>Gln283Pro</td>\n",
       "      <td>Gln</td>\n",
       "      <td>283</td>\n",
       "      <td>Pro</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>15062</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_020779.4(WDR35):c.2590G&gt;A (p.Ala864Thr)</td>\n",
       "      <td>57539</td>\n",
       "      <td>WDR35</td>\n",
       "      <td>HGNC:29250</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>May 25, 2017</td>\n",
       "      <td>267607175</td>\n",
       "      <td>...</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>2590</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Ala864Thr</td>\n",
       "      <td>Ala</td>\n",
       "      <td>864</td>\n",
       "      <td>Thr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>15072</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_138413.4(HOGA1):c.209G&gt;C (p.Arg70Pro)</td>\n",
       "      <td>112817</td>\n",
       "      <td>HOGA1</td>\n",
       "      <td>HGNC:25155</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Sep 19, 2021</td>\n",
       "      <td>267606763</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>209</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Arg70Pro</td>\n",
       "      <td>Arg</td>\n",
       "      <td>70</td>\n",
       "      <td>Pro</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>15087</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_001199107.2(TBC1D24):c.439G&gt;C (p.Asp147His)</td>\n",
       "      <td>57465</td>\n",
       "      <td>TBC1D24</td>\n",
       "      <td>HGNC:29203</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Sep 01, 2021</td>\n",
       "      <td>267607103</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>439</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Asp147His</td>\n",
       "      <td>Asp</td>\n",
       "      <td>147</td>\n",
       "      <td>His</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>15111</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000374.5(UROD):c.942G&gt;A (p.Glu314=)</td>\n",
       "      <td>7389</td>\n",
       "      <td>UROD</td>\n",
       "      <td>HGNC:12591</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Aug 13, 2021</td>\n",
       "      <td>121918062</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>942</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Glu314=</td>\n",
       "      <td>Glu</td>\n",
       "      <td>314</td>\n",
       "      <td>Glu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251142</th>\n",
       "      <td>1863387</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000093.5(COL5A1):c.1433G&gt;C (p.Gly478Ala)</td>\n",
       "      <td>1289</td>\n",
       "      <td>COL5A1</td>\n",
       "      <td>HGNC:2209</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>May 21, 2020</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1433</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Gly478Ala</td>\n",
       "      <td>Gly</td>\n",
       "      <td>478</td>\n",
       "      <td>Ala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251148</th>\n",
       "      <td>1863390</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000474.4(TWIST1):c.200G&gt;C (p.Gly67Ala)</td>\n",
       "      <td>7291</td>\n",
       "      <td>TWIST1</td>\n",
       "      <td>HGNC:12428</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>0</td>\n",
       "      <td>May 06, 2021</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>200</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Gly67Ala</td>\n",
       "      <td>Gly</td>\n",
       "      <td>67</td>\n",
       "      <td>Ala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251152</th>\n",
       "      <td>1863392</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_015915.5(ATL1):c.458G&gt;C (p.Ser153Thr)</td>\n",
       "      <td>51062</td>\n",
       "      <td>ATL1</td>\n",
       "      <td>HGNC:11231</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>May 06, 2021</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>458</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Ser153Thr</td>\n",
       "      <td>Ser</td>\n",
       "      <td>153</td>\n",
       "      <td>Thr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251154</th>\n",
       "      <td>1863393</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_001040142.2(SCN2A):c.5311T&gt;A (p.Tyr1771Asn)</td>\n",
       "      <td>6326</td>\n",
       "      <td>SCN2A</td>\n",
       "      <td>HGNC:10588</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>May 17, 2021</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>5311</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Tyr1771Asn</td>\n",
       "      <td>Tyr</td>\n",
       "      <td>1771</td>\n",
       "      <td>Asn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251170</th>\n",
       "      <td>1863408</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_181303.2(NLGN3):c.2222T&gt;G (p.Leu741Arg)</td>\n",
       "      <td>54413</td>\n",
       "      <td>NLGN3</td>\n",
       "      <td>HGNC:14289</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Nov 15, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>2222</td>\n",
       "      <td>T/G</td>\n",
       "      <td>Leu741Arg</td>\n",
       "      <td>Leu</td>\n",
       "      <td>741</td>\n",
       "      <td>Arg</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>379772 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #AlleleID                       Type  \\\n",
       "33           15058  single nucleotide variant   \n",
       "41           15062  single nucleotide variant   \n",
       "61           15072  single nucleotide variant   \n",
       "91           15087  single nucleotide variant   \n",
       "139          15111  single nucleotide variant   \n",
       "...            ...                        ...   \n",
       "3251142    1863387  single nucleotide variant   \n",
       "3251148    1863390  single nucleotide variant   \n",
       "3251152    1863392  single nucleotide variant   \n",
       "3251154    1863393  single nucleotide variant   \n",
       "3251170    1863408  single nucleotide variant   \n",
       "\n",
       "                                                   Name  GeneID GeneSymbol  \\\n",
       "33              NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n",
       "41           NM_020779.4(WDR35):c.2590G>A (p.Ala864Thr)   57539      WDR35   \n",
       "61             NM_138413.4(HOGA1):c.209G>C (p.Arg70Pro)  112817      HOGA1   \n",
       "91       NM_001199107.2(TBC1D24):c.439G>C (p.Asp147His)   57465    TBC1D24   \n",
       "139              NM_000374.5(UROD):c.942G>A (p.Glu314=)    7389       UROD   \n",
       "...                                                 ...     ...        ...   \n",
       "3251142     NM_000093.5(COL5A1):c.1433G>C (p.Gly478Ala)    1289     COL5A1   \n",
       "3251148       NM_000474.4(TWIST1):c.200G>C (p.Gly67Ala)    7291     TWIST1   \n",
       "3251152        NM_015915.5(ATL1):c.458G>C (p.Ser153Thr)   51062       ATL1   \n",
       "3251154  NM_001040142.2(SCN2A):c.5311T>A (p.Tyr1771Asn)    6326      SCN2A   \n",
       "3251170      NM_181303.2(NLGN3):c.2222T>G (p.Leu741Arg)   54413      NLGN3   \n",
       "\n",
       "            HGNC_ID ClinicalSignificance  ClinSigSimple LastEvaluated  \\\n",
       "33        HGNC:4886           Pathogenic              1  Sep 04, 2021   \n",
       "41       HGNC:29250    Likely pathogenic              1  May 25, 2017   \n",
       "61       HGNC:25155           Pathogenic              1  Sep 19, 2021   \n",
       "91       HGNC:29203           Pathogenic              1  Sep 01, 2021   \n",
       "139      HGNC:12591    Likely pathogenic              1  Aug 13, 2021   \n",
       "...             ...                  ...            ...           ...   \n",
       "3251142   HGNC:2209    Likely pathogenic              1  May 21, 2020   \n",
       "3251148  HGNC:12428        Likely benign              0  May 06, 2021   \n",
       "3251152  HGNC:11231    Likely pathogenic              1  May 06, 2021   \n",
       "3251154  HGNC:10588    Likely pathogenic              1  May 17, 2021   \n",
       "3251170  HGNC:14289    Likely pathogenic              1  Nov 15, 2022   \n",
       "\n",
       "         RS# (dbSNP)  ... ReferenceAlleleVCF AlternateAlleleVCF stars  \\\n",
       "33         111033563  ...                  A                  C     1   \n",
       "41         267607175  ...                  C                  T     1   \n",
       "61         267606763  ...                  G                  C     1   \n",
       "91         267607103  ...                  G                  C     1   \n",
       "139        121918062  ...                  G                  A     1   \n",
       "...              ...  ...                ...                ...   ...   \n",
       "3251142           -1  ...                  G                  C     1   \n",
       "3251148           -1  ...                  C                  G     1   \n",
       "3251152           -1  ...                  G                  C     1   \n",
       "3251154           -1  ...                  T                  A     1   \n",
       "3251170           -1  ...                  T                  G     1   \n",
       "\n",
       "                   classes CDS_position AlleleChange Protein_Patterns  \\\n",
       "33              Pathogenic          848          A/C        Gln283Pro   \n",
       "41       Likely pathogenic         2590          G/A        Ala864Thr   \n",
       "61              Pathogenic          209          G/C         Arg70Pro   \n",
       "91              Pathogenic          439          G/C        Asp147His   \n",
       "139      Likely pathogenic          942          G/A          Glu314=   \n",
       "...                    ...          ...          ...              ...   \n",
       "3251142  Likely pathogenic         1433          G/C        Gly478Ala   \n",
       "3251148      Likely benign          200          G/C         Gly67Ala   \n",
       "3251152  Likely pathogenic          458          G/C        Ser153Thr   \n",
       "3251154  Likely pathogenic         5311          T/A       Tyr1771Asn   \n",
       "3251170  Likely pathogenic         2222          T/G        Leu741Arg   \n",
       "\n",
       "        Amino_acids_before_3letters Protein_positions  \\\n",
       "33                              Gln               283   \n",
       "41                              Ala               864   \n",
       "61                              Arg                70   \n",
       "91                              Asp               147   \n",
       "139                             Glu               314   \n",
       "...                             ...               ...   \n",
       "3251142                         Gly               478   \n",
       "3251148                         Gly                67   \n",
       "3251152                         Ser               153   \n",
       "3251154                         Tyr              1771   \n",
       "3251170                         Leu               741   \n",
       "\n",
       "         Amino_acids_after_3letters  \n",
       "33                              Pro  \n",
       "41                              Thr  \n",
       "61                              Pro  \n",
       "91                              His  \n",
       "139                             Glu  \n",
       "...                             ...  \n",
       "3251142                         Ala  \n",
       "3251148                         Ala  \n",
       "3251152                         Thr  \n",
       "3251154                         Asn  \n",
       "3251170                         Arg  \n",
       "\n",
       "[379772 rows x 41 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop introns and premature stop codon\n",
    "df_meta = df_meta[(df_meta['Protein_Patterns'].notnull()) & (df_meta['Amino_acids_after_3letters'] != 'Ter')]\n",
    "df_meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/pbs.463587.hn-10-03/ipykernel_34616/4248943374.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_meta['Amino_acids_before'] = df_meta['Amino_acids_before_3letters'].map(amino_acids_dict)\n",
      "/var/tmp/pbs.463587.hn-10-03/ipykernel_34616/4248943374.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_meta['Amino_acids_after'] = df_meta['Amino_acids_after_3letters'].map(amino_acids_dict)\n",
      "/var/tmp/pbs.463587.hn-10-03/ipykernel_34616/4248943374.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_meta['Amino_acids'] = df_meta['Amino_acids_before'] + '/' + df_meta['Amino_acids_after']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>...</th>\n",
       "      <th>classes</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>AlleleChange</th>\n",
       "      <th>Protein_Patterns</th>\n",
       "      <th>Amino_acids_before_3letters</th>\n",
       "      <th>Protein_positions</th>\n",
       "      <th>Amino_acids_after_3letters</th>\n",
       "      <th>Amino_acids_before</th>\n",
       "      <th>Amino_acids_after</th>\n",
       "      <th>Amino_acids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>15058</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n",
       "      <td>3077</td>\n",
       "      <td>HFE</td>\n",
       "      <td>HGNC:4886</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Sep 04, 2021</td>\n",
       "      <td>111033563</td>\n",
       "      <td>...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>848</td>\n",
       "      <td>A/C</td>\n",
       "      <td>Gln283Pro</td>\n",
       "      <td>Gln</td>\n",
       "      <td>283</td>\n",
       "      <td>Pro</td>\n",
       "      <td>Q</td>\n",
       "      <td>P</td>\n",
       "      <td>Q/P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>15062</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_020779.4(WDR35):c.2590G&gt;A (p.Ala864Thr)</td>\n",
       "      <td>57539</td>\n",
       "      <td>WDR35</td>\n",
       "      <td>HGNC:29250</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>May 25, 2017</td>\n",
       "      <td>267607175</td>\n",
       "      <td>...</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>2590</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Ala864Thr</td>\n",
       "      <td>Ala</td>\n",
       "      <td>864</td>\n",
       "      <td>Thr</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>A/T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>15072</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_138413.4(HOGA1):c.209G&gt;C (p.Arg70Pro)</td>\n",
       "      <td>112817</td>\n",
       "      <td>HOGA1</td>\n",
       "      <td>HGNC:25155</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Sep 19, 2021</td>\n",
       "      <td>267606763</td>\n",
       "      <td>...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>209</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Arg70Pro</td>\n",
       "      <td>Arg</td>\n",
       "      <td>70</td>\n",
       "      <td>Pro</td>\n",
       "      <td>R</td>\n",
       "      <td>P</td>\n",
       "      <td>R/P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>15087</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_001199107.2(TBC1D24):c.439G&gt;C (p.Asp147His)</td>\n",
       "      <td>57465</td>\n",
       "      <td>TBC1D24</td>\n",
       "      <td>HGNC:29203</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Sep 01, 2021</td>\n",
       "      <td>267607103</td>\n",
       "      <td>...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>439</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Asp147His</td>\n",
       "      <td>Asp</td>\n",
       "      <td>147</td>\n",
       "      <td>His</td>\n",
       "      <td>D</td>\n",
       "      <td>H</td>\n",
       "      <td>D/H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>15114</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000374.5(UROD):c.912C&gt;A (p.Asn304Lys)</td>\n",
       "      <td>7389</td>\n",
       "      <td>UROD</td>\n",
       "      <td>HGNC:12591</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 14, 2021</td>\n",
       "      <td>121918065</td>\n",
       "      <td>...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>912</td>\n",
       "      <td>C/A</td>\n",
       "      <td>Asn304Lys</td>\n",
       "      <td>Asn</td>\n",
       "      <td>304</td>\n",
       "      <td>Lys</td>\n",
       "      <td>N</td>\n",
       "      <td>K</td>\n",
       "      <td>N/K</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251142</th>\n",
       "      <td>1863387</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000093.5(COL5A1):c.1433G&gt;C (p.Gly478Ala)</td>\n",
       "      <td>1289</td>\n",
       "      <td>COL5A1</td>\n",
       "      <td>HGNC:2209</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>May 21, 2020</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1433</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Gly478Ala</td>\n",
       "      <td>Gly</td>\n",
       "      <td>478</td>\n",
       "      <td>Ala</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G/A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251148</th>\n",
       "      <td>1863390</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000474.4(TWIST1):c.200G&gt;C (p.Gly67Ala)</td>\n",
       "      <td>7291</td>\n",
       "      <td>TWIST1</td>\n",
       "      <td>HGNC:12428</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>0</td>\n",
       "      <td>May 06, 2021</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>200</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Gly67Ala</td>\n",
       "      <td>Gly</td>\n",
       "      <td>67</td>\n",
       "      <td>Ala</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G/A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251152</th>\n",
       "      <td>1863392</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_015915.5(ATL1):c.458G&gt;C (p.Ser153Thr)</td>\n",
       "      <td>51062</td>\n",
       "      <td>ATL1</td>\n",
       "      <td>HGNC:11231</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>May 06, 2021</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>458</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Ser153Thr</td>\n",
       "      <td>Ser</td>\n",
       "      <td>153</td>\n",
       "      <td>Thr</td>\n",
       "      <td>S</td>\n",
       "      <td>T</td>\n",
       "      <td>S/T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251154</th>\n",
       "      <td>1863393</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_001040142.2(SCN2A):c.5311T&gt;A (p.Tyr1771Asn)</td>\n",
       "      <td>6326</td>\n",
       "      <td>SCN2A</td>\n",
       "      <td>HGNC:10588</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>May 17, 2021</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>5311</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Tyr1771Asn</td>\n",
       "      <td>Tyr</td>\n",
       "      <td>1771</td>\n",
       "      <td>Asn</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y/N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3251170</th>\n",
       "      <td>1863408</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_181303.2(NLGN3):c.2222T&gt;G (p.Leu741Arg)</td>\n",
       "      <td>54413</td>\n",
       "      <td>NLGN3</td>\n",
       "      <td>HGNC:14289</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Nov 15, 2022</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>2222</td>\n",
       "      <td>T/G</td>\n",
       "      <td>Leu741Arg</td>\n",
       "      <td>Leu</td>\n",
       "      <td>741</td>\n",
       "      <td>Arg</td>\n",
       "      <td>L</td>\n",
       "      <td>R</td>\n",
       "      <td>L/R</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>109856 rows × 44 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #AlleleID                       Type  \\\n",
       "33           15058  single nucleotide variant   \n",
       "41           15062  single nucleotide variant   \n",
       "61           15072  single nucleotide variant   \n",
       "91           15087  single nucleotide variant   \n",
       "145          15114  single nucleotide variant   \n",
       "...            ...                        ...   \n",
       "3251142    1863387  single nucleotide variant   \n",
       "3251148    1863390  single nucleotide variant   \n",
       "3251152    1863392  single nucleotide variant   \n",
       "3251154    1863393  single nucleotide variant   \n",
       "3251170    1863408  single nucleotide variant   \n",
       "\n",
       "                                                   Name  GeneID GeneSymbol  \\\n",
       "33              NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n",
       "41           NM_020779.4(WDR35):c.2590G>A (p.Ala864Thr)   57539      WDR35   \n",
       "61             NM_138413.4(HOGA1):c.209G>C (p.Arg70Pro)  112817      HOGA1   \n",
       "91       NM_001199107.2(TBC1D24):c.439G>C (p.Asp147His)   57465    TBC1D24   \n",
       "145            NM_000374.5(UROD):c.912C>A (p.Asn304Lys)    7389       UROD   \n",
       "...                                                 ...     ...        ...   \n",
       "3251142     NM_000093.5(COL5A1):c.1433G>C (p.Gly478Ala)    1289     COL5A1   \n",
       "3251148       NM_000474.4(TWIST1):c.200G>C (p.Gly67Ala)    7291     TWIST1   \n",
       "3251152        NM_015915.5(ATL1):c.458G>C (p.Ser153Thr)   51062       ATL1   \n",
       "3251154  NM_001040142.2(SCN2A):c.5311T>A (p.Tyr1771Asn)    6326      SCN2A   \n",
       "3251170      NM_181303.2(NLGN3):c.2222T>G (p.Leu741Arg)   54413      NLGN3   \n",
       "\n",
       "            HGNC_ID ClinicalSignificance  ClinSigSimple LastEvaluated  \\\n",
       "33        HGNC:4886           Pathogenic              1  Sep 04, 2021   \n",
       "41       HGNC:29250    Likely pathogenic              1  May 25, 2017   \n",
       "61       HGNC:25155           Pathogenic              1  Sep 19, 2021   \n",
       "91       HGNC:29203           Pathogenic              1  Sep 01, 2021   \n",
       "145      HGNC:12591           Pathogenic              1  Dec 14, 2021   \n",
       "...             ...                  ...            ...           ...   \n",
       "3251142   HGNC:2209    Likely pathogenic              1  May 21, 2020   \n",
       "3251148  HGNC:12428        Likely benign              0  May 06, 2021   \n",
       "3251152  HGNC:11231    Likely pathogenic              1  May 06, 2021   \n",
       "3251154  HGNC:10588    Likely pathogenic              1  May 17, 2021   \n",
       "3251170  HGNC:14289    Likely pathogenic              1  Nov 15, 2022   \n",
       "\n",
       "         RS# (dbSNP)  ...            classes CDS_position AlleleChange  \\\n",
       "33         111033563  ...         Pathogenic          848          A/C   \n",
       "41         267607175  ...  Likely pathogenic         2590          G/A   \n",
       "61         267606763  ...         Pathogenic          209          G/C   \n",
       "91         267607103  ...         Pathogenic          439          G/C   \n",
       "145        121918065  ...         Pathogenic          912          C/A   \n",
       "...              ...  ...                ...          ...          ...   \n",
       "3251142           -1  ...  Likely pathogenic         1433          G/C   \n",
       "3251148           -1  ...      Likely benign          200          G/C   \n",
       "3251152           -1  ...  Likely pathogenic          458          G/C   \n",
       "3251154           -1  ...  Likely pathogenic         5311          T/A   \n",
       "3251170           -1  ...  Likely pathogenic         2222          T/G   \n",
       "\n",
       "        Protein_Patterns Amino_acids_before_3letters Protein_positions  \\\n",
       "33             Gln283Pro                         Gln               283   \n",
       "41             Ala864Thr                         Ala               864   \n",
       "61              Arg70Pro                         Arg                70   \n",
       "91             Asp147His                         Asp               147   \n",
       "145            Asn304Lys                         Asn               304   \n",
       "...                  ...                         ...               ...   \n",
       "3251142        Gly478Ala                         Gly               478   \n",
       "3251148         Gly67Ala                         Gly                67   \n",
       "3251152        Ser153Thr                         Ser               153   \n",
       "3251154       Tyr1771Asn                         Tyr              1771   \n",
       "3251170        Leu741Arg                         Leu               741   \n",
       "\n",
       "        Amino_acids_after_3letters Amino_acids_before Amino_acids_after  \\\n",
       "33                             Pro                  Q                 P   \n",
       "41                             Thr                  A                 T   \n",
       "61                             Pro                  R                 P   \n",
       "91                             His                  D                 H   \n",
       "145                            Lys                  N                 K   \n",
       "...                            ...                ...               ...   \n",
       "3251142                        Ala                  G                 A   \n",
       "3251148                        Ala                  G                 A   \n",
       "3251152                        Thr                  S                 T   \n",
       "3251154                        Asn                  Y                 N   \n",
       "3251170                        Arg                  L                 R   \n",
       "\n",
       "         Amino_acids  \n",
       "33               Q/P  \n",
       "41               A/T  \n",
       "61               R/P  \n",
       "91               D/H  \n",
       "145              N/K  \n",
       "...              ...  \n",
       "3251142          G/A  \n",
       "3251148          G/A  \n",
       "3251152          S/T  \n",
       "3251154          Y/N  \n",
       "3251170          L/R  \n",
       "\n",
       "[109856 rows x 44 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# keep only nonsynonymous changes\n",
    "amino_acids_dict = {'Ala': 'A', 'Arg': 'R', 'Asn': 'N', 'Asp': 'D', 'Cys': 'C', 'Glu': 'E', 'Gln': 'Q', 'Gly': 'G', 'His': 'H', 'Ile': 'I', 'Leu': 'L', 'Lys': 'K', 'Met': 'M', 'Phe': 'F', 'Pro': 'P', 'Ser': 'S', 'Thr': 'T', 'Trp': 'W', 'Tyr': 'Y', 'Val': 'V'}\n",
    "df_meta['Amino_acids_before'] = df_meta['Amino_acids_before_3letters'].map(amino_acids_dict)\n",
    "df_meta['Amino_acids_after'] = df_meta['Amino_acids_after_3letters'].map(amino_acids_dict)\n",
    "df_meta = df_meta[df_meta['Amino_acids_before'] != df_meta['Amino_acids_after']]\n",
    "df_meta['Amino_acids'] = df_meta['Amino_acids_before'] + '/' + df_meta['Amino_acids_after']\n",
    "#df_subset.drop(['Amino_acids_after_3letters', 'Amino_acids_before_3letters', 'Amino_acids_before', 'Amino_acids_after'], axis=1, inplace=True)\n",
    "df_meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/pbs.463587.hn-10-03/ipykernel_34616/1721707379.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_subset['uniprot_acc'] = df_subset['OtherIDs'].str.extract(uniprot_id_regex).fillna('NA')\n",
      "/var/tmp/pbs.463587.hn-10-03/ipykernel_34616/1721707379.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_subset['variant_id'] = df_subset['OtherIDs'].str.extract(variant_id_regex).fillna('NA')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>...</th>\n",
       "      <th>AlleleChange</th>\n",
       "      <th>Protein_Patterns</th>\n",
       "      <th>Amino_acids_before_3letters</th>\n",
       "      <th>Protein_positions</th>\n",
       "      <th>Amino_acids_after_3letters</th>\n",
       "      <th>Amino_acids_before</th>\n",
       "      <th>Amino_acids_after</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>uniprot_acc</th>\n",
       "      <th>variant_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>15058</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000410.4(HFE):c.848A&gt;C (p.Gln283Pro)</td>\n",
       "      <td>3077</td>\n",
       "      <td>HFE</td>\n",
       "      <td>HGNC:4886</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Sep 04, 2021</td>\n",
       "      <td>111033563</td>\n",
       "      <td>...</td>\n",
       "      <td>A/C</td>\n",
       "      <td>Gln283Pro</td>\n",
       "      <td>Gln</td>\n",
       "      <td>283</td>\n",
       "      <td>Pro</td>\n",
       "      <td>Q</td>\n",
       "      <td>P</td>\n",
       "      <td>Q/P</td>\n",
       "      <td>Q30201</td>\n",
       "      <td>VAR_037304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>15062</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_020779.4(WDR35):c.2590G&gt;A (p.Ala864Thr)</td>\n",
       "      <td>57539</td>\n",
       "      <td>WDR35</td>\n",
       "      <td>HGNC:29250</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>May 25, 2017</td>\n",
       "      <td>267607175</td>\n",
       "      <td>...</td>\n",
       "      <td>G/A</td>\n",
       "      <td>Ala864Thr</td>\n",
       "      <td>Ala</td>\n",
       "      <td>864</td>\n",
       "      <td>Thr</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>A/T</td>\n",
       "      <td>Q9P2L0</td>\n",
       "      <td>VAR_064582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>15087</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_001199107.2(TBC1D24):c.439G&gt;C (p.Asp147His)</td>\n",
       "      <td>57465</td>\n",
       "      <td>TBC1D24</td>\n",
       "      <td>HGNC:29203</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Sep 01, 2021</td>\n",
       "      <td>267607103</td>\n",
       "      <td>...</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Asp147His</td>\n",
       "      <td>Asp</td>\n",
       "      <td>147</td>\n",
       "      <td>His</td>\n",
       "      <td>D</td>\n",
       "      <td>H</td>\n",
       "      <td>D/H</td>\n",
       "      <td>Q9ULP9</td>\n",
       "      <td>VAR_064365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>15114</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000374.5(UROD):c.912C&gt;A (p.Asn304Lys)</td>\n",
       "      <td>7389</td>\n",
       "      <td>UROD</td>\n",
       "      <td>HGNC:12591</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 14, 2021</td>\n",
       "      <td>121918065</td>\n",
       "      <td>...</td>\n",
       "      <td>C/A</td>\n",
       "      <td>Asn304Lys</td>\n",
       "      <td>Asn</td>\n",
       "      <td>304</td>\n",
       "      <td>Lys</td>\n",
       "      <td>N</td>\n",
       "      <td>K</td>\n",
       "      <td>N/K</td>\n",
       "      <td>P06132</td>\n",
       "      <td>VAR_007914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>15136</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_000355.4(TCN2):c.776G&gt;C (p.Arg259Pro)</td>\n",
       "      <td>6948</td>\n",
       "      <td>TCN2</td>\n",
       "      <td>HGNC:11653</td>\n",
       "      <td>Benign</td>\n",
       "      <td>0</td>\n",
       "      <td>Jan 06, 2022</td>\n",
       "      <td>1801198</td>\n",
       "      <td>...</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Arg259Pro</td>\n",
       "      <td>Arg</td>\n",
       "      <td>259</td>\n",
       "      <td>Pro</td>\n",
       "      <td>R</td>\n",
       "      <td>P</td>\n",
       "      <td>R/P</td>\n",
       "      <td>P20062</td>\n",
       "      <td>VAR_001640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510517</th>\n",
       "      <td>354137</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_023067.4(FOXL2):c.316C&gt;T (p.Leu106Phe)</td>\n",
       "      <td>668</td>\n",
       "      <td>FOXL2</td>\n",
       "      <td>HGNC:1092</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Nov 03, 2016</td>\n",
       "      <td>1057516156</td>\n",
       "      <td>...</td>\n",
       "      <td>C/T</td>\n",
       "      <td>Leu106Phe</td>\n",
       "      <td>Leu</td>\n",
       "      <td>106</td>\n",
       "      <td>Phe</td>\n",
       "      <td>L</td>\n",
       "      <td>F</td>\n",
       "      <td>L/F</td>\n",
       "      <td>P58012</td>\n",
       "      <td>VAR_016885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510521</th>\n",
       "      <td>354139</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_023067.4(FOXL2):c.311A&gt;G (p.His104Arg)</td>\n",
       "      <td>668</td>\n",
       "      <td>FOXL2</td>\n",
       "      <td>HGNC:1092</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Nov 03, 2016</td>\n",
       "      <td>1057516153</td>\n",
       "      <td>...</td>\n",
       "      <td>A/G</td>\n",
       "      <td>His104Arg</td>\n",
       "      <td>His</td>\n",
       "      <td>104</td>\n",
       "      <td>Arg</td>\n",
       "      <td>H</td>\n",
       "      <td>R</td>\n",
       "      <td>H/R</td>\n",
       "      <td>P58012</td>\n",
       "      <td>VAR_021199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510527</th>\n",
       "      <td>354142</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_023067.4(FOXL2):c.303C&gt;G (p.Ser101Arg)</td>\n",
       "      <td>668</td>\n",
       "      <td>FOXL2</td>\n",
       "      <td>HGNC:1092</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Nov 03, 2016</td>\n",
       "      <td>1057516151</td>\n",
       "      <td>...</td>\n",
       "      <td>C/G</td>\n",
       "      <td>Ser101Arg</td>\n",
       "      <td>Ser</td>\n",
       "      <td>101</td>\n",
       "      <td>Arg</td>\n",
       "      <td>S</td>\n",
       "      <td>R</td>\n",
       "      <td>S/R</td>\n",
       "      <td>P58012</td>\n",
       "      <td>VAR_046495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510531</th>\n",
       "      <td>354144</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_023067.4(FOXL2):c.292T&gt;C (p.Trp98Arg)</td>\n",
       "      <td>668</td>\n",
       "      <td>FOXL2</td>\n",
       "      <td>HGNC:1092</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Nov 03, 2016</td>\n",
       "      <td>1057516149</td>\n",
       "      <td>...</td>\n",
       "      <td>T/C</td>\n",
       "      <td>Trp98Arg</td>\n",
       "      <td>Trp</td>\n",
       "      <td>98</td>\n",
       "      <td>Arg</td>\n",
       "      <td>W</td>\n",
       "      <td>R</td>\n",
       "      <td>W/R</td>\n",
       "      <td>P58012</td>\n",
       "      <td>VAR_062546</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510668</th>\n",
       "      <td>354275</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_006912.6(RIT1):c.246T&gt;A (p.Phe82Leu)</td>\n",
       "      <td>6016</td>\n",
       "      <td>RIT1</td>\n",
       "      <td>HGNC:10023</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Aug 12, 2021</td>\n",
       "      <td>730881014</td>\n",
       "      <td>...</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Phe82Leu</td>\n",
       "      <td>Phe</td>\n",
       "      <td>82</td>\n",
       "      <td>Leu</td>\n",
       "      <td>F</td>\n",
       "      <td>L</td>\n",
       "      <td>F/L</td>\n",
       "      <td>Q92963</td>\n",
       "      <td>VAR_070152</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7049 rows × 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        #AlleleID                       Type  \\\n",
       "33          15058  single nucleotide variant   \n",
       "41          15062  single nucleotide variant   \n",
       "91          15087  single nucleotide variant   \n",
       "145         15114  single nucleotide variant   \n",
       "185         15136  single nucleotide variant   \n",
       "...           ...                        ...   \n",
       "510517     354137  single nucleotide variant   \n",
       "510521     354139  single nucleotide variant   \n",
       "510527     354142  single nucleotide variant   \n",
       "510531     354144  single nucleotide variant   \n",
       "510668     354275  single nucleotide variant   \n",
       "\n",
       "                                                  Name  GeneID GeneSymbol  \\\n",
       "33             NM_000410.4(HFE):c.848A>C (p.Gln283Pro)    3077        HFE   \n",
       "41          NM_020779.4(WDR35):c.2590G>A (p.Ala864Thr)   57539      WDR35   \n",
       "91      NM_001199107.2(TBC1D24):c.439G>C (p.Asp147His)   57465    TBC1D24   \n",
       "145           NM_000374.5(UROD):c.912C>A (p.Asn304Lys)    7389       UROD   \n",
       "185           NM_000355.4(TCN2):c.776G>C (p.Arg259Pro)    6948       TCN2   \n",
       "...                                                ...     ...        ...   \n",
       "510517       NM_023067.4(FOXL2):c.316C>T (p.Leu106Phe)     668      FOXL2   \n",
       "510521       NM_023067.4(FOXL2):c.311A>G (p.His104Arg)     668      FOXL2   \n",
       "510527       NM_023067.4(FOXL2):c.303C>G (p.Ser101Arg)     668      FOXL2   \n",
       "510531        NM_023067.4(FOXL2):c.292T>C (p.Trp98Arg)     668      FOXL2   \n",
       "510668         NM_006912.6(RIT1):c.246T>A (p.Phe82Leu)    6016       RIT1   \n",
       "\n",
       "           HGNC_ID ClinicalSignificance  ClinSigSimple LastEvaluated  \\\n",
       "33       HGNC:4886           Pathogenic              1  Sep 04, 2021   \n",
       "41      HGNC:29250    Likely pathogenic              1  May 25, 2017   \n",
       "91      HGNC:29203           Pathogenic              1  Sep 01, 2021   \n",
       "145     HGNC:12591           Pathogenic              1  Dec 14, 2021   \n",
       "185     HGNC:11653               Benign              0  Jan 06, 2022   \n",
       "...            ...                  ...            ...           ...   \n",
       "510517   HGNC:1092    Likely pathogenic              1  Nov 03, 2016   \n",
       "510521   HGNC:1092    Likely pathogenic              1  Nov 03, 2016   \n",
       "510527   HGNC:1092    Likely pathogenic              1  Nov 03, 2016   \n",
       "510531   HGNC:1092    Likely pathogenic              1  Nov 03, 2016   \n",
       "510668  HGNC:10023           Pathogenic              1  Aug 12, 2021   \n",
       "\n",
       "        RS# (dbSNP)  ... AlleleChange Protein_Patterns  \\\n",
       "33        111033563  ...          A/C        Gln283Pro   \n",
       "41        267607175  ...          G/A        Ala864Thr   \n",
       "91        267607103  ...          G/C        Asp147His   \n",
       "145       121918065  ...          C/A        Asn304Lys   \n",
       "185         1801198  ...          G/C        Arg259Pro   \n",
       "...             ...  ...          ...              ...   \n",
       "510517   1057516156  ...          C/T        Leu106Phe   \n",
       "510521   1057516153  ...          A/G        His104Arg   \n",
       "510527   1057516151  ...          C/G        Ser101Arg   \n",
       "510531   1057516149  ...          T/C         Trp98Arg   \n",
       "510668    730881014  ...          T/A         Phe82Leu   \n",
       "\n",
       "       Amino_acids_before_3letters Protein_positions  \\\n",
       "33                             Gln               283   \n",
       "41                             Ala               864   \n",
       "91                             Asp               147   \n",
       "145                            Asn               304   \n",
       "185                            Arg               259   \n",
       "...                            ...               ...   \n",
       "510517                         Leu               106   \n",
       "510521                         His               104   \n",
       "510527                         Ser               101   \n",
       "510531                         Trp                98   \n",
       "510668                         Phe                82   \n",
       "\n",
       "       Amino_acids_after_3letters Amino_acids_before Amino_acids_after  \\\n",
       "33                            Pro                  Q                 P   \n",
       "41                            Thr                  A                 T   \n",
       "91                            His                  D                 H   \n",
       "145                           Lys                  N                 K   \n",
       "185                           Pro                  R                 P   \n",
       "...                           ...                ...               ...   \n",
       "510517                        Phe                  L                 F   \n",
       "510521                        Arg                  H                 R   \n",
       "510527                        Arg                  S                 R   \n",
       "510531                        Arg                  W                 R   \n",
       "510668                        Leu                  F                 L   \n",
       "\n",
       "       Amino_acids uniprot_acc  variant_id  \n",
       "33             Q/P      Q30201  VAR_037304  \n",
       "41             A/T      Q9P2L0  VAR_064582  \n",
       "91             D/H      Q9ULP9  VAR_064365  \n",
       "145            N/K      P06132  VAR_007914  \n",
       "185            R/P      P20062  VAR_001640  \n",
       "...            ...         ...         ...  \n",
       "510517         L/F      P58012  VAR_016885  \n",
       "510521         H/R      P58012  VAR_021199  \n",
       "510527         S/R      P58012  VAR_046495  \n",
       "510531         W/R      P58012  VAR_062546  \n",
       "510668         F/L      Q92963  VAR_070152  \n",
       "\n",
       "[7049 rows x 46 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# uniprot_entry\n",
    "df_subset = df_meta[df_meta['OtherIDs'].str.contains('UniProtKB')]\n",
    "uniprot_id_regex = r'UniProtKB:(.+?)#VAR'\n",
    "variant_id_regex = r'(VAR_\\d+)'\n",
    "df_subset['uniprot_acc'] = df_subset['OtherIDs'].str.extract(uniprot_id_regex).fillna('NA')\n",
    "df_subset['variant_id'] = df_subset['OtherIDs'].str.extract(variant_id_regex).fillna('NA')\n",
    "df_subset"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "df = pd.DataFrame({'data': ['ClinGen:CA251380,UniProtKB:P06132#VAR_007914,OMIM:613521.0011',\n",
    "                            'ClinGen:CA113840,OMIM:613468.0002',\n",
    "                            'ClinGen:CA285111,UniProtKB/Swiss-Prot:VAR_029686']})\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "uni_prot_regex = r'UniProtKB:(.+?)#VAR'\n",
    "variant_id_regex = r'(VAR_\\d+)'\n",
    "df['uniprot_acc'] = df['data'].str.extract(uni_prot_regex)\n",
    "df['variant_id'] = df['data'].str.extract(variant_id_regex)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprot_acc</th>\n",
       "      <th>Amino_acids_before</th>\n",
       "      <th>Amino_acids_after</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Protein_positions</th>\n",
       "      <th>classes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Q30201</td>\n",
       "      <td>Q</td>\n",
       "      <td>P</td>\n",
       "      <td>Q/P</td>\n",
       "      <td>283</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Q9P2L0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>A/T</td>\n",
       "      <td>864</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>Q9ULP9</td>\n",
       "      <td>D</td>\n",
       "      <td>H</td>\n",
       "      <td>D/H</td>\n",
       "      <td>147</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>P06132</td>\n",
       "      <td>N</td>\n",
       "      <td>K</td>\n",
       "      <td>N/K</td>\n",
       "      <td>304</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>P20062</td>\n",
       "      <td>R</td>\n",
       "      <td>P</td>\n",
       "      <td>R/P</td>\n",
       "      <td>259</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510517</th>\n",
       "      <td>P58012</td>\n",
       "      <td>L</td>\n",
       "      <td>F</td>\n",
       "      <td>L/F</td>\n",
       "      <td>106</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510521</th>\n",
       "      <td>P58012</td>\n",
       "      <td>H</td>\n",
       "      <td>R</td>\n",
       "      <td>H/R</td>\n",
       "      <td>104</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510527</th>\n",
       "      <td>P58012</td>\n",
       "      <td>S</td>\n",
       "      <td>R</td>\n",
       "      <td>S/R</td>\n",
       "      <td>101</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510531</th>\n",
       "      <td>P58012</td>\n",
       "      <td>W</td>\n",
       "      <td>R</td>\n",
       "      <td>W/R</td>\n",
       "      <td>98</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510668</th>\n",
       "      <td>Q92963</td>\n",
       "      <td>F</td>\n",
       "      <td>L</td>\n",
       "      <td>F/L</td>\n",
       "      <td>82</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7049 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       uniprot_acc Amino_acids_before Amino_acids_after Amino_acids  \\\n",
       "33          Q30201                  Q                 P         Q/P   \n",
       "41          Q9P2L0                  A                 T         A/T   \n",
       "91          Q9ULP9                  D                 H         D/H   \n",
       "145         P06132                  N                 K         N/K   \n",
       "185         P20062                  R                 P         R/P   \n",
       "...            ...                ...               ...         ...   \n",
       "510517      P58012                  L                 F         L/F   \n",
       "510521      P58012                  H                 R         H/R   \n",
       "510527      P58012                  S                 R         S/R   \n",
       "510531      P58012                  W                 R         W/R   \n",
       "510668      Q92963                  F                 L         F/L   \n",
       "\n",
       "        Protein_positions            classes  \n",
       "33                    283         Pathogenic  \n",
       "41                    864  Likely pathogenic  \n",
       "91                    147         Pathogenic  \n",
       "145                   304         Pathogenic  \n",
       "185                   259             Benign  \n",
       "...                   ...                ...  \n",
       "510517                106  Likely pathogenic  \n",
       "510521                104  Likely pathogenic  \n",
       "510527                101  Likely pathogenic  \n",
       "510531                 98  Likely pathogenic  \n",
       "510668                 82         Pathogenic  \n",
       "\n",
       "[7049 rows x 6 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_preliminary_check = df_subset[['uniprot_acc', 'Amino_acids_before', 'Amino_acids_after', 'Amino_acids', 'Protein_positions', 'classes']]\n",
    "df_preliminary_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "df_preliminary_check.to_csv('./7049_preliminary_check.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>...</th>\n",
       "      <th>AlleleChange</th>\n",
       "      <th>Protein_Patterns</th>\n",
       "      <th>Amino_acids_before_3letters</th>\n",
       "      <th>Protein_positions</th>\n",
       "      <th>Amino_acids_after_3letters</th>\n",
       "      <th>Amino_acids_before</th>\n",
       "      <th>Amino_acids_after</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>uniprot_acc</th>\n",
       "      <th>variant_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [#AlleleID, Type, Name, GeneID, GeneSymbol, HGNC_ID, ClinicalSignificance, ClinSigSimple, LastEvaluated, RS# (dbSNP), nsv/esv (dbVar), RCVaccession, PhenotypeIDS, PhenotypeList, Origin, OriginSimple, Assembly, ChromosomeAccession, Chromosome, Start, Stop, ReferenceAllele, AlternateAllele, Cytogenetic, NumberSubmitters, Guidelines, TestedInGTR, OtherIDs, SubmitterCategories, VariationID, PositionVCF, ReferenceAlleleVCF, AlternateAlleleVCF, stars, classes, CDS_position, AlleleChange, Protein_Patterns, Amino_acids_before_3letters, Protein_positions, Amino_acids_after_3letters, Amino_acids_before, Amino_acids_after, Amino_acids, uniprot_acc, variant_id]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 46 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_need_map = df_subset[(df_subset['OtherIDs'].str.contains('UniProtKB/Swiss-Prot')) & (df_subset['uniprot_acc'].isna())]\n",
    "df_need_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "# processing uniprotkb variant metadata into a dictionary\n",
    "import itertools\n",
    "import re\n",
    "uniprot_variant_dict = {}\n",
    "len_of_disease_entries = 0\n",
    "with open('./data/metas/humsavar.txt') as f:\n",
    "    for line in itertools.islice(f, 44, 81966):\n",
    "        # check weekly record\n",
    "        # the gene_name seems never exceed 10 character, use ' ' cannot strip between gene name and accession number\n",
    "        gene_name = line[:10].strip()\n",
    "        fields = re.split(r'\\s+', line[10:].strip())\n",
    "        try:\n",
    "            key = fields[0]+'/'+fields[1]+'/'+gene_name\n",
    "            if key in uniprot_variant_dict.keys():\n",
    "                # if this critera is hit, this variant has multiple disease OMIM\n",
    "                uniprot_variant_dict[key]['Disease_name'].append(' '.join(fields[5:]))\n",
    "                len_of_disease_entries += 1\n",
    "                #print(uniprot_variant_dict[key], line)\n",
    "                pass\n",
    "            else:\n",
    "                uniprot_variant_dict[key] = {}\n",
    "                uniprot_variant_dict[key]['Gene_name'] = gene_name\n",
    "                uniprot_variant_dict[key]['Accession'] = fields[0]\n",
    "                uniprot_variant_dict[key]['Variant_FTId'] = fields[1]\n",
    "                uniprot_variant_dict[key]['Amino_acid_change'] = fields[2]\n",
    "                uniprot_variant_dict[key]['Variant_category'] = fields[3]\n",
    "                uniprot_variant_dict[key]['dbSNP'] = fields[4]\n",
    "                uniprot_variant_dict[key]['Disease_name'] = [' '.join(fields[5:])]\n",
    "        except IndexError:\n",
    "            print(line)\n",
    "uniprot_variant_dict = {k: {**v, 'Disease_name':'; '.join(v['Disease_name'])} for k, v in uniprot_variant_dict.items()}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_name</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Variant_FTId</th>\n",
       "      <th>Amino_acid_change</th>\n",
       "      <th>Variant_category</th>\n",
       "      <th>dbSNP</th>\n",
       "      <th>Disease_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P04217/VAR_018369/A1BG</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>P04217</td>\n",
       "      <td>VAR_018369</td>\n",
       "      <td>p.His52Arg</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs893184</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P04217/VAR_018370/A1BG</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>P04217</td>\n",
       "      <td>VAR_018370</td>\n",
       "      <td>p.His395Arg</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs2241788</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q9NQ94/VAR_052201/A1CF</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>Q9NQ94</td>\n",
       "      <td>VAR_052201</td>\n",
       "      <td>p.Val555Met</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs9073</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q9NQ94/VAR_059821/A1CF</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>Q9NQ94</td>\n",
       "      <td>VAR_059821</td>\n",
       "      <td>p.Ala558Ser</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs11817448</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P01023/VAR_000012/A2M</th>\n",
       "      <td>A2M</td>\n",
       "      <td>P01023</td>\n",
       "      <td>VAR_000012</td>\n",
       "      <td>p.Arg704His</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs1800434</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q8N402/VAR_042675/-</th>\n",
       "      <td>-</td>\n",
       "      <td>Q8N402</td>\n",
       "      <td>VAR_042675</td>\n",
       "      <td>p.Thr131Pro</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs6519442</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q96M66/VAR_039178/-</th>\n",
       "      <td>-</td>\n",
       "      <td>Q96M66</td>\n",
       "      <td>VAR_039178</td>\n",
       "      <td>p.Arg37His</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs350229</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q96M66/VAR_039179/-</th>\n",
       "      <td>-</td>\n",
       "      <td>Q96M66</td>\n",
       "      <td>VAR_039179</td>\n",
       "      <td>p.Arg171Ser</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs11648228</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q9N2K0/VAR_017799/-</th>\n",
       "      <td>-</td>\n",
       "      <td>Q9N2K0</td>\n",
       "      <td>VAR_017799</td>\n",
       "      <td>p.Val81Leu</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q9N2K0/VAR_017800/-</th>\n",
       "      <td>-</td>\n",
       "      <td>Q9N2K0</td>\n",
       "      <td>VAR_017800</td>\n",
       "      <td>p.Phe150Leu</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>80854 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       Gene_name Accession Variant_FTId Amino_acid_change  \\\n",
       "P04217/VAR_018369/A1BG      A1BG    P04217   VAR_018369        p.His52Arg   \n",
       "P04217/VAR_018370/A1BG      A1BG    P04217   VAR_018370       p.His395Arg   \n",
       "Q9NQ94/VAR_052201/A1CF      A1CF    Q9NQ94   VAR_052201       p.Val555Met   \n",
       "Q9NQ94/VAR_059821/A1CF      A1CF    Q9NQ94   VAR_059821       p.Ala558Ser   \n",
       "P01023/VAR_000012/A2M        A2M    P01023   VAR_000012       p.Arg704His   \n",
       "...                          ...       ...          ...               ...   \n",
       "Q8N402/VAR_042675/-            -    Q8N402   VAR_042675       p.Thr131Pro   \n",
       "Q96M66/VAR_039178/-            -    Q96M66   VAR_039178        p.Arg37His   \n",
       "Q96M66/VAR_039179/-            -    Q96M66   VAR_039179       p.Arg171Ser   \n",
       "Q9N2K0/VAR_017799/-            -    Q9N2K0   VAR_017799        p.Val81Leu   \n",
       "Q9N2K0/VAR_017800/-            -    Q9N2K0   VAR_017800       p.Phe150Leu   \n",
       "\n",
       "                       Variant_category       dbSNP Disease_name  \n",
       "P04217/VAR_018369/A1BG             LB/B    rs893184            -  \n",
       "P04217/VAR_018370/A1BG             LB/B   rs2241788            -  \n",
       "Q9NQ94/VAR_052201/A1CF             LB/B      rs9073            -  \n",
       "Q9NQ94/VAR_059821/A1CF             LB/B  rs11817448            -  \n",
       "P01023/VAR_000012/A2M              LB/B   rs1800434            -  \n",
       "...                                 ...         ...          ...  \n",
       "Q8N402/VAR_042675/-                LB/B   rs6519442            -  \n",
       "Q96M66/VAR_039178/-                LB/B    rs350229            -  \n",
       "Q96M66/VAR_039179/-                LB/B  rs11648228            -  \n",
       "Q9N2K0/VAR_017799/-                LB/B           -            -  \n",
       "Q9N2K0/VAR_017800/-                LB/B           -            -  \n",
       "\n",
       "[80854 rows x 7 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# make the above info into dataframe\n",
    "df_var = pd.DataFrame.from_dict(uniprot_variant_dict, orient='index')\n",
    "df_var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1068\n"
     ]
    }
   ],
   "source": [
    "print(len_of_disease_entries)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_name</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Variant_FTId</th>\n",
       "      <th>Amino_acid_change</th>\n",
       "      <th>Variant_category</th>\n",
       "      <th>dbSNP</th>\n",
       "      <th>Disease_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P35498_VAR_064269</th>\n",
       "      <td>SCN1A</td>\n",
       "      <td>P35498</td>\n",
       "      <td>VAR_064269</td>\n",
       "      <td>p.Arg1645Gln</td>\n",
       "      <td>LP/P</td>\n",
       "      <td>rs121917976</td>\n",
       "      <td>Dravet syndrome (DRVT) [MIM:607208]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Gene_name Accession Variant_FTId Amino_acid_change  \\\n",
       "P35498_VAR_064269     SCN1A    P35498   VAR_064269      p.Arg1645Gln   \n",
       "\n",
       "                  Variant_category        dbSNP  \\\n",
       "P35498_VAR_064269             LP/P  rs121917976   \n",
       "\n",
       "                                          Disease_name  \n",
       "P35498_VAR_064269  Dravet syndrome (DRVT) [MIM:607208]  "
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# ClinGen:CA284991,UniProtKB/Swiss-Prot:VAR_064269\n",
    "df_var[df_var['Variant_FTId'] == 'VAR_064269']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_name</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Variant_FTId</th>\n",
       "      <th>Amino_acid_change</th>\n",
       "      <th>Variant_category</th>\n",
       "      <th>dbSNP</th>\n",
       "      <th>Disease_name</th>\n",
       "      <th>Amino_acids_before_3letters</th>\n",
       "      <th>Protein_positions</th>\n",
       "      <th>Amino_acids_after_3letters</th>\n",
       "      <th>Amino_acids_before</th>\n",
       "      <th>Amino_acids_after</th>\n",
       "      <th>Amino_acids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P04217/VAR_018369/A1BG</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>P04217</td>\n",
       "      <td>VAR_018369</td>\n",
       "      <td>p.His52Arg</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs893184</td>\n",
       "      <td>-</td>\n",
       "      <td>His</td>\n",
       "      <td>52</td>\n",
       "      <td>Arg</td>\n",
       "      <td>H</td>\n",
       "      <td>R</td>\n",
       "      <td>H/R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P04217/VAR_018370/A1BG</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>P04217</td>\n",
       "      <td>VAR_018370</td>\n",
       "      <td>p.His395Arg</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs2241788</td>\n",
       "      <td>-</td>\n",
       "      <td>His</td>\n",
       "      <td>395</td>\n",
       "      <td>Arg</td>\n",
       "      <td>H</td>\n",
       "      <td>R</td>\n",
       "      <td>H/R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q9NQ94/VAR_052201/A1CF</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>Q9NQ94</td>\n",
       "      <td>VAR_052201</td>\n",
       "      <td>p.Val555Met</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs9073</td>\n",
       "      <td>-</td>\n",
       "      <td>Val</td>\n",
       "      <td>555</td>\n",
       "      <td>Met</td>\n",
       "      <td>V</td>\n",
       "      <td>M</td>\n",
       "      <td>V/M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q9NQ94/VAR_059821/A1CF</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>Q9NQ94</td>\n",
       "      <td>VAR_059821</td>\n",
       "      <td>p.Ala558Ser</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs11817448</td>\n",
       "      <td>-</td>\n",
       "      <td>Ala</td>\n",
       "      <td>558</td>\n",
       "      <td>Ser</td>\n",
       "      <td>A</td>\n",
       "      <td>S</td>\n",
       "      <td>A/S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P01023/VAR_000012/A2M</th>\n",
       "      <td>A2M</td>\n",
       "      <td>P01023</td>\n",
       "      <td>VAR_000012</td>\n",
       "      <td>p.Arg704His</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs1800434</td>\n",
       "      <td>-</td>\n",
       "      <td>Arg</td>\n",
       "      <td>704</td>\n",
       "      <td>His</td>\n",
       "      <td>R</td>\n",
       "      <td>H</td>\n",
       "      <td>R/H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q8N402/VAR_042675/-</th>\n",
       "      <td>-</td>\n",
       "      <td>Q8N402</td>\n",
       "      <td>VAR_042675</td>\n",
       "      <td>p.Thr131Pro</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs6519442</td>\n",
       "      <td>-</td>\n",
       "      <td>Thr</td>\n",
       "      <td>131</td>\n",
       "      <td>Pro</td>\n",
       "      <td>T</td>\n",
       "      <td>P</td>\n",
       "      <td>T/P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q96M66/VAR_039178/-</th>\n",
       "      <td>-</td>\n",
       "      <td>Q96M66</td>\n",
       "      <td>VAR_039178</td>\n",
       "      <td>p.Arg37His</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs350229</td>\n",
       "      <td>-</td>\n",
       "      <td>Arg</td>\n",
       "      <td>37</td>\n",
       "      <td>His</td>\n",
       "      <td>R</td>\n",
       "      <td>H</td>\n",
       "      <td>R/H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q96M66/VAR_039179/-</th>\n",
       "      <td>-</td>\n",
       "      <td>Q96M66</td>\n",
       "      <td>VAR_039179</td>\n",
       "      <td>p.Arg171Ser</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>rs11648228</td>\n",
       "      <td>-</td>\n",
       "      <td>Arg</td>\n",
       "      <td>171</td>\n",
       "      <td>Ser</td>\n",
       "      <td>R</td>\n",
       "      <td>S</td>\n",
       "      <td>R/S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q9N2K0/VAR_017799/-</th>\n",
       "      <td>-</td>\n",
       "      <td>Q9N2K0</td>\n",
       "      <td>VAR_017799</td>\n",
       "      <td>p.Val81Leu</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Val</td>\n",
       "      <td>81</td>\n",
       "      <td>Leu</td>\n",
       "      <td>V</td>\n",
       "      <td>L</td>\n",
       "      <td>V/L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q9N2K0/VAR_017800/-</th>\n",
       "      <td>-</td>\n",
       "      <td>Q9N2K0</td>\n",
       "      <td>VAR_017800</td>\n",
       "      <td>p.Phe150Leu</td>\n",
       "      <td>LB/B</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Phe</td>\n",
       "      <td>150</td>\n",
       "      <td>Leu</td>\n",
       "      <td>F</td>\n",
       "      <td>L</td>\n",
       "      <td>F/L</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>80854 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       Gene_name Accession Variant_FTId Amino_acid_change  \\\n",
       "P04217/VAR_018369/A1BG      A1BG    P04217   VAR_018369        p.His52Arg   \n",
       "P04217/VAR_018370/A1BG      A1BG    P04217   VAR_018370       p.His395Arg   \n",
       "Q9NQ94/VAR_052201/A1CF      A1CF    Q9NQ94   VAR_052201       p.Val555Met   \n",
       "Q9NQ94/VAR_059821/A1CF      A1CF    Q9NQ94   VAR_059821       p.Ala558Ser   \n",
       "P01023/VAR_000012/A2M        A2M    P01023   VAR_000012       p.Arg704His   \n",
       "...                          ...       ...          ...               ...   \n",
       "Q8N402/VAR_042675/-            -    Q8N402   VAR_042675       p.Thr131Pro   \n",
       "Q96M66/VAR_039178/-            -    Q96M66   VAR_039178        p.Arg37His   \n",
       "Q96M66/VAR_039179/-            -    Q96M66   VAR_039179       p.Arg171Ser   \n",
       "Q9N2K0/VAR_017799/-            -    Q9N2K0   VAR_017799        p.Val81Leu   \n",
       "Q9N2K0/VAR_017800/-            -    Q9N2K0   VAR_017800       p.Phe150Leu   \n",
       "\n",
       "                       Variant_category       dbSNP Disease_name  \\\n",
       "P04217/VAR_018369/A1BG             LB/B    rs893184            -   \n",
       "P04217/VAR_018370/A1BG             LB/B   rs2241788            -   \n",
       "Q9NQ94/VAR_052201/A1CF             LB/B      rs9073            -   \n",
       "Q9NQ94/VAR_059821/A1CF             LB/B  rs11817448            -   \n",
       "P01023/VAR_000012/A2M              LB/B   rs1800434            -   \n",
       "...                                 ...         ...          ...   \n",
       "Q8N402/VAR_042675/-                LB/B   rs6519442            -   \n",
       "Q96M66/VAR_039178/-                LB/B    rs350229            -   \n",
       "Q96M66/VAR_039179/-                LB/B  rs11648228            -   \n",
       "Q9N2K0/VAR_017799/-                LB/B           -            -   \n",
       "Q9N2K0/VAR_017800/-                LB/B           -            -   \n",
       "\n",
       "                       Amino_acids_before_3letters  Protein_positions  \\\n",
       "P04217/VAR_018369/A1BG                         His                 52   \n",
       "P04217/VAR_018370/A1BG                         His                395   \n",
       "Q9NQ94/VAR_052201/A1CF                         Val                555   \n",
       "Q9NQ94/VAR_059821/A1CF                         Ala                558   \n",
       "P01023/VAR_000012/A2M                          Arg                704   \n",
       "...                                            ...                ...   \n",
       "Q8N402/VAR_042675/-                            Thr                131   \n",
       "Q96M66/VAR_039178/-                            Arg                 37   \n",
       "Q96M66/VAR_039179/-                            Arg                171   \n",
       "Q9N2K0/VAR_017799/-                            Val                 81   \n",
       "Q9N2K0/VAR_017800/-                            Phe                150   \n",
       "\n",
       "                       Amino_acids_after_3letters Amino_acids_before  \\\n",
       "P04217/VAR_018369/A1BG                        Arg                  H   \n",
       "P04217/VAR_018370/A1BG                        Arg                  H   \n",
       "Q9NQ94/VAR_052201/A1CF                        Met                  V   \n",
       "Q9NQ94/VAR_059821/A1CF                        Ser                  A   \n",
       "P01023/VAR_000012/A2M                         His                  R   \n",
       "...                                           ...                ...   \n",
       "Q8N402/VAR_042675/-                           Pro                  T   \n",
       "Q96M66/VAR_039178/-                           His                  R   \n",
       "Q96M66/VAR_039179/-                           Ser                  R   \n",
       "Q9N2K0/VAR_017799/-                           Leu                  V   \n",
       "Q9N2K0/VAR_017800/-                           Leu                  F   \n",
       "\n",
       "                       Amino_acids_after Amino_acids  \n",
       "P04217/VAR_018369/A1BG                 R         H/R  \n",
       "P04217/VAR_018370/A1BG                 R         H/R  \n",
       "Q9NQ94/VAR_052201/A1CF                 M         V/M  \n",
       "Q9NQ94/VAR_059821/A1CF                 S         A/S  \n",
       "P01023/VAR_000012/A2M                  H         R/H  \n",
       "...                                  ...         ...  \n",
       "Q8N402/VAR_042675/-                    P         T/P  \n",
       "Q96M66/VAR_039178/-                    H         R/H  \n",
       "Q96M66/VAR_039179/-                    S         R/S  \n",
       "Q9N2K0/VAR_017799/-                    L         V/L  \n",
       "Q9N2K0/VAR_017800/-                    L         F/L  \n",
       "\n",
       "[80854 rows x 13 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# need to consider conditions of:\n",
    "# p.Tyr5033=  synonymous mutation\n",
    "# p.Ter5039*  premature stop codon\n",
    "\n",
    "def extract_amino_acid(row, before = False, pos = False, after = False):\n",
    "    import re\n",
    "    try:\n",
    "        if before:\n",
    "            first_aa = row[2:5]\n",
    "            return first_aa\n",
    "        if pos:\n",
    "            pos = int(re.search(r'\\d+', row).group())\n",
    "            return pos\n",
    "        # the following command extract the index within the string where the position number ends\n",
    "        if after:\n",
    "            first_aa = row[2:5]\n",
    "            position_end_index = re.search(r'\\d+', row).span()[-1]\n",
    "            later_part = row[position_end_index:]\n",
    "            if later_part == '=':\n",
    "                return first_aa\n",
    "            elif later_part == '*':\n",
    "                return 'STOP'\n",
    "            else:\n",
    "                return later_part\n",
    "    except TypeError:\n",
    "        import numpy as np\n",
    "        return np.nan\n",
    "\n",
    "df_var['Amino_acids_before_3letters'] = df_var['Amino_acid_change'].apply(extract_amino_acid, before = True)\n",
    "df_var['Protein_positions'] = df_var['Amino_acid_change'].apply(extract_amino_acid, pos = True)\n",
    "df_var['Amino_acids_after_3letters'] = df_var['Amino_acid_change'].apply(extract_amino_acid, after = True)\n",
    "df_var['Protein_positions'] = df_var['Protein_positions'].astype('Int64')\n",
    "# drop introns and premature stop codon\n",
    "df_var = df_var[(df_var['Amino_acid_change'].notnull()) & (df_var['Amino_acids_after_3letters'] != 'Ter')]\n",
    "# no premature termination\n",
    "# keep only nonsynonymous changes\n",
    "amino_acids_dict = {'Ala': 'A', 'Arg': 'R', 'Asn': 'N', 'Asp': 'D', 'Cys': 'C', 'Glu': 'E', 'Gln': 'Q', 'Gly': 'G', 'His': 'H', 'Ile': 'I', 'Leu': 'L', 'Lys': 'K', 'Met': 'M', 'Phe': 'F', 'Pro': 'P', 'Ser': 'S', 'Thr': 'T', 'Trp': 'W', 'Tyr': 'Y', 'Val': 'V'}\n",
    "df_var['Amino_acids_before'] = df_var['Amino_acids_before_3letters'].map(amino_acids_dict)\n",
    "df_var['Amino_acids_after'] = df_var['Amino_acids_after_3letters'].map(amino_acids_dict)\n",
    "df_var = df_var[df_var['Amino_acids_before'] != df_var['Amino_acids_after']]\n",
    "df_var['Amino_acids'] = df_var['Amino_acids_before'] + '/' + df_var['Amino_acids_after']\n",
    "df_var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Q30201 Q30201\n",
      "Q9ULP9 Q9ULP9\n",
      "P06132 P06132\n",
      "P20062 P20062\n",
      "P35520 P35520\n",
      "P35520 P35520\n",
      "P04181 P04181\n",
      "P04181 P04181\n",
      "P04181 P04181\n",
      "P04181 P04181\n",
      "P04181 P04181\n",
      "P04181 P04181\n",
      "P04181 P04181\n",
      "P04181 P04181\n",
      "P04181 P04181\n",
      "Q9P0N5 Q9P0N5\n",
      "P12955 P12955\n",
      "P12955 P12955\n",
      "P20933 P20933\n",
      "P20933 P20933\n",
      "P00387 P00387\n",
      "Q8WVQ1 Q8WVQ1\n",
      "P04275 P04275\n",
      "P04275 P04275\n",
      "P04275 P04275\n",
      "P04275 P04275\n",
      "P04275 P04275\n",
      "P21359 P21359\n",
      "P21359 P21359\n",
      "P21359 P21359\n",
      "P21359 P21359\n",
      "Q9NRR6 Q9NRR6\n",
      "Q9NRR6 Q9NRR6\n",
      "Q9NX18 Q9NX18\n",
      "P15259 P15259\n",
      "Q9H4B7 Q9H4B7\n",
      "Q9BVA1 Q9BVA1\n",
      "Q6UXH8 Q6UXH8\n",
      "Q6UXH8 Q6UXH8\n",
      "P05062 P05062\n",
      "P05062 P05062\n",
      "P05062 P05062\n",
      "Q03393 Q03393\n",
      "Q03393 Q03393\n",
      "Q03393 Q03393\n",
      "Q03393 Q03393\n",
      "Q03393 Q03393\n",
      "P09417 P09417\n",
      "P35556 P35556\n",
      "Q9BZD2 Q9BZD2\n",
      "Q9BZD2 Q9BZD2\n",
      "Q5TEU4 Q5TEU4\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P00439 P00439\n",
      "P12259 P12259\n",
      "P12259 P12259\n",
      "P12259 P12259\n",
      "P04070 P04070\n",
      "P04070 P04070\n",
      "P04070 P04070\n",
      "P04066 P04066\n",
      "Q86TH1 Q86TH1\n",
      "P34059 P34059\n",
      "P34059 P34059\n",
      "P34059 P34059\n",
      "P34059 P34059\n",
      "P34059 P34059\n",
      "P34059 P34059\n",
      "O14949 O14949\n",
      "Q9BTV4 Q9BTV4\n",
      "P09467 P09467\n",
      "P15848 P15848\n",
      "P08236 P08236\n",
      "P08236 P08236\n",
      "P08236 P08236\n",
      "P08236 P08236\n",
      "Q6NT55 Q6NT55\n",
      "Q6NT55 Q6NT55\n",
      "Q04671 Q04671\n",
      "Q04671 Q04671\n",
      "Q04671 Q04671\n",
      "Q04671 Q04671\n",
      "P0C7Q2 P0C7Q2\n",
      "Q8IXQ5 Q8IXQ5\n",
      "Q8IXQ5 Q8IXQ5\n",
      "Q9Y6X0 Q9Y6X0\n",
      "Q9Y6X0 Q9Y6X0\n",
      "Q9BUM1 Q9BUM1\n",
      "Q27J81 Q27J81\n",
      "Q27J81 Q27J81\n",
      "Q6PI48 Q6PI48\n",
      "Q6PI48 Q6PI48\n",
      "Q68CZ1 Q68CZ1\n",
      "Q68CZ1 Q68CZ1\n",
      "Q68CZ1 Q68CZ1\n",
      "O60393 O60393\n",
      "Q676U5 Q676U5\n",
      "Q12768 Q12768\n",
      "P00748 P00748\n",
      "P00748 P00748\n",
      "P15538 P15538\n",
      "P15538 P15538\n",
      "P15538 P15538\n",
      "P15538 P15538\n",
      "P15538 P15538\n",
      "Q00266 Q00266\n",
      "Q00266 Q00266\n",
      "Q00266 Q00266\n",
      "Q00266 Q00266\n",
      "Q15437 Q15437\n",
      "Q15437 Q15437\n",
      "O75648 O75648\n",
      "Q0ZLH3 Q0ZLH3\n",
      "Q8TAM1 Q8TAM1\n",
      "O15078 O15078\n",
      "Q8TAA9 Q8TAA9\n",
      "P51649 P51649\n",
      "Q5HYA8 Q5HYA8\n",
      "Q5HYA8 Q5HYA8\n",
      "Q5HYA8 Q5HYA8\n",
      "Q9Y4U1 Q9Y4U1\n",
      "Q8N130 Q8N130\n",
      "Q8N130 Q8N130\n",
      "Q8N130 Q8N130\n",
      "P08397 P08397\n",
      "P08397 P08397\n",
      "P08397 P08397\n",
      "P08397 P08397\n",
      "P08397 P08397\n",
      "P08397 P08397\n",
      "P08397 P08397\n",
      "P30613 P30613\n",
      "P30613 P30613\n",
      "P06858 P06858\n",
      "P06858 P06858\n",
      "P06858 P06858\n",
      "P06858 P06858\n",
      "P06858 P06858\n",
      "P54802 P54802\n",
      "P54802 P54802\n",
      "P54802 P54802\n",
      "P54802 P54802\n",
      "P54802 P54802\n",
      "Q8WX94 Q8WX94\n",
      "Q9H9P8 Q9H9P8\n",
      "Q9H9P8 Q9H9P8\n",
      "Q14999 Q14999\n",
      "P28330 P28330\n",
      "P49748 P49748\n",
      "P49748 P49748\n",
      "P49748 P49748\n",
      "P51648 P51648\n",
      "O15528 O15528\n",
      "O15528 O15528\n",
      "O00754 O00754\n",
      "Q6ZVD7 Q6ZVD7\n",
      "Q0D2K0 Q0D2K0\n",
      "P27352 P27352\n",
      "P43246 P43246\n",
      "P43246 P43246\n",
      "P43246 P43246\n",
      "P43246 P43246\n",
      "P43246 P43246\n",
      "P05093 P05093\n",
      "P05093 P05093\n",
      "P05093 P05093\n",
      "P05093 P05093\n",
      "P05093 P05093\n",
      "P05093 P05093\n",
      "Q96AD5 Q96AD5\n",
      "P22033 P22033\n",
      "P22033 P22033\n",
      "P22033 P22033\n",
      "P22033 P22033\n",
      "P22033 P22033\n",
      "P22033 P22033\n",
      "Q96PL5 Q96PL5\n",
      "Q9HCC0 Q9HCC0\n",
      "Q9HCC0 Q9HCC0\n",
      "Q9HCC0 Q9HCC0\n",
      "Q9HCC0 Q9HCC0\n",
      "Q96RQ3 Q96RQ3\n",
      "Q5S007 Q5S007\n",
      "Q5S007 Q5S007\n",
      "Q5S007 Q5S007\n",
      "P00813 P00813\n",
      "P00813 P00813\n",
      "P00813 P00813\n",
      "P00813 P00813\n",
      "P00813 P00813\n",
      "Q3SXY8 Q3SXY8\n",
      "Q9P2D1 Q9P2D1\n",
      "Q86YN6 Q86YN6\n",
      "Q9H0F7 Q9H0F7\n",
      "Q9H0F7 Q9H0F7\n",
      "Q96NR8 Q96NR8\n",
      "Q96NR8 Q96NR8\n",
      "Q96NR8 Q96NR8\n",
      "Q5JVL4 Q5JVL4\n",
      "Q5T442 Q5T442\n",
      "Q5T442 Q5T442\n",
      "Q92947 Q92947\n",
      "Q92947 Q92947\n",
      "Q92947 Q92947\n",
      "Q92947 Q92947\n",
      "Q92947 Q92947\n",
      "Q8NFD2 Q8NFD2\n",
      "Q7Z412 Q7Z412\n",
      "Q96MS0 Q96MS0\n",
      "Q96MS0 Q96MS0\n",
      "Q96MS0 Q96MS0\n",
      "Q96MS0 Q96MS0\n",
      "Q96MS0 Q96MS0\n",
      "Q96MS0 Q96MS0\n",
      "Q6ZWL3 Q6ZWL3\n",
      "Q7Z406 Q7Z406\n",
      "P40337 P40337\n",
      "P40337 P40337\n",
      "P40337 P40337\n",
      "P40337 P40337\n",
      "P40337 P40337\n",
      "P40337 P40337\n",
      "P40337 P40337\n",
      "P40337 P40337\n",
      "P40337 P40337\n",
      "P40337 P40337\n",
      "P40337 P40337\n",
      "P40337 P40337\n",
      "P13498 P13498\n",
      "P13498 P13498\n",
      "O95140 O95140\n",
      "O95140 O95140\n",
      "O95140 O95140\n",
      "O95140 O95140\n",
      "O95140 O95140\n",
      "O95140 O95140\n",
      "O95140 O95140\n",
      "O95140 O95140\n",
      "O95140 O95140\n",
      "Q96MT3 Q96MT3\n",
      "Q7Z333 Q7Z333\n",
      "Q7Z333 Q7Z333\n",
      "Q7Z333 Q7Z333\n",
      "P11217 P11217\n",
      "O95571 O95571\n",
      "Q8NCH0 Q8NCH0\n",
      "Q8NCH0 Q8NCH0\n",
      "O75445 O75445\n",
      "O75445 O75445\n",
      "O75445 O75445\n",
      "O75445 O75445\n",
      "Q6ZVN8 Q6ZVN8\n",
      "Q6ZVN8 Q6ZVN8\n",
      "Q6ZVN8 Q6ZVN8\n",
      "P12694 P12694\n",
      "P05089 P05089\n",
      "P04424 P04424\n",
      "P04424 P04424\n",
      "Q9BXM7 Q9BXM7\n",
      "Q9BXM7 Q9BXM7\n",
      "P31327 P31327\n",
      "P31327 P31327\n",
      "Q7Z4S6 Q7Z4S6\n",
      "Q7Z4S6 Q7Z4S6\n",
      "Q99519 Q99519\n",
      "P30566 P30566\n",
      "P30566 P30566\n",
      "Q16394 Q16394\n",
      "Q16394 Q16394\n",
      "P48436 P48436\n",
      "P48436 P48436\n",
      "Q7RTP0 Q7RTP0\n",
      "Q7RTP0 Q7RTP0\n",
      "Q7RTP0 Q7RTP0\n",
      "Q9H1B5 Q9H1B5\n",
      "Q86Y38 Q86Y38\n",
      "Q6VVB1 Q6VVB1\n",
      "Q6VVB1 Q6VVB1\n",
      "Q6VVB1 Q6VVB1\n",
      "P13804 P13804\n",
      "P13804 P13804\n",
      "P45381 P45381\n",
      "P45381 P45381\n",
      "P45381 P45381\n",
      "P45381 P45381\n",
      "Q9UJY1 Q9UJY1\n",
      "O14773 O14773\n",
      "O14773 O14773\n",
      "O14773 O14773\n",
      "O14773 O14773\n",
      "Q8NBK3 Q8NBK3\n",
      "Q8NBK3 Q8NBK3\n",
      "O95180 O95180\n",
      "O76090 O76090\n",
      "O76090 O76090\n",
      "O76090 O76090\n",
      "O76090 O76090\n",
      "O76090 O76090\n",
      "O76090 O76090\n",
      "O76090 O76090\n",
      "O76090 O76090\n",
      "Q04446 Q04446\n",
      "Q9UJJ9 Q9UJJ9\n",
      "Q9UBY8 Q9UBY8\n",
      "Q9UBY8 Q9UBY8\n",
      "P24752 P24752\n",
      "P24752 P24752\n",
      "Q9H2F3 Q9H2F3\n",
      "P08514 P08514\n",
      "P08514 P08514\n",
      "P08514 P08514\n",
      "P08514 P08514\n",
      "P08514 P08514\n",
      "P08514 P08514\n",
      "Q9UDY2 Q9UDY2\n",
      "P32929 P32929\n",
      "O15118 O15118\n",
      "O15118 O15118\n",
      "O15118 O15118\n",
      "O15118 O15118\n",
      "P35527 P35527\n",
      "P35527 P35527\n",
      "P35527 P35527\n",
      "Q8IWZ6 Q8IWZ6\n",
      "Q13315 Q13315\n",
      "Q13315 Q13315\n",
      "Q13315 Q13315\n",
      "O95278 O95278\n",
      "O95278 O95278\n",
      "Q5VWK5 Q5VWK5\n",
      "Q8NDX2 Q8NDX2\n",
      "P42704 P42704\n",
      "P51787 P51787\n",
      "P51787 P51787\n",
      "P51787 P51787\n",
      "P51787 P51787\n",
      "P51787 P51787\n",
      "P51787 P51787\n",
      "P51787 P51787\n",
      "P51787 P51787\n",
      "P51787 P51787\n",
      "P51787 P51787\n",
      "P51787 P51787\n",
      "P51787 P51787\n",
      "P51787 P51787\n",
      "Q93099 Q93099\n",
      "Q93099 Q93099\n",
      "Q93099 Q93099\n",
      "Q93099 Q93099\n",
      "O75072 O75072\n",
      "O75072 O75072\n",
      "O75072 O75072\n",
      "Q9UKY4 Q9UKY4\n",
      "Q9Y6A1 Q9Y6A1\n",
      "Q01718 Q01718\n",
      "Q01718 Q01718\n",
      "Q6P1J9 Q6P1J9\n",
      "O43395 O43395\n",
      "Q6P2Q9 Q6P2Q9\n",
      "Q6P2Q9 Q6P2Q9\n",
      "Q6P2Q9 Q6P2Q9\n",
      "Q6P2Q9 Q6P2Q9\n",
      "Q6P2Q9 Q6P2Q9\n",
      "O14896 O14896\n",
      "O14896 O14896\n",
      "O14896 O14896\n",
      "Q9NUW8 Q9NUW8\n",
      "Q96HD1 Q96HD1\n",
      "Q9BV10 Q9BV10\n",
      "Q96S37 Q96S37\n",
      "Q96S37 Q96S37\n",
      "P42898 P42898\n",
      "P42898 P42898\n",
      "P42898 P42898\n",
      "Q13286 Q13286\n",
      "P11310 P11310\n",
      "P11310 P11310\n",
      "P07902 P07902\n",
      "P07902 P07902\n",
      "P07902 P07902\n",
      "P01130 P01130\n",
      "P01130 P01130\n",
      "P01130 P01130\n",
      "P01130 P01130\n",
      "P01130 P01130\n",
      "P01130 P01130\n",
      "P01130 P01130\n",
      "P01130 P01130\n",
      "P01130 P01130\n",
      "P10746 P10746\n",
      "P10746 P10746\n",
      "P10746 P10746\n",
      "P10746 P10746\n",
      "P14679 P14679\n",
      "P14679 P14679\n",
      "P14679 P14679\n",
      "P14679 P14679\n",
      "P14679 P14679\n",
      "P14679 P14679\n",
      "P14679 P14679\n",
      "P14679 P14679\n",
      "P14679 P14679\n",
      "P14679 P14679\n",
      "P14679 P14679\n",
      "P14679 P14679\n",
      "P14679 P14679\n",
      "P14679 P14679\n",
      "P51687 P51687\n",
      "P16219 P16219\n",
      "P35670 P35670\n",
      "P35670 P35670\n",
      "P35670 P35670\n",
      "P35670 P35670\n",
      "P35670 P35670\n",
      "P35670 P35670\n",
      "P35670 P35670\n",
      "P35670 P35670\n",
      "P06865 P06865\n",
      "P06865 P06865\n",
      "P06865 P06865\n",
      "P06865 P06865\n",
      "P06865 P06865\n",
      "P06865 P06865\n",
      "P06865 P06865\n",
      "P05155 P05155\n",
      "P05155 P05155\n",
      "Q8WZA1 Q8WZA1\n",
      "P10253 P10253\n",
      "P10253 P10253\n",
      "P10253 P10253\n",
      "P10253 P10253\n",
      "P10253 P10253\n",
      "P10253 P10253\n",
      "P10253 P10253\n",
      "P10253 P10253\n",
      "P07202 P07202\n",
      "P07202 P07202\n",
      "Q9NRD8 Q9NRD8\n",
      "O43897 O43897\n",
      "P50443 P50443\n",
      "P50443 P50443\n",
      "P08F94 P08F94\n",
      "P07359 P07359\n",
      "P07359 P07359\n",
      "Q96RP9 Q96RP9\n",
      "Q9C000 Q9C000\n",
      "Q9NZC9 Q9NZC9\n",
      "Q9NZC9 Q9NZC9\n",
      "Q9NSU2 Q9NSU2\n",
      "Q9NSU2 Q9NSU2\n",
      "Q8TB36 Q8TB36\n",
      "P23760 P23760\n",
      "Q9H9S5 Q9H9S5\n",
      "Q9H9S5 Q9H9S5\n",
      "Q9H9S5 Q9H9S5\n",
      "Q9H9S5 Q9H9S5\n",
      "Q9H6K4 Q9H6K4\n",
      "Q02318 Q02318\n",
      "O75844 O75844\n",
      "P04062 P04062\n",
      "P04062 P04062\n",
      "P04062 P04062\n",
      "P04062 P04062\n",
      "P04062 P04062\n",
      "P04062 P04062\n",
      "P04062 P04062\n",
      "P04062 P04062\n",
      "P04062 P04062\n",
      "P04062 P04062\n",
      "P04062 P04062\n",
      "P04062 P04062\n",
      "P04062 P04062\n",
      "P04062 P04062\n",
      "P04062 P04062\n",
      "P49770 P49770\n",
      "P49770 P49770\n",
      "O43464 O43464\n",
      "Q8WXF7 Q8WXF7\n",
      "Q8WXF7 Q8WXF7\n",
      "Q8WXF7 Q8WXF7\n",
      "Q8WXF7 Q8WXF7\n",
      "Q15392 Q15392\n",
      "Q8WZ55 Q8WZ55\n",
      "Q8WZ55 Q8WZ55\n",
      "P58418 P58418\n",
      "P58418 P58418\n",
      "Q6Q788 Q6Q788\n",
      "O43586 O43586\n",
      "O60931 O60931\n",
      "O60931 O60931\n",
      "O60931 O60931\n",
      "P58753 P58753\n",
      "Q9UPY3 Q9UPY3\n",
      "O76024 O76024\n",
      "Q9BZV2 Q9BZV2\n",
      "Q9BZV2 Q9BZV2\n",
      "P55000 P55000\n",
      "P55000 P55000\n",
      "Q969F9 Q969F9\n",
      "Q96RR1 Q96RR1\n",
      "Q9BT22 Q9BT22\n",
      "Q96A29 Q96A29\n",
      "Q9NPC2 Q9NPC2\n",
      "Q9UI17 Q9UI17\n",
      "P82650 P82650\n",
      "Q9BXJ7 Q9BXJ7\n",
      "O15270 O15270\n",
      "O15269 O15269\n",
      "O15269 O15269\n",
      "O95292 O95292\n",
      "O43511 O43511\n",
      "O43511 O43511\n",
      "O43511 O43511\n",
      "O43511 O43511\n",
      "O43511 O43511\n",
      "O43511 O43511\n",
      "O43511 O43511\n",
      "Q9NP91 Q9NP91\n",
      "Q96T60 Q96T60\n",
      "O95500 O95500\n",
      "P37058 P37058\n",
      "P37058 P37058\n",
      "P37058 P37058\n",
      "Q96QU1 Q96QU1\n",
      "P57727 P57727\n",
      "Q9HBA0 Q9HBA0\n",
      "Q9HBA0 Q9HBA0\n",
      "Q9HBA0 Q9HBA0\n",
      "Q9HBA0 Q9HBA0\n",
      "Q9HBA0 Q9HBA0\n",
      "Q9GZV9 Q9GZV9\n",
      "Q9H2C0 Q9H2C0\n",
      "Q9NRG9 Q9NRG9\n",
      "Q9BQ52 Q9BQ52\n",
      "Q9BQ52 Q9BQ52\n",
      "O15409 O15409\n",
      "Q9GZX3 Q9GZX3\n",
      "Q9GZX3 Q9GZX3\n",
      "Q9GZX3 Q9GZX3\n",
      "Q92574 Q92574\n",
      "Q92574 Q92574\n",
      "P51688 P51688\n",
      "P51688 P51688\n",
      "Q9GZX7 Q9GZX7\n",
      "Q9GZU1 Q9GZU1\n",
      "Q9HBG4 Q9HBG4\n",
      "O14656 O14656\n",
      "O43291 O43291\n",
      "Q9NQW8 Q9NQW8\n",
      "Q13148 Q13148\n",
      "Q13148 Q13148\n",
      "Q13148 Q13148\n",
      "Q9NZR4 Q9NZR4\n",
      "Q9NZR4 Q9NZR4\n",
      "Q9NQ38 Q9NQ38\n",
      "O15020 O15020\n",
      "Q9NPJ1 Q9NPJ1\n",
      "Q9NPJ1 Q9NPJ1\n",
      "Q9Y4Z2 Q9Y4Z2\n",
      "P57679 P57679\n",
      "Q9UEF7 Q9UEF7\n",
      "Q8WTS1 Q8WTS1\n",
      "Q9NP85 Q9NP85\n",
      "Q7LG56 Q7LG56\n",
      "Q9NP59 Q9NP59\n",
      "O43707 O43707\n",
      "O43707 O43707\n",
      "Q12988 Q12988\n",
      "O95970 O95970\n",
      "Q9Y4W6 Q9Y4W6\n",
      "Q9NZJ4 Q9NZJ4\n",
      "Q9Y5X4 Q9Y5X4\n",
      "O95452 O95452\n",
      "O95452 O95452\n",
      "O96017 O96017\n",
      "Q9UI46 Q9UI46\n",
      "O43490 O43490\n",
      "Q9NRA2 Q9NRA2\n",
      "Q9NRA2 Q9NRA2\n",
      "Q9BSI4 Q9BSI4\n",
      "Q9BSI4 Q9BSI4\n",
      "P51570 P51570\n",
      "P01034 P01034\n",
      "Q9Y620 Q9Y620\n",
      "P21549 P21549\n",
      "P21549 P21549\n",
      "P21549 P21549\n",
      "P21549 P21549\n",
      "P21549 P21549\n",
      "Q9UBP0 Q9UBP0\n",
      "Q9UBP0 Q9UBP0\n",
      "Q9UBP0 Q9UBP0\n",
      "Q9UBP0 Q9UBP0\n",
      "O43819 O43819\n",
      "P82279 P82279\n",
      "P82279 P82279\n",
      "Q6P1M0 Q6P1M0\n",
      "Q9H015 Q9H015\n",
      "P82251 P82251\n",
      "Q76LX8 Q76LX8\n",
      "Q76LX8 Q76LX8\n",
      "Q9Y653 Q9Y653\n",
      "Q9UNE0 Q9UNE0\n",
      "Q9UNE0 Q9UNE0\n",
      "O75916 O75916\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "P35499 P35499\n",
      "Q9Y5I7 Q9Y5I7\n",
      "Q9Y5I7 Q9Y5I7\n",
      "Q9Y5I7 Q9Y5I7\n",
      "Q13144 Q13144\n",
      "Q13144 Q13144\n",
      "O60779 O60779\n",
      "Q14520 Q14520\n",
      "Q9P2R7 Q9P2R7\n",
      "O95817 O95817\n",
      "Q9Y619 Q9Y619\n",
      "Q9Y619 Q9Y619\n",
      "Q9Y619 Q9Y619\n",
      "Q9Y223 Q9Y223\n",
      "Q9Y223 Q9Y223\n",
      "Q9Y223 Q9Y223\n",
      "Q9Y223 Q9Y223\n",
      "O43548 O43548\n",
      "Q99732 Q99732\n",
      "Q9Y6H5 Q9Y6H5\n",
      "O95343 O95343\n",
      "O95343 O95343\n",
      "O95343 O95343\n",
      "Q9NZB8 Q9NZB8\n",
      "Q8WW38 Q8WW38\n",
      "Q8WW38 Q8WW38\n",
      "Q9HC10 Q9HC10\n",
      "Q9Y210 Q9Y210\n",
      "Q8N3I7 Q8N3I7\n",
      "Q9Y276 Q9Y276\n",
      "Q9Y276 Q9Y276\n",
      "Q9Y276 Q9Y276\n",
      "Q7KZN9 Q7KZN9\n",
      "O75880 O75880\n",
      "P54577 P54577\n",
      "P54577 P54577\n",
      "O60733 O60733\n",
      "O60733 O60733\n",
      "Q9UM01 Q9UM01\n",
      "P56696 P56696\n",
      "P56696 P56696\n",
      "P56696 P56696\n",
      "P56696 P56696\n",
      "P56696 P56696\n",
      "O75197 O75197\n",
      "O75197 O75197\n",
      "O75197 O75197\n",
      "O60762 O60762\n",
      "P39019 P39019\n",
      "P39019 P39019\n",
      "P00966 P00966\n",
      "P00966 P00966\n",
      "P00966 P00966\n",
      "P00966 P00966\n",
      "O75569 O75569\n",
      "Q15858 Q15858\n",
      "O75369 O75369\n",
      "O75369 O75369\n",
      "O75369 O75369\n",
      "O75369 O75369\n",
      "O76082 O76082\n",
      "O76082 O76082\n",
      "O76082 O76082\n",
      "Q53GS7 Q53GS7\n",
      "Q99814 Q99814\n",
      "O14901 O14901\n",
      "Q8NCM8 Q8NCM8\n",
      "Q8NCM8 Q8NCM8\n",
      "Q9NVS9 Q9NVS9\n",
      "Q9H3D4 Q9H3D4\n",
      "Q9H3D4 Q9H3D4\n",
      "Q9H3D4 Q9H3D4\n",
      "Q9H3D4 Q9H3D4\n",
      "Q9H3D4 Q9H3D4\n",
      "Q9H3D4 Q9H3D4\n",
      "Q9H3D4 Q9H3D4\n",
      "O95255 O95255\n",
      "O95255 O95255\n",
      "O95255 O95255\n",
      "O95255 O95255\n",
      "O95255 O95255\n",
      "O95255 O95255\n",
      "O95255 O95255\n",
      "O95255 O95255\n",
      "O95255 O95255\n",
      "O95342 O95342\n",
      "O95342 O95342\n",
      "O43405 O43405\n",
      "O43405 O43405\n",
      "P15169 P15169\n",
      "Q9Y215 Q9Y215\n",
      "Q13253 Q13253\n",
      "P61764 P61764\n",
      "O60566 O60566\n",
      "Q9UBM7 Q9UBM7\n",
      "Q9UBM7 Q9UBM7\n",
      "Q9UBM7 Q9UBM7\n",
      "Q9UBM7 Q9UBM7\n",
      "Q12840 Q12840\n",
      "Q9UQ13 Q9UQ13\n",
      "O43502 O43502\n",
      "P78386 P78386\n",
      "P78385 P78385\n",
      "O15228 O15228\n",
      "Q9UGJ0 Q9UGJ0\n",
      "Q9UGJ0 Q9UGJ0\n",
      "Q9UGJ0 Q9UGJ0\n",
      "Q9UGJ0 Q9UGJ0\n",
      "P51798 P51798\n",
      "O14521 O14521\n",
      "O14521 O14521\n",
      "O14521 O14521\n",
      "NA O43826\n"
     ]
    },
    {
     "ename": "AssertionError",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
      "Input \u001b[0;32mIn [34]\u001b[0m, in \u001b[0;36m<cell line: 4>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     19\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m     20\u001b[0m         \u001b[38;5;28mprint\u001b[39m(row[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124muniprot_acc\u001b[39m\u001b[38;5;124m'\u001b[39m], uniprot_acc_from_uniprot_var_db)\n\u001b[0;32m---> 21\u001b[0m         \u001b[38;5;28;01massert\u001b[39;00m row[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124muniprot_acc\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m==\u001b[39m uniprot_acc_from_uniprot_var_db\n\u001b[1;32m     22\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mIndexError\u001b[39;00m:\n\u001b[1;32m     23\u001b[0m     \u001b[38;5;28;01mpass\u001b[39;00m\n",
      "\u001b[0;31mAssertionError\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# cross mapping between clinvar and uniprotkb variant database for those don't have uniprotkb id\n",
    "# df_subset from clinvar\n",
    "# df_var from uniprot\n",
    "for index, row in df_subset.iterrows():\n",
    "    var = row['variant_id']\n",
    "    #uniprot_acc = row['uniprot_acc']\n",
    "    amino_acid_before = row['Amino_acids_before']\n",
    "    amino_acid_after = row['Amino_acids_after']\n",
    "    position_of_change = row['Protein_positions']\n",
    "    try:\n",
    "        uniprot_acc_from_uniprot_var_db = df_var.loc[(df_var['Variant_FTId'] == var) &\n",
    "                                                 (df_var['Amino_acids_before'] == amino_acid_before) &\n",
    "                                                 (df_var['Amino_acids_after'] == amino_acid_after) &\n",
    "                                                 (df_var['Protein_positions'] == position_of_change), 'Accession'].values[0]\n",
    "        if row['uniprot_acc'] == np.nan:\n",
    "            print(uniprot_acc_from_uniprot_var_db, row)\n",
    "            row['uniprot_acc'] = uniprot_acc_from_uniprot_var_db\n",
    "            print(uniprot_acc_from_uniprot_var_db, row)\n",
    "        else:\n",
    "            #print(row['uniprot_acc'], uniprot_acc_from_uniprot_var_db)\n",
    "            assert row['uniprot_acc'] == uniprot_acc_from_uniprot_var_db\n",
    "    except IndexError:\n",
    "        pass\n",
    "        #print((df_var['Variant_FTId'] == var))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    },
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
